{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cobrakbase 0.2.3\n"
     ]
    }
   ],
   "source": [
    "import logging\n",
    "import os\n",
    "import pymongo\n",
    "import cobrakbase\n",
    "import time\n",
    "\n",
    "logger = logging.getLogger(__name__)\n",
    "%run ../../scripts/bios_utils.py\n",
    "%run ../../annotation-server/annotation_api_neo4j.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../../annotation-server/annotation_api.py\n",
    "#aclient = pymongo.MongoClient('mongodb://127.0.0.1:27017/')\n",
    "aclient = pymongo.MongoClient(\"mongodb+srv://server:dx75S3HBXX6h2U3D@bios-dk66o.gcp.mongodb.net/test?retryWrites=true&w=majority\")\n",
    "annotation_api_atlas = AnnotationApi(aclient)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../../annotation-server/annotation_api_neo4j.py\n",
    "#THIS MAC\n",
    "host, port, user, pwd = (\"0.0.0.0\", 7687, \"neo4j\", \"123585\")\n",
    "#TK?\n",
    "#host, port, user, pwd = (\"192.168.1.19\", 7687, \"neo4j\", \"123585\")\n",
    "#anl\n",
    "#host, port, user, pwd = (\"sequoia.mcs.anl.gov\", 7687, \"neo4j\", \"SLEEP-CESS!\")\n",
    "\n",
    "annotation_api = AnnotationApiNeo4j(user=user, pwd=pwd, port=port, host=host)\n",
    "annotation_api.neo4j_graph = Graph(\"http://neo4j:123585@0.0.0.0:7474\")\n",
    "annotation_api.matcher = NodeMatcher(annotation_api.neo4j_graph)\n",
    "annotation_api.r_matcher = RelationshipMatcher(annotation_api.neo4j_graph)\n",
    "#annotation_api.init_constraints()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "        let function_ids = [];\n",
    "        _.each(e1, function(function_data, function_name) {\n",
    "          function_ids.push(function_data['id']);\n",
    "        });"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(_85480:KeggOrthology {created_at: 1570258118739, key: 'K01181', updated_at: 1570258118739})"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#K01181 3.2.1.8,  xynA endo-1,4-beta-xylanase\n",
    "#K13465 3.2.1.8,  EIX  ethylene-1,4-beta-xylanase\n",
    "#K20845 3.2.1.32, txyA endo-1,3-beta-xylanase\n",
    "annotation_api.get_ko('K01181')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "K01181"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K02169 {'total': 56, 'has_function': 56}\n"
     ]
    }
   ],
   "source": [
    "o = annotation_api.get_reaction_annotation_data('rxn41452', None, 10, {'K02169' : True})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "target_template_id = 'fungi'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "res = annotation_api.neo4j_graph.nodes.match(\"TemplateReactionAnnotation\")\n",
    "for o in res:\n",
    "    rxn_id = o['reaction']\n",
    "    rxn_key, template_id = o['key'].split('@')\n",
    "    if o['key'] == 'rxn12512_c@GramNegModelTemplateV2' or False:\n",
    "        if o['key'].endswith('@GramNegModelTemplateV2'):\n",
    "            #print(o)\n",
    "            for rel in annotation_api.neo4j_graph.match((o, ), r_type=\"has_function\", ):\n",
    "                function_or_complex = rel.end_node\n",
    "                if 'Function' in function_or_complex.labels:\n",
    "                    function_id = function_or_complex.identity\n",
    "                    prev_score = annotation_api_atlas.get_template_reaction_function_score(rxn_id, 'fungi', function_id)\n",
    "                    if prev_score == None:\n",
    "                        annotation_api_atlas.add_function_to_template_rxn(function_id, rxn_id, 'system', 'fungi', 'opt_score1')\n",
    "                        print(rxn_key, function_id, function_or_complex['key'], prev_score)\n",
    "                elif 'FunctionComplex' in function_or_complex.labels:\n",
    "                    for rel_complex_function in annotation_api.neo4j_graph.match((function_or_complex, ), r_type=\"has_function\", ):\n",
    "                        complex_function = rel_complex_function.end_node\n",
    "                        function_id = complex_function.identity\n",
    "                        prev_score = annotation_api_atlas.get_template_reaction_function_score(rxn_id, 'fungi', function_id)\n",
    "                        if prev_score == None:\n",
    "                            annotation_api_atlas.add_function_to_template_rxn(function_id, rxn_id, 'system', 'fungi', 'opt_score1')\n",
    "                            print(rxn_key, function_id, complex_function['key'], prev_score)\n",
    "                        #print(complex_function)\n",
    "                else:\n",
    "                    print(function_or_complex)\n",
    "        "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 84,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'rxn00973'"
      ]
     },
     "execution_count": 84,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'opt_score2'"
      ]
     },
     "execution_count": 81,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reaction_id = 'rxn00973'\n",
    "function_id = '327401'\n",
    "\n",
    "\n",
    "annotation_api_atlas.get_template_reaction_function_score(reaction_id, 'fungi', 12637)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "template_id = 'fungi'\n",
    "function_id = '5549230' #benzoate 1,2-dioxygenase (EC 1.14.12.10)\n",
    "annotation_api_atlas.get_rxn_with_function(function_id, \n",
    "                                           template_id)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_id': 'rxn03729@fungi',\n",
       " 'functions': {'5549230': 'opt_score1'},\n",
       " 'log': [{'timestamp': 1578248672,\n",
       "   'user_id': 'system',\n",
       "   'action': 'opt_score1',\n",
       "   'target': 5549230}]}"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "doc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#o-succinylbenzoate synthase (EC 4.2.1.113)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def list_genomes(annotation_api):\n",
    "    l = {}\n",
    "    res = annotation_api.page_nodes('RefSeqGenome', 0, 10)\n",
    "    page = 0\n",
    "    while not res == None:\n",
    "        for r in res:\n",
    "            l[r['n']['key']] = r['n']['scientific_name']\n",
    "        page += 1\n",
    "        res = annotation_api.page_nodes('RefSeqGenome', page, 10)\n",
    "    return l\n",
    "loaded_genomes = list_genomes(annotation_api)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1675"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(loaded_genomes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": [
    "#test K01692\n",
    "res = annotation_api.get_functional_roles2('K01692')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "res = annotation_api.get_functional_roles2('K12525')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K00873\n",
      "K00873 {'total': 150, 'has_function': 2}\n",
      "K12406\n",
      "K12406 {'total': 0, 'has_function': 0}\n"
     ]
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotation_api.get_annotation_function(key):\n",
    "    return Neo4jAnnotationFunction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K01739 {'total': 63, 'has_function': 52}\n"
     ]
    }
   ],
   "source": [
    "function_data = annotation_api.get_reaction_annotation_data('rxn15395')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Cystathionine gamma-synthase (EC 2.5.1.48)': {'id': 18173,\n",
       "  'hits': [{'score': 44, 'source': ['KEGG', 'K01739']}]},\n",
       " 'Cystathionine gamma-synthase (EC 2.5.1.48) @ O-acetylhomoserine sulfhydrylase (EC 2.5.1.49)': {'id': 351391,\n",
       "  'hits': [{'score': 6, 'source': ['KEGG', 'K01739']}]},\n",
       " 'O-acetylhomoserine sulfhydrylase (EC 2.5.1.49)': {'id': 18177,\n",
       "  'hits': [{'score': 7, 'source': ['KEGG', 'K01739']}]},\n",
       " 'Cystathionine gamma-lyase (EC 4.4.1.1)': {'id': 13836,\n",
       "  'hits': [{'score': 6, 'source': ['KEGG', 'K01739']}]},\n",
       " 'Cystathionine beta-lyase (EC 4.4.1.8)': {'id': 321777,\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K01739']}],\n",
       "  'sources': {'KBASE_RAST': ['times annotated in genome',\n",
       "    'times in genes',\n",
       "    'ten example genes']},\n",
       "  'subsystems': {}}}"
      ]
     },
     "execution_count": 77,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "function_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K14272 {'total': 0, 'has_function': 0}\n",
      "K00827 {'total': 0, 'has_function': 0}\n",
      "K00830 {'total': 21, 'has_function': 21}\n"
     ]
    },
    {
     "ename": "AttributeError",
     "evalue": "'NoneType' object has no attribute 'identity'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-76-d23230b695e9>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mannotation_api\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_reaction_annotation_data\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'rxn00272'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/workspace/jupyter/python3/annotation-server/annotation_api_neo4j.py\u001b[0m in \u001b[0;36mget_reaction_annotation_data\u001b[0;34m(self, rxn_id, example_genes)\u001b[0m\n\u001b[1;32m    655\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mfunction_data\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    656\u001b[0m             \u001b[0mnode\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmatcher\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmatch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Function\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfirst\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 657\u001b[0;31m             \u001b[0mannotation\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mNeo4jAnnotationFunction\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnode\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    658\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    659\u001b[0m             \u001b[0msubsystem_tags\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/workspace/jupyter/python3/annotation-server/annotation_api_neo4j.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, node)\u001b[0m\n\u001b[1;32m     60\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     61\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnode\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 62\u001b[0;31m         \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m__init__\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mnode\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0midentity\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnode\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'key'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     63\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnode\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mnode\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     64\u001b[0m         \u001b[0;31m#print(node)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mAttributeError\u001b[0m: 'NoneType' object has no attribute 'identity'"
     ]
    }
   ],
   "source": [
    "annotation_api.get_reaction_annotation_data('rxn00272', 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "pair rxn00272_c@GramNegModelTemplate\n",
      "5-aminovalerate transaminase (EC 2.6.1.48)\n",
      "pair rxn00272_c@GramNegModelTemplateV2\n",
      "5-aminovalerate transaminase (EC 2.6.1.48)\n"
     ]
    }
   ],
   "source": [
    "template_id = 'GramNegModelTemplateV2'\n",
    "rxn_id = 'rxn00272'\n",
    "for template_set in annotation_api.page_nodes('TemplateSet', 0, 10):\n",
    "    template_id = template_set['n']['key']\n",
    "    res = annotation_api.get_template_reaction_data(template_id, rxn_id)\n",
    "    for r in res:\n",
    "        print('pair', r['n2']['key'])\n",
    "        node = annotation_api.neo4j_graph.nodes[r['n2'].id]\n",
    "        for rel in annotation_api.neo4j_graph.match((node, ), r_type=\"has_function\", ):\n",
    "            if rel.end_node.has_label('FunctionComplex'):\n",
    "                for rel_cpx in annotation_api.neo4j_graph.match((rel.end_node, ), r_type=\"has_function\", ):\n",
    "                    function = Neo4jAnnotationFunction(rel_cpx.end_node)\n",
    "                    if not function.value in function_data:\n",
    "                        function_data[function.value] = {\n",
    "                            'id' : function.id,\n",
    "                            'hits' : []\n",
    "                        }\n",
    "                    function_data[function.value]['hits'].append({\n",
    "                        'score' : 0,\n",
    "                        'source' : ['template', template_id]\n",
    "                    })\n",
    "            else:\n",
    "                function = Neo4jAnnotationFunction(rel.end_node)\n",
    "                print(function)\n",
    "                if not function.value in function_data:\n",
    "                    function_data[function.value] = {\n",
    "                        'id' : function.id,\n",
    "                        'hits' : []\n",
    "                    }\n",
    "                function_data[function.value]['hits'].append({\n",
    "                    'score' : 0,\n",
    "                    'source' : ['template', template_id]\n",
    "                })"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "5546181"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "function.id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotation_api."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K19726 {'total': 0, 'has_function': 0}\n",
      "K00640 {'total': 117, 'has_function': 115}\n",
      "K00640 Serine acetyltransferase (EC 2.3.1.30)\n",
      "K00640 Hexapeptide transferase family protein\n",
      "K00640 serine acetyltransferase\n",
      "K00640 serine O-acetyltransferase\n",
      "K00640 hypothetical protein\n",
      "K00640 Serine acetyltransferase( EC:2.3.1.30 )\n",
      "K00640 serine O-acetyltransferase( EC:2.3.1.30 )\n",
      "K00640 Serine acetyltransferase\n",
      "scan GramNegModelTemplate\n",
      "pair rxn00423_c@GramNegModelTemplate\n",
      "scan GramNegModelTemplateV2\n",
      "pair rxn00423_c@GramNegModelTemplateV2\n",
      "Serine acetyltransferase (EC 2.3.1.30) dict_keys(['BETA_RAST', 'KBASE_RAST'])\n",
      "Hexapeptide transferase family protein dict_keys(['KBASE_RAST', 'BETA_RAST'])\n",
      "serine acetyltransferase dict_keys(['KBASE_RAST'])\n",
      "serine O-acetyltransferase dict_keys(['KBASE_RAST', 'BETA_RAST'])\n",
      "hypothetical protein dict_keys(['BETA_RAST', 'KBASE_RAST'])\n",
      "Serine acetyltransferase( EC:2.3.1.30 ) dict_keys(['KBASE_RAST'])\n",
      "serine O-acetyltransferase( EC:2.3.1.30 ) dict_keys(['KBASE_RAST'])\n",
      "Serine acetyltransferase dict_keys(['KBASE_RAST'])\n",
      "D-tryptophan N-acetyltransferase (EC 2.3.1.34) dict_keys([])\n"
     ]
    }
   ],
   "source": [
    "def get_reaction_annotation_data2(annotation_api, rxn_id, example_genes = 3):\n",
    "    kos = annotation_api.get_ko_by_seed_id(rxn_id)\n",
    "    function_data = {}\n",
    "    for ko in kos:\n",
    "        functions, functions_data, metadata = annotation_api.get_functional_roles2(ko)\n",
    "        print(ko, metadata)\n",
    "        for f in functions:\n",
    "            print(ko, functions_data[f]['value'])\n",
    "            function = functions_data[f]['value']\n",
    "            if not function in function_data:\n",
    "                function_data[function] = {\n",
    "                    'id' : f,\n",
    "                    'hits' : []\n",
    "                }\n",
    "            function_data[function]['hits'].append({\n",
    "                'score' : len(functions[f]),\n",
    "                'source' : ['KEGG', ko]\n",
    "            })\n",
    "\n",
    "    #MISSING ADD TEMPLATE DATA\n",
    "    for template_set in annotation_api.page_nodes('TemplateSet', 0, 10):\n",
    "        template_id = template_set['n']['key']\n",
    "        print('scan', template_id)\n",
    "        res = get_template_reaction_data(annotation_api, template_id, rxn_id)\n",
    "        for r in res:\n",
    "            print('pair', r['n2']['key'])\n",
    "            node = annotation_api.neo4j_graph.nodes[r['n2'].id]\n",
    "            for rel in annotation_api.neo4j_graph.match((node, ), r_type=\"has_function\", ):\n",
    "                function = Neo4jAnnotationFunction(rel.end_node)\n",
    "                if not function.value in function_data:\n",
    "                    function_data[function.value] = {\n",
    "                        'id' : function.id,\n",
    "                        'hits' : []\n",
    "                    }\n",
    "                function_data[function.value]['hits'].append({\n",
    "                    'score' : 0,\n",
    "                    'source' : ['template', template_id]\n",
    "                })\n",
    "    #MISSING ADD SBML DATA\n",
    "    \n",
    "    search_value_to_function = {}\n",
    "\n",
    "    for f in function_data:\n",
    "        node = annotation_api.matcher.match(\"Function\", key=f).first()\n",
    "        annotation = Neo4jAnnotationFunction(node)\n",
    "        if not annotation.search_value in search_value_to_function:\n",
    "            search_value_to_function[annotation.search_value] = set()\n",
    "        search_value_to_function[annotation.search_value].add(annotation.value)\n",
    "\n",
    "        subsystem_tags = {}\n",
    "        for ss in annotation.function_group:\n",
    "            subsystem_tags[ss] = {}\n",
    "\n",
    "        fcount = get_function_count(annotation_api, annotation)\n",
    "        print(f, fcount.keys())\n",
    "        source_tags = {}\n",
    "        #sources = annotation.neo4j_load_source()\n",
    "        for s in fcount:\n",
    "            source_tags[s] = [len(fcount[s]['genomes']), len(fcount[s]['genes']), list(fcount[s]['genes'])[:example_genes]]\n",
    "        \n",
    "        function_data[f]['sources'] = source_tags\n",
    "        function_data[f]['subsystems'] = subsystem_tags\n",
    "\n",
    "    return function_data, search_value_to_function\n",
    "\n",
    "#rxn15395 @\n",
    "#rxn00423 mult search names\n",
    "function_data, search_value_to_function = get_reaction_annotation_data2(annotation_api, 'rxn00423', 1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 71,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Serine acetyltransferase (EC 2.3.1.30)': {'id': 13414,\n",
       "  'hits': [{'score': 107, 'source': ['KEGG', 'K00640']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplate']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplateV2']}],\n",
       "  'sources': {'BETA_RAST': [136, 170, ['CIMIT_RS09635@GCF_000739455.1']],\n",
       "   'KBASE_RAST': [1111, 1480, ['CSAL_RS12925@GCF_000055785.1']]},\n",
       "  'subsystems': {'Cysteine synthesis': {}}},\n",
       " 'Hexapeptide transferase family protein': {'id': 361500,\n",
       "  'hits': [{'score': 2, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [13, 14, ['ACIS_RS01195@GCF_000024505.1']],\n",
       "   'BETA_RAST': [1, 1, ['I597_RS07610@GCF_001653755.1']]},\n",
       "  'subsystems': {}},\n",
       " 'serine acetyltransferase': {'id': 491604,\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [5, 5, ['ECHVI_RS14465@GCF_000325705.1']]},\n",
       "  'subsystems': {}},\n",
       " 'serine O-acetyltransferase': {'id': 892117,\n",
       "  'hits': [{'score': 2, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [8, 14, ['MPAL_RS02950@GCF_000021965.1']],\n",
       "   'BETA_RAST': [1, 3, ['MPAL_RS02950@GCF_000021965.1']]},\n",
       "  'subsystems': {}},\n",
       " 'hypothetical protein': {'id': 315778,\n",
       "  'hits': [{'score': 2, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'BETA_RAST': [201, 210778, ['RBI_RS12985@GCF_000723465.1']],\n",
       "   'KBASE_RAST': [1562, 1196033, ['TS85_RS17155@GCF_000935025.1']]},\n",
       "  'subsystems': {}},\n",
       " 'Serine acetyltransferase( EC:2.3.1.30 )': {'id': 1384573,\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [3, 3, ['LA_1821@GCF_000092565.1']]},\n",
       "  'subsystems': {}},\n",
       " 'serine O-acetyltransferase( EC:2.3.1.30 )': {'id': 1394321,\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [2, 2, ['OEOE_RS00365@GCF_000014385.1']]},\n",
       "  'subsystems': {}},\n",
       " 'Serine acetyltransferase': {'id': 682230,\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [9, 9, ['RLEG3_RS22835@GCF_000517605.1']]},\n",
       "  'subsystems': {}},\n",
       " 'D-tryptophan N-acetyltransferase (EC 2.3.1.34)': {'id': 5546181,\n",
       "  'hits': [{'score': 0, 'source': ['template', 'GramNegModelTemplate']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplateV2']}],\n",
       "  'sources': {},\n",
       "  'subsystems': {}}}"
      ]
     },
     "execution_count": 71,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "function_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'Serine acetyltransferase (EC 2.3.1.30)': {'id': 13414,\n",
       "  'hits': [{'score': 107, 'source': ['KEGG', 'K00640']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplate']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplateV2']}],\n",
       "  'sources': {'BETA_RAST': [136, 170, ['CIMIT_RS09635@GCF_000739455.1']],\n",
       "   'KBASE_RAST': [1111, 1480, ['CSAL_RS12925@GCF_000055785.1']]},\n",
       "  'subsystems': {'Cysteine synthesis': {}}},\n",
       " 'Hexapeptide transferase family protein': {'id': 361500,\n",
       "  'hits': [{'score': 2, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [13, 14, ['ACIS_RS01195@GCF_000024505.1']],\n",
       "   'BETA_RAST': [1, 1, ['I597_RS07610@GCF_001653755.1']]},\n",
       "  'subsystems': {}},\n",
       " 'serine acetyltransferase': {'id': 491604,\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [5, 5, ['ECHVI_RS14465@GCF_000325705.1']]},\n",
       "  'subsystems': {}},\n",
       " 'serine O-acetyltransferase': {'id': 892117,\n",
       "  'hits': [{'score': 2, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [8, 14, ['MPAL_RS02950@GCF_000021965.1']],\n",
       "   'BETA_RAST': [1, 3, ['MPAL_RS02950@GCF_000021965.1']]},\n",
       "  'subsystems': {}},\n",
       " 'Serine acetyltransferase( EC:2.3.1.30 )': {'id': 1384573,\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [3, 3, ['LA_1821@GCF_000092565.1']]},\n",
       "  'subsystems': {}},\n",
       " 'hypothetical protein': {'id': 315778,\n",
       "  'hits': [{'score': 2, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'BETA_RAST': [201, 210778, ['RBI_RS12985@GCF_000723465.1']],\n",
       "   'KBASE_RAST': [1562, 1196033, ['TS85_RS17155@GCF_000935025.1']]},\n",
       "  'subsystems': {}},\n",
       " 'serine O-acetyltransferase( EC:2.3.1.30 )': {'id': 1394321,\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [2, 2, ['OEOE_RS00365@GCF_000014385.1']]},\n",
       "  'subsystems': {}},\n",
       " 'Serine acetyltransferase': {'id': 682230,\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K00640']}],\n",
       "  'sources': {'KBASE_RAST': [9, 9, ['RLEG3_RS22835@GCF_000517605.1']]},\n",
       "  'subsystems': {}},\n",
       " 'D-tryptophan N-acetyltransferase (EC 2.3.1.34)': {'id': 'D-tryptophan N-acetyltransferase (EC 2.3.1.34)',\n",
       "  'hits': [{'score': 0, 'source': ['template', 'GramNegModelTemplate']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplateV2']}],\n",
       "  'sources': {},\n",
       "  'subsystems': {}}}"
      ]
     },
     "execution_count": 66,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "function_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cysteinesynthasebec25147\n",
      "\t Cysteine synthase B (EC 2.5.1.47)\n",
      "\t {'id': 13418, 'hits': [{'score': 17, 'source': ['KEGG', 'K01738']}], 'sources': {'KBASE_RAST': [364, 459, ['RHE_RS09075@GCF_000092045.1']]}, 'subsystems': {'Cysteine synthesis': {}}}\n",
      "cysteinesynthaseec25147\n",
      "\t Cysteine synthase (EC 2.5.1.47)\n",
      "\t {'id': 13416, 'hits': [{'score': 105, 'source': ['KEGG', 'K01738']}], 'sources': {'KBASE_RAST': [1008, 1556, ['DMR_RS17370@GCF_000010665.1']]}, 'subsystems': {'Cysteine synthesis': {}}}\n",
      "pyridoxal5phosphatedependentenzymebetasubunit\n",
      "\t Pyridoxal-5'-phosphate-dependent enzyme, beta subunit\n",
      "\t {'id': 317316, 'hits': [{'score': 6, 'source': ['KEGG', 'K01738']}], 'sources': {'KBASE_RAST': [33, 33, ['VC1061@GCF_000006745.1']]}, 'subsystems': {}}\n",
      "cystathioninebetasynthaseec42122\n",
      "\t Cystathionine beta-synthase (EC 4.2.1.22)\n",
      "\t {'id': 13838, 'hits': [{'score': 10, 'source': ['KEGG', 'K01738']}], 'sources': {'KBASE_RAST': [362, 392, ['AA314_RS45255@GCF_001027285.1']]}, 'subsystems': {'Cysteine synthesis': {}}}\n",
      "putativelyase\n",
      "\t putative lyase\n",
      "\t {'id': 373522, 'hits': [{'score': 1, 'source': ['KEGG', 'K01738']}], 'sources': {'KBASE_RAST': [111, 168, ['ABM27_RS13780@GCF_001013905.1']]}, 'subsystems': {}}\n",
      "cysteinesynthase\n",
      "\t cysteine synthase\n",
      "\t {'id': 785210, 'hits': [{'score': 1, 'source': ['KEGG', 'K01738']}], 'sources': {'KBASE_RAST': [19, 19, ['FACI_RS09240@GCF_000152265.2']]}, 'subsystems': {}}\n"
     ]
    }
   ],
   "source": [
    "for sv in search_value_to_function:\n",
    "    print(sv)\n",
    "    for f in search_value_to_function[sv]:\n",
    "        print('\\t', f)\n",
    "        print('\\t', function_data[f])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [],
   "source": [
    "def _fetch_gene_annotation_relationships(tx, function_str):\n",
    "    query = \"MATCH (n:KBaseGene)-[r:has_annotation]->(f:Function {key:$f})\" \\\n",
    "            \"RETURN n.key AS gene_id, r.function_source AS function_source\"\n",
    "    #ORDER BY n.key SKIP $skip LIMIT $page_size\n",
    "    result = tx.run(query, f=function_str)\n",
    "    return list(result.records())\n",
    "\n",
    "def fetch_gene_annotation_relationships(annotation_api, function_str):\n",
    "    result = None\n",
    "    with annotation_api.driver.session() as session:\n",
    "        result = session.read_transaction(_fetch_gene_annotation_relationships, function_str)\n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "DSHI_RS04555@GCF_000018145.1 ['KBASE_RAST']\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "1744"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res = annotation_api.fetch_gene_annotation_relationships(f)\n",
    "for record in res:\n",
    "    print(record['gene_id'], record['function_source'].split(';'))\n",
    "    break\n",
    "len(res)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "hypothetical protein\n",
      "KBASE_RAST 1129218 1483\n",
      "CPU times: user 24 s, sys: 1.1 s, total: 25.1 s\n",
      "Wall time: 29 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "#f = 'Allophanate hydrolase 2 subunit 2 (EC 3.5.1.54)'\n",
    "f = 'hypothetical protein'\n",
    "#f = 'Cysteine synthase (EC 2.5.1.47)'\n",
    "node = annotation_api.matcher.match(\"Function\", key=f).first()\n",
    "annotation = Neo4jAnnotationFunction(node)\n",
    "print(annotation)\n",
    "\n",
    "result = get_function_count(annotation_api, annotation)\n",
    "for s in result:\n",
    "    print(s, len(result[s]['genes']), len(result[s]['genomes']))\n",
    "    \n",
    "#Cysteine synthase (EC 2.5.1.47)\n",
    "#KBASE_RAST 1744 1129\n",
    "#hypothetical protein\n",
    "#KBASE_RAST 1118753 1471"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "    #if search_str load function group\n",
    "    #return Neo4jAnnotationFunction(!)\n",
    "annotation_function = get_annotation_function(annotation_api, 'Pyruvate Kinase', search_str=True)\n",
    "if not annotation_function == None:\n",
    "    print(annotation_function)\n",
    "    print(annotation_function.synonyms)\n",
    "    print(annotation_function.function_group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "matcher = NodeMatcher(annotation_api.neo4j_graph)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "for rel in node.graph.match((node, ), r_type=\"has_annotation\", ):\n",
    "    annotation = Neo4jAnnotationFunction(rel.end_node)\n",
    "    annotation.neo4j_load_source()\n",
    "    print(annotation.value, annotation.source)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 244,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotation_api.matcher.match('KBaseGene', \"MG_RS01230@GCF_000027325.1\").first()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 247,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def exists(aaaa, node_type, key):\n",
    "    res = aaaa.matcher.match(node_type, key).first()\n",
    "    return not res == None\n",
    "exists(annotation_api, 'KBaseGene', 'MG_RS01230@GCF_000027325.1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 285,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "[<Record r=<Relationship id=658960 nodes=(<Node id=75898 labels=set() properties={}>, <Node id=412153 labels=set() properties={}>) type='has_gene' properties={'test': 'oh nores'}>>]"
      ]
     },
     "execution_count": 285,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.set_relationship_property(annotation_api, 658960, \"test\", \"oh nores\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 311,
   "metadata": {},
   "outputs": [],
   "source": [
    "feature = KBaseGenomeFeature(list(filter(lambda x : x['id'] == \"MG_RS01230\", genome.features))[0])\n",
    "res = annotation_api.add_genome_feature(feature, 'GCF_000027325.1', 'BETA_RASTxxx')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 259,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#THIS IS TRASH!\n",
    "def update_relationship(annotation_api, relationship, props = {}):\n",
    "    if len(props) == 0:\n",
    "        return False\n",
    "    if not relationship == None:\n",
    "        relationship.update(props)\n",
    "        tx = annotation_api.neo4j_graph.begin()\n",
    "        tx.merge(relationship)\n",
    "        tx.graph.push(relationship)\n",
    "        tx.commit()\n",
    "        return True\n",
    "    return False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 312,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 313,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_data = read_json('../../../data/kbase_data/jplfaria_narrative_1524466549180/GCF_000027325.1.RAST.json')\n",
    "genome = cobrakbase.core.KBaseGenome(genome_data)\n",
    "genome_node = annotation_api.add_kbase_genome(genome, 'GCF_000027325.1', 'KBASE_RAST')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 184,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_data = read_json('../../../data/kbase_data/filipeliu_narrative_1569597697005/GCF_000027325.1.RAST2.json')\n",
    "genome = cobrakbase.core.KBaseGenome(genome_data)\n",
    "genome_node = annotation_api.add_kbase_genome(genome, 'GCF_000027325.1', 'BETA_RAST')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "507"
      ]
     },
     "execution_count": 185,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(genome.features)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 192,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'MG_RS00005'"
      ]
     },
     "execution_count": 192,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r.get('id')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 175,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 175,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'function_source' in r"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "'RAST' in r['function_source']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_items([('function_source', 'KBASE_RAST')])"
      ]
     },
     "execution_count": 167,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r.items()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 165,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 165,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "update_relationship(annotation_api, r, 1, {'function_source' : 'KBASE_RAST'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 170,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
