{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "import redis\n",
    "import logging\n",
    "import jsonpickle\n",
    "import pymongo\n",
    "from py2neo import Graph, NodeMatcher, RelationshipMatcher\n",
    "logging.getLogger('__main__').setLevel(logging.DEBUG)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../curation_api.py\n",
    "mclient = pymongo.MongoClient('mongodb://127.0.0.1:27017/')\n",
    "annotation_api_atlas = CurationApi(mclient)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "#cache = redis.Redis(host='localhost', port=6379, db=0)\n",
    "cache = redis.Redis(host='192.168.1.19', port=6379, db=0) #TK"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:get_reaction_annotation_data3_2::cache lookup\n",
      "DEBUG:__main__:get_cache [get_reaction_annotation_data3_2$rxn16573:K09789#K02170#K19561#K19560:None:10]\n",
      "DEBUG:__main__:get_reaction_annotation_data3_2::cache return\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 10.9 ms, sys: 1.94 ms, total: 12.8 ms\n",
      "Wall time: 19.9 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "rxn_id = 'rxn16573'\n",
    "body = {\n",
    "    'genome_set_id' : 'ModelSEED2'\n",
    "}\n",
    "rxn_annotation_manual_ko = {'ko' : {}}\n",
    "rxn_annotation_manual_function = {'functions' : {}}\n",
    "genome_set_id = None\n",
    "rxn_annotation = annotation_api.get_reaction_annotation_data3(rxn_id, genome_set_id, 10, \n",
    "                                                             rxn_annotation_manual_ko['ko'], \n",
    "                                                             rxn_annotation_manual_function['functions'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "rxn41452"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "rxn02264"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:get_reaction_annotation_data3_2::cache lookup\n",
      "DEBUG:__main__:get_cache [get_reaction_annotation_data3_2$rxn41452::None:10]\n",
      "DEBUG:__main__:get_reaction_annotation_data3_2::cache not found get live\n",
      "DEBUG:__main__:get_reaction_annotation_data3_2::cache live data\n",
      "DEBUG:__main__:set_cache [get_reaction_annotation_data3_2$rxn41452::None:10]\n"
     ]
    }
   ],
   "source": [
    "rxn_id = 'rxn41452'\n",
    "rxn_annotation_manual_ko = {'ko' : {}}\n",
    "rxn_annotation_manual_function = {'functions' : {}}\n",
    "genome_set_id = None\n",
    "rxn_annotation = annotation_api.get_reaction_annotation_data3(rxn_id, genome_set_id, 10, \n",
    "                                                             rxn_annotation_manual_ko['ko'], \n",
    "                                                             rxn_annotation_manual_function['functions'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:get_reaction_annotation_data3_2::cache lookup\n",
      "DEBUG:__main__:get_cache [get_reaction_annotation_data3_2$rxn02264:K13542#K01719#K13543:None:10]\n",
      "DEBUG:__main__:get_reaction_annotation_data3_2::cache return\n"
     ]
    }
   ],
   "source": [
    "rxn_id = 'rxn02264'\n",
    "rxn_annotation_manual_ko = {'ko' : {'K13543' : True}}\n",
    "rxn_annotation_manual_function = {'functions' : {}}\n",
    "genome_set_id = None\n",
    "rxn_annotation = annotation_api.get_reaction_annotation_data3(rxn_id, genome_set_id, 10, \n",
    "                                                             rxn_annotation_manual_ko['ko'], \n",
    "                                                             rxn_annotation_manual_function['functions'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "template_id = 'fungi'\n",
    "reaction_template_id = '{}@{}'.format(rxn_id, template_id)\n",
    "rxn_annotation_curation = annotation_api_atlas.collection_templates_reactions.find_one({'_id' : reaction_template_id})\n",
    "rxn_annotation_functions_rxn = {}\n",
    "function_ids = set()\n",
    "for name in rxn_annotation:\n",
    "    function_ids.add(rxn_annotation[name]['id'])\n",
    "for function_id in function_ids:\n",
    "    res = annotation_api_atlas.get_rxn_with_function(function_id, template_id)\n",
    "    if res == None:\n",
    "        res = {}\n",
    "    rxn_annotation_functions_rxn[function_id] = res"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": [
    "rxn_annotation_functions_rxn[1] = \"\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'{\"aa\": {\"1\": \"b\"}, \"bb\": {\"1\": \"c\"}}'"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from flask import Flask, request, jsonify\n",
    "json.dumps(a)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'aa': {1: 'b'}, 'bb': {'1': 'c'}}"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = {\n",
    "    'aa' : {1 : 'b'},\n",
    "    'bb' : {\"1\" : 'c'}\n",
    "}\n",
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cobrakbase 0.2.4\n"
     ]
    }
   ],
   "source": [
    "%run ../../annotation-server/annotation_api_neo4j.py\n",
    "%run ../../annotation-server/annotation_api_redis.py\n",
    "#THIS MAC\n",
    "host, port, user, pwd = (\"0.0.0.0\", 7687, \"neo4j\", \"123585\")\n",
    "#TK?\n",
    "#host, port, user, pwd = (\"192.168.1.19\", 7687, \"neo4j\", \"123585\")\n",
    "#anl\n",
    "#host, port, user, pwd = (\"sequoia.mcs.anl.gov\", 7687, \"neo4j\", \"SLEEP-CESS!\")\n",
    "\n",
    "annotation_api = AnnotationApiRedisCache(cache, user=user, pwd=pwd, port=port, host=host)\n",
    "annotation_api.neo4j_graph = Graph(\"http://neo4j:123585@0.0.0.0:7474\")\n",
    "annotation_api.matcher = NodeMatcher(annotation_api.neo4j_graph)\n",
    "annotation_api.r_matcher = RelationshipMatcher(annotation_api.neo4j_graph)\n",
    "#annotation_api.init_constraints()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:set_cache [temp_key]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.set_cache('temp_key', rxn_annotation, 600)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "''"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "genome_set\n",
    "genome_set_arg = \"None\" if genome_set == None else genome_set\n",
    "reaction_annotation_data_key = \"reaction_annotation_data:{}\".fomart()\n",
    "reaction_annotation_data_key"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:get_cache [temp_key]\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rxn_annotation == annotation_api.get_cache('temp_key')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:get_genome_set::cache lookup\n",
      "DEBUG:__main__:get_genome_set::cache return\n",
      "DEBUG:__main__:get_ko_function_data::cache lookup\n",
      "DEBUG:__main__:get_cache [ko_function_data::K00161#K00162#K00163#K01568#K01652#K01653#K11258]\n",
      "DEBUG:__main__:get_ko_function_data::cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [15553]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [15605]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [317129]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [15595]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [360186]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [15603]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [13764]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [315778]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [323968]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [15549]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [15591]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [317131]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [15551]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [15593]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [502403]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [15563]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [357189]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [357188]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [13762]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [440067]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [5550153]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [321588]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [9293]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [339567]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [17791]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [401563]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [321298]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [340272]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [564667]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [382053]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [5541983]\n",
      "DEBUG:__main__:cache not found get live\n",
      "DEBUG:__main__:cache live data\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [17310]\n",
      "DEBUG:__main__:cache not found get live\n",
      "DEBUG:__main__:cache live data\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [5545457]\n",
      "DEBUG:__main__:cache not found get live\n",
      "DEBUG:__main__:cache live data\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [5544669]\n",
      "DEBUG:__main__:cache not found get live\n",
      "DEBUG:__main__:cache live data\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [17314]\n",
      "DEBUG:__main__:cache not found get live\n",
      "DEBUG:__main__:cache live data\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [4292224]\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:get_function_count::cache lookup\n",
      "DEBUG:__main__:get_cache_get_function_count [17312]\n",
      "DEBUG:__main__:cache not found get live\n",
      "DEBUG:__main__:cache live data\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 3.54 s, sys: 278 ms, total: 3.82 s\n",
      "Wall time: 5.28 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "rxn_id = 'rxn00011'\n",
    "body = {\n",
    "    'genome_set_id' : 'ModelSEED2'\n",
    "}\n",
    "rxn_annotation_manual_ko = {'ko' : {}}\n",
    "rxn_annotation_manual_function = {'functions' : {}}\n",
    "genome_set = None\n",
    "if 'genome_set_id' in body:\n",
    "    genome_set = annotation_api.get_genome_set(body['genome_set_id'])\n",
    "rxn_annotation = annotation_api.get_reaction_annotation_data(rxn_id, genome_set, 10, \n",
    "                                                             rxn_annotation_manual_ko['ko'], \n",
    "                                                             rxn_annotation_manual_function['functions'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "a, b, c = annotation_api.get_functional_roles2('K00873')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:cache lookup\n",
      "DEBUG:__main__:cache return\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "336\n",
      "186\n",
      "CPU times: user 5.76 ms, sys: 1.61 ms, total: 7.37 ms\n",
      "Wall time: 15.3 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "function_str = 'Biotin carboxyl carrier protein'\n",
    "gene_function = annotation_api.get_function(function_str)\n",
    "fcount = annotation_api.get_function_count(gene_function)\n",
    "print(len(fcount['BETA_RAST']['genes']))\n",
    "print(len(fcount['KBASE_RAST']['genes']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:cache lookup\n",
      "DEBUG:__main__:cache not found get live\n",
      "DEBUG:__main__:cache live data\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1747551\n",
      "1217551\n",
      "CPU times: user 51.1 s, sys: 661 ms, total: 51.7 s\n",
      "Wall time: 3min 11s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "function_str = 'hypothetical protein'\n",
    "gene_function = annotation_api.get_function(function_str)\n",
    "fcount = annotation_api.get_function_count(gene_function)\n",
    "print(len(fcount['BETA_RAST']['genes']))\n",
    "print(len(fcount['KBASE_RAST']['genes']))\n",
    "#1747551 1196033"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:cache lookup\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:cache lookup\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:cache lookup\n",
      "DEBUG:__main__:cache not found get live\n",
      "DEBUG:__main__:cache live data\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K01689 {'total': 124, 'has_function': 124}\n",
      "CPU times: user 411 ms, sys: 26.7 ms, total: 437 ms\n",
      "Wall time: 719 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "rxn_id = 'rxn00459'\n",
    "body = {\n",
    "    'genome_set_id' : 'ModelSEED2'\n",
    "}\n",
    "rxn_annotation_manual_ko = {'ko' : {}}\n",
    "rxn_annotation_manual_function = {'functions' : {}}\n",
    "genome_set = None\n",
    "if 'genome_set_id' in body:\n",
    "    genome_set = annotation_api.get_genome_set(body['genome_set_id'])\n",
    "rxn_annotation = annotation_api.get_reaction_annotation_data(rxn_id, genome_set, 10, \n",
    "                                                             rxn_annotation_manual_ko['ko'], \n",
    "                                                             rxn_annotation_manual_function['functions'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "neo4j.types.graph.Node"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(gene_function)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_ko_genes(aaaaaaaaaa, ko_id):\n",
    "    m = aaaaaaaaaa.neo4j_graph.nodes.match(\"KeggOrthology\", key=ko_id)\n",
    "    if len(m) < 1:\n",
    "        return None\n",
    "    ko_node = m.first()\n",
    "    gene_nodes = set()\n",
    "    #KeggOrthology -[:has_gene]-> Node\n",
    "    for rel in aaaaaaaaaa.neo4j_graph.match((ko_node, ), r_type=\"has_gene\", ):\n",
    "        gene_nodes.add(rel.end_node)\n",
    "        \n",
    "    if cache:\n",
    "        get_ko_genes_cache(cache, ko_id)\n",
    "    else:\n",
    "        \n",
    "    return gene_nodes\n",
    "\n",
    "def get_ko_genes_cache(cache, ko_id):\n",
    "    \n",
    "    ko_genes = get_ko_genes(aaaaaaaaaa, ko_id)\n",
    "    set_ko_genes_cache(cache, ko_id, ko_genes)\n",
    "    \n",
    "    return None\n",
    "def set_ko_genes_cache(cache, ko_id, genes):\n",
    "    return None\n",
    "\n",
    "def get_functional_roles3(aaaaaaaaaa, ko_id):\n",
    "    m = aaaaaaaaaa.neo4j_graph.nodes.match(\"KeggOrthology\", key=ko_id)\n",
    "    if len(m) < 1:\n",
    "        pass\n",
    "    ko_node = m.first()\n",
    "    gene_nodes = set()\n",
    "    #KeggOrthology -[:has_gene]-> Node\n",
    "    for rel in aaaaaaaaaa.neo4j_graph.match((ko_node, ), r_type=\"has_gene\", ):\n",
    "        gene_nodes.add(rel.end_node)\n",
    "\n",
    "    functions = {}\n",
    "    functions_data = {}\n",
    "\n",
    "    found = 0\n",
    "    for gene_node in gene_nodes:\n",
    "        #print(gene_node)\n",
    "        gene_functions = set()\n",
    "        #gene_node -[:has_annotatation]-> gene_function\n",
    "        for rel in aaaaaaaaaa.neo4j_graph.match((gene_node, ), r_type=\"has_annotation\", ):\n",
    "            function = Neo4jAnnotationFunction(rel.end_node)\n",
    "            gene_functions.add(function)\n",
    "        if not len(gene_functions) == 0:\n",
    "            found+= 1\n",
    "            for gene_function in gene_functions:\n",
    "                if not gene_function.id in functions:\n",
    "                    functions[gene_function.id] = set()\n",
    "                functions[gene_function.id].add(gene_node['key'])\n",
    "                functions_data[gene_function.id] = {'value' : gene_function.value}\n",
    "                for sub_function in gene_function.sub_functions:\n",
    "                    if not sub_function.id in functions:\n",
    "                        functions[sub_function.id] = set()\n",
    "                    functions[sub_function.id].add(gene_node['key'])\n",
    "                    functions_data[sub_function.id] = {'value' : sub_function.value}\n",
    "                #print(gene_node['id'], gene_function['key'])\n",
    "\n",
    "    return functions, functions_data, {'total' : len(gene_nodes), 'has_function' : found}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'gene:140586405441208'"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "k = 'gene:' + str(id(gene_node))\n",
    "k"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cache.expire(k, 10)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "-2"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cache.ttl(k)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{}"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cache.hgetall(k)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cache.hmset('gene:' + str(id(gene_node)), dict(gene_node.items()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 48,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_functions(gene):\n",
    "    pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "140586405441208\n",
      "Pyruvate kinase (EC 2.7.1.40) / Phosphohistidine swiveling domain\n",
      "Pyruvate kinase (EC 2.7.1.40)\n"
     ]
    }
   ],
   "source": [
    "for gene_node in gene_nodes:\n",
    "    print(id(gene_node))\n",
    "    gene_functions = set()\n",
    "    for rel in annotation_api.neo4j_graph.match((gene_node, ), r_type=\"has_annotation\", ):\n",
    "        function = Neo4jAnnotationFunction(rel.end_node)\n",
    "        print(function)\n",
    "        gene_functions.add(function)\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "150"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_nodes = get_ko_genes(annotation_api, 'K00873')\n",
    "len(gene_nodes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../../annotation-server/annotation_api_neo4j.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "a, b, c = get_functional_roles3(annotation_api, 'K00873')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_ko_genes(annotation_api, ko_id):\n",
    "    res = None\n",
    "    if annotation_api.:\n",
    "        res = annotation_api.cache.get_cache_ko_genes(ko_id)\n",
    "        if res:\n",
    "            return res\n",
    "    \n",
    "    res = annotation_api.get_ko_genes():\n",
    "    \n",
    "    res = annotation_api.cache.set_cache_ko_genes(ko_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "    \n",
    "def get_functional_roles2(annotation_api, ko_id):\n",
    "    m = annotation_api.neo4j_graph.nodes.match(\"KeggOrthology\", key=ko_id)\n",
    "    if len(m) < 1:\n",
    "        pass\n",
    "    ko_node = m.first()\n",
    "    gene_nodes = set()\n",
    "    #KeggOrthology -[:has_gene]-> Node\n",
    "    for rel in annotation_api.neo4j_graph.match((ko_node, ), r_type=\"has_gene\", ):\n",
    "        gene_nodes.add(rel.end_node)\n",
    "\n",
    "    functions = {}\n",
    "    functions_data = {}\n",
    "\n",
    "    found = 0\n",
    "    for gene_node in gene_nodes:\n",
    "        print(id(gene_node))\n",
    "        gene_functions = set()\n",
    "        #gene_node -[:has_annotatation]-> gene_function\n",
    "        for rel in annotation_api.neo4j_graph.match((gene_node, ), r_type=\"has_annotation\", ):\n",
    "            function = Neo4jAnnotationFunction(rel.end_node)\n",
    "            gene_functions.add(function)\n",
    "            \n",
    "        print(id(gene_node), len(gene_functions))\n",
    "        if not len(gene_functions) == 0:\n",
    "            found+= 1\n",
    "            for gene_function in gene_functions:\n",
    "                if not gene_function.id in functions:\n",
    "                    functions[gene_function.id] = set()\n",
    "                functions[gene_function.id].add(gene_node['key'])\n",
    "                functions_data[gene_function.id] = {'value' : gene_function.value}\n",
    "                for sub_function in gene_function.sub_functions:\n",
    "                    if not sub_function.id in functions:\n",
    "                        functions[sub_function.id] = set()\n",
    "                    functions[sub_function.id].add(gene_node['key'])\n",
    "                    functions_data[sub_function.id] = {'value' : sub_function.value}\n",
    "                #print(gene_node['id'], gene_function['key'])\n",
    "\n",
    "    return functions, functions_data, {'total' : len(gene_nodes), 'has_function' : found}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [],
   "source": [
    "class KBaseGene:\n",
    "    \n",
    "    def __init__(self, data):\n",
    "        self.data = data\n",
    "        \n",
    "    @property\n",
    "    def gene_id(self):\n",
    "        if 'id' in self.data:\n",
    "            return self.data['id']\n",
    "        return '?'\n",
    "    \n",
    "    def __str__(self):\n",
    "        return self.gene_id\n",
    "    \n",
    "    def __repr__(self):\n",
    "        return \"{}:KBaseGene({})\".format(self.data['database_id'], self.gene_id)\n",
    "\n",
    "\n",
    "\n",
    "#get_ko_genes(annotation_api, 'K05349')\n",
    "a = get_ko_genes(annotation_api, 'K00200')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{97788, 97789}"
      ]
     },
     "execution_count": 44,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'time' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-20-8b0b0c6c945e>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mstart_time\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m \u001b[0ma\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mget_functional_roles2\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mannotation_api\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'K05349'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mend_time\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mtime\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtime\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mend_time\u001b[0m \u001b[0;34m-\u001b[0m \u001b[0mstart_time\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'time' is not defined"
     ]
    }
   ],
   "source": [
    "start_time = time.time()\n",
    "a = get_functional_roles2(annotation_api, 'K05349')\n",
    "end_time = time.time()\n",
    "print(end_time - start_time)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": [
    "import time\n",
    "start_time = time.time()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_reaction_annotation_data(annotation_api, rxn_id, example_genes = 10):\n",
    "    kos = annotation_api.get_ko_by_seed_id(rxn_id)\n",
    "    function_data = {}\n",
    "    for ko in kos:\n",
    "        start_time = time.time()\n",
    "        functions, functions_data, metadata = annotation_api.get_functional_roles2(ko)\n",
    "        print(ko, metadata)\n",
    "        for f in functions:\n",
    "            function = functions_data[f]['value']\n",
    "            if not function in function_data:\n",
    "                function_data[function] = {\n",
    "                    'id' : f,\n",
    "                    'hits' : []\n",
    "                }\n",
    "            function_data[function]['hits'].append({\n",
    "                'score' : len(functions[f]),\n",
    "                'source' : ['KEGG', ko]\n",
    "            })\n",
    "        end_time = time.time()\n",
    "        print('#ko', ko, end_time - start_time)\n",
    "\n",
    "    #MISSING ADD TEMPLATE DATA\n",
    "    for template_set in annotation_api.page_nodes('TemplateSet', 0, 10):\n",
    "        start_time = time.time()\n",
    "        template_id = template_set['n']['key']\n",
    "        res = annotation_api.get_template_reaction_data(template_id, rxn_id)\n",
    "        for r in res:\n",
    "            #print('pair', r['n2']['key'])\n",
    "            node = annotation_api.neo4j_graph.nodes[r['n2'].id]\n",
    "            for rel in annotation_api.neo4j_graph.match((node, ), r_type=\"has_function\", ):\n",
    "                if rel.end_node.has_label('FunctionComplex'):\n",
    "                    for rel_cpx in annotation_api.neo4j_graph.match((rel.end_node, ), r_type=\"has_function\", ):\n",
    "                        function = Neo4jAnnotationFunction(rel_cpx.end_node)\n",
    "                        if not function.value in function_data:\n",
    "                            function_data[function.value] = {\n",
    "                                'id' : function.id,\n",
    "                                'hits' : []\n",
    "                            }\n",
    "                        function_data[function.value]['hits'].append({\n",
    "                            'score' : 0,\n",
    "                            'source' : ['template', template_id]\n",
    "                        })\n",
    "                else:\n",
    "                    function = Neo4jAnnotationFunction(rel.end_node)\n",
    "                    if not function.value in function_data:\n",
    "                        function_data[function.value] = {\n",
    "                            'id' : function.id,\n",
    "                            'hits' : []\n",
    "                        }\n",
    "                    function_data[function.value]['hits'].append({\n",
    "                        'score' : 0,\n",
    "                        'source' : ['template', template_id]\n",
    "                    })\n",
    "        end_time = time.time()\n",
    "        print('#template_set', template_set, end_time - start_time)\n",
    "    #MISSING ADD SBML DATA\n",
    "\n",
    "    for f in function_data:\n",
    "        node = annotation_api.matcher.match(\"Function\", key=f).first()\n",
    "        annotation = Neo4jAnnotationFunction(node)\n",
    "\n",
    "        subsystem_tags = {}\n",
    "        for ss in annotation.function_group:\n",
    "            subsystem_tags[ss] = {}\n",
    "\n",
    "        fcount = annotation_api.get_function_count(annotation)\n",
    "\n",
    "        source_tags = {}\n",
    "        for s in fcount:\n",
    "            source_tags[s] = [len(fcount[s]['genomes']), \n",
    "                              len(fcount[s]['genes']), \n",
    "                              list(fcount[s]['genes'])[:example_genes]]\n",
    "\n",
    "        function_data[f]['sources'] = source_tags\n",
    "        function_data[f]['subsystems'] = subsystem_tags\n",
    "\n",
    "    return function_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "K01188 {'total': 0, 'has_function': 0}\n",
      "#ko K01188 0.03233504295349121\n",
      "K05349 {'total': 111, 'has_function': 111}\n",
      "#ko K05349 1.2055320739746094\n",
      "K05350 {'total': 35, 'has_function': 35}\n",
      "#ko K05350 0.6186277866363525\n",
      "#template_set <Record n=<Node id=5561658 labels={'TemplateSet'} properties={'updated_at': 1570538930130, 'created_at': 1570538930130, 'key': 'GramNegModelTemplate'}>> 0.023299694061279297\n",
      "#template_set <Record n=<Node id=5570337 labels={'TemplateSet'} properties={'updated_at': 1570539971948, 'created_at': 1570539971948, 'key': 'GramNegModelTemplateV2'}>> 0.023211002349853516\n",
      "2.4638850688934326\n"
     ]
    }
   ],
   "source": [
    "start_time = time.time()\n",
    "res = get_reaction_annotation_data(annotation_api, 'rxn00020')\n",
    "end_time = time.time()\n",
    "print(end_time - start_time)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "node = annotation_api.matcher.match(\"Function\", key='hypothetical protein').first()\n",
    "#annotation = Neo4jAnnotationFunction(node)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "node = annotation_api.matcher.match(\"Function\", key='loool').first()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(_315778:Function {key: 'hypothetical protein'})"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 66,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation.function_group"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "fcount = annotation_api.get_function_count(annotation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 70,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "dict_keys(['genes', 'genomes'])"
      ]
     },
     "execution_count": 70,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fcount['BETA_RAST'].keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'hypothetical protein'"
      ]
     },
     "execution_count": 59,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation.value"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "315778"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation.id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "BETA_RAST <class 'dict'>\n",
      "KBASE_RAST <class 'dict'>\n"
     ]
    }
   ],
   "source": [
    "for k in fcount:\n",
    "    print(k, type(fcount[k]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 81,
   "metadata": {},
   "outputs": [],
   "source": [
    "fcount_wrap = {}\n",
    "for annotation_source in fcount:\n",
    "    for sub_key in fcount[annotation_source]:\n",
    "        fcount_wrap[annotation_source + '#' + sub_key] = '#'.join(fcount[annotation_source][sub_key])\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "False"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cache.expire('!!!', 11)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "b'aaa#aaab'"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cache.get('genome_set:fungi')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "cache.set('genome_set:fungi', 'aaa#aaab')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [],
   "source": [
    "n = annotation_api.get_function()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "#cache.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:cache lookup\n",
      "DEBUG:__main__:cache return\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1747551\n",
      "1217551\n",
      "CPU times: user 625 ms, sys: 151 ms, total: 776 ms\n",
      "Wall time: 1.71 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "function_str = 'hypothetical protein'\n",
    "gene_function = annotation_api.get_function(function_str)\n",
    "fcount = annotation_api.get_function_count(gene_function)\n",
    "print(len(fcount['BETA_RAST']['genes']))\n",
    "print(len(fcount['KBASE_RAST']['genes']))\n",
    "#1747551 1196033"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:cache lookup\n",
      "DEBUG:__main__:cache return\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 2.48 ms, sys: 2.34 ms, total: 4.82 ms\n",
      "Wall time: 4.39 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "genome_set = annotation_api.get_genome_set('ModelSEED2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1747551\n",
      "1196033\n",
      "CPU times: user 1.35 s, sys: 15.2 ms, total: 1.36 s\n",
      "Wall time: 1.36 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "fcount = annotation_api.filter_genome_set(fcount, genome_set)\n",
    "print(len(fcount['BETA_RAST']['genes']))\n",
    "print(len(fcount['KBASE_RAST']['genes']))\n",
    "#1747551 1196033"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:cache lookup\n",
      "DEBUG:__main__:cache return\n",
      "DEBUG:__main__:cache lookup\n",
      "DEBUG:__main__:cache return\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1747551\n",
      "1217551\n",
      "12400\n",
      "28496\n",
      "12\n",
      "33\n",
      "CPU times: user 2.34 s, sys: 247 ms, total: 2.58 s\n",
      "Wall time: 3.48 s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "function_str = 'hypothetical protein'\n",
    "gene_function = annotation_api.get_function(function_str)\n",
    "fcount = annotation_api.get_function_count(gene_function)\n",
    "print(len(fcount['BETA_RAST']['genes']))\n",
    "print(len(fcount['KBASE_RAST']['genes']))\n",
    "genome_set = annotation_api.get_genome_set('TNSeq')\n",
    "fcount = annotation_api.filter_genome_set(fcount, genome_set)\n",
    "print(len(fcount['BETA_RAST']['genes']))\n",
    "print(len(fcount['KBASE_RAST']['genes']))\n",
    "print(len(fcount['BETA_RAST']['genomes']))\n",
    "print(len(fcount['KBASE_RAST']['genomes']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "DEBUG:__main__:cache lookup\n",
      "DEBUG:__main__:cache return\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1675\n",
      "CPU times: user 2.57 ms, sys: 1.56 ms, total: 4.12 ms\n",
      "Wall time: 3.36 ms\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "genome_set = annotation_api.get_genome_set('ModelSEED2')\n",
    "print(len(genome_set))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
