{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [],
   "source": [
    "import logging\n",
    "import os\n",
    "import pymongo\n",
    "import cobrakbase\n",
    "import time\n",
    "%run ../../scripts/bios_utils.py\n",
    "%run ../../annotation-server/annotation_api.py\n",
    "\n",
    "#mclient = pymongo.MongoClient('mongodb://192.168.1.10:27017/')\n",
    "mclient = pymongo.MongoClient('mongodb://0.0.0.0:27017/')\n",
    "annotation_api = AnnotationApi(mclient)\n",
    "#aclient = pymongo.MongoClient(\"mongodb+srv://server:dx75S3HBXX6h2U3D@bios-dk66o.gcp.mongodb.net/test?retryWrites=true&w=majority\")\n",
    "#annotation_api_atlas = AnnotationApi(aclient)\n",
    "kbase = cobrakbase.KBaseAPI(token='YAFOCRSMRNDXZ7KMW7GCK5AC3SBNTEFD')\n",
    "logger = logging.getLogger(__name__)\n",
    "database = mclient['KBase']\n",
    "mdb_kbase_ko_to_genes = database['ko_gene_mapping']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'_id': 'rxn00148', 'kegg_ids': ['R00200'], 'kegg_kos': ['K00873', 'K12406'], 'name': 'ATP:pyruvate 2-O-phosphotransferase'}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'K00873', 'K12406'}"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.get_ko_by_seed_id('rxn00148')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<__main__.AnnotationApi at 0x7fd2b010a400>"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "a, b, c = get_functional_roles2('K00873', annotation_api)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'_id': 'rxn00148', 'kegg_ids': ['R00200'], 'kegg_kos': ['K00873', 'K12406'], 'name': 'ATP:pyruvate 2-O-phosphotransferase'}\n",
      "K00873 {'total': 150, 'has_function': 150}\n",
      "K12406 {}\n"
     ]
    }
   ],
   "source": [
    "def get_reaction_annotation_data(rxn_id):\n",
    "    kos = annotation_api.get_ko_by_seed_id(rxn_id)\n",
    "    function_data = {}\n",
    "    for ko in kos:\n",
    "        #print(ko)\n",
    "        functions, functions_data, metadata = get_functional_roles2(ko, annotation_api)\n",
    "        print(ko, metadata)\n",
    "        for f in functions:\n",
    "            function = functions_data[f]['value']\n",
    "            if not function in function_data:\n",
    "                function_data[function] = {\n",
    "                    'id' : f,\n",
    "                    'hits' : []\n",
    "                }\n",
    "            function_data[function]['hits'].append({\n",
    "                'score' : len(functions[f]),\n",
    "                'source' : ['KEGG', ko]\n",
    "            })\n",
    "    for template_doc in annotation_api.collection_templates.find():\n",
    "        if rxn_id in template_doc['reactions']:\n",
    "            template_id = template_doc['_id']\n",
    "            template_reaction_doc = template_doc['reactions'][rxn_id]\n",
    "            for and_rule in template_reaction_doc['functions']:\n",
    "                for function_id in and_rule:\n",
    "                    function_doc = annotation_api.collection_functions.find_one({'_id' : function_id})\n",
    "                    function = function_doc['value']\n",
    "                    if not function in function_data:\n",
    "                        function_data[function] = {\n",
    "                            'id' : function_id,\n",
    "                            'hits' : []\n",
    "                        }\n",
    "                    function_data[function]['hits'].append({\n",
    "                        'score' : 0,\n",
    "                        'source' : ['template', template_id]\n",
    "                    })\n",
    "                    #print(template_id, function_doc['value'])\n",
    "        \n",
    "    for function in function_data:\n",
    "        function_id = function_data[function]['id']\n",
    "        #print(function_id)\n",
    "        function_doc = annotation_api.collection_functions.find_one({'_id' : function_id})\n",
    "        function_metadata = annotation_api.collection_functions_data.find_one({'_id' : function_id})\n",
    "        \n",
    "        subsystem_tags = {}\n",
    "        if 'subsystem' in function_doc:\n",
    "            for subsystem_tag in function_doc['subsystem']:\n",
    "                subsystem_tags[subsystem_tag] = [function_doc['subsystem'][subsystem_tag]]\n",
    "                if 'subsystem_class' in function_doc and subsystem_tag in function_doc['subsystem_class']:\n",
    "                    subsystem_tags[subsystem_tag].append(function_doc['subsystem_class'][subsystem_tag])\n",
    "        #print(subsystem_tags)\n",
    "        \n",
    "        \n",
    "        source_tags = {}\n",
    "        if not function_metadata == None:\n",
    "            sources = function_metadata['sources']\n",
    "            for s in sources:\n",
    "                source_tags[s] = [len(sources[s]['genomes']), len(sources[s]['genes']), sources[s]['genes'][:10]]\n",
    "            #print(source_tags)\n",
    "        function_data[function]['sources'] = source_tags\n",
    "        function_data[function]['subsystems'] = subsystem_tags\n",
    "    return function_data\n",
    "function_data = get_reaction_annotation_data('rxn00148')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': '2d48fd430d9958479c298e77125615444a61025cc405a2197e3c34a2c11b3234',\n",
       " 'hits': [{'score': 126, 'source': ['KEGG', 'K00873']},\n",
       "  {'score': 0, 'source': ['template', 'GramNegModelTemplate']},\n",
       "  {'score': 0, 'source': ['template', 'GramNegModelTemplateV2']}],\n",
       " 'sources': {'KBASE_RAST': [214,\n",
       "   253,\n",
       "   ['MCRO_RS02685@GCF_000025845.1',\n",
       "    'TBIS_RS07295@GCF_000092645.1',\n",
       "    'HDEN_RS13045@GCF_000143145.1',\n",
       "    'CCA_RS03390@GCF_000007605.1',\n",
       "    'NOCA_RS16785@GCF_000015265.1',\n",
       "    'WANG_RS03905@GCF_000214785.1',\n",
       "    'VSVS12_RS13960@GCF_001685465.1',\n",
       "    'VSVS12_RS18120@GCF_001685465.1',\n",
       "    'WGL_RS00790@GCF_000008885.1',\n",
       "    'W908_RS07935@GCF_001281385.1']]},\n",
       " 'subsystems': {'patric': ['Glycolysis and Gluconeogenesis, including Archaeal enzymes',\n",
       "   'Central Metabolism']}}"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "function_data['Pyruvate kinase (EC 2.7.1.40)']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "write_json(ko_to_genes, '../../data/ko_kegg_genes.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "ko_to_genes = {}\n",
    "for c in mdb_kbase_ko_to_genes.find():\n",
    "    ko_to_genes[c['_id']] = c['genes']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../../annotation-server/annotation_api_neo4j.py\n",
    "from neo4j import GraphDatabase\n",
    "driver = GraphDatabase.driver(\"bolt://127.0.0.1:7687\", auth=(\"neo4j\", \"\"))\n",
    "neo_annotation_api = AnnotationApiNeo4j(driver)\n",
    "#neo_annotation_api.init_constraints()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 99,
   "metadata": {},
   "outputs": [],
   "source": [
    "result = neo_annotation_api.get_node('RefSeqGenome', 'GCF_000027305.1')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Record n=<Node id=558 labels={'RefSeqGenome'} properties={'genetic_code': 11, 'notes': '', 'release': '1', 'created_at': 1563767048645, 'scientific_name': 'Haemophilus influenzae Rd KW20', 'source': 'refseq', 'taxonomy': 'cellular organisms; Bacteria; Proteobacteria; Gammaproteobacteria; Pasteurellales; Pasteurellaceae; Haemophilus; Haemophilus influenzae', 'type': 'Reference', 'external_source_origination_date': '14-APR-2016', 'count_features': 1765, 'workspace_id': 'jplfaria:narrative_1524466549180', 'dna_size': 1830138, 'original_source_file_name': 'GCF_000027305.1_ASM2730v1_genomic.gbff', 'reference_annotation': 1, 'updated_at': 1563767048645, 'num_contigs': 1, 'assembly_ref': '19217/45552/1', 'domain': 'Bacteria', 'source_id': 'NC_000907', 'id': 'GCF_000027305.1', 'taxon_ref': '12570/465385/1', 'genbank_handle_ref': 'KBH_233040', 'gc_content': 0.38147888301319355, 'key': 'GCF_000027305.1'}>>"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 357,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Node id=20 labels={'Fruit'} properties={'created_at': 1563766639414, 'key_str': 'ya!', 'updated_at': 1563766642893, 'price': 0.5, 'key': 'Banana', 'int': 3535}>"
      ]
     },
     "execution_count": 357,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "neo_annotation_api.add_node('Fruit', 'Banana', props)\n",
    "#1563766585626 1563766585626\n",
    "#1563766354122 1563766354122"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 353,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'{key_str:\"ya!\", int:3535, price:0.5}'"
      ]
     },
     "execution_count": 353,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "props = {\n",
    "    'key_str' : 'ya!',\n",
    "    'int' : 3535,\n",
    "    'price' : 0.50\n",
    "}\n",
    "to_str_dict(props)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "with driver.session() as session:\n",
    "    session.read_transaction(lambda tx : tx.run('MATCH (n) RETURN LIMIT 5'))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 361,
   "metadata": {},
   "outputs": [],
   "source": [
    "for genome_doc in annotation_api.collection_genomes.find():\n",
    "    key = genome_doc['_id']\n",
    "    del genome_doc['_id']\n",
    "    neo_annotation_api.add_node('RefSeqGenome', key, genome_doc)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 396,
   "metadata": {},
   "outputs": [],
   "source": [
    "alles_genes = list(map(lambda x : x, annotation_api.collection_genes.find()))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 180,
   "metadata": {},
   "outputs": [],
   "source": [
    "import copy\n",
    "def prep_feature(feature):    \n",
    "    feature_prep = copy.deepcopy(feature)\n",
    "    \n",
    "    if 'quality' in feature_prep:\n",
    "        del feature_prep['quality']\n",
    "    if 'ontology_terms' in feature_prep:\n",
    "        del feature_prep['ontology_terms']\n",
    "    \n",
    "    for k in feature_prep:\n",
    "        if k == 'location':\n",
    "            loc = []\n",
    "            for location in feature_prep['location']:\n",
    "                location = map(str, location)\n",
    "                loc.append(':'.join(location))\n",
    "            feature_prep['location'] = loc\n",
    "        if type(feature_prep[k]) == list:\n",
    "            feature_prep[k] = ';'.join(feature_prep[k])\n",
    "    return feature_prep\n",
    "#prep_feature(feature)\n",
    "#neo_annotation_api.to_str_dict(prep_feature(feature))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'alles_genes' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-181-85a63fd01867>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mgene_doc\u001b[0m \u001b[0;32min\u001b[0m \u001b[0malles_genes\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m     \u001b[0mkey\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgene_doc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0mgene_id\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenome_id\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mkey\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'@'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0;32mdel\u001b[0m \u001b[0mgene_doc\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'_id'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      5\u001b[0m     \u001b[0;32mfor\u001b[0m \u001b[0mk\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgene_doc\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'alles_genes' is not defined"
     ]
    }
   ],
   "source": [
    "for gene_doc in alles_genes:\n",
    "    key = gene_doc['_id']\n",
    "    gene_id, genome_id = key.split('@')\n",
    "    del gene_doc['_id']\n",
    "    for k in gene_doc:\n",
    "        if k == 'location':\n",
    "            loc = []\n",
    "            for location in gene_doc['location']:\n",
    "                location = map(str, location)\n",
    "                loc.append(':'.join(location))\n",
    "            gene_doc['location'] = loc\n",
    "        if type(gene_doc[k]) == list:\n",
    "            #print(k, gene_doc[k])\n",
    "            gene_doc[k] = ';'.join(gene_doc[k])\n",
    "            \n",
    "    genome_node = neo_annotation_api.get_genome(genome_id)\n",
    "    gene_node = neo_annotation_api.add_node('KBaseGene', key, gene_doc)\n",
    "    #print(genome_node)\n",
    "    #print(gene_node)\n",
    "    neo_annotation_api.link_nodes(genome_node, gene_node, 'has_gene')\n",
    "    #print(gene_id, genome_id)\n",
    "    #print(gene_doc)\n",
    "    #print(neo_annotation_api.to_str_dict(gene_doc))\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_annotation2f(f):\n",
    "    annotation = set([None])\n",
    "    if 'functions' in f:\n",
    "        annotation = set(f['functions'])\n",
    "    elif 'function' in f:\n",
    "        annotation = set([f['function']])\n",
    "    return annotation\n",
    "#get_annotation2(genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 163,
   "metadata": {},
   "outputs": [],
   "source": [
    "function_source_node = neo_annotation_api.get_node('FunctionSource', 'KBASE_RASTs')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 164,
   "metadata": {},
   "outputs": [],
   "source": [
    "function_source_node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Node id=74534 labels={'FunctionSource'} properties={'updated_at': 1563777594588, 'created_at': 1563774149176, 'key': 'KBASE_RAST'}>"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 179,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_genome_annotation(function_source, genome_id, genome):\n",
    "    \n",
    "    for feature in genome.features:\n",
    "        gene_genome_id = \"{}@{}\".format(feature['id'], genome_id)\n",
    "        annotation = get_annotation2f(feature)\n",
    "    pass\n",
    "\n",
    "def add_genome(genome_id, genome, function_source = None):\n",
    "    function_source_node = None\n",
    "    if not function_source == None:\n",
    "        function_source_node = neo_annotation_api.add_node('FunctionSource', function_source)\n",
    "    \n",
    "    genome_node = neo_annotation_api.get_genome(genome_id)\n",
    "    if genome_node == None:\n",
    "        genome_node_props = convert_to_mongo_document(genome_id, genome)\n",
    "        \n",
    "    if genome_node == None:\n",
    "        logger.error('failed to add genome')\n",
    "        return\n",
    "    \n",
    "    for feature in genome.features:\n",
    "        gene_genome_id = \"{}@{}\".format(feature['id'], genome_id)\n",
    "        feature_prep = prep_feature(feature)\n",
    "        kbase_gene_node = neo_annotation_api.add_node('KBaseGene', gene_genome_id, feature_prep)\n",
    "        neo_annotation_api.link_nodes(genome_node, kbase_gene_node, 'has_gene')\n",
    "        \n",
    "    if not function_source == None:\n",
    "        add_genome_annotation(function_source, genome_id, genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [],
   "source": [
    "o = 'GCF_000005845.2.json'\n",
    "genome_data = read_json('../../../data/kbase_data/' + o)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_id': 'wut',\n",
       " 'assembly_ref': '19217/45552/1',\n",
       " 'dna_size': 1830138,\n",
       " 'domain': 'Bacteria',\n",
       " 'external_source_origination_date': '14-APR-2016',\n",
       " 'gc_content': 0.382,\n",
       " 'genbank_handle_ref': 'KBH_1243438',\n",
       " 'genetic_code': 11,\n",
       " 'id': 'GCF_000027305.1.RAST',\n",
       " 'md5': 'e059f4bce671b3e39e4d0cc8e9689f84',\n",
       " 'notes': '',\n",
       " 'num_contigs': 1,\n",
       " 'original_source_file_name': 'GCF_000027305.1_ASM2730v1_genomic.gbff',\n",
       " 'reference_annotation': 1,\n",
       " 'release': '1',\n",
       " 'scientific_name': 'Haemophilus influenzae Rd KW20',\n",
       " 'source': 'refseq',\n",
       " 'source_id': 'NC_000907',\n",
       " 'taxon_ref': '12570/465385/1',\n",
       " 'taxonomy': 'cellular organisms; Bacteria; Proteobacteria; Gammaproteobacteria; Pasteurellales; Pasteurellaceae; Haemophilus; Haemophilus influenzae',\n",
       " 'type': 'Reference'}"
      ]
     },
     "execution_count": 177,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def convert_to_mongo_document(id, genome):\n",
    "    genome_doc = {}\n",
    "    #genome_doc['workspace_id'] = 'jplfaria:narrative_1524466549180'\n",
    "    genome_doc['_id'] = id\n",
    "    for k in genome.data:\n",
    "        if not type(genome.data[k]) == list:\n",
    "            genome_doc[k] = genome.data[k]\n",
    "    return genome_doc\n",
    "\n",
    "\n",
    "genome_data = read_json('../../../data/kbase_data/' + o)\n",
    "genome = cobrakbase.core.kbasegenomesgenome.KBaseGenome(genome_data)\n",
    "convert_to_mongo_document('wut', genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "        \n",
    "for o in os.listdir('../../../data/kbase_data'):\n",
    "    function_source_node = neo_annotation_api.add_node('FunctionSource', 'KBASE_RAST')\n",
    "    genome_data = read_json('../../../data/kbase_data/' + o)\n",
    "    genome = cobrakbase.core.kbasegenomesgenome.KBaseGenome(genome_data)\n",
    "    #annotation = get_annotation2(genome)\n",
    "    genome_id = genome.id[:-5]\n",
    "    genome_node = neo_annotation_api.get_genome(genome_id)\n",
    "    \n",
    "    for feature in genome.features:\n",
    "        gene_genome_id = \"{}@{}\".format(feature['id'], genome_id)\n",
    "        feature_prep = prep_feature(feature)\n",
    "        kbase_gene_node = neo_annotation_api.add_node('KBaseGene', gene_genome_id, feature_prep)\n",
    "        neo_annotation_api.link_nodes(genome_node, kbase_gene_node, 'has_gene')\n",
    "        #print(gene_genome_id, feature)\n",
    "        annotation = get_annotation2f(feature)\n",
    "        neo_annotation_api.add_annotation(gene_genome_id, annotation, 'KBASE_RAST')\n",
    "        break\n",
    "        #break\n",
    "    #for gene_id in annotation:\n",
    "    #    gene_genome_id = \"{}@{}\"\n",
    "        #add_annotation(gene_genome_id, annotation[gene_id], 'KBASE_RAST')\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'aliases': [['gene_synonym', 'ECK0001; JW4367'], ['gene', 'thrL'], ['locus_tag', 'b0001'], ['protein_id', 'NP_414542.1']], 'cdss': ['b0001_CDS_1'], 'db_xrefs': [['GeneID', '944742'], ['EcoGene', 'EG11277']], 'dna_sequence': 'ATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGA', 'dna_sequence_length': 66, 'functions': ['Thr operon leader peptide'], 'id': 'b0001', 'location': [['NC_000913.3', 190, '+', 66]], 'md5': '13fe4258b37e32edda386faa50ec5fdf', 'ontology_terms': {'GO': {'GO:0009088': [0]}}, 'protein_md5': '13fe4258b37e32edda386faa50ec5fdf', 'protein_translation': 'MKRISTTITTTITITTGNGAG', 'protein_translation_length': 21, 'quality': {'hit_count': 13, 'weighted_hit_count': 22.729}}\n"
     ]
    }
   ],
   "source": [
    "\n",
    "genome3 = kbase.get_object('test_nokmers1', 'jplfaria:narrative_1561579862828')\n",
    "\n",
    "\n",
    "#jplfaria:narrative_1561579862828"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome2_script = kbase.get_object('GCF_000005845.2.RAST2', 'filipeliu:narrative_1562086644968')\n",
    "genome2_raw = kbase.get_object('GCF_000005845.2', 'filipeliu:narrative_1562086644968')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome1_rast = kbase.get_object('GCF_000005845.2.RAST', 'jplfaria:narrative_1524466549180')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 301,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "4319"
      ]
     },
     "execution_count": 301,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(list(genome3['features']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_zip = read_json('../../../data/kbase_data/GCF_000005845.2.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0 0 4498\n",
      "4498\n"
     ]
    }
   ],
   "source": [
    "compare(get_annotation2(cobrakbase.core.KBaseGenome(genome1_rast)), get_annotation2(cobrakbase.core.KBaseGenome(genome_zip)))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'UPF0174 protein YaaW'}"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "get_annotation2(cobrakbase.core.KBaseGenome(genome_zip))['b0011']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "new_beta_annotation = get_annotation(genome3)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'aliases': ['b0006', 'UniProtKB/Swiss-Prot:P0A8I3', 'NP_414547.1', 'GeneID:944749', 'GI:16128000', 'yaaA', 'JW0005', 'EcoGene:EG10011', 'ECK0006', 'ASAP:ABE-0000018'], 'cdss': ['NP_414547.1'], 'dna_sequence': 'ATGCTGATTCTTATTTCACCTGCGAAAACGCTTGATTACCAAAGCCCGTTGACCACCACGCGCTATACGCTGCCGGAGCTGTTAGACAATTCCCAGCAGTTGATCCATGAGGCGCGGAAACTGACGCCTCCGCAGATTAGCACGCTGATGCGCATCAGCGACAAACTGGCGGGTATCAACGCCGCTCGCTTTCATGACTGGCAGCCAGATTTCACGCCGGCGAATGCCCGCCAGGCGATTCTGGCGTTTAAAGGTGATGTCTACACCGGCTTGCAGGCCGAAACCTTCAGCGAAGACGATTTCGATTTTGCCCAACAGCATTTGCGAATGCTTTCCGGCTTGTATGGCGTACTCCGCCCGCTCGATTTAATGCAGCCTTATCGTCTGGAAATGGGGATCCGTCTTGAGAATGCCCGAGGGAAAGATCTGTATCAATTCTGGGGAGATATCATCACCAACAAGCTGAACGAGGCGCTCGCAGCACAAGGCGATAATGTGGTGATTAATCTGGCGTCAGATGAGTATTTTAAATCGGTGAAGCCGAAGAAATTGAATGCCGAGATTATCAAGCCGGTGTTCCTCGATGAGAAGAACGGCAAATTTAAGATCATCAGCTTCTACGCTAAGAAAGCACGCGGTCTGATGAGTCGTTTCATTATTGAAAATCGGCTGACCAAACCAGAGCAACTGACTGGTTTTAATAGCGAAGGTTACTTCTTTGATGAAGATTCCTCCAGCAATGGCGAACTGGTGTTTAAACGCTACGAGCAGCGTTAA', 'dna_sequence_length': 777, 'function': 'peroxide resistance protein, lowers intracellular iron', 'functions': ['UPF0246 protein YaaA'], 'id': 'b0006', 'location': [['NC_000913', 6459, '-', 777]], 'md5': '526cc558d00c8b6a111dc18ae36caa22', 'mrnas': ['mRNA_6'], 'protein_translation': 'MLILISPAKTLDYQSPLTTTRYTLPELLDNSQQLIHEARKLTPPQISTLMRISDKLAGINAARFHDWQPDFTPANARQAILAFKGDVYTGLQAETFSEDDFDFAQQHLRMLSGLYGVLRPLDLMQPYRLEMGIRLENARGKDLYQFWGDIITNKLNEALAAQGDNVVINLASDEYFKSVKPKKLNAEIIKPVFLDEKNGKFKIISFYAKKARGLMSRFIIENRLTKPEQLTGFNSEGYFFDEDSSSNGELVFKRYEQR', 'protein_translation_length': 258, 'type': 'gene'}\n"
     ]
    }
   ],
   "source": [
    "for f in genome2_script['features']:\n",
    "    gene_id = f['id']\n",
    "    if gene_id =='b0006':\n",
    "        print(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'aliases': ['b0006', 'UniProtKB/Swiss-Prot:P0A8I3', 'NP_414547.1', 'GeneID:944749', 'GI:16128000', 'yaaA', 'JW0005', 'EcoGene:EG10011', 'ECK0006', 'ASAP:ABE-0000018'], 'cdss': ['NP_414547.1'], 'dna_sequence': 'ATGCTGATTCTTATTTCACCTGCGAAAACGCTTGATTACCAAAGCCCGTTGACCACCACGCGCTATACGCTGCCGGAGCTGTTAGACAATTCCCAGCAGTTGATCCATGAGGCGCGGAAACTGACGCCTCCGCAGATTAGCACGCTGATGCGCATCAGCGACAAACTGGCGGGTATCAACGCCGCTCGCTTTCATGACTGGCAGCCAGATTTCACGCCGGCGAATGCCCGCCAGGCGATTCTGGCGTTTAAAGGTGATGTCTACACCGGCTTGCAGGCCGAAACCTTCAGCGAAGACGATTTCGATTTTGCCCAACAGCATTTGCGAATGCTTTCCGGCTTGTATGGCGTACTCCGCCCGCTCGATTTAATGCAGCCTTATCGTCTGGAAATGGGGATCCGTCTTGAGAATGCCCGAGGGAAAGATCTGTATCAATTCTGGGGAGATATCATCACCAACAAGCTGAACGAGGCGCTCGCAGCACAAGGCGATAATGTGGTGATTAATCTGGCGTCAGATGAGTATTTTAAATCGGTGAAGCCGAAGAAATTGAATGCCGAGATTATCAAGCCGGTGTTCCTCGATGAGAAGAACGGCAAATTTAAGATCATCAGCTTCTACGCTAAGAAAGCACGCGGTCTGATGAGTCGTTTCATTATTGAAAATCGGCTGACCAAACCAGAGCAACTGACTGGTTTTAATAGCGAAGGTTACTTCTTTGATGAAGATTCCTCCAGCAATGGCGAACTGGTGTTTAAACGCTACGAGCAGCGTTAA', 'dna_sequence_length': 777, 'function': 'UPF0246 protein YaaA', 'id': 'b0006', 'location': [['NC_000913', 6459, '-', 777]], 'md5': '3918eea8540ab18805c356c3e3096954', 'mrnas': ['mRNA_6'], 'ontology_terms': {'SSO': {'SSO:000008666': {'evidence': [{'method': 'annotate_genomes', 'method_version': '1', 'timestamp': '2018-04-23T06:59:27'}], 'id': 'SSO:000008666', 'ontology_ref': '17159/4/1', 'term_lineage': [], 'term_name': 'UPF0246 protein YaaA'}}}, 'protein_translation': 'MLILISPAKTLDYQSPLTTTRYTLPELLDNSQQLIHEARKLTPPQISTLMRISDKLAGINAARFHDWQPDFTPANARQAILAFKGDVYTGLQAETFSEDDFDFAQQHLRMLSGLYGVLRPLDLMQPYRLEMGIRLENARGKDLYQFWGDIITNKLNEALAAQGDNVVINLASDEYFKSVKPKKLNAEIIKPVFLDEKNGKFKIISFYAKKARGLMSRFIIENRLTKPEQLTGFNSEGYFFDEDSSSNGELVFKRYEQR', 'protein_translation_length': 258, 'quality': {'hit_count': 246, 'weighted_hit_count': 665.650208}, 'type': 'gene'}\n"
     ]
    }
   ],
   "source": [
    "for f in genome1_rast['features']:\n",
    "    gene_id = f['id']\n",
    "    if gene_id =='b0006':\n",
    "        print(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "assembly_ref <class 'str'>\n",
      "cdss <class 'list'>\n",
      "contig_ids <class 'list'>\n",
      "contig_lengths <class 'list'>\n",
      "dna_size <class 'int'>\n",
      "domain <class 'str'>\n",
      "external_source_origination_date <class 'str'>\n",
      "feature_counts <class 'dict'>\n",
      "features <class 'list'>\n",
      "gc_content <class 'float'>\n",
      "genbank_handle_ref <class 'str'>\n",
      "genetic_code <class 'int'>\n",
      "genome_tiers <class 'list'>\n",
      "id <class 'str'>\n",
      "md5 <class 'str'>\n",
      "molecule_type <class 'str'>\n",
      "mrnas <class 'list'>\n",
      "non_coding_features <class 'list'>\n",
      "notes <class 'str'>\n",
      "num_contigs <class 'int'>\n",
      "ontologies_present <class 'dict'>\n",
      "ontology_events <class 'list'>\n",
      "original_source_file_name <class 'str'>\n",
      "publications <class 'list'>\n",
      "scientific_name <class 'str'>\n",
      "source <class 'str'>\n",
      "source_id <class 'str'>\n",
      "taxon_ref <class 'str'>\n",
      "taxonomy <class 'str'>\n",
      "warnings <class 'list'>\n"
     ]
    }
   ],
   "source": [
    "for k in genome3.keys():\n",
    "    print(k, type(genome3[k]))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 154,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'aliases': [['gene', 'ralA'], ['locus_tag', 'b4714'], ['gene_synonym', 'ECK4571; rcbA']], 'db_xrefs': [['EcoGene', 'EG31192'], ['GeneID', '20160642']], 'dna_sequence': 'GAGGACTGAAGTTTCTCGCAATTAAAATTTATCAGTTTTACTTTCTGCTCTCTGGAAACGCCTGCTTCTTTTTTACCTGAGAGCATTTTTTCGCATTCTGATTTCGTTAGTTTAGATTTTGAATATCTTGTCCAGTTAGTAGGAGTGCCACCTTCCTTTTCAATAGTGGCGGTAATTTT', 'dna_sequence_length': 179, 'id': 'b4714', 'location': [['NC_000913.3', 1413556, '+', 179]], 'md5': '31ad409db33a242a4d1b81b78e157423', 'note': 'Antisense sRNA antitoxin for the RalR toxin', 'type': 'gene'}\n"
     ]
    }
   ],
   "source": [
    "for f in genome3['non_coding_features']:\n",
    "    gene_id = f['id']\n",
    "    if gene_id == 'b4714':\n",
    "        print(f)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "metadata": {},
   "outputs": [],
   "source": [
    "write_json(genome3, '../../../data/genomes/genome3.json', True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 168,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome2_raw_annotation = get_annotation(genome2_raw)\n",
    "genome2_script_annotation = get_annotation(genome2_script)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['a', 'assd', 'asd@as', 'asd/sa /']"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#>>> a='Beautiful, is; better*than\\nugly'\n",
    "import re\n",
    "re.split(' / | @ ', \"a / assd @ asd@as / asd/sa /\")\n",
    "#['Beautiful', 'is', 'better', 'than', 'ugly']\n",
    "#\"a / assd @ asd\".split(' / ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'aa', 'function'}"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def split_annotation(annotation):\n",
    "    split = set()\n",
    "    for s in annotation:\n",
    "        if not s == None:\n",
    "            for comp in re.split(' / | @ |; ', s):\n",
    "                split.add(comp)\n",
    "        else:\n",
    "            split.add(s)\n",
    "    return split\n",
    "\n",
    "def split_function(function):\n",
    "    split = set()\n",
    "    if not function == None:\n",
    "        for comp in re.split(' / | @ |; ', function):\n",
    "            split.add(comp)\n",
    "    else:\n",
    "        split.add(s)\n",
    "    return split\n",
    "\n",
    "split_function(\"function @ aa\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "def compare(annotation1, annotation2):\n",
    "    genes_both = set(annotation1) & set(annotation2)\n",
    "    unique1 = set(annotation1) - set(annotation2)\n",
    "    unique2 = set(annotation2) - set(annotation1)\n",
    "    print(len(unique1), len(unique2), len(genes_both))\n",
    "    \n",
    "    equal_annotation = set()\n",
    "    \n",
    "    for gene_id in genes_both:\n",
    "        functions1 = split_annotation(annotation1[gene_id])\n",
    "        functions2 = split_annotation(annotation2[gene_id])\n",
    "        if functions1 == functions2:\n",
    "            equal_annotation.add(gene_id)\n",
    "        else:\n",
    "            print(gene_id, functions1, functions2)\n",
    "            #break\n",
    "    print(len(equal_annotation))\n",
    "    \n",
    "def compare_multi(annotations, labels = []):\n",
    "    genes_both = set(annotations[0])\n",
    "    for annotation_set in annotations:\n",
    "        genes_both &= set(annotation_set)\n",
    "    #unique1 = set(annotation1) - set(annotation2)\n",
    "    #unique2 = set(annotation2) - set(annotation1)\n",
    "    print(len(genes_both))\n",
    "    \n",
    "    equal_annotation = set()\n",
    "    \n",
    "    function = {}\n",
    "    for gene_id in genes_both:\n",
    "        \n",
    "        function[gene_id] = {}\n",
    "        for i in range(len(annotations)):\n",
    "            function[gene_id][labels[i]] = split_annotation(annotations[i][gene_id])\n",
    "\n",
    "    return function\n",
    "#compare(new_beta_annotation, genome2_raw_annotation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 218,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "4140\n"
     ]
    }
   ],
   "source": [
    "function_comp = compare_multi([new_beta_annotation, genome_annotation['KBASE_RAST'], genome2_script_annotation], ['beta', 'old', 'script'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2080 5 2055\n"
     ]
    }
   ],
   "source": [
    "match = set()\n",
    "match_partial = set()\n",
    "match_none = set()\n",
    "for gene_id in function_comp:\n",
    "    \n",
    "    common = set(function_comp[gene_id][set_id])\n",
    "    for set_id in function_comp[gene_id]:\n",
    "        common &= function_comp[gene_id][set_id]\n",
    "    \n",
    "    is_match = True\n",
    "    for set_id in function_comp[gene_id]:\n",
    "        is_match &= function_comp[gene_id][set_id] == common\n",
    "    \n",
    "    if is_match:\n",
    "        match.add(gene_id)\n",
    "    elif len(common) > 1:\n",
    "        match_partial.add(gene_id)\n",
    "    else:\n",
    "        match_none.add(gene_id)\n",
    "    #if not is_match:\n",
    "    #    print(gene_id, common, function_comp[gene_id])\n",
    "    #    break|\n",
    "print(len(match), len(match_partial), len(match_none))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 302,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('beta', 'old') 39\n",
      "('beta', 'old') b4682 {'beta': {'hypothetical protein'}, 'old': {'hypothetical protein'}, 'script': {'FIG00638198: hypothetical protein'}}\n",
      "('beta', 'old') b4669 {'beta': {'hypothetical protein'}, 'old': {'hypothetical protein'}, 'script': {'uncharacterized protein'}}\n",
      "('beta', 'old') b0280 {'beta': {'hypothetical protein'}, 'old': {'hypothetical protein'}, 'script': {'uncharacterized protein', 'CP4-6 prophage'}}\n",
      "\n",
      "('beta', 'script') 1656\n",
      "('beta', 'script') b3141 {'beta': {'Galactosamine-6-phosphate isomerase [deaminating]'}, 'old': {'Galactosamine-6-phosphate isomerase (galactosamine-6-phosphate deaminase) (EC 5.3.1.-)'}, 'script': {'Galactosamine-6-phosphate isomerase [deaminating]'}}\n",
      "('beta', 'script') b2552 {'beta': {'Flavohemoglobin', 'Nitric oxide dioxygenase (EC 1.14.12.17)'}, 'old': {'Flavohemoprotein (Hemoglobin-like protein) (Flavohemoglobin) (Nitric oxide dioxygenase) (EC 1.14.12.17)'}, 'script': {'Flavohemoglobin', 'Nitric oxide dioxygenase (EC 1.14.12.17)'}}\n",
      "('beta', 'script') b3961 {'beta': {'Hydrogen peroxide-inducible genes activator => OxyR'}, 'old': {'Hydrogen peroxide-inducible genes activator'}, 'script': {'Hydrogen peroxide-inducible genes activator => OxyR'}}\n",
      "\n",
      "('old', 'script') 217\n",
      "('old', 'script') b1247 {'beta': {'Oligopeptide ABC transporter, ATP-binding protein OppF (TC 3.A.1.5.1)'}, 'old': {'Oligopeptide transport ATP-binding protein OppF (TC 3.A.1.5.1)'}, 'script': {'Oligopeptide transport ATP-binding protein OppF (TC 3.A.1.5.1)'}}\n",
      "('old', 'script') b0428 {'beta': {'Heme O synthase, protoheme IX farnesyltransferase, COX10-CtaB'}, 'old': {'Heme O synthase, protoheme IX farnesyltransferase (EC 2.5.1.-) COX10-CtaB'}, 'script': {'Heme O synthase, protoheme IX farnesyltransferase (EC 2.5.1.-) COX10-CtaB'}}\n",
      "('old', 'script') b1069 {'beta': {'Peptidoglycan lipid II flippase MurJ'}, 'old': {'Proposed peptidoglycan lipid II flippase MurJ'}, 'script': {'Proposed peptidoglycan lipid II flippase MurJ'}}\n",
      "\n",
      "no_match 143\n",
      "no_match b1562 {'beta': {'hypothetical protein'}, 'old': {'IS, phage, Tn', 'Transposon-related functions'}, 'script': {'RelF inactive antibacterial toxin protein'}}\n",
      "no_match b1705 {'beta': {'Hemin uptake protein HemP/HmuP'}, 'old': {'Hemin uptake protein HemP'}, 'script': {'Uncharacterized protein YdiE'}}\n",
      "no_match b1328 {'beta': {'Transcriptional regulator PgrR, LysR family'}, 'old': {'LysR family transcriptional regulator YcjZ'}, 'script': {'LysR family transcriptional regulator PgrR'}}\n",
      "\n"
     ]
    }
   ],
   "source": [
    "pair_comp = [('beta', 'old'), ('beta', 'script'), ('old', 'script')]\n",
    "pair_comp_match_count = {}\n",
    "for p in pair_comp:\n",
    "    pair_comp_match_count[p] = set()\n",
    "pair_comp_match_count[\"no_match\"] = set()\n",
    "for gene_id in match_none:\n",
    "    #print(gene_id)\n",
    "    valid_pairs = set()\n",
    "    for p in pair_comp:\n",
    "        p1 = function_comp[gene_id][p[0]]\n",
    "        p2 = function_comp[gene_id][p[1]]\n",
    "        if p1 == p2:\n",
    "            pair_comp_match_count[p].add(gene_id)\n",
    "            valid_pairs.add(p)\n",
    "    if len(valid_pairs) == 0:\n",
    "        pair_comp_match_count[\"no_match\"].add(gene_id)\n",
    "        #print(p, p1, p2)\n",
    "    #for set_id in function_comp[gene_id]:\n",
    "    #    print(set_id, function_comp[gene_id][set_id])\n",
    "    #break\n",
    "for p in pair_comp_match_count:\n",
    "    print(p, len(pair_comp_match_count[p]))\n",
    "    \n",
    "    for gene_id in list(pair_comp_match_count[p])[:3]:\n",
    "        print(p, gene_id, function_comp[gene_id])\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 269,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'aa'"
      ]
     },
     "execution_count": 269,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a = \"Aa\"\n",
    "a.lower()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "for annotation in [new_beta_annotation, genome_annotation['KBASE_RAST'], genome2_script_annotation]:\n",
    "    for gene_id in \n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 299,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1470 22 2311\n",
      "2071 6 2015\n",
      "2063 6 2027\n"
     ]
    }
   ],
   "source": [
    "def ss_match(new_beta_annotation):\n",
    "    match_ss = set()\n",
    "    match_ss_sn = set()\n",
    "    match_fail = set()\n",
    "    for gene_id in new_beta_annotation:\n",
    "        for f in new_beta_annotation[gene_id]:\n",
    "            if not f == None:\n",
    "                for f_split in split_function(f):\n",
    "                    #print(f_split)\n",
    "                    if f_split in ss_roles:\n",
    "                        match_ss.add(f_split)\n",
    "                    elif normalize_role(f_split) in ss_roles_sn:\n",
    "                        match_ss_sn.add(f_split)\n",
    "                    else:\n",
    "                        match_fail.add(f_split)\n",
    "\n",
    "    return match_ss, match_ss_sn, match_fail\n",
    "\n",
    "match_ss, match_ss_sn, match_fail = ss_match(genome_annotation['KBASE_RAST'])\n",
    "print(len(match_ss), len(match_ss_sn), len(match_fail))\n",
    "match_ss, match_ss_sn, match_fail = ss_match(genome2_script_annotation)\n",
    "print(len(match_ss), len(match_ss_sn), len(match_fail))\n",
    "match_ss, match_ss_sn, match_fail = ss_match(new_beta_annotation)\n",
    "print(len(match_ss), len(match_ss_sn), len(match_fail))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "9104 9091\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "ss_roles = set()\n",
    "ss_roles_sn = set()\n",
    "for r in substem_ref['response']['docs']:\n",
    "    #print(r)\n",
    "    #break\n",
    "    if 'role_id' in r:\n",
    "        for role_id in r['role_id']:\n",
    "            ss_roles.add(role_id)\n",
    "            ss_roles_sn.add(normalize_role(role_id))\n",
    "        #break\n",
    "print(len(ss_roles), len(ss_roles_sn))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{' ', '@', 'a'}"
      ]
     },
     "execution_count": 290,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "split_annotation(\"a @ a\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 306,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['ironsulfurclusterregulatorsufr',\n",
       " 'atpdependentdnahelicaseuvrdpcraclostridialparalog2',\n",
       " 'dihydroanticapsin7dehydrogenaseec111385bacc',\n",
       " 'n5methyltetrahydromethanopterincoenzymemmethyltransferasesubunitcec21186']"
      ]
     },
     "execution_count": 306,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "list(ss_roles_sn)[:4]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 307,
   "metadata": {},
   "outputs": [],
   "source": [
    "#compare(new_beta_annotation, genome2_raw_annotation)\n",
    "#compare(new_beta_annotation, genome2_script_annotation)\n",
    "#compare(new_beta_annotation, genome_annotation['KBASE_RAST'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'genome' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-117-de3638a8b3a2>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mtype\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenome\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m: name 'genome' is not defined"
     ]
    }
   ],
   "source": [
    "type(genome)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 88,
   "metadata": {},
   "outputs": [],
   "source": [
    "ecoli_genome = annotation_api.collection_genomes_genes.find_one({'_id' : 'GCF_000005845.2'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 173,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_annotation = {}\n",
    "no_annotation = set()\n",
    "for gene_id in ecoli_genome['features']:\n",
    "    gene_genome_id = \"{}@{}\".format(gene_id, 'GCF_000005845.2')\n",
    "    f = annotation_api.collection_gene_functions.find_one({'_id' : gene_genome_id})\n",
    "    if not f == None:\n",
    "        for s in f['function']:\n",
    "            if not s in genome_annotation:\n",
    "                genome_annotation[s] = {}\n",
    "            f_data = annotation_api.collection_functions.find_one({'_id' : f['function'][s]})\n",
    "            if not f_data == None:\n",
    "                genome_annotation[s][gene_id] = set([f_data['value']])\n",
    "            else:\n",
    "                genome_annotation[s][gene_id] = set([None])\n",
    "    else:\n",
    "        no_annotation.add(gene_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome_rast2=mclient['Models']['genomes'].find_one({'_id' : 'GCF_000005845.2'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b4296 {'RAST': ObjectId('5bad031edb6c6111032cca2d'), 'RAST2': ObjectId('5bad031edb6c6111032cca2d'), 'ko': ['K03299']}\n"
     ]
    }
   ],
   "source": [
    "for f in genome_rast2['features']:\n",
    "    print(f, genome_rast2['features'][f])\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 100,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "358"
      ]
     },
     "execution_count": 100,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(no_annotation)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 83,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 12 µs, sys: 1e+03 ns, total: 13 µs\n",
      "Wall time: 16 µs\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'688787d8ff144c502c7f5cffaafe2cc588d86079f9de88304c26b0cb99ce91c6'"
      ]
     },
     "execution_count": 83,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "annotation_api.hash_value('asd')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [],
   "source": [
    "#@staticmethod\n",
    "def add_value_node(tx, label, key):\n",
    "    result = tx.run(\"MERGE (n:\" + label + \" {key: $key}) \"\n",
    "                    \"ON CREATE SET n.created_at = timestamp(), n.updated_at = timestamp()\"\n",
    "                    \"ON MATCH SET n.updated_at = timestamp() \"\n",
    "                    \"RETURN n\", key=key)\n",
    "    return result.single()[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_relationship(tx, node1, node2, relationship):\n",
    "    result = tx.run(\"MATCH (n1) WHERE id(n1)=$node1_id \"\n",
    "                    \"WITH n1 \"\n",
    "                    \"MATCH (n2) WHERE id(n2)=$node2_id \"\n",
    "                    \"CREATE UNIQUE (n1)-[r:\" + relationship + \"]->(n2) \", \n",
    "                    node1_id=node1.id, node2_id=node2.id)\n",
    "    return result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "MATCH (n1) WHERE id(n1)=1\n",
    "WITH n1\n",
    "MATCH (n2) WHERE id(n2)=16\n",
    "MERGE (n1)-[r:has_function]->(n2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "MATCH (a:FunctionGroup {value: $value1}),MERGE (a)-[r:LOVES]->(b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [],
   "source": [
    "substem_ref = read_json('../../../data/tnseq/patric_subsystems/substem_ref.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 30.7 s, sys: 2.97 s, total: 33.7 s\n",
      "Wall time: 3min 39s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "for doc in substem_ref['response']['docs']:\n",
    "    value = doc['subsystem_name']\n",
    "    \n",
    "    ss_node = None\n",
    "    \n",
    "    with driver.session() as session:\n",
    "        ss_node = session.write_transaction(add_value_node, 'FunctionGroup', value)\n",
    "    \n",
    "    if 'role_name' in doc:\n",
    "        for role in doc['role_name']:\n",
    "            role = role.strip()\n",
    "            if len(role) > 0:\n",
    "                with driver.session() as session:\n",
    "                    function_node = session.write_transaction(add_value_node, 'Function', role)\n",
    "                    session.write_transaction(add_relationship, \n",
    "                                    ss_node, function_node, 'has_function')\n",
    "    #break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "session.read_transaction"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#add genome:kegg -> ncbitaxonomy\n",
    "#add genome -[has_gene]-> gene\n",
    "#           -[has_taxonomy]-> ncbitaxonomy\n",
    "#add gene -[has_function]-> function\n",
    "#add rxn:seed -> rxn:kegg\n",
    "#add rxn:kegg -> ortholog:kegg\n",
    "#add ortholog:kegg -> gene\n",
    "#template -[function, compartment]-> rxn:seed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "#genome -[has_gene]-> gene <-[has_gene]- ortholog:kegg -[has_reaction]-> rxn:kegg <-[has_reference]- rxn:seed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "metadata": {},
   "outputs": [],
   "source": [
    "for i in out.records():\n",
    "    print(i)\n",
    "    pass\n",
    "out."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {},
   "outputs": [],
   "source": [
    "with driver.session() as session:\n",
    "    out = session.write_transaction(add_relationship, \n",
    "                                    ss_node, function_node, 'has_function')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "for doc in substem_ref['response']['docs']:\n",
    "    collection_functions_group.update_one('')\n",
    "    break\n",
    "    update_data = {\n",
    "        'subsystem.patric' : doc['subsystem_name']\n",
    "    }\n",
    "    if 'subclass' in doc:\n",
    "        update_data['subsystem_class.patric'] = doc['subclass']\n",
    "    #print(update_data)\n",
    "    if 'role_name' in doc:\n",
    "        for role in doc['role_name']:\n",
    "            function_doc = annotation_api.get_function(role)\n",
    "            if not function_doc == None:\n",
    "                annotation_api.collection_functions.update_one(\n",
    "                    {'_id' : function_doc['_id']}, \n",
    "                    {'$set' : update_data})\n",
    "    else:\n",
    "        pass\n",
    "        #print(function_doc)\n",
    "    #print(doc)\n",
    "    #break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'subsystem_id': 'Conjugative_transposon,_Bacteroidales',\n",
       " 'description': '\\n',\n",
       " 'role_id': ['Putative mobilization protein BF0133',\n",
       "  'Putative conjugative transposon mobilization protein BF0132',\n",
       "  'hypothetical protein clusted with conjugative transposons, BF0131',\n",
       "  'Conjugative transposon protein TraA',\n",
       "  'Conjugative transposon protein TraB',\n",
       "  'Conjugative transposon protein TraC',\n",
       "  'Conjugative transposon protein TraD',\n",
       "  'Conjugative transposon protein TraE',\n",
       "  'Conjugative transposon protein TraF',\n",
       "  'Conjugative transposon protein TraG',\n",
       "  'Conjugative transposon protein TraH',\n",
       "  'Conjugative transposon protein TraI',\n",
       "  'Conjugative transposon protein TraJ',\n",
       "  'Conjugative transposon protein TraK',\n",
       "  'Conjugative transposon protein TraL',\n",
       "  'Conjugative transposon protein TraM',\n",
       "  'Conjugative transposon protein TraN',\n",
       "  'Conjugative transposon protein TraO',\n",
       "  'DNA primase, TraP-type',\n",
       "  'Conjugative transposon protein TraQ'],\n",
       " 'subsystem_name': 'Conjugative transposon, Bacteroidales',\n",
       " 'role_name': ['Putative mobilization protein BF0133',\n",
       "  'Putative conjugative transposon mobilization protein BF0132',\n",
       "  'hypothetical protein clusted with conjugative transposons, BF0131',\n",
       "  'Conjugative transposon protein TraA',\n",
       "  'Conjugative transposon protein TraB',\n",
       "  'Conjugative transposon protein TraC',\n",
       "  'Conjugative transposon protein TraD',\n",
       "  'Conjugative transposon protein TraE',\n",
       "  'Conjugative transposon protein TraF',\n",
       "  'Conjugative transposon protein TraG',\n",
       "  'Conjugative transposon protein TraH',\n",
       "  'Conjugative transposon protein TraI',\n",
       "  'Conjugative transposon protein TraJ',\n",
       "  'Conjugative transposon protein TraK',\n",
       "  'Conjugative transposon protein TraL',\n",
       "  'Conjugative transposon protein TraM',\n",
       "  'Conjugative transposon protein TraN',\n",
       "  'Conjugative transposon protein TraO',\n",
       "  'DNA primase, TraP-type',\n",
       "  'Conjugative transposon protein TraQ'],\n",
       " 'subclass': 'Transposable elements',\n",
       " 'notes': [\"Defined by CTnDOT of Bacteroides fragilis YCH46:\\n\\nBonheyo, G., Graham, D., Shoemaker, N. B., and Salyers, A. A..  2001.  Transfer region of a Bacteroides conjugative transposon, CTnDOT.  Plasmid 45, 41�51.\\n\\nTraA is similar to chromosome partitioning protein ParA\\n\\nTraC is an imperfect direct repeat of TraB and is not present in most CTn's\\n\\nTraG is similar to IncF conjugation system protein TraC = IncP system protein TrbE = VirB4\\n\\nTraP is similar to DNA primase, and is not present in most of the CTn's\\n\\nEffect of gene disruptions on transposition as multiple of wild-type frequency (Bonheyo, G. T., Hund, B. D., Shoemaker, N. B, and Salyers, A. A.  2001.  Transfer region of a Bacteroides conjugative transposon contains regulatory as well as structural genes.  Plasmid 46, 202�209):\\n\\nG  <0.001\\nH    0.001-0.01\\nI   <0.001\\nJ   <0.001\\nM  <0.001\\nN   0.001-0.01\\nO   10-100\\nP   100-1000\\nQ  100-1000\\n\"],\n",
       " 'superclass': 'Miscellaneous',\n",
       " 'class': 'Prophages, Transposable elements, Plasmids',\n",
       " 'id': '000a3c14-9b53-40a4-b1d0-b2f1942bf75c',\n",
       " 'date_inserted': '2018-03-21T18:40:25.135Z',\n",
       " 'date_modified': '2018-03-21T18:40:25.135Z',\n",
       " '_version_': 1631710205186146304}"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "doc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['annotation', 'admin', 'local']"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "aclient.list_database_names()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "#mongodb+srv://<username>:<password>@bios-dk66o.gcp.mongodb.net/test?retryWrites=true&w=majority"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "catalog = {}\n",
    "for f in os.listdir('/Users/fliu/workspace/jupyter/data/www/viewer/data/ModelSEED/'):\n",
    "    map_id = f[:-5]\n",
    "    catalog[map_id] = {\n",
    "        \"src\" : \"data/ModelSEED/\" + f\n",
    "    }\n",
    "write_json(catalog, '../../../data/www/viewer/data/catalog.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "annotation_api_atlas.add_function_to_template_rxn('6559f6aeb335466fea1c9b0a35762b9decb6b3a0daa88dffa06cb33b902154c9', 'rxn00952', 'fliu', 'template_v3', True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'_id': 'rxn00952@template_v3',\n",
       " 'functions': ['6559f6aeb335466fea1c9b0a35762b9decb6b3a0daa88dffa06cb33b902154c9'],\n",
       " 'log': [{'timestamp': 1562840194, 'user_id': 'fliu', 'action': True}]}"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api_atlas.collection_templates_reactions.find_one({'_id' : 'rxn00952@template_v3'})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "mdb_kbase_ko_to_genes = mclient['KBase']['ko_gene_mapping']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'_id': 'rxn00002', 'name': 'urea-1-carboxylate amidohydrolase', 'kegg_ids': ['R00005'], 'kegg_kos': ['K14541', 'K01457']}\n",
      "K01457 {'total': 16, 'has_function': 16}\n",
      "K14541 {}\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'amidase': {'id': '1f809c1d70582d0ed3286a66c4c62ddaf530596b10a6a8fd1d2b425196f77e09',\n",
       "  'hits': [{'score': 3, 'source': ['KEGG', 'K01457']}],\n",
       "  'sources': {'RAST2': [126,\n",
       "    373,\n",
       "    ['MCRO_RS02635@GCF_000025845.1',\n",
       "     'TBIS_RS04645@GCF_000092645.1',\n",
       "     'TBIS_RS07155@GCF_000092645.1',\n",
       "     'TBIS_RS08725@GCF_000092645.1',\n",
       "     'TBIS_RS12565@GCF_000092645.1',\n",
       "     'TBIS_RS15950@GCF_000092645.1',\n",
       "     'HDEN_RS01390@GCF_000143145.1',\n",
       "     'NOCA_RS04770@GCF_000015265.1',\n",
       "     'NOCA_RS07625@GCF_000015265.1',\n",
       "     'NOCA_RS07990@GCF_000015265.1']],\n",
       "   'KBASE_RAST': [4,\n",
       "    4,\n",
       "    ['HDEN_RS01390@GCF_000143145.1',\n",
       "     'NOCA_RS07625@GCF_000015265.1',\n",
       "     'AMK58_RS24340@GCF_001315015.1',\n",
       "     'TPAU_RS03900@GCF_000092225.1']]},\n",
       "  'subsystems': {}},\n",
       " 'Allophanate hydrolase (EC 3.5.1.54)': {'id': 'ce22b1ae7c021defb5375f493afc05d450697a8313d0fbed2a6be4262432458c',\n",
       "  'hits': [{'score': 9, 'source': ['KEGG', 'K01457']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplate']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplateV2']}],\n",
       "  'sources': {'KBASE_RAST': [25,\n",
       "    28,\n",
       "    ['SAVERM_RS34405@GCF_000009765.2',\n",
       "     'CR3_RS21045@GCF_001281465.1',\n",
       "     'bll1044@GCF_000011365.1',\n",
       "     'bll5806@GCF_000011365.1',\n",
       "     'VARPA_RS01360@GCF_000184745.1',\n",
       "     'CACI_RS18535@GCF_000024025.1',\n",
       "     'F506_RS18665@GCF_001267925.1',\n",
       "     'PC1_RS10755@GCF_000023605.1',\n",
       "     'Psyr_1170@GCF_000012245.1',\n",
       "     'Psyr_3978@GCF_000012245.1']]},\n",
       "  'subsystems': {'patric': ['Urea carboxylase and Allophanate hydrolase cluster',\n",
       "    'Arginine; urea cycle, creatine, polyamines']}},\n",
       " 'allophanate hydrolase': {'id': '07ab0ccce738bef67942870b69bf6a237fda6f73b84ef07635b9077dc96b0127',\n",
       "  'hits': [{'score': 10, 'source': ['KEGG', 'K01457']}],\n",
       "  'sources': {'RAST2': [63,\n",
       "    114,\n",
       "    ['TBIS_RS12890@GCF_000092645.1',\n",
       "     'TBIS_RS12895@GCF_000092645.1',\n",
       "     'NOCA_RS01855@GCF_000015265.1',\n",
       "     'NOCA_RS16560@GCF_000015265.1',\n",
       "     'NOCA_RS16565@GCF_000015265.1',\n",
       "     'VSVS12_RS18575@GCF_001685465.1',\n",
       "     'VSVS12_RS18580@GCF_001685465.1',\n",
       "     'UL81_RS02580@GCF_000980815.1',\n",
       "     'SAVERM_RS34405@GCF_000009765.2',\n",
       "     'SAVERM_RS35635@GCF_000009765.2']]},\n",
       "  'subsystems': {}},\n",
       " 'Allophanate hydrolase 2 subunit 2 (EC 3.5.1.54)': {'id': '3c713e36b3bc772a6a695c47919c1cb74805cf3048d88f0e474504499cd74522',\n",
       "  'hits': [{'score': 3, 'source': ['KEGG', 'K01457']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplate']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplateV2']}],\n",
       "  'sources': {'KBASE_RAST': [78,\n",
       "    89,\n",
       "    ['TBIS_RS12895@GCF_000092645.1',\n",
       "     'NOCA_RS01855@GCF_000015265.1',\n",
       "     'NOCA_RS16560@GCF_000015265.1',\n",
       "     'VSVS12_RS18575@GCF_001685465.1',\n",
       "     'SAVERM_RS35635@GCF_000009765.2',\n",
       "     'ACFER_RS09570@GCF_000025305.1',\n",
       "     'SWOO_RS00045@GCF_000019525.1',\n",
       "     'SWOO_RS08940@GCF_000019525.1',\n",
       "     'CR3_RS00250@GCF_001281465.1',\n",
       "     'F7308_RS00355@GCF_000219045.1']]},\n",
       "  'subsystems': {'patric': ['Urea carboxylase and Allophanate hydrolase cluster',\n",
       "    'Arginine; urea cycle, creatine, polyamines']}},\n",
       " 'Putative allophanate hydrolase subunit 2': {'id': 'aecf7eba159db47bfb4dc59f749de84b23ced189331ca8829b2278dd19f95d13',\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K01457']}],\n",
       "  'sources': {'RAST2': [1, 1, ['RSP_3449@GCF_000012905.2']]},\n",
       "  'subsystems': {}},\n",
       " 'allophanate hydrolase subunit 2': {'id': 'f9ec46ef1dfdfddddaa87b553433b170f60f485f1b2fa1ab0384ec3de2f19c59',\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K01457']}],\n",
       "  'sources': {'RAST2': [4,\n",
       "    4,\n",
       "    ['NCgl0960@GCF_000011325.1',\n",
       "     'TPAU_RS17590@GCF_000092225.1',\n",
       "     'Rru_A1505@GCF_000013085.1',\n",
       "     'MSMEG_0435@GCF_000015005.1']]},\n",
       "  'subsystems': {}},\n",
       " 'urea short-chain amide or branched-chain amino acid uptake ABC transporter periplasmic solute-binding protein': {'id': '29794baf163dd1900b948f2142bae071259b8729661f41b91e7701a68899dfa5',\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K01457']}],\n",
       "  'sources': {'RAST2': [1,\n",
       "    3,\n",
       "    ['Psyr_0499@GCF_000012245.1',\n",
       "     'Psyr_2720@GCF_000012245.1',\n",
       "     'Psyr_3977@GCF_000012245.1']]},\n",
       "  'subsystems': {}},\n",
       " 'Allophanate hydrolase 2 subunit 1 (EC 3.5.1.54)': {'id': '28ee8c93653456e2ce3a48ddd1663a92392264d9becd018bc077c1eca3cc5b4e',\n",
       "  'hits': [{'score': 0, 'source': ['template', 'GramNegModelTemplate']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplateV2']}],\n",
       "  'sources': {'KBASE_RAST': [75,\n",
       "    85,\n",
       "    ['TBIS_RS12890@GCF_000092645.1',\n",
       "     'NOCA_RS16565@GCF_000015265.1',\n",
       "     'VSVS12_RS18580@GCF_001685465.1',\n",
       "     'UL81_RS02580@GCF_000980815.1',\n",
       "     'SAVERM_RS35640@GCF_000009765.2',\n",
       "     'ACFER_RS09575@GCF_000025305.1',\n",
       "     'DC74_RS08300@GCF_000695235.1',\n",
       "     'SWOO_RS00050@GCF_000019525.1',\n",
       "     'SWOO_RS08935@GCF_000019525.1',\n",
       "     'CACET_RS06490@GCF_001042715.1']]},\n",
       "  'subsystems': {'patric': ['Urea carboxylase and Allophanate hydrolase cluster',\n",
       "    'Arginine; urea cycle, creatine, polyamines']}},\n",
       " 'N-benzyloxycarbonylglycine hydrolase (EC 3.5.1.58)': {'id': '24ccdf6695e0c53e5bf29bf986596546e275884d3ff533abc91055239cedcbd3',\n",
       "  'hits': [{'score': 0, 'source': ['template', 'GramNegModelTemplate']},\n",
       "   {'score': 0, 'source': ['template', 'GramNegModelTemplateV2']}],\n",
       "  'sources': {},\n",
       "  'subsystems': {}}}"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_functional_roles2(ko_id, annotation_api):\n",
    "    functions = {}\n",
    "    doc = mdb_kbase_ko_to_genes.find_one({'_id' : ko_id})\n",
    "    if doc == None:\n",
    "        return {}, {}, {}\n",
    "    gene_mapping = doc['genes']\n",
    "    \n",
    "    found = 0\n",
    "    for tup in gene_mapping:\n",
    "        gene_id = \"{}@{}\".format(tup[1], tup[0])\n",
    "        gene_functions = annotation_api.collection_gene_functions.find_one({'_id' : gene_id})\n",
    "        \n",
    "        if not gene_functions == None:\n",
    "            found += 1\n",
    "            for annotation_id in gene_functions['function']:\n",
    "                function_id = gene_functions['function'][annotation_id]\n",
    "                if not function_id in functions:\n",
    "                    functions[function_id] = set()\n",
    "                functions[function_id].add(gene_id)\n",
    "\n",
    "    functions_data = {}\n",
    "    \n",
    "    for function_id in functions:\n",
    "        function_doc = annotation_api.collection_functions.find_one({'_id' : function_id})\n",
    "        if not function_doc == None:\n",
    "            functions_data[function_id] = function_doc\n",
    "    \n",
    "    return functions, functions_data, {'total' : len(gene_mapping), 'has_function' : found}\n",
    "\n",
    "def get_reaction_annotation_data(rxn_id):\n",
    "    kos = annotation_api.get_ko_by_seed_id(rxn_id)\n",
    "    function_data = {}\n",
    "    for ko in kos:\n",
    "        #print(ko)\n",
    "        functions, functions_data, metadata = get_functional_roles2(ko, annotation_api)\n",
    "        print(ko, metadata)\n",
    "        for f in functions:\n",
    "            function = functions_data[f]['value']\n",
    "            if not function in function_data:\n",
    "                function_data[function] = {\n",
    "                    'id' : f,\n",
    "                    'hits' : []\n",
    "                }\n",
    "            function_data[function]['hits'].append({\n",
    "                'score' : len(functions[f]),\n",
    "                'source' : ['KEGG', ko]\n",
    "            })\n",
    "    for template_doc in annotation_api.collection_templates.find():\n",
    "        if rxn_id in template_doc['reactions']:\n",
    "            template_id = template_doc['_id']\n",
    "            template_reaction_doc = template_doc['reactions'][rxn_id]\n",
    "            for and_rule in template_reaction_doc['functions']:\n",
    "                for function_id in and_rule:\n",
    "                    function_doc = annotation_api.collection_functions.find_one({'_id' : function_id})\n",
    "                    \n",
    "                    function = function_doc['value']\n",
    "                    if not function in function_data:\n",
    "                        function_data[function] = {\n",
    "                            'id' : function_id,\n",
    "                            'hits' : []\n",
    "                        }\n",
    "                    function_data[function]['hits'].append({\n",
    "                        'score' : 0,\n",
    "                        'source' : ['template', template_id]\n",
    "                    })\n",
    "                    #print(template_id, function_doc['value'])\n",
    "        \n",
    "    for function in function_data:\n",
    "        function_id = function_data[function]['id']\n",
    "        #print(function_id)\n",
    "        function_doc = annotation_api.collection_functions.find_one({'_id' : function_id})\n",
    "        function_metadata = annotation_api.collection_functions_data.find_one({'_id' : function_id})\n",
    "        \n",
    "        subsystem_tags = {}\n",
    "        if 'subsystem' in function_doc:\n",
    "            for subsystem_tag in function_doc['subsystem']:\n",
    "                subsystem_tags[subsystem_tag] = [function_doc['subsystem'][subsystem_tag]]\n",
    "                if 'subsystem_class' in function_doc and subsystem_tag in function_doc['subsystem_class']:\n",
    "                    subsystem_tags[subsystem_tag].append(function_doc['subsystem_class'][subsystem_tag])\n",
    "        #print(subsystem_tags)\n",
    "        \n",
    "        \n",
    "        source_tags = {}\n",
    "        if not function_metadata == None:\n",
    "            sources = function_metadata['sources']\n",
    "            for s in sources:\n",
    "                source_tags[s] = [len(sources[s]['genomes']), len(sources[s]['genes']), sources[s]['genes'][:10]]\n",
    "            #print(source_tags)\n",
    "        function_data[function]['sources'] = source_tags\n",
    "        function_data[function]['subsystems'] = subsystem_tags\n",
    "    return function_data\n",
    "get_reaction_annotation_data('rxn00002')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'_id': 'rxn00002', 'name': 'urea-1-carboxylate amidohydrolase', 'kegg_ids': ['R00005'], 'kegg_kos': ['K14541', 'K01457']}\n",
      "K01457\n",
      "{'total': 16, 'has_function': 16}\n",
      "K14541\n",
      "{}\n",
      "1f809c1d70582d0ed3286a66c4c62ddaf530596b10a6a8fd1d2b425196f77e09\n",
      "ce22b1ae7c021defb5375f493afc05d450697a8313d0fbed2a6be4262432458c\n",
      "07ab0ccce738bef67942870b69bf6a237fda6f73b84ef07635b9077dc96b0127\n",
      "3c713e36b3bc772a6a695c47919c1cb74805cf3048d88f0e474504499cd74522\n",
      "aecf7eba159db47bfb4dc59f749de84b23ced189331ca8829b2278dd19f95d13\n",
      "f9ec46ef1dfdfddddaa87b553433b170f60f485f1b2fa1ab0384ec3de2f19c59\n",
      "29794baf163dd1900b948f2142bae071259b8729661f41b91e7701a68899dfa5\n",
      "28ee8c93653456e2ce3a48ddd1663a92392264d9becd018bc077c1eca3cc5b4e\n",
      "24ccdf6695e0c53e5bf29bf986596546e275884d3ff533abc91055239cedcbd3\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'amidase': {'id': '1f809c1d70582d0ed3286a66c4c62ddaf530596b10a6a8fd1d2b425196f77e09',\n",
       "  'hits': [{'score': 3, 'source': ['KEGG', 'K01457']}]},\n",
       " 'Allophanate hydrolase (EC 3.5.1.54)': {'id': 'ce22b1ae7c021defb5375f493afc05d450697a8313d0fbed2a6be4262432458c',\n",
       "  'hits': [{'score': 9, 'source': ['KEGG', 'K01457']},\n",
       "   {'score': 0, 'source': 'GramNegModelTemplateV2'},\n",
       "   {'score': 0, 'source': 'GramNegModelTemplate'}]},\n",
       " 'allophanate hydrolase': {'id': '07ab0ccce738bef67942870b69bf6a237fda6f73b84ef07635b9077dc96b0127',\n",
       "  'hits': [{'score': 10, 'source': ['KEGG', 'K01457']}]},\n",
       " 'Allophanate hydrolase 2 subunit 2 (EC 3.5.1.54)': {'id': '3c713e36b3bc772a6a695c47919c1cb74805cf3048d88f0e474504499cd74522',\n",
       "  'hits': [{'score': 3, 'source': ['KEGG', 'K01457']},\n",
       "   {'score': 0, 'source': 'GramNegModelTemplateV2'},\n",
       "   {'score': 0, 'source': 'GramNegModelTemplate'}]},\n",
       " 'Putative allophanate hydrolase subunit 2': {'id': 'aecf7eba159db47bfb4dc59f749de84b23ced189331ca8829b2278dd19f95d13',\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K01457']}]},\n",
       " 'allophanate hydrolase subunit 2': {'id': 'f9ec46ef1dfdfddddaa87b553433b170f60f485f1b2fa1ab0384ec3de2f19c59',\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K01457']}]},\n",
       " 'urea short-chain amide or branched-chain amino acid uptake ABC transporter periplasmic solute-binding protein': {'id': '29794baf163dd1900b948f2142bae071259b8729661f41b91e7701a68899dfa5',\n",
       "  'hits': [{'score': 1, 'source': ['KEGG', 'K01457']}]},\n",
       " 'Allophanate hydrolase 2 subunit 1 (EC 3.5.1.54)': {'id': '28ee8c93653456e2ce3a48ddd1663a92392264d9becd018bc077c1eca3cc5b4e',\n",
       "  'hits': [{'score': 0, 'source': 'GramNegModelTemplateV2'},\n",
       "   {'score': 0, 'source': 'GramNegModelTemplate'}]},\n",
       " 'N-benzyloxycarbonylglycine hydrolase (EC 3.5.1.58)': {'id': '24ccdf6695e0c53e5bf29bf986596546e275884d3ff533abc91055239cedcbd3',\n",
       "  'hits': [{'score': 0, 'source': 'GramNegModelTemplateV2'},\n",
       "   {'score': 0, 'source': 'GramNegModelTemplate'}]}}"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
