{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "import logging\n",
    "import os\n",
    "import pymongo\n",
    "import cobrakbase\n",
    "import time\n",
    "\n",
    "logger = logging.getLogger(__name__)\n",
    "%run ../../scripts/bios_utils.py\n",
    "%run ../../annotation-server/annotation_api_neo4j.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../../annotation-server/annotation_api_neo4j.py\n",
    "\n",
    "host, port, user, pwd = (\"0.0.0.0\", 7687, \"neo4j\", \"123585\")\n",
    "#driver = GraphDatabase.driver(\"bolt://192.168.1.19:7687\", auth=(\"neo4j\", \"123585\"))\n",
    "#driver = GraphDatabase.driver(\"bolt://0.0.0.0:7687\", auth=(\"neo4j\", \"123585\"))\n",
    "#driver = GraphDatabase.driver(\"bolt://sequoia.mcs.anl.gov:7687\", auth=(\"neo4j\", \"SLEEP-CESS!\"))\n",
    "#annotation_api = AnnotationApiNeo4j(driver)\n",
    "#neo_annotation_api.init_constraints()\n",
    "\n",
    "#annotation_api.neo4j_graph = Graph(\"http://neo4j:SLEEP-CESS!@sequoia.mcs.anl.gov:7474\")\n",
    "def init_annotation_api(host, port, user, pwd):\n",
    "    annotation_api = AnnotationApiNeo4j(user=user, pwd=pwd, port=port, host=host)\n",
    "    annotation_api.neo4j_graph = Graph(\"http://neo4j:123585@0.0.0.0:7474\")\n",
    "    annotation_api.matcher = NodeMatcher(annotation_api.neo4j_graph)\n",
    "    annotation_api.r_matcher = RelationshipMatcher(annotation_api.neo4j_graph)\n",
    "    annotation_api.init_constraints()\n",
    "    return annotation_api\n",
    "\n",
    "annotation_api = init_annotation_api(host, port, user, pwd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def list_genomes(annotation_api):\n",
    "    l = {}\n",
    "    res = annotation_api.page_nodes('RefSeqGenome', 0, 10)\n",
    "    page = 0\n",
    "    while not res == None:\n",
    "        for r in res:\n",
    "            l[r['n']['key']] = r['n']['scientific_name']\n",
    "        page += 1\n",
    "        res = annotation_api.page_nodes('RefSeqGenome', page, 10)\n",
    "    return l\n",
    "loaded_genomes = list_genomes(annotation_api)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "load: ../../../ModelSEEDDatabase/Biochemistry/reactions.tsv\n",
      "load: ../../../ModelSEEDDatabase/Biochemistry/compounds.tsv\n",
      "load: ../../../ModelSEEDDatabase/Biochemistry/Structures/Unique_ModelSEED_Structures.txt\n",
      "load: ../../../ModelSEEDDatabase/Biochemistry/Aliases/Unique_ModelSEED_Compound_Aliases.txt\n",
      "load: ../../../ModelSEEDDatabase/Biochemistry/Aliases/Unique_ModelSEED_Reaction_Aliases.txt\n",
      "load: ../../../ModelSEEDDatabase/Biochemistry/Aliases/Unique_ModelSEED_Reaction_ECs.txt\n"
     ]
    }
   ],
   "source": [
    "modelseed_local = cobrakbase.modelseed.from_local('../../../ModelSEEDDatabase')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "substem_ref = read_json('../../../data/tnseq/patric_subsystems/substem_ref.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 2min 18s, sys: 11.3 s, total: 2min 29s\n",
      "Wall time: 14min 15s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "for doc in substem_ref['response']['docs']:\n",
    "    value = doc['subsystem_name']\n",
    "    \n",
    "    subclass = None if not 'subclass' in doc else doc['subclass']\n",
    "    \n",
    "    ss_node = annotation_api.add_node('FunctionGroup', value)\n",
    "    \n",
    "    if 'role_name' in doc:\n",
    "        for role in doc['role_name']:\n",
    "            role = role.strip()\n",
    "            if len(role) > 0:\n",
    "                function_nodes = annotation_api.add_annotation([role], 'PATRIC')\n",
    "                for function_node in function_nodes:\n",
    "                    annotation_api.link_nodes(ss_node, function_node, 'has_function')\n",
    "                    annotation_api.link_nodes(function_node, ss_node, 'has_function_group')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Node id=57 labels={'RefSeqGenome'} properties={'genetic_code': 11, 'notes': '', 'release': '1', 'created_at': 1563767041245, 'scientific_name': 'Rickettsia conorii str. Malish 7', 'source': 'refseq', 'taxonomy': 'cellular organisms; Bacteria; Proteobacteria; Alphaproteobacteria; Rickettsiales; Rickettsiaceae; Rickettsieae; Rickettsia; spotted fever group; Rickettsia conorii', 'type': 'Representative', 'external_source_origination_date': '16-NOV-2016', 'count_features': 1532, 'workspace_id': 'jplfaria:narrative_1524466549180', 'dna_size': 1268755, 'original_source_file_name': 'GCF_000007025.1_ASM702v1_genomic.gbff', 'reference_annotation': 0, 'updated_at': 1563767041245, 'num_contigs': 1, 'assembly_ref': '19217/46804/1', 'domain': 'Bacteria', 'source_id': 'NC_003103', 'id': 'GCF_000007025.1', 'taxon_ref': '12570/666753/1', 'genbank_handle_ref': 'KBH_235629', 'gc_content': 0.3243880812292365, 'key': 'GCF_000007025.1'}>\n"
     ]
    }
   ],
   "source": [
    "node = annotation_api.get_node('RefSeqGenome', 'GCF_000007025.1')\n",
    "print(node)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "for doc in substem_ref['response']['docs']:\n",
    "    value = doc['subsystem_name']\n",
    "    subclass = None if not 'subclass' in doc else doc['subclass']\n",
    "    #print(value, '#######',subclass)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'{name:\"a \\\\\"banana\\\\\"\", key:\"banana\"}'"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "props_str = to_str_dict(props)\n",
    "props_str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "props = {\n",
    "    'desc' : 'what is this?',\n",
    "    'bzzzz' : 'yes'\n",
    "}\n",
    "annotation_api.update_node('Fruit', 'Fruztz', props)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': 'rxn10881',\n",
       " 'name': 'streptomycin 3\"-adenylyltransferase',\n",
       " 'equation': '(1) cpd00002[0] + (1) cpd00067[0] + (1) cpd00328[0] <= (1) cpd00012[0] + (1) cpd16061[0]',\n",
       " 'definition': '(1) ATP[0] + (1) H+[0] + (1) Streptomycin[0] <= (1) PPi[0] + (1) 3\"-Adenylylstreptomycin[0]',\n",
       " 'key': 'rxn10881'}"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "props"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load ModelSEED Reactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 31s, sys: 7.62 s, total: 1min 39s\n",
      "Wall time: 11min 8s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "for seed_id in modelseed_local.reactions:\n",
    "    rxn = modelseed_local.get_seed_reaction(seed_id)\n",
    "    rxn_node = annotation_api.get_node('ModelSeedReaction', seed_id)\n",
    "    if rxn_node == None:\n",
    "        delete = set()\n",
    "        for k in rxn.data:\n",
    "            if pd.isna(rxn.data[k]):\n",
    "                delete.add(k)\n",
    "        for k in delete:\n",
    "            del rxn.data[k]\n",
    "        annotation_api.add_node('ModelSeedReaction', seed_id, rxn.data)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load KEGG Orthologs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 50s, sys: 8.79 s, total: 1min 59s\n",
      "Wall time: 17min 31s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "df_kegg_ko = pd.read_csv('../../data/ko_kegg.tsv', sep='\\t')\n",
    "for row_id, d in df_kegg_ko.iterrows():\n",
    "    kegg_ko = d['kegg_ko']\n",
    "    kegg_rxns = d['kegg_rxns']\n",
    "    \n",
    "    ko_node = annotation_api.get_node('KeggOrthology', kegg_ko)\n",
    "    if ko_node == None:\n",
    "        annotation_api.add_node('KeggOrthology', kegg_ko, {})\n",
    "        ko_node = annotation_api.get_node('KeggOrthology', kegg_ko)\n",
    "        \n",
    "    if not pd.isna(kegg_rxns):\n",
    "        for rxn_id in kegg_rxns.split(';'):\n",
    "            rxn_node = annotation_api.get_node('LigandReaction', seed_id)\n",
    "            if rxn_node == None:\n",
    "                annotation_api.add_node('LigandReaction', rxn_id, {})\n",
    "                rxn_node = annotation_api.get_node('LigandReaction', rxn_id)\n",
    "            annotation_api.link_nodes(ko_node, rxn_node, 'has_reaction')\n",
    "            annotation_api.link_nodes(rxn_node, ko_node, 'has_ortholog')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 18min 42s, sys: 1min 33s, total: 20min 16s\n",
      "Wall time: 2h 4min 11s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "ko_kegg_genes = read_json('../../data/ko_kegg_genes.json')\n",
    "for ko in ko_kegg_genes:\n",
    "    ko_node = annotation_api.get_node('KeggOrthology', ko)\n",
    "    if not ko_node == None:\n",
    "        for genome_id, gene_id in ko_kegg_genes[ko]:\n",
    "            gene_node = annotation_api.get_node('KBaseGene', \"{}@{}\".format(gene_id, genome_id))\n",
    "            if gene_node == None:\n",
    "                gene_node = annotation_api.add_node('KBaseGene', \"{}@{}\".format(gene_id, genome_id))\n",
    "                \n",
    "            annotation_api.link_nodes(gene_node, ko_node, 'has_ortholog')\n",
    "            annotation_api.link_nodes(ko_node, gene_node, 'has_gene')\n",
    "    else:\n",
    "        logger.debug('not found: %s', ko)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load KEGG Orthologs to SEED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 48.4 s, sys: 3.66 s, total: 52 s\n",
      "Wall time: 2min 18s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "for seed_id in modelseed_local.reactions:\n",
    "    rxn = modelseed_local.get_seed_reaction(seed_id)\n",
    "    seed_rxn_node = annotation_api.get_node('ModelSeedReaction', seed_id)\n",
    "    if seed_id in modelseed_local.reaction_aliases and 'KEGG' in modelseed_local.reaction_aliases[seed_id]:\n",
    "        for kegg_id in modelseed_local.reaction_aliases[seed_id]['KEGG']:\n",
    "            kegg_rxn_node = annotation_api.get_node('LigandReaction', kegg_id)\n",
    "            if not kegg_rxn_node == None and not seed_rxn_node == None:\n",
    "                annotation_api.link_nodes(seed_rxn_node, kegg_rxn_node, 'has_crossreference_to')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load Templates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 76,
   "metadata": {},
   "outputs": [],
   "source": [
    "kbase = cobrakbase.KBaseAPI(token='YAFOCRSMRNDXZ7KMW7GCK5AC3SBNTEFD')\n",
    "templates = {}\n",
    "templates['GramNegModelTemplateV2'] = kbase.get_object('GramNegModelTemplateV2', 'NewKBaseModelTemplates')\n",
    "templates['GramNegModelTemplate'] = kbase.get_object('GramNegModelTemplate', 'NewKBaseModelTemplates')\n",
    "#template_doc = annotation_api.add_template('GramNegModelTemplateV2', templates['GramNegModelTemplateV2'])\n",
    "#template_doc = annotation_api.add_template('GramNegModelTemplate', templates['GramNegModelTemplate'])"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "def add_template(self, template_id, template):\n",
    "        roles = {}\n",
    "        function_uids = {}\n",
    "        compcompound_compartment = {}\n",
    "\n",
    "        for compcompound in template['compcompounds']:\n",
    "            compcompound_compartment[compcompound['id']] = compcompound['templatecompartment_ref'].split('/')[-1]\n",
    "\n",
    "        for role in template['roles']:\n",
    "            roles[role['id']] = role['name']\n",
    "        for role_id in roles:\n",
    "            function = roles[role_id]\n",
    "            #print(function)\n",
    "            function_doc = self.get_function(function)\n",
    "            if function_doc['value'] == function:\n",
    "                function_uids[function] = function_doc['_id']\n",
    "            else:\n",
    "                print('errooo!')\n",
    "        missing_function = set()\n",
    "        for role_id in roles:\n",
    "            function = roles[role_id]\n",
    "            if not function in function_uids:\n",
    "                missing_function.add(function)\n",
    "\n",
    "        print(len(missing_function))\n",
    "\n",
    "        cpx_to_function_uids = {}\n",
    "        for cpx in template['complexes']:\n",
    "            uids = set()\n",
    "            for complexrole in cpx['complexroles']:\n",
    "                role_id = complexrole['templaterole_ref'].split('/')[-1]\n",
    "                if role_id in roles:\n",
    "                    uids.add(function_uids[roles[role_id]])\n",
    "                else:\n",
    "                    print('errooo!')\n",
    "\n",
    "            cpx_to_function_uids[cpx['id']] = uids\n",
    "\n",
    "        reactions = {}\n",
    "        for rxn in template['reactions']:\n",
    "            rxn_id = rxn['reaction_ref'].split('/')[-1]\n",
    "            rxn_compartments = set()\n",
    "            for templateReactionReagent in rxn['templateReactionReagents']:\n",
    "                templatecompcompound_ref = templateReactionReagent['templatecompcompound_ref'].split('/')[-1]\n",
    "                rxn_compartments.add(compcompound_compartment[templatecompcompound_ref])\n",
    "            or_rule = set()\n",
    "            for complex_ref in rxn['templatecomplex_refs']:\n",
    "                complex_id = complex_ref.split('/')[-1]\n",
    "                and_rule = set()\n",
    "                if complex_id in cpx_to_function_uids:\n",
    "                    and_rule = frozenset(cpx_to_function_uids[complex_id])\n",
    "                or_rule.add(and_rule)\n",
    "\n",
    "            reactions[rxn_id] = {\n",
    "                'functions' : list(map(lambda x : list(x), or_rule)),\n",
    "                'compartment' : list(rxn_compartments),\n",
    "                'base_cost' : rxn['base_cost'],\n",
    "                'direction' : rxn['direction'],\n",
    "                'forward_penalty' : rxn['forward_penalty'],\n",
    "                'reverse_penalty' : rxn['reverse_penalty'],\n",
    "                'type' : rxn['type'],\n",
    "                'GapfillDirection' : rxn['GapfillDirection'],\n",
    "\n",
    "            }\n",
    "\n",
    "        template_doc = {\n",
    "            '_id' : template_id,\n",
    "            'reactions' : reactions\n",
    "        }\n",
    "\n",
    "        self.collection_templates.insert_one(template_doc)\n",
    "\n",
    "        return template_doc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 143,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Node id=172061 labels={'TemplateSet'} properties={'updated_at': 1568620759396, 'created_at': 1568620759396, 'key': 'GramNegModelTemplate'}>"
      ]
     },
     "execution_count": 143,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "template_node = annotation_api.get_node('TemplateSet', template_id)\n",
    "if template_node == None:\n",
    "    annotation_api.add_node('TemplateSet', template_id, {})\n",
    "    template_node = annotation_api.get_node('TemplateSet', template_id)\n",
    "template_node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Node id=185606 labels={'TemplateReactionAnnotation'} properties={'reaction': 'rxn00879', 'updated_at': 1568635675119, 'forward_penalty': 0, 'compartment': 'c', 'base_cost': 2, 'reverse_penalty': 0, 'created_at': 1568635675119, 'GapfillDirection': '=', 'type': 'conditional', 'key': 'rxn09225_c', 'direction': '='}>"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "add_template_reaction_annotation(annotation_api, template_node, rxn_node, rxn_compartments, or_rule, props)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'template_node' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-6-06ea6c8bf3fc>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m()\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0;32mfor\u001b[0m \u001b[0mr\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mtemplate_node\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mgraph\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrelationships\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      2\u001b[0m     \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mr\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'template_node' is not defined"
     ]
    }
   ],
   "source": [
    "for r in template_node.graph.relationships:\n",
    "    print(r)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Node id=87034 labels={'ModelSeedReaction'} properties={'updated_at': 1568609814074, 'equation': '(1) cpd02570[0] <=> (2) cpd00020[0]', 'name': '4-hydroxy-4-methyl-2-oxoglutarate pyruvate-lyase (pyruvate-forming)', 'created_at': 1568609814074, 'definition': '(1) Parapyruvate[0] <=> (2) Pyruvate[0]', 'id': 'rxn00004', 'key': 'rxn00004'}>\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_ko_by_seed_id(annotation_api, seed_id):\n",
    "    rxn_doc = annotation_api.get_node('ModelSeedReaction', seed_id)\n",
    "    print(rxn_doc)\n",
    "    if not rxn_doc == None:\n",
    "        for r in rxn_doc.graph.relationships:\n",
    "            print(r)\n",
    "        if 'kegg_kos' in rxn_doc:\n",
    "            return set(rxn_doc['kegg_kos'])\n",
    "        else:\n",
    "            return set()\n",
    "    return None\n",
    "get_ko_by_seed_id(annotation_api, 'rxn00004')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_reaction_annotation_data(rxn_id):\n",
    "    kos = annotation_api.get_ko_by_seed_id(rxn_id)\n",
    "    function_data = {}\n",
    "    for ko in kos:\n",
    "        #print(ko)\n",
    "        functions, functions_data, metadata = get_functional_roles2(ko, annotation_api)\n",
    "        print(ko, metadata)\n",
    "        for f in functions:\n",
    "            function = functions_data[f]['value']\n",
    "            if not function in function_data:\n",
    "                function_data[function] = {\n",
    "                    'id' : f,\n",
    "                    'hits' : []\n",
    "                }\n",
    "            function_data[function]['hits'].append({\n",
    "                'score' : len(functions[f]),\n",
    "                'source' : ['KEGG', ko]\n",
    "            })\n",
    "    for template_doc in annotation_api.collection_templates.find():\n",
    "        if rxn_id in template_doc['reactions']:\n",
    "            template_id = template_doc['_id']\n",
    "            template_reaction_doc = template_doc['reactions'][rxn_id]\n",
    "            for and_rule in template_reaction_doc['functions']:\n",
    "                for function_id in and_rule:\n",
    "                    function_doc = annotation_api.collection_functions.find_one({'_id' : function_id})\n",
    "                    function = function_doc['value']\n",
    "                    if not function in function_data:\n",
    "                        function_data[function] = {\n",
    "                            'id' : function_id,\n",
    "                            'hits' : []\n",
    "                        }\n",
    "                    function_data[function]['hits'].append({\n",
    "                        'score' : 0,\n",
    "                        'source' : ['template', template_id]\n",
    "                    })\n",
    "                    #print(template_id, function_doc['value'])\n",
    "        \n",
    "    for function in function_data:\n",
    "        function_id = function_data[function]['id']\n",
    "        #print(function_id)\n",
    "        function_doc = annotation_api.collection_functions.find_one({'_id' : function_id})\n",
    "        function_metadata = annotation_api.collection_functions_data.find_one({'_id' : function_id})\n",
    "        \n",
    "        subsystem_tags = {}\n",
    "        if 'subsystem' in function_doc:\n",
    "            for subsystem_tag in function_doc['subsystem']:\n",
    "                subsystem_tags[subsystem_tag] = [function_doc['subsystem'][subsystem_tag]]\n",
    "                if 'subsystem_class' in function_doc and subsystem_tag in function_doc['subsystem_class']:\n",
    "                    subsystem_tags[subsystem_tag].append(function_doc['subsystem_class'][subsystem_tag])\n",
    "        #print(subsystem_tags)\n",
    "        \n",
    "        \n",
    "        source_tags = {}\n",
    "        if not function_metadata == None:\n",
    "            sources = function_metadata['sources']\n",
    "            for s in sources:\n",
    "                source_tags[s] = [len(sources[s]['genomes']), len(sources[s]['genes']), sources[s]['genes'][:10]]\n",
    "            #print(source_tags)\n",
    "        function_data[function]['sources'] = source_tags\n",
    "        function_data[function]['subsystems'] = subsystem_tags\n",
    "    return function_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "rxn_record = annotation_api.get_node('ModelSeedReaction', 'rxn00148')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(<Node id=87270 labels=set() properties={}>, <Node id=133756 labels=set() properties={}>)\n"
     ]
    }
   ],
   "source": [
    "def get_relationship(tx, key, label, target_label, relationship_type):\n",
    "    if not target_label == None:\n",
    "        target_label = \":\" + target_label\n",
    "    else:\n",
    "        target_label = \"\"\n",
    "    if not label == None:\n",
    "        label = \":\" + label\n",
    "    else:\n",
    "        label = \"\"\n",
    "    if not relationship_type == None:\n",
    "        relationship_type = \":\" + relationship_type\n",
    "    else:\n",
    "        relationship_type = \"\"\n",
    "    cypher_query = \"MATCH (n1\" + label + \" {key:$key})-[r\" + relationship_type + \"]->(n2\" + target_label + \") RETURN r as relationships\"\n",
    "    logger.debug('Query: %s', cypher_query)\n",
    "    result = tx.run(cypher_query, key = key)\n",
    "    return list(result.records())\n",
    "\n",
    "#rxn_doc = annotation_api.get_node('ModelSeedReaction', seed_id)\n",
    "def get_r_nodes(annotation_api, key, label, target_label=None, relationship_type = None):\n",
    "    result = None\n",
    "    with annotation_api.driver.session() as session:\n",
    "        query = session.read_transaction(get_relationship, key, label, target_label, relationship_type)\n",
    "    return query\n",
    "\n",
    "results = get_r_nodes(annotation_api, 'rxn00148', 'ModelSeedReaction')\n",
    "node_ids = set()\n",
    "for result in results:\n",
    "    r = result['relationships']\n",
    "    #r = r.data().get('r')\n",
    "    node_pair = (r.start_node, r.end_node)\n",
    "    node_ids.add(r.end_node.id)\n",
    "    print(node_pair)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\"MATCH (n1:LigandReaction)-[r2]-(n2:KeggOrthology) WHERE id(n1)= 133756 RETURN n2\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{133756}"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "node_ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "99759"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [],
   "source": [
    "kos = annotation_api.get_ko_by_seed_id('rxn00148')\n",
    "function_data = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Node id=87270 labels={'ModelSeedReaction'} properties={'updated_at': 1568609816198, 'equation': '(1) cpd00002[0] + (1) cpd00020[0] <=> (1) cpd00008[0] + (1) cpd00061[0] + (1) cpd00067[0]', 'name': 'ATP:pyruvate 2-O-phosphotransferase', 'created_at': 1568609816198, 'definition': '(1) ATP[0] + (1) Pyruvate[0] <=> (1) ADP[0] + (1) Phosphoenolpyruvate[0] + (1) H+[0]', 'id': 'rxn00148', 'key': 'rxn00148'}>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rxn_record."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "a = Node(\"Person\", name=\"Alice\")\n",
    "b = Node(\"Person\", name=\"Bob\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "ab = Relationship(a, \"KNOWS\", b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a.relationships"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "m = neo4j_graph.match((Node(\"ModelSeedReaction\", key=\"rxn00148\"), ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(_133756:LigandReaction {created_at: 1568617528505, key: 'R00200', updated_at: 1568618373969})"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "node = neo4j_graph.nodes[133756]\n",
    "#node.__primarykey__ = \"key\"\n",
    "node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{<__main__.Neo4jAnnotationFunction at 0x7fb0c3d00400>,\n",
       " <__main__.Neo4jAnnotationFunction at 0x7fb0a7043160>}"
      ]
     },
     "execution_count": 290,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "        \n",
    "node = matcher.match('Function', key='Aspartokinase (EC 2.7.2.4) / Homoserine dehydrogenase (EC 1.1.1.3)').first()\n",
    "node\n",
    "function = Neo4jAnnotationFunction(node)\n",
    "function.sub_functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 273,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "node.has_label('Function')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aspartokinase (EC 2.7.2.4) / Homoserine dehydrogenase (EC 1.1.1.3) Aspartokinase (EC 2.7.2.4) / Homoserine dehydrogenase (EC 1.1.1.3) set()\n",
      "Homoserine dehydrogenase (EC 1.1.1.3) homoserinedehydrogenaseec1113 {'Sulfur metabolism related cluster', 'Aspartate to Threonine Module', 'Aspartate to Homoserine module'}\n",
      "Aspartokinase (EC 2.7.2.4) aspartokinaseec2724 {'Aspartate to Threonine Module', 'Peptidase clustering with DAP', 'Lysine DAP biosynthetic pathway', 'Diaminopimelate Synthesis ', 'Aspartate to Homoserine module', 'Sulfur metabolism related cluster'}\n"
     ]
    }
   ],
   "source": [
    "print(function, function.search_value, function.function_group)\n",
    "for f in function.sub_functions:\n",
    "    print(f, f.search_value, f.function_group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{(_74620:Function {created_at: 1563774166329, key: 'Aspartokinase (EC 2.7.2.4) / Homoserine dehydrogenase (EC 1.1.1.3)', updated_at: 1563778320381}): {(_75944:Function {created_at: 1563774195041, key: 'Aspartokinase (EC 2.7.2.4)', updated_at: 1563778378559}),\n",
       "  (_83913:Function {created_at: 1563774552943, key: 'Homoserine dehydrogenase (EC 1.1.1.3)', updated_at: 1563774578474})}}"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_gene_functions(gene_id):\n",
    "    functions = {}\n",
    "    matcher = NodeMatcher(annotation_api.neo4j_graph)\n",
    "    node = matcher.match('KBaseGene', key=gene_id).first()\n",
    "    if not node == None:\n",
    "        for rel in annotation_api.neo4j_graph.match((node, ), r_type=\"has_annotation\", ):\n",
    "            base_annotation = rel.end_node\n",
    "            functions[base_annotation] = set()\n",
    "            for sub_rel in annotation_api.neo4j_graph.match((base_annotation, ), r_type=\"has_subfunction\", ):\n",
    "                sub_function = sub_rel.end_node\n",
    "                functions[base_annotation].add(sub_function)\n",
    "                \n",
    "    return functions\n",
    "get_gene_functions('b3940@GCF_000005845.2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "({74620: {'b3940@GCF_000005845.2'},\n",
       "  75944: {'b3940@GCF_000005845.2'},\n",
       "  83913: {'b3940@GCF_000005845.2'}},\n",
       " {74620: {'value': 'Aspartokinase (EC 2.7.2.4) / Homoserine dehydrogenase (EC 1.1.1.3)'},\n",
       "  75944: {'value': 'Aspartokinase (EC 2.7.2.4)'},\n",
       "  83913: {'value': 'Homoserine dehydrogenase (EC 1.1.1.3)'}},\n",
       " {'total': 20, 'has_function': 1})"
      ]
     },
     "execution_count": 284,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'DVU2514@GCF_000195755.1'"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "216826"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_node.identity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7194380194339715759"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hash(gene_node)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'HI1573'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_node['id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "kbase = cobrakbase.KBaseAPI(token='UGOG6KLAWTCYI2ASYECYHNIIFTEXGA2J')\n",
    "genome1_rast   = kbase.get_object('GCF_000005845.2.RAST', 'jplfaria:narrative_1524466549180')\n",
    "genome2_script = kbase.get_object('GCF_000005845.2.RAST2', 'filipeliu:narrative_1562086644968')\n",
    "genome2_app    = kbase.get_object('test_nokmers1', 'jplfaria:narrative_1561579862828')\n",
    "genomes = {\n",
    "    'rast1' : cobrakbase.core.KBaseGenome(genome1_rast),\n",
    "    'rast2_script' : cobrakbase.core.KBaseGenome(genome2_script),\n",
    "    'rast2_app' : cobrakbase.core.KBaseGenome(genome2_app),\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'aliases': ['GeneID:944742', 'thrL', 'b0001', 'NP_414542.1', 'UniProtKB/Swiss-Prot:P0AD86', 'ASAP:ABE-0000006', 'JW4367', 'GI:16127995', 'ECK0001', 'EcoGene:EG11277'], 'cdss': ['NP_414542.1'], 'dna_sequence': 'ATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGA', 'dna_sequence_length': 66, 'function': 'Thr operon leader peptide', 'id': 'b0001', 'location': [['NC_000913', 190, '+', 66]], 'md5': '13fe4258b37e32edda386faa50ec5fdf', 'mrnas': ['mRNA_1'], 'protein_translation': 'MKRISTTITTTITITTGNGAG', 'protein_translation_length': 21, 'quality': {'hit_count': 7, 'weighted_hit_count': 9.1488}, 'type': 'gene'}\n"
     ]
    }
   ],
   "source": [
    "genome = genomes['rast1']\n",
    "for f in genome.features:\n",
    "    print(f)\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Match (p: Person)\n",
    "WHERE p.id > 0\n",
    "Return p\n",
    "ORDER BY p.id\n",
    "Skip 0 LIMIT 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MATCH (n:RefSeqGenome) RETURN n ORDER BY n.key SKIP $skip LIMIT $page_size\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "dict_keys(['GCF_000005825.2', 'GCF_000005845.2', 'GCF_000006175.1', 'GCF_000006605.1', 'GCF_000006685.1', 'GCF_000006725.1', 'GCF_000006745.1', 'GCF_000006765.1', 'GCF_000006785.2', 'GCF_000006845.1'])"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res = get_page_nodes(annotation_api, 'RefSeqGenome', 0, 10)\n",
    "os = {}\n",
    "for o in res:\n",
    "    n = o.data()['n']\n",
    "    os[n.get('key')] = dict(n.items())\n",
    "os.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MATCH (n:RefSeqGenome) WHERE n.scientific_name =~ '.*coli.*' RETURN n ORDER BY n.key SKIP $skip LIMIT $page_size\n",
      "Escherichia coli str. K-12 substr. MG1655\n",
      "Escherichia coli O157:H7 str. Sakai\n",
      "Yersinia enterocolitica subsp. enterocolitica 8081\n",
      "Escherichia coli UMN026\n",
      "Escherichia coli IAI39\n",
      "Escherichia coli O83:H1 str. NRG 857C\n",
      "Syntrophobotulus glycolicus DSM 8271\n",
      "Yersinia enterocolitica subsp. palearctica Y11\n",
      "Escherichia coli O104:H4 str. 2011C-3493\n",
      "Brachyspira pilosicoli P43/6/78\n",
      "Campylobacter coli\n"
     ]
    }
   ],
   "source": [
    "def search_genomes(search_str, limit = None):\n",
    "    limit = 1000 if limit == None else limit\n",
    "    res = annotation_api.page_nodes('RefSeqGenome', 0, limit, \"WHERE n.scientific_name =~ '.*\" + search_str.strip() + \".*'\")\n",
    "    os = {}\n",
    "    if not res == None:\n",
    "        for o in res:\n",
    "            n = o.data()['n']\n",
    "            os[n.get('key')] = dict(n.items())\n",
    "    return os\n",
    "res = search_genomes('coli')\n",
    "for id in res:\n",
    "    print(res[id]['scientific_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MATCH (n:RefSeqGenome) RETURN n ORDER BY n.key SKIP $skip LIMIT $page_size\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'GCF_000005825.2': {'genetic_code': 11,\n",
       "  'notes': '',\n",
       "  'release': '2',\n",
       "  'created_at': 1563767037784,\n",
       "  'scientific_name': 'Bacillus pseudofirmus OF4',\n",
       "  'source': 'refseq',\n",
       "  'taxonomy': 'cellular organisms; Bacteria; Terrabacteria group; Firmicutes; Bacilli; Bacillales; Bacillaceae; Bacillus; Bacillus pseudofirmus',\n",
       "  'type': 'Representative',\n",
       "  'external_source_origination_date': '30-JUL-2015',\n",
       "  'count_features': 4211,\n",
       "  'workspace_id': 'jplfaria:narrative_1524466549180',\n",
       "  'dna_size': 4249248,\n",
       "  'original_source_file_name': 'GCF_000005825.2_ASM582v2_genomic.gbff',\n",
       "  'reference_annotation': 0,\n",
       "  'updated_at': 1563767040915,\n",
       "  'num_contigs': 3,\n",
       "  'assembly_ref': '19217/82559/1',\n",
       "  'domain': 'Bacteria',\n",
       "  'source_id': 'NC_013791 (2 more accessions)',\n",
       "  'id': 'GCF_000005825.2',\n",
       "  'taxon_ref': '12570/597018/1',\n",
       "  'genbank_handle_ref': 'KBH_310617',\n",
       "  'gc_content': 0.3986055885653179,\n",
       "  'key': 'GCF_000005825.2'}}"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def list_genomes(limit = None):\n",
    "    limit = 1000 if limit == None else limit\n",
    "    res = annotation_api.page_nodes('RefSeqGenome', 0, limit)\n",
    "    os = {}\n",
    "    if not res == None:\n",
    "        for o in res:\n",
    "            n = o.data()['n']\n",
    "            os[n.get('key')] = dict(n.items())\n",
    "    return os\n",
    "list_genomes(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_data():\n",
    "    pass\n",
    "def get_genome():\n",
    "    pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'rast1': <cobrakbase.core.kbasegenomesgenome.KBaseGenome at 0x7fdaa8b10a58>,\n",
       " 'rast2_script': <cobrakbase.core.kbasegenomesgenome.KBaseGenome at 0x7fdaa9374f60>,\n",
       " 'rast2_app': <cobrakbase.core.kbasegenomesgenome.KBaseGenome at 0x7fdaa8527320>}"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def add_genome(genome):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "print(process_list([['1', '3', 1], '2']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'assembly_ref': '19217/231473/2',\n",
       " 'contig_ids': 'NC_000913.3',\n",
       " 'dna_size': 4641652,\n",
       " 'domain': 'Bacteria',\n",
       " 'external_source_origination_date': '08-Aug-2016',\n",
       " 'gc_content': 0.508,\n",
       " 'genbank_handle_ref': 'KBH_3448578',\n",
       " 'genetic_code': 11,\n",
       " 'genome_tiers': 'ExternalDB',\n",
       " 'id': 'test_nokmers1',\n",
       " 'md5': '482a2b04485ec8c4b5f4eaba2c2002da',\n",
       " 'molecule_type': 'DNA',\n",
       " 'mrnas': '',\n",
       " 'notes': 'REVIEWED REFSEQ: This record has been curated by NCBI staff. The\\nreference sequence is identical to U00096.\\nOn Nov 3, 2013 this sequence version replaced NC_000913.2.\\nRefSeq Category: Reference Genome\\n            FGS: First Genome sequenced\\n            MOD: Model Organism\\n            PHY: Based on Phylogenetics\\n            UPR: UniProt Genome\\nCurrent U00096 annotation updates are derived from EcoGene\\nhttp://ecogene.org. Suggestions for updates can be sent to Dr.\\nKenneth Rudd (krudd@miami.edu). These updates are being generated\\nfrom a collaboration that also includes ASAP/ERIC, the Coli Genetic\\nStock Center, EcoliHub, EcoCyc, RegulonDB and UniProtKB/Swiss-Prot.\\nCOMPLETENESS: full length.',\n",
       " 'num_contigs': 1,\n",
       " 'original_source_file_name': 'GCF_000005845.2_ASM584v2_genomic.gbff',\n",
       " 'scientific_name': 'Escherichia coli str. K-12 substr. MG1655',\n",
       " 'source': 'RefSeq',\n",
       " 'source_id': 'NC_000913',\n",
       " 'taxon_ref': '12570/625052/2',\n",
       " 'taxonomy': 'cellular organisms; Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacterales; Enterobacteriaceae; Escherichia; Escherichia coli; Escherichia coli K-12',\n",
       " 'warnings': ''}"
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "process_dict(genome.data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{<Node id=414883 labels={'Function'} properties={'updated_at': 1569876120585, 'created_at': 1569876120585, 'key': 'fake_function1 @ very_fake_function2'}>}"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.add_annotation(['fake_function1 @ very_fake_function2'], 'Test')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{<Node id=185611 labels={'Function'} properties={'updated_at': 1569877637469, 'created_at': 1569877637469, 'key': 'fake function 1'}>}"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.add_annotation(['very fake function 2'], 'Test2')\n",
    "annotation_api.add_annotation(['fake function 1'], 'Test2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{<Node id=414887 labels={'Function'} properties={'updated_at': 1569877575598, 'created_at': 1569877575598, 'key': 'fake function1 @ very fake function2'}>}"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.add_annotation(['fake function1 @ very fake function2'], 'Test3')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{<Node id=185610 labels={'Function'} properties={'updated_at': 1569877486050, 'created_at': 1569877486050, 'key': 'very fake function 2'}>}"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.add_annotation(['very fake function 2'], 'Test4')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:__main__:ADD BASE Function: fake_function1 @ very_fake_function2\n",
      "WARNING:__main__:ADD SUB Function: very_fake_function2\n",
      "WARNING:__main__:ADD SEARCH SUB Function: veryfakefunction2\n",
      "WARNING:__main__:ADD SUB Function: fake_function1\n",
      "WARNING:__main__:ADD SEARCH SUB Function: fakefunction1\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{<Node id=414883 labels={'Function'} properties={'updated_at': 1569876120585, 'created_at': 1569876120585, 'key': 'fake_function1 @ very_fake_function2'}>}"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "add_annotation(annotation_api, ['fake_function1 @ very_fake_function2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(_364723:KBaseGene {aliases: 'GeneID:944742 ### thrL ### b0001 ### NP_414542.1 ### UniProtKB/Swiss-Prot:P0AD86 ### ASAP:ABE-0000006 ### JW4367 ### GI:16127995 ### ECK0001 ### EcoGene:EG11277', cdss: 'NP_414542.1', created_at: 1569833182468, dna_sequence: 'ATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGA', dna_sequence_length: 66, function: 'Thr operon leader peptide', id: 'b0001', key: 'b0001@GCF_000005845.2', md5: '13fe4258b37e32edda386faa50ec5fdf', mrnas: 'mRNA_1', protein_translation: 'MKRISTTITTTITITTGNGAG', protein_translation_length: 21, type: 'gene', updated_at: 1569879680696})"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from py2neo import Graph, NodeMatcher\n",
    "matcher = NodeMatcher(annotation_api.neo4j_graph)\n",
    "node = matcher.match(\"KBaseGene\", key=feature_id).first()\n",
    "node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'aliases': 'GeneID:944742 ### thrL ### b0001 ### NP_414542.1 ### UniProtKB/Swiss-Prot:P0AD86 ### ASAP:ABE-0000006 ### JW4367 ### GI:16127995 ### ECK0001 ### EcoGene:EG11277',\n",
       " 'cdss': 'NP_414542.1',\n",
       " 'dna_sequence': 'ATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGA',\n",
       " 'dna_sequence_length': 66,\n",
       " 'function': 'Thr operon leader peptide',\n",
       " 'id': 'b0001',\n",
       " 'md5': '13fe4258b37e32edda386faa50ec5fdf',\n",
       " 'mrnas': 'mRNA_1',\n",
       " 'protein_translation': 'MKRISTTITTTITITTGNGAG',\n",
       " 'protein_translation_length': 21,\n",
       " 'type': 'gene',\n",
       " 'key': 'b0001@GCF_000005845.2'}"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "node = feature_props"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "genome = genomes['rast2_app']\n",
    "add_kbase_genome(annotation_api, genome, 'GCF_000005845.2', 'BETA_RAST')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome = genomes['rast1']\n",
    "add_kbase_genome(annotation_api, genome, 'GCF_000005845.2', 'KBASE_RAST')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_annotation_old(annotation_api, gene_genome_id, functions, source):\n",
    "    #[KBaseGene] -[has_function]-> [Function] <-[has_function]- [FunctionSource]\n",
    "\n",
    "    kbase_gene_node = self.get_node('KBaseGene', gene_genome_id)\n",
    "    function_source_node = self.get_node('FunctionSource', source)\n",
    "\n",
    "    if function_source_node == None:\n",
    "        print('not found', source)\n",
    "        return    \n",
    "    if kbase_gene_node == None:\n",
    "        print('not found', gene_genome_id)\n",
    "        return\n",
    "\n",
    "    for f in functions:\n",
    "        if not f == None:\n",
    "            function_node = annotation_api.add_node('Function', f)\n",
    "            annotation_api.link_nodes(function_source_node, function_node, 'has_function')\n",
    "            annotation_api.link_nodes(kbase_gene_node, function_node, 'has_function')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "aliases <class 'list'>\n",
      "cdss <class 'list'>\n",
      "location <class 'list'>\n",
      "mrnas <class 'list'>\n",
      "quality <class 'dict'>\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'{aliases:, cdss:, dna_sequence:\"ATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGA\", dna_sequence_length:66, function:\"Thr operon leader peptide\", id:\"b0001\", location:, md5:\"13fe4258b37e32edda386faa50ec5fdf\", mrnas:, protein_translation:\"MKRISTTITTTITITTGNGAG\", protein_translation_length:21, quality:, type:\"gene\"}'"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.to_str_dict(feature.data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(_41:RefSeqGenome {assembly_ref: '19217/231473/1', count_features: 4498, created_at: 1563767040940, dna_size: 4641652, domain: 'Bacteria', external_source_origination_date: '08-AUG-2016', gc_content: 0.5079070985933456, genbank_handle_ref: 'KBH_644247', genetic_code: 11, id: 'GCF_000005845.2', key: 'GCF_000005845.2', notes: '', num_contigs: 1, original_source_file_name: 'GCF_000005845.2_ASM584v2_genomic.gbff', reference_annotation: 1, release: '2', scientific_name: 'Escherichia coli str. K-12 substr. MG1655', source: 'refseq', source_id: 'NC_000913', taxon_ref: '12570/625052/2', taxonomy: 'cellular organisms; Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacterales; Enterobacteriaceae; Escherichia; Escherichia coli; Escherichia coli K-12', type: 'Reference', updated_at: 1563767040940, workspace_id: 'jplfaria:narrative_1524466549180'})"
      ]
     },
     "execution_count": 292,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{None}"
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature.functions_unsplit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_nodes = annotation_api.add_annotation(feature.functions_unsplit, 'KBASE_RAST')\n",
    "annotation_nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "kbase = cobrakbase.KBaseAPI(token='UGOG6KLAWTCYI2ASYECYHNIIFTEXGA2J')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load Current RAST Genomes\n",
    "jplfaria:narrative_1524466549180"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#genomes 1675\n"
     ]
    }
   ],
   "source": [
    "genome_objects = {}\n",
    "for o in kbase.list_objects('jplfaria:narrative_1524466549180'):\n",
    "    if o[1].endswith('RAST') and o[2].startswith(\"KBaseGenomes.Genome\"):\n",
    "        genome_id = o[1][:-5]\n",
    "        genome_objects[genome_id] = o[1]\n",
    "        \n",
    "print('#genomes', len(genome_objects))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 426,
   "metadata": {},
   "outputs": [],
   "source": [
    "for genome_id in genome_objects:\n",
    "    object_id = genome_objects[genome_id]\n",
    "    data = kbase.get_object(object_id, 'jplfaria:narrative_1524466549180')\n",
    "    write_json(data, '../../../data/kbase_data/jplfaria_narrative_1524466549180/' + object_id + '.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:__main__:ADD RefSeqGenome: GCF_000240185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000026325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000026345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000183345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000091545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000006765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000299455.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000064305.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007985.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000069185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000185905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000146165.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000203855.3\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000191145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000006905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007565.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000006945.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196035.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000006965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000174395.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000318015.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317935.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195715.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196115.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011445.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000027345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000026745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000215745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000005845.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000165905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195835.1\n"
     ]
    },
    {
     "ename": "CypherSyntaxError",
     "evalue": "Invalid input '%': expected '\\', ''', '\"', 'b', 'f', 'n', 'r', 't', UTF16 or UTF32 (line 1, column 1510 (offset: 1509))\n\"MERGE (n:KBaseGene {aliases:\"GI:31793105 ### Mb1947c ### GeneID:1093437 ### fadB5 ### 1.-.-.- ### NP_855598.1\", cdss:\"NP_855598.1\", dna_sequence:\"ATGCGAGCAGTGGTCATCACCAAACATGGCGACCCATCGGTCTTGCAGGTGCGGCAGCGACCGGACCCGCCGCCACCGGGCCCGGGCCAGCTGCGGGTCGCCGTCCGCGCAGCAGGGGTGAACTTCGCTGACCATCTCGCCCGCGTCGGCCTGTACCCAGACGCGCCGAAACTTCCGGCGGTGGTCGGATACGAAGTCGCTGGGACGGTCGAGGCTGTCGGTGATGGGGTCGACCCGAACCGGGTCGGCGAACGAGTCCTGGCCGGTACACGATTTGGTGGCTACTGCGAGATCGTCAACGTTGCGGCCACCGACTCGGTTGTGCTCCCCGATGCGCTGAGCTTCGAACAGGGTGCCGCGGTCCCGGTGAATTACGCGACCGCCTGGGCGGCGCTGCACGGCTACGGATCGTTGCGCGCCGGTGAGCGGGTGCTGATTCACGCCGCGGCCGGTGGAGTCGGCATCGCGGCGGTCCAATTCGCGAAAGCAGCCAAGGCCGAAGTGCACGGCACCGCATCACCCCAAAAACATCAGAAGCTGGCCGAGTTCGGTGTGGACCGCGCGATCGACTACCGCCGGGACGGCTGGTGGCAGGGATTGGGCCCGTATGACGTCGTGCTTGACGCGCTCGGCGGCACCTCGCTGCGGCGGTCCTACACTCTGCTGCGCCCGGGTGGAAGGCTGGTTGGCTACGGGATTTCGAATATGCAGCACGGCGAGAAACGATCGATGCGCAGGGTGGCGCCCCACGCGTTGTCAATGCTGCGCGGCTTTAACCTGATGAAACAACTCGAGGAGTCGAAAACCGTGATCGGTCTTAACATGCTGCGGTTGTGGGACGATCGCCGCACCCTTGAACCCTGGATCGCGCCGCTGACCAAGGCGCTCAACGACGGAACGATCCTGCCGATCGTTCATGCAATCGTGCCGTTCGCCGAAGCTCCTGAAGCACATCGGATTCTGGCCGCACGGGAGAACGTCGGCAAGGTGGTGCTGGTACCGTGA\", dna_sequence_length:1005, function:\"(MTCY180.06), len: 334. fadB5, Possible oxidoreductase, similar to hydroxyacyl-CoA dehydrogenase, quinone oxidoreductases (EC 1.6.5.5) and polyketide synthases.Contains quinone oxidoreductase / zeta-crystallin signature(PS01162). FASTA results: Q53927 HYDROXYACYL-COA DEHYDROGENASE. (329 aa) opt: 621; E(): 2e-30; 39.5\\% identity in349 aa overlap. Similar to many hypothetical M. tuberculosisproteins including: MTCY24G1.09, MTCY13D12.11, MTCY19H9.01, MTCY24G1.03, MTCY03A2.17c, etc. TBparse score is 0.921\", id:\"Mb1947c\", md5:\"9c10d1749848e1238a18af5456114300\", mrnas:\"mRNA_1927\", protein_translation:\"MRAVVITKHGDPSVLQVRQRPDPPPPGPGQLRVAVRAAGVNFADHLARVGLYPDAPKLPAVVGYEVAGTVEAVGDGVDPNRVGERVLAGTRFGGYCEIVNVAATDSVVLPDALSFEQGAAVPVNYATAWAALHGYGSLRAGERVLIHAAAGGVGIAAVQFAKAAKAEVHGTASPQKHQKLAEFGVDRAIDYRRDGWWQGLGPYDVVLDALGGTSLRRSYTLLRPGGRLVGYGISNMQHGEKRSMRRVAPHALSMLRGFNLMKQLEESKTVIGLNMLRLWDDRRTLEPWIAPLTKALNDGTILPIVHAIVPFAEAPEAHRILAARENVGKVVLVP\", protein_translation_length:334, type:\"gene\", key:\"Mb1947c@GCF_000195835.1\"}) ON CREATE SET n.created_at = timestamp(), n.updated_at = timestamp()ON MATCH  SET n.updated_at = timestamp() RETURN n\"\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      ^",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mCypherSyntaxError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-23-997b0c6841c0>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      3\u001b[0m     \u001b[0mgenome_data\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mread_json\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'../../../data/kbase_data/jplfaria_narrative_1524466549180/'\u001b[0m \u001b[0;34m+\u001b[0m \u001b[0mobject_id\u001b[0m \u001b[0;34m+\u001b[0m\u001b[0;34m'.json'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      4\u001b[0m     \u001b[0mgenome\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mcobrakbase\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcore\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mKBaseGenome\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenome_data\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 5\u001b[0;31m     \u001b[0mgenome_node\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mannotation_api\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madd_kbase_genome\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenome\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenome_id\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'KBASE_RAST'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/workspace/jupyter/python3/annotation-server/annotation_api_neo4j.py\u001b[0m in \u001b[0;36madd_kbase_genome\u001b[0;34m(self, genome, genome_id, annotation_source)\u001b[0m\n\u001b[1;32m    526\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mgenome\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfeatures\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    527\u001b[0m             \u001b[0mfeature\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mKBaseGenomeFeature\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mf\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 528\u001b[0;31m             \u001b[0mfeature_node\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madd_genome_feature\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfeature\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgenome_id\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mannotation_source\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    529\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlink_nodes\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgenome_node\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfeature_node\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'has_gene'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    530\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/workspace/jupyter/python3/annotation-server/annotation_api_neo4j.py\u001b[0m in \u001b[0;36madd_genome_feature\u001b[0;34m(self, feature, genome_id, annotation_source)\u001b[0m\n\u001b[1;32m    488\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mfeature_node\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    489\u001b[0m             \u001b[0mlogger\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdebug\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'ADD KBaseGene: %s'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfeature_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 490\u001b[0;31m             \u001b[0mfeature_node\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madd_node\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'KBaseGene'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfeature_id\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfeature_props\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    491\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    492\u001b[0m             \u001b[0mlogger\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdebug\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'UPDATE KBaseGene: %s'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mfeature_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/workspace/jupyter/python3/annotation-server/annotation_api_neo4j.py\u001b[0m in \u001b[0;36madd_node\u001b[0;34m(self, node_type, key, props)\u001b[0m\n\u001b[1;32m    184\u001b[0m         \u001b[0;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdriver\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msession\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    185\u001b[0m             \u001b[0mlogger\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdebug\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'write_transaction %s %s'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnode_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprops_str\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 186\u001b[0;31m             \u001b[0mquery\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msession\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrite_transaction\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_add_value_node\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mnode_type\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mprops_str\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    187\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mquery\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    188\u001b[0m                 \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mquery\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'n'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/neo4j/__init__.py\u001b[0m in \u001b[0;36mwrite_transaction\u001b[0;34m(self, unit_of_work, *args, **kwargs)\u001b[0m\n\u001b[1;32m    699\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mwrite_transaction\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0munit_of_work\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    700\u001b[0m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_assert_open\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 701\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_run_transaction\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mWRITE_ACCESS\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0munit_of_work\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    702\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    703\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_assert_open\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/neo4j/__init__.py\u001b[0m in \u001b[0;36m_run_transaction\u001b[0;34m(self, access_mode, unit_of_work, *args, **kwargs)\u001b[0m\n\u001b[1;32m    665\u001b[0m                 \u001b[0mtx\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_transaction\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    666\u001b[0m                 \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 667\u001b[0;31m                     \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0munit_of_work\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtx\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    668\u001b[0m                 \u001b[0;32mexcept\u001b[0m \u001b[0mException\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    669\u001b[0m                     \u001b[0;32mif\u001b[0m \u001b[0mtx\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msuccess\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/workspace/jupyter/python3/annotation-server/annotation_api_neo4j.py\u001b[0m in \u001b[0;36m_add_value_node\u001b[0;34m(tx, label, props_str)\u001b[0m\n\u001b[1;32m    282\u001b[0m                         \u001b[0;34m\"ON MATCH  SET n.updated_at = timestamp() \"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    283\u001b[0m                         \"RETURN n\")\n\u001b[0;32m--> 284\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresult\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mrecords\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    285\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    286\u001b[0m     \u001b[0;34m@\u001b[0m\u001b[0mstaticmethod\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/neo4j/__init__.py\u001b[0m in \u001b[0;36mrecords\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    923\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_session\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    924\u001b[0m         \u001b[0;32mwhile\u001b[0m \u001b[0mattached\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 925\u001b[0;31m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_session\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfetch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    926\u001b[0m             \u001b[0;32mwhile\u001b[0m \u001b[0mrecords\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    927\u001b[0m                 \u001b[0;32myield\u001b[0m \u001b[0mnext_record\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/neo4j/__init__.py\u001b[0m in \u001b[0;36mfetch\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    516\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_connection\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    517\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 518\u001b[0;31m                 \u001b[0mdetail_count\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0m_\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_connection\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfetch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    519\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mConnectionExpired\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    520\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0mSessionExpired\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0merror\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/neobolt/direct.py\u001b[0m in \u001b[0;36mfetch\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    420\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mfetch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    421\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 422\u001b[0;31m             \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_fetch\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    423\u001b[0m         \u001b[0;32mexcept\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merror_handler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mknown_errors\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merror\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    424\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0merror_handler\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhandle\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0merror\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0munresolved_address\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/neobolt/direct.py\u001b[0m in \u001b[0;36m_fetch\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    462\u001b[0m         \u001b[0;32melif\u001b[0m \u001b[0msummary_signature\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34mb\"\\x7F\"\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    463\u001b[0m             \u001b[0mlog_debug\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"[#%04X]  S: FAILURE %r\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mlocal_port\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0msummary_metadata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 464\u001b[0;31m             \u001b[0mresponse\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mon_failure\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msummary_metadata\u001b[0m \u001b[0;32mor\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    465\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    466\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mProtocolError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Unexpected response message with signature %02X\"\u001b[0m \u001b[0;34m%\u001b[0m \u001b[0msummary_signature\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/anaconda3/lib/python3.6/site-packages/neobolt/direct.py\u001b[0m in \u001b[0;36mon_failure\u001b[0;34m(self, metadata)\u001b[0m\n\u001b[1;32m    757\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandler\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    758\u001b[0m             \u001b[0mhandler\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 759\u001b[0;31m         \u001b[0;32mraise\u001b[0m \u001b[0mCypherError\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhydrate\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m**\u001b[0m\u001b[0mmetadata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    760\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    761\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mon_ignored\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmetadata\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mCypherSyntaxError\u001b[0m: Invalid input '%': expected '\\', ''', '\"', 'b', 'f', 'n', 'r', 't', UTF16 or UTF32 (line 1, column 1510 (offset: 1509))\n\"MERGE (n:KBaseGene {aliases:\"GI:31793105 ### Mb1947c ### GeneID:1093437 ### fadB5 ### 1.-.-.- ### NP_855598.1\", cdss:\"NP_855598.1\", dna_sequence:\"ATGCGAGCAGTGGTCATCACCAAACATGGCGACCCATCGGTCTTGCAGGTGCGGCAGCGACCGGACCCGCCGCCACCGGGCCCGGGCCAGCTGCGGGTCGCCGTCCGCGCAGCAGGGGTGAACTTCGCTGACCATCTCGCCCGCGTCGGCCTGTACCCAGACGCGCCGAAACTTCCGGCGGTGGTCGGATACGAAGTCGCTGGGACGGTCGAGGCTGTCGGTGATGGGGTCGACCCGAACCGGGTCGGCGAACGAGTCCTGGCCGGTACACGATTTGGTGGCTACTGCGAGATCGTCAACGTTGCGGCCACCGACTCGGTTGTGCTCCCCGATGCGCTGAGCTTCGAACAGGGTGCCGCGGTCCCGGTGAATTACGCGACCGCCTGGGCGGCGCTGCACGGCTACGGATCGTTGCGCGCCGGTGAGCGGGTGCTGATTCACGCCGCGGCCGGTGGAGTCGGCATCGCGGCGGTCCAATTCGCGAAAGCAGCCAAGGCCGAAGTGCACGGCACCGCATCACCCCAAAAACATCAGAAGCTGGCCGAGTTCGGTGTGGACCGCGCGATCGACTACCGCCGGGACGGCTGGTGGCAGGGATTGGGCCCGTATGACGTCGTGCTTGACGCGCTCGGCGGCACCTCGCTGCGGCGGTCCTACACTCTGCTGCGCCCGGGTGGAAGGCTGGTTGGCTACGGGATTTCGAATATGCAGCACGGCGAGAAACGATCGATGCGCAGGGTGGCGCCCCACGCGTTGTCAATGCTGCGCGGCTTTAACCTGATGAAACAACTCGAGGAGTCGAAAACCGTGATCGGTCTTAACATGCTGCGGTTGTGGGACGATCGCCGCACCCTTGAACCCTGGATCGCGCCGCTGACCAAGGCGCTCAACGACGGAACGATCCTGCCGATCGTTCATGCAATCGTGCCGTTCGCCGAAGCTCCTGAAGCACATCGGATTCTGGCCGCACGGGAGAACGTCGGCAAGGTGGTGCTGGTACCGTGA\", dna_sequence_length:1005, function:\"(MTCY180.06), len: 334. fadB5, Possible oxidoreductase, similar to hydroxyacyl-CoA dehydrogenase, quinone oxidoreductases (EC 1.6.5.5) and polyketide synthases.Contains quinone oxidoreductase / zeta-crystallin signature(PS01162). FASTA results: Q53927 HYDROXYACYL-COA DEHYDROGENASE. (329 aa) opt: 621; E(): 2e-30; 39.5\\% identity in349 aa overlap. Similar to many hypothetical M. tuberculosisproteins including: MTCY24G1.09, MTCY13D12.11, MTCY19H9.01, MTCY24G1.03, MTCY03A2.17c, etc. TBparse score is 0.921\", id:\"Mb1947c\", md5:\"9c10d1749848e1238a18af5456114300\", mrnas:\"mRNA_1927\", protein_translation:\"MRAVVITKHGDPSVLQVRQRPDPPPPGPGQLRVAVRAAGVNFADHLARVGLYPDAPKLPAVVGYEVAGTVEAVGDGVDPNRVGERVLAGTRFGGYCEIVNVAATDSVVLPDALSFEQGAAVPVNYATAWAALHGYGSLRAGERVLIHAAAGGVGIAAVQFAKAAKAEVHGTASPQKHQKLAEFGVDRAIDYRRDGWWQGLGPYDVVLDALGGTSLRRSYTLLRPGGRLVGYGISNMQHGEKRSMRRVAPHALSMLRGFNLMKQLEESKTVIGLNMLRLWDDRRTLEPWIAPLTKALNDGTILPIVHAIVPFAEAPEAHRILAARENVGKVVLVP\", protein_translation_length:334, type:\"gene\", key:\"Mb1947c@GCF_000195835.1\"}) ON CREATE SET n.created_at = timestamp(), n.updated_at = timestamp()ON MATCH  SET n.updated_at = timestamp() RETURN n\"\n                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                      ^"
     ]
    }
   ],
   "source": [
    "loaded_genomes = list_genomes(annotation_api)\n",
    "for genome_id in genome_objects:\n",
    "    if not genome_id in loaded_genomes:\n",
    "        object_id = genome_objects[genome_id]\n",
    "        genome_data = read_json('../../../data/kbase_data/jplfaria_narrative_1524466549180/' + object_id +'.json')\n",
    "        genome = cobrakbase.core.KBaseGenome(genome_data)\n",
    "        genome_node = annotation_api.add_kbase_genome(genome, genome_id, 'KBASE_RAST')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092585.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001027025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001465835.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000185885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000661895.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001750165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000214725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000147055.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001026985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_002211785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000447675.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000828915.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000264495.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000183725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001021085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000214095.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000347595.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017585.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000817955.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000306725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000761155.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000576555.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001190745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000063505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000964565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001412615.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000764535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000300005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000253015.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000237085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000165715.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001038625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000316605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000224985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000803645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000508245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000504125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195335.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000313635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001465855.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000238215.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000973505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196095.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001586195.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000816145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001458475.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000973545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000144625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317855.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000319385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001189295.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001642655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_002234495.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025405.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001887285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000247605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000585495.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000785515.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000230715.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196835.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000270085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000262715.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000184685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001456315.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000184435.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000190535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000626635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000737865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008685.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000180175.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000723165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196295.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000005825.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000190735.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000484535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000191585.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196235.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001547975.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015585.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000148645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196515.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000143965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000739085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000246855.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001027545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001043175.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001655245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000812185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000186345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000583875.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000091325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001189495.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000147875.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000271665.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317495.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000784965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000510265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000550765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000829035.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000763575.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001652565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001444425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000237865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001578185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001025175.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000226295.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000941055.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000243115.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000967425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000304355.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001507665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000225325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000500935.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000725425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000830985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000006175.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000192745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000063585.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000157355.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000600005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000265465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000376585.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000143165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000478905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000512205.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000184345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000512355.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000091665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000027145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000299355.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001028645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196355.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000300075.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000185985.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000970205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195975.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001307195.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000251105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000145235.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001488575.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_002222615.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000833105.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196855.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000747525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001693675.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000725365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000240165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000183135.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000325745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000146065.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000270245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000218545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000512735.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000214415.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000212695.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001908275.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000213825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000828635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001454945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000470775.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000183385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000304455.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001267865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000400935.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001685465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000496595.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000026105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001553605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001653755.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000973105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001314995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000263735.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000226315.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000445425.4\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000484505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001558415.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001051995.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317835.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000304735.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000313175.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000297055.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000259255.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000219175.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000511385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001514435.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000968195.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001610975.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317695.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001042675.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000151105.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317475.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000270145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000187005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000177635.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000236705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196895.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317515.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000178115.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000263195.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001020985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000750535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000147835.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001729485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000159155.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000165505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000253395.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000212395.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000215105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000389965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000299095.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000144645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000258405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001652465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000590475.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000969905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000941075.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000739475.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196135.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_900086555.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000146505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001908775.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000177535.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000199675.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000404225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000316515.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001005905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001042635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000093065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000214825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000294515.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000525655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000550805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000204645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000328565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001886715.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000166415.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007465.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001553195.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317675.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000819565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000224105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001698145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000194605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000147715.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000294715.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001013905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196735.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001444405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284615.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000477435.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000192845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009765.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284155.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000152265.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000230735.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001021935.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000259275.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000590555.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000063605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000225445.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317795.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195915.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000179915.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000973085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000217995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001514455.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001431725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000801295.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001593245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001420915.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000568815.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000950575.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001190945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000758685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001682385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000321415.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000175255.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000737575.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000875755.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001007995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001641285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000967915.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001021385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001887595.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000215705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000316575.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000953195.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000223395.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012585.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000147335.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000828715.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000223375.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000265365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000213805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001447335.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000283595.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000833575.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000348785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000827005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001548015.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000152825.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000155735.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000186885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000477415.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001269425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000953475.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000026045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000828835.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000235405.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000517425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000949425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000632845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001518815.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000260985.4\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001606005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000220625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000233915.3\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000154785.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000145295.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000734015.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000316645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001562115.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000266945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000231385.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000737325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001886695.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000734895.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000184745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001698225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000311765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000473305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000743945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000027305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014585.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001750785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000190635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000147695.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001305595.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000215995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000190575.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000227705.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000265525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000807255.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000164675.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000695095.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000513475.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000389635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001021975.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010585.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000328685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000815185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000306785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000769655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000242635.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000175095.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000970285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000063525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001008165.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195275.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000152245.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000348725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001787335.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000283515.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000730385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000194625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001281385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000270205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000763535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001439585.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000828515.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000277125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000242255.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000177235.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000219805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000186365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000224475.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196615.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000970085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000325705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000090965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000800295.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001584185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000439435.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000953655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001421015.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196815.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000767055.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000764555.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000287335.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001042595.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000422165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000767275.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000148385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001941345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000178875.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195315.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_002222635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001542565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000179395.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001267175.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001318345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000046705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000818035.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000953695.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001020955.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000327485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_002240355.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001955715.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000214785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014445.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000439455.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000237995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000212735.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000069925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001050475.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001187785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_900078695.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000968535.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000006745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000063445.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000214155.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000767465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000521565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000455605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_002215215.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001953195.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000328765.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000829395.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000736415.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_002211765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000591055.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000204565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001483385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000175575.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196015.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000960975.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000599985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001190755.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000835165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000403645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000802245.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000166055.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000341345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000341355.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000225345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000344805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000328625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001685435.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000331995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000283555.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000219355.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000307105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001951175.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000063545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001586215.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023445.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000178835.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000832305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000231405.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001077715.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000816085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000247565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000147355.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000830005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001687565.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001273775.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001274875.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012085.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001610955.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000354175.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000334405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001078055.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000153405.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000219105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000163895.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000507245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001676765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000204155.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000164695.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001618885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001636015.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001280225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001708485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001686985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000215975.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000091125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000493735.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000305935.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000828855.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001594265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000832985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001605965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196695.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000265385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284315.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000800475.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000340435.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000828655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000832905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000176915.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001499615.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000143085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001007935.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000304215.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195735.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001040945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000577895.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000550785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000327045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000250675.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000612055.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001558935.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001011035.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000471025.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000499765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001606025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000210915.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000980835.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196275.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000179575.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000619905.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000153485.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000172995.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000689415.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000724485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001456215.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000815065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000006785.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001460635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000970325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000179035.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000227745.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284035.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000511355.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000178955.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000463355.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_900116045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000525675.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000807275.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000832605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000731315.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001578105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000974425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000183745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000243135.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000056065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001677275.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001457475.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000227665.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000026005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000026125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001042715.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000953635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000219585.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001660045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000218625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000300295.3\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000253035.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000006605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000194135.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001307545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000218875.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000069025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001456065.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000091565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000093085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000695235.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000253055.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000143145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000340885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000816845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000612685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284095.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001543175.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000524555.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000412695.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001305655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000253295.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000153165.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000204415.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000242935.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000739455.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000747345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000494755.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000246985.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000260985.3\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000067205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000242455.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000723425.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000969965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001482385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025225.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000236685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001308105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000981545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001017435.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000203835.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000940995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000250635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000215085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000227685.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000517445.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000143845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001477625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001263395.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000255115.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195755.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000306885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000165485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196455.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001889105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001042405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000299235.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000300235.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000283655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000756615.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000523235.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000759475.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000223905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000213255.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001275345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001281465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000981765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000190555.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000237205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000224085.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000146185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000316685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000511405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000441555.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001310255.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000219045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001262055.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001767275.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001267155.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000789395.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001010285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001046955.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001023575.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001278035.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000186265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_900093775.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001597285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_002214645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001553955.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284515.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000953435.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284235.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000175215.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000183425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000309885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000812665.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001185205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_002073715.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000349945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015445.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001572725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001262035.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000344785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001314945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000186985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000144605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000648515.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000328725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000183405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000043285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000953015.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000330885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000189295.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001854245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196315.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000222485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000970025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000349845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001277995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000300295.4\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011965.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001021065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000026205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000027325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001078275.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001547995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000829415.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022605.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000754265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196395.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001709315.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000759535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000055785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000265295.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000828615.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000730245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000813245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000632805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001652485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000959245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000341385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000233435.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000027165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000306765.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000298115.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000632985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000972865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001499655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000828895.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000972765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000192865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000987835.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000348805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000332115.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000470655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000277795.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007905.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000269945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001021045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001314305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000091305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317975.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000204925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284375.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000831005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000328665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001702135.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000202635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196495.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001558255.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000739065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000217675.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010065.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000237145.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000300135.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000236665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000706685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000970045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000517605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000027225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001700965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001011155.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000762265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001187845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001050115.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195955.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001444365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000062885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000816345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000732945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000058485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000473245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000969645.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000968135.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013445.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000261025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000230895.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195775.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000217715.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001010805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000785105.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001027285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000193395.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000287355.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000219605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000235605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000208385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000959725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000300255.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284335.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000612485.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000243155.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000340825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317615.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000969885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000757785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000400955.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001042695.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000204255.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000972245.3\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001050435.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000213215.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000967305.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000219535.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000511305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000468615.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000590925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012905.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000294775.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021285.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000828675.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000212415.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001314975.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001442535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000332735.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000189775.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195295.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001586255.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000165465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000993805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001281045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000017045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000166395.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000191545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001483965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000250655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000189415.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000266885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001021025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000725405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000238995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000699505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001688625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000233715.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000757825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001693385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000982715.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000212675.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001578205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000006845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000968175.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000444995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001302585.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000801275.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001676785.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001941825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000737535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001702175.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000347635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001880325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000177615.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000145275.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000218565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000248095.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000742835.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000145615.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284255.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001412575.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000444875.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000525635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000214375.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000144405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317435.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000307165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001941945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001305615.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000091465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001006045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000305785.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001182745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001402875.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001886815.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019945.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001308145.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000209655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000214665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020385.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195935.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000147815.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000158275.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000953535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001902315.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000940805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000182745.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000024565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000217635.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001553935.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000009965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000281175.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000006685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018685.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001189535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000255535.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000230955.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000214215.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001443605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001482365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000830885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000265405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000953735.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_002075285.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001278055.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013185.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000317125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000060345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016325.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000164985.3\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000316625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000442645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000018405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000300455.3\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000164865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000010785.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019745.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007765.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000269985.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000069225.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000331735.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000800805.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000747315.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000217795.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_900011245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000513295.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000801315.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000503895.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001936235.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000814825.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000754275.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000013125.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000021725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000093165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000186245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000007045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000828975.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000963865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000209675.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001477655.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000212375.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015345.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001188915.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000772105.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000225465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001543265.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001483845.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001191005.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000196675.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001922025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016605.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000222305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000214355.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016545.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284115.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000284015.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000011465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000758725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000022045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000230275.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000980815.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000012765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000397205.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001583415.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000265425.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000328705.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000219725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195995.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001688625.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000019725.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000299115.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000023025.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000968055.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001548075.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000226565.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000092245.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000191045.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001719165.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000008765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000015885.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000732925.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000236405.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001028625.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195855.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001729525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000006865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000016525.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000325665.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000829195.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000418365.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001878675.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000195875.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000020465.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000723505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001483865.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000327505.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000785705.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001705175.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000350305.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000144915.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_001865855.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000014965.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000025645.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000224005.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000299335.2\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000969765.1\n",
      "WARNING:__main__:ADD RefSeqGenome: GCF_000211855.2\n"
     ]
    }
   ],
   "source": [
    "from multiprocessing import Pool\n",
    "\n",
    "to_load = set(genome_objects)\n",
    "loaded_genomes = list_genomes(annotation_api)\n",
    "to_load = to_load - set(loaded_genomes)\n",
    "\n",
    "def load_data(genome_id):\n",
    "    object_id = genome_objects[genome_id]\n",
    "    genome_data = read_json('../../../data/kbase_data/jplfaria_narrative_1524466549180/' + object_id +'.json')\n",
    "    genome = cobrakbase.core.KBaseGenome(genome_data)\n",
    "    a = init_annotation_api(host, port, user, pwd)\n",
    "    genome_node = a.add_kbase_genome(genome, genome_id, 'KBASE_RAST')\n",
    "    return True\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    p = Pool(processes=5)\n",
    "    inputs = list(range(20))\n",
    "    i = p.map_async(load_data, to_load, chunksize = 2)\n",
    "    i.wait()\n",
    "    print(i.get())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "threading"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 2\n",
      "5 8\n",
      "3 4\n",
      "4 6\n",
      "6 10\n",
      "7 12\n",
      "1 0\n",
      "10 18\n",
      "11 5\n",
      "9 16\n",
      "8 14\n",
      "14 9\n",
      "16 11\n",
      "12 7\n",
      "15 13\n",
      "13 1\n",
      "18 15\n",
      "17 3\n",
      "19 17\n",
      "20 19\n",
      "[0, 10, 20, 30, 40, 50, 60, 70, 80, 90, 100, 110, 120, 130, 140, 150, 160, 170, 180, 190]\n"
     ]
    }
   ],
   "source": [
    "from multiprocessing import Pool, Value\n",
    "from time import sleep\n",
    "\n",
    "counter = None\n",
    "\n",
    "def init(args):\n",
    "    ''' store the counter for later use '''\n",
    "    global counter\n",
    "    counter = args\n",
    "\n",
    "def analyze_data(args):\n",
    "    ''' increment the global counter, do something with the input '''\n",
    "    global counter\n",
    "    # += operation is not atomic, so we need to get a lock:\n",
    "    with counter.get_lock():\n",
    "        counter.value += 1\n",
    "    print(counter.value, args)\n",
    "    return args * 10\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    #inputs = os.listdir(some_directory)\n",
    "\n",
    "    #\n",
    "    # initialize a cross-process counter and the input lists\n",
    "    #\n",
    "    counter = Value('i', 0)\n",
    "    inputs = list(range(20))\n",
    "\n",
    "    #\n",
    "    # create the pool of workers, ensuring each one receives the counter \n",
    "    # as it starts. \n",
    "    #\n",
    "    p = Pool(initializer = init, initargs = (counter, ))\n",
    "    i = p.map_async(analyze_data, inputs, chunksize = 2)\n",
    "    i.wait()\n",
    "    print(i.get())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Worker 9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Process Process-7:\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Worker 9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Process Process-8:\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Worker 9\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/anaconda3/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "Traceback (most recent call last):\n",
      "Process Process-9:\n",
      "  File \"/anaconda3/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-7-b78d20e6a0da>\", line 13, in worker\n",
      "    running_jobs -= 1\n",
      "  File \"/anaconda3/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "Traceback (most recent call last):\n",
      "UnboundLocalError: local variable 'running_jobs' referenced before assignment\n",
      "  File \"/anaconda3/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"/anaconda3/lib/python3.6/multiprocessing/process.py\", line 258, in _bootstrap\n",
      "    self.run()\n",
      "  File \"<ipython-input-7-b78d20e6a0da>\", line 13, in worker\n",
      "    running_jobs -= 1\n",
      "UnboundLocalError: local variable 'running_jobs' referenced before assignment\n",
      "  File \"/anaconda3/lib/python3.6/multiprocessing/process.py\", line 93, in run\n",
      "    self._target(*self._args, **self._kwargs)\n",
      "  File \"<ipython-input-7-b78d20e6a0da>\", line 13, in worker\n",
      "    running_jobs -= 1\n",
      "UnboundLocalError: local variable 'running_jobs' referenced before assignment\n"
     ]
    },
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-7-b78d20e6a0da>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     20\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     21\u001b[0m     \u001b[0mlock\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mthreading\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mLock\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 22\u001b[0;31m     \u001b[0;32mwhile\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mwork\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m0\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     23\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mrunning_jobs\u001b[0m \u001b[0;34m<\u001b[0m \u001b[0;36m3\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     24\u001b[0m             \u001b[0mrunning_jobs\u001b[0m \u001b[0;34m+=\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "import multiprocessing\n",
    "#loaded_genomes = list_genomes(annotation_api)\n",
    "work = list(range(10))\n",
    "def worker():\n",
    "    \"\"\"worker function\"\"\"\n",
    "    lock.acquire() # will block if lock is already held\n",
    "    w = work.pop()\n",
    "    print('Worker', w)\n",
    "    lock.release()\n",
    "    \n",
    "    \n",
    "    lock.acquire() # will block if lock is already held\n",
    "    running_jobs -= 1\n",
    "    lock.release()\n",
    "    return\n",
    "\n",
    "running_jobs = 0\n",
    "if __name__ == '__main__':\n",
    "    jobs = []\n",
    "    \n",
    "    lock = threading.Lock()\n",
    "    while len(work) > 0:\n",
    "        if running_jobs < 3:\n",
    "            running_jobs += 1\n",
    "            p = multiprocessing.Process(target=worker)\n",
    "            p.start()\n",
    "        "
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load Updated RAST Genomes\n",
    "filipeliu:narrative_1569597697005 KBASE2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 427,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#genomes 1676\n"
     ]
    }
   ],
   "source": [
    "genome_objects = {}\n",
    "for o in kbase.list_objects('filipeliu:narrative_1569597697005'):\n",
    "    if o[1].endswith('RAST2') and o[2].startswith(\"KBaseGenomes.Genome\"):\n",
    "        genome_id = o[1][:-6]\n",
    "        genome_objects[genome_id] = o[1]\n",
    "        \n",
    "print('#genomes', len(genome_objects))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 428,
   "metadata": {},
   "outputs": [],
   "source": [
    "for genome_id in genome_objects:\n",
    "    object_id = genome_objects[genome_id]\n",
    "    data = kbase.get_object(object_id, 'filipeliu:narrative_1569597697005')\n",
    "    write_json(data, '../../../data/kbase_data/filipeliu_narrative_1569597697005/' + object_id + '.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'genetic_code': 11,\n",
       " 'notes': '',\n",
       " 'release': '2',\n",
       " 'created_at': 1563767040940,\n",
       " 'scientific_name': 'Escherichia coli str. K-12 substr. MG1655',\n",
       " 'source': 'refseq',\n",
       " 'taxonomy': 'cellular organisms; Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacterales; Enterobacteriaceae; Escherichia; Escherichia coli; Escherichia coli K-12',\n",
       " 'type': 'Reference',\n",
       " 'external_source_origination_date': '08-AUG-2016',\n",
       " 'count_features': 4498,\n",
       " 'workspace_id': 'jplfaria:narrative_1524466549180',\n",
       " 'dna_size': 4641652,\n",
       " 'original_source_file_name': 'GCF_000005845.2_ASM584v2_genomic.gbff',\n",
       " 'reference_annotation': 1,\n",
       " 'updated_at': 1563767040940,\n",
       " 'num_contigs': 1,\n",
       " 'assembly_ref': '19217/231473/1',\n",
       " 'domain': 'Bacteria',\n",
       " 'source_id': 'NC_000913',\n",
       " 'id': 'GCF_000005845.2',\n",
       " 'taxon_ref': '12570/625052/2',\n",
       " 'genbank_handle_ref': 'KBH_644247',\n",
       " 'gc_content': 0.5079070985933456,\n",
       " 'key': 'GCF_000005845.2'}"
      ]
     },
     "execution_count": 359,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "class Neo4jGenome():\n",
    "    \n",
    "    def __init__(self, node):\n",
    "        self.node = node\n",
    "        self.data = dict(self.node.items())\n",
    "        self.features = {}\n",
    "        self.features_annotation = {}\n",
    "        #print(node)\n",
    "        self.populate_from_graph()\n",
    "        \n",
    "    def populate_from_graph(self):\n",
    "        for rel in self.node.graph.match((self.node, ), r_type=\"has_gene\", ):\n",
    "            gene_id = rel.end_node['id']\n",
    "            self.features[gene_id] = rel.end_node\n",
    "            self.features_annotation[gene_id] = set()\n",
    "            for srel in self.node.graph.match((rel.end_node, ), r_type=\"has_annotation\", ):\n",
    "                annotation = Neo4jAnnotationFunction(srel.end_node)\n",
    "                self.features_annotation[gene_id].add(annotation)\n",
    "                #print(annotation)\n",
    "            \n",
    "        for rel in self.node.graph.match((self.node, ), r_type=\"search_function\", ):\n",
    "            self.search_value = rel.end_node['key']\n",
    "            for srel in self.node.graph.match((rel.end_node, ), r_type=\"has_function\", ):\n",
    "                n = srel.end_node\n",
    "                if not n['key'] == self.value:\n",
    "                    self.synonyms.add(n['key'])\n",
    "                    \n",
    "        for rel in self.node.graph.match((None, self.node), r_type=\"has_function\", ):\n",
    "            n = rel.start_node\n",
    "            if n.has_label('FunctionGroup'):\n",
    "                self.function_group.add(n['key'])\n",
    "\n",
    "    def __str__(self):\n",
    "        return self.value\n",
    "    \n",
    "node = matcher.match(\"RefSeqGenome\", key='GCF_000005845.2').first()\n",
    "neo_object = Neo4jGenome(node)\n",
    "neo_object.data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Two-component response regulator CreC {'KBASE_RAST'}\n",
      "Sensory histidine kinase CreC of two-component signal transduction system CreBC {'BETA_RAST'}\n"
     ]
    }
   ],
   "source": [
    "node = matcher.match(\"KBaseGene\", key='b4399@GCF_000005845.2').first()\n",
    "\n",
    "for rel in node.graph.match((node, ), r_type=\"has_annotation\", ):\n",
    "    annotation = Neo4jAnnotationFunction(rel.end_node)\n",
    "    annotation.neo4j_load_source()\n",
    "    print(annotation.value, annotation.source)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 352,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation.sub_functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_kegg_orthologs(genome):\n",
    "    genome.features_ko = {}\n",
    "    for gene_id in genome.features:\n",
    "        genome.features_ko[gene_id] = {}\n",
    "        #print(gene_id, genome.features[gene_id])\n",
    "        for rel in genome.node.graph.match((genome.features[gene_id], ), r_type=\"has_ortholog\", ):\n",
    "            ko_node = rel.end_node\n",
    "            genome.features_ko[gene_id][ko_node['key']] = {}\n",
    "            for srel in ko_node.graph.match((ko_node, ), r_type=\"has_reaction\", ):\n",
    "                kegg_rn_node = srel.end_node\n",
    "                genome.features_ko[gene_id][ko_node['key']][kegg_rn_node['key']] = {}\n",
    "                for seed_rel in ko_node.graph.match((None, kegg_rn_node), r_type=\"has_crossreference_to\", ):\n",
    "                    seed_node = seed_rel.start_node\n",
    "                    genome.features_ko[gene_id][ko_node['key']][kegg_rn_node['key']][seed_node['key']] = seed_node\n",
    "    pass\n",
    "load_kegg_orthologs(neo_object)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 380,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b4403 687\n",
      "\t K02533\n",
      "\t {'KBASE_RAST'} RNA methyltransferase, TrmH family, group 1 set()\n",
      "\t\t set()\n",
      "\t {'BETA_RAST'} Uncharacterized tRNA/rRNA methyltransferase LasT set()\n",
      "\t\t set()\n"
     ]
    }
   ],
   "source": [
    "for gene_id in neo_object.features:\n",
    "    gene_data = neo_object.features[gene_id]\n",
    "    print(gene_id, gene_data['dna_sequence_length'])\n",
    "    for ko in neo_object.features_ko[gene_id]:\n",
    "        print('\\t', ko)\n",
    "    for annotation in neo_object.features_annotation[gene_id]:\n",
    "        load_source(annotation)\n",
    "        if len(annotation.sub_functions) > 0:\n",
    "            print('\\t', annotation.source, annotation.value, annotation.synonyms)\n",
    "            for function in annotation.sub_functions:\n",
    "                print('\\t\\t', function.value, function.synonyms)\n",
    "                print('\\t\\t\\t', function.function_group)\n",
    "        else:\n",
    "            print('\\t', annotation.source, annotation.value, annotation.synonyms)\n",
    "            print('\\t\\t', annotation.function_group)\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 378,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'K02533': {}}"
      ]
     },
     "execution_count": 378,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
