{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cobrakbase 0.2.4\n"
     ]
    }
   ],
   "source": [
    "import logging\n",
    "import os\n",
    "import pymongo\n",
    "import cobrakbase\n",
    "import time\n",
    "\n",
    "logger = logging.getLogger(__name__)\n",
    "%run ../../scripts/bios_utils.py\n",
    "%run ../../annotation-server/annotation_api_neo4j.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../../annotation-server/annotation_api_neo4j.py\n",
    "\n",
    "host, port, user, pwd = (\"0.0.0.0\", 7687, \"neo4j\", \"123585\")\n",
    "#host, port, user, pwd = (\"sequoia.mcs.anl.gov\", 7687, \"neo4j\", \"SLEEP-CESS!\")\n",
    "#driver = GraphDatabase.driver(\"bolt://192.168.1.19:7687\", auth=(\"neo4j\", \"123585\"))\n",
    "#driver = GraphDatabase.driver(\"bolt://0.0.0.0:7687\", auth=(\"neo4j\", \"123585\"))\n",
    "#driver = GraphDatabase.driver(\"bolt://sequoia.mcs.anl.gov:7687\", auth=(\"neo4j\", \"SLEEP-CESS!\"))\n",
    "#annotation_api = AnnotationApiNeo4j(driver)\n",
    "#neo_annotation_api.init_constraints()\n",
    "\n",
    "#annotation_api.neo4j_graph = Graph(\"http://neo4j:SLEEP-CESS!@sequoia.mcs.anl.gov:7474\")\n",
    "def init_annotation_api(host, port, user, pwd):\n",
    "    annotation_api = AnnotationApiNeo4j(user=user, pwd=pwd, port=port, host=host)\n",
    "    annotation_api.neo4j_graph = Graph(\"http://neo4j:\" + pwd + \"@\" + host + \":7474\")\n",
    "    annotation_api.matcher = NodeMatcher(annotation_api.neo4j_graph)\n",
    "    annotation_api.r_matcher = RelationshipMatcher(annotation_api.neo4j_graph)\n",
    "    annotation_api.init_constraints()\n",
    "    return annotation_api\n",
    "\n",
    "annotation_api = init_annotation_api(host, port, user, pwd)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "def list_genomes(annotation_api):\n",
    "    l = {}\n",
    "    res = annotation_api.page_nodes('RefSeqGenome', 0, 10)\n",
    "    page = 0\n",
    "    while not res == None:\n",
    "        for r in res:\n",
    "            l[r['n']['key']] = r['n']['scientific_name']\n",
    "        page += 1\n",
    "        res = annotation_api.page_nodes('RefSeqGenome', page, 10)\n",
    "    return l\n",
    "loaded_genomes = list_genomes(annotation_api)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "kbase = cobrakbase.KBaseAPI(token='4VW3677NZK7HX7Q3AHELGVVI55TMVDC5')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "load: ../../../ModelSEEDDatabase/Biochemistry/reactions.tsv\n",
      "load: ../../../ModelSEEDDatabase/Biochemistry/compounds.tsv\n",
      "load: ../../../ModelSEEDDatabase/Biochemistry/Structures/Unique_ModelSEED_Structures.txt\n",
      "load: ../../../ModelSEEDDatabase/Biochemistry/Aliases/Unique_ModelSEED_Compound_Aliases.txt\n",
      "load: ../../../ModelSEEDDatabase/Biochemistry/Aliases/Unique_ModelSEED_Reaction_Aliases.txt\n",
      "load: ../../../ModelSEEDDatabase/Biochemistry/Aliases/Unique_ModelSEED_Reaction_ECs.txt\n"
     ]
    }
   ],
   "source": [
    "modelseed_local = cobrakbase.modelseed.from_local('../../../ModelSEEDDatabase')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "substem_ref = read_json('../../../data/tnseq/patric_subsystems/substem_ref.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 2min 18s, sys: 11.3 s, total: 2min 29s\n",
      "Wall time: 14min 15s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "for doc in substem_ref['response']['docs']:\n",
    "    value = doc['subsystem_name']\n",
    "    \n",
    "    subclass = None if not 'subclass' in doc else doc['subclass']\n",
    "    \n",
    "    ss_node = annotation_api.add_node('FunctionGroup', value)\n",
    "    \n",
    "    if 'role_name' in doc:\n",
    "        for role in doc['role_name']:\n",
    "            role = role.strip()\n",
    "            if len(role) > 0:\n",
    "                function_nodes = annotation_api.add_annotation([role], 'PATRIC')\n",
    "                for function_node in function_nodes:\n",
    "                    annotation_api.link_nodes(ss_node, function_node, 'has_function')\n",
    "                    annotation_api.link_nodes(function_node, ss_node, 'has_function_group')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Node id=57 labels={'RefSeqGenome'} properties={'genetic_code': 11, 'notes': '', 'release': '1', 'created_at': 1563767041245, 'scientific_name': 'Rickettsia conorii str. Malish 7', 'source': 'refseq', 'taxonomy': 'cellular organisms; Bacteria; Proteobacteria; Alphaproteobacteria; Rickettsiales; Rickettsiaceae; Rickettsieae; Rickettsia; spotted fever group; Rickettsia conorii', 'type': 'Representative', 'external_source_origination_date': '16-NOV-2016', 'count_features': 1532, 'workspace_id': 'jplfaria:narrative_1524466549180', 'dna_size': 1268755, 'original_source_file_name': 'GCF_000007025.1_ASM702v1_genomic.gbff', 'reference_annotation': 0, 'updated_at': 1563767041245, 'num_contigs': 1, 'assembly_ref': '19217/46804/1', 'domain': 'Bacteria', 'source_id': 'NC_003103', 'id': 'GCF_000007025.1', 'taxon_ref': '12570/666753/1', 'genbank_handle_ref': 'KBH_235629', 'gc_content': 0.3243880812292365, 'key': 'GCF_000007025.1'}>\n"
     ]
    }
   ],
   "source": [
    "node = annotation_api.get_node('RefSeqGenome', 'GCF_000007025.1')\n",
    "print(node)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "for doc in substem_ref['response']['docs']:\n",
    "    value = doc['subsystem_name']\n",
    "    subclass = None if not 'subclass' in doc else doc['subclass']\n",
    "    #print(value, '#######',subclass)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'{name:\"a \\\\\"banana\\\\\"\", key:\"banana\"}'"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "props_str = to_str_dict(props)\n",
    "props_str"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 169,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "props = {\n",
    "    'desc' : 'what is this?',\n",
    "    'bzzzz' : 'yes'\n",
    "}\n",
    "annotation_api.update_node('Fruit', 'Fruztz', props)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'id': 'rxn10881',\n",
       " 'name': 'streptomycin 3\"-adenylyltransferase',\n",
       " 'equation': '(1) cpd00002[0] + (1) cpd00067[0] + (1) cpd00328[0] <= (1) cpd00012[0] + (1) cpd16061[0]',\n",
       " 'definition': '(1) ATP[0] + (1) H+[0] + (1) Streptomycin[0] <= (1) PPi[0] + (1) 3\"-Adenylylstreptomycin[0]',\n",
       " 'key': 'rxn10881'}"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "props"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "1"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load ModelSEED Reactions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 31s, sys: 7.62 s, total: 1min 39s\n",
      "Wall time: 11min 8s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "for seed_id in modelseed_local.reactions:\n",
    "    rxn = modelseed_local.get_seed_reaction(seed_id)\n",
    "    rxn_node = annotation_api.get_node('ModelSeedReaction', seed_id)\n",
    "    if rxn_node == None:\n",
    "        delete = set()\n",
    "        for k in rxn.data:\n",
    "            if pd.isna(rxn.data[k]):\n",
    "                delete.add(k)\n",
    "        for k in delete:\n",
    "            del rxn.data[k]\n",
    "        annotation_api.add_node('ModelSeedReaction', seed_id, rxn.data)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load KEGG Orthologs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 50s, sys: 8.79 s, total: 1min 59s\n",
      "Wall time: 17min 31s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "df_kegg_ko = pd.read_csv('../../data/ko_kegg.tsv', sep='\\t')\n",
    "for row_id, d in df_kegg_ko.iterrows():\n",
    "    kegg_ko = d['kegg_ko']\n",
    "    kegg_rxns = d['kegg_rxns']\n",
    "    \n",
    "    ko_node = annotation_api.get_node('KeggOrthology', kegg_ko)\n",
    "    if ko_node == None:\n",
    "        annotation_api.add_node('KeggOrthology', kegg_ko, {})\n",
    "        ko_node = annotation_api.get_node('KeggOrthology', kegg_ko)\n",
    "        \n",
    "    if not pd.isna(kegg_rxns):\n",
    "        for rxn_id in kegg_rxns.split(';'):\n",
    "            rxn_node = annotation_api.get_node('LigandReaction', seed_id)\n",
    "            if rxn_node == None:\n",
    "                annotation_api.add_node('LigandReaction', rxn_id, {})\n",
    "                rxn_node = annotation_api.get_node('LigandReaction', rxn_id)\n",
    "            annotation_api.link_nodes(ko_node, rxn_node, 'has_reaction')\n",
    "            annotation_api.link_nodes(rxn_node, ko_node, 'has_ortholog')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 18min 42s, sys: 1min 33s, total: 20min 16s\n",
      "Wall time: 2h 4min 11s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "ko_kegg_genes = read_json('../../data/ko_kegg_genes.json')\n",
    "for ko in ko_kegg_genes:\n",
    "    ko_node = annotation_api.get_node('KeggOrthology', ko)\n",
    "    if not ko_node == None:\n",
    "        for genome_id, gene_id in ko_kegg_genes[ko]:\n",
    "            gene_node = annotation_api.get_node('KBaseGene', \"{}@{}\".format(gene_id, genome_id))\n",
    "            if gene_node == None:\n",
    "                gene_node = annotation_api.add_node('KBaseGene', \"{}@{}\".format(gene_id, genome_id))\n",
    "                \n",
    "            annotation_api.link_nodes(gene_node, ko_node, 'has_ortholog')\n",
    "            annotation_api.link_nodes(ko_node, gene_node, 'has_gene')\n",
    "    else:\n",
    "        logger.debug('not found: %s', ko)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load KEGG Orthologs to SEED"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 48.4 s, sys: 3.66 s, total: 52 s\n",
      "Wall time: 2min 18s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "for seed_id in modelseed_local.reactions:\n",
    "    rxn = modelseed_local.get_seed_reaction(seed_id)\n",
    "    seed_rxn_node = annotation_api.get_node('ModelSeedReaction', seed_id)\n",
    "    if seed_id in modelseed_local.reaction_aliases and 'KEGG' in modelseed_local.reaction_aliases[seed_id]:\n",
    "        for kegg_id in modelseed_local.reaction_aliases[seed_id]['KEGG']:\n",
    "            kegg_rxn_node = annotation_api.get_node('LigandReaction', kegg_id)\n",
    "            if not kegg_rxn_node == None and not seed_rxn_node == None:\n",
    "                annotation_api.link_nodes(seed_rxn_node, kegg_rxn_node, 'has_crossreference_to')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Load Templates"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "templates = {}\n",
    "templates['GramNegModelTemplateV2'] = kbase.get_object('GramNegModelTemplateV2', 'NewKBaseModelTemplates')\n",
    "templates['GramNegModelTemplate'] = kbase.get_object('GramNegModelTemplate', 'NewKBaseModelTemplates')\n",
    "#template_doc = annotation_api.add_template('GramNegModelTemplateV2', templates['GramNegModelTemplateV2'])\n",
    "#template_doc = annotation_api.add_template('GramNegModelTemplate', templates['GramNegModelTemplate'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "#list(filter(lambda x : 'rxn05528' in x['id'] , templates['GramNegModelTemplate']['reactions']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 86,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:__main__:19936 0\n",
      "WARNING:__main__:cpx00250@GramNegModelTemplate\n",
      "WARNING:__main__:cpx01214@GramNegModelTemplate\n",
      "WARNING:__main__:cpx01497@GramNegModelTemplate\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 25s, sys: 7.22 s, total: 1min 32s\n",
      "Wall time: 8min 14s\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Node id=5561658 labels={'TemplateSet'} properties={'updated_at': 1570538930130, 'created_at': 1570538930130, 'key': 'GramNegModelTemplate'}>"
      ]
     },
     "execution_count": 86,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "annotation_api.add_template('GramNegModelTemplate', templates['GramNegModelTemplate'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 87,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:__main__:19957 0\n",
      "WARNING:__main__:cpx00250@GramNegModelTemplateV2\n",
      "WARNING:__main__:cpx01214@GramNegModelTemplateV2\n",
      "WARNING:__main__:cpx01497@GramNegModelTemplateV2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 1min 24s, sys: 7.51 s, total: 1min 32s\n",
      "Wall time: 8min 24s\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "<Node id=5570337 labels={'TemplateSet'} properties={'updated_at': 1570539971948, 'created_at': 1570539971948, 'key': 'GramNegModelTemplateV2'}>"
      ]
     },
     "execution_count": 87,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "%%time\n",
    "annotation_api.add_template('GramNegModelTemplateV2', templates['GramNegModelTemplateV2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 73,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "cpx01799@GramNegModelTemplate Succinate dehydrogenase cytochrome b558 subunit\n",
      "cpx01799@GramNegModelTemplate Succinate dehydrogenase iron-sulfur protein (EC 1.3.99.1)\n",
      "cpx01799@GramNegModelTemplate Succinate dehydrogenase flavoprotein subunit (EC 1.3.99.1)\n"
     ]
    }
   ],
   "source": [
    "for cpd_id in cpx_to_function_uids:\n",
    "    if len(cpx_to_function_uids[cpd_id]) > 1:\n",
    "        for o in cpx_to_function_uids[cpd_id]:\n",
    "            print(cpd_id, o['key'])\n",
    "        break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 74,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:__main__:19936 0\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'self' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-38-3ba29111ca2f>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     98\u001b[0m                                                                 rxn_compartments, or_rule, props)\n\u001b[1;32m     99\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 100\u001b[0;31m \u001b[0madd_template\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mannotation_api\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'GramNegModelTemplate'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtemplates\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'GramNegModelTemplate'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m<ipython-input-38-3ba29111ca2f>\u001b[0m in \u001b[0;36madd_template\u001b[0;34m(annotation_api, template_id, template)\u001b[0m\n\u001b[1;32m     47\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mcpx_id\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mcpx_to_function_uids\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     48\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcpx_to_function_uids\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mcpx_id\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 49\u001b[0;31m                 \u001b[0mcomplex_node\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_node\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'FunctionComplex'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcpx_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     50\u001b[0m                 \u001b[0;32mif\u001b[0m \u001b[0mcomplex_node\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     51\u001b[0m                     \u001b[0mcomplex_node\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0madd_node\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'FunctionComplex'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcpx_id\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'self' is not defined"
     ]
    }
   ],
   "source": [
    "reactions = {}\n",
    "for rxn in template['reactions']:\n",
    "    rxn_id = rxn['id']\n",
    "    if '_' in rxn_id:\n",
    "        rxn_id = rxn_id.split('_')[0]\n",
    "    rxn_id += '@' + template_id\n",
    "    rxn_node = annotation_api.get_node('ModelSeedReaction', rxn_id)\n",
    "    if not rxn_node == None:\n",
    "        #print(rxn_node)\n",
    "    #rxn_id = rxn['reaction_ref'].split('/')[-1]\n",
    "        rxn_compartments = set()\n",
    "        for templateReactionReagent in rxn['templateReactionReagents']:\n",
    "            templatecompcompound_ref = templateReactionReagent['templatecompcompound_ref'].split('/')[-1]\n",
    "            rxn_compartments.add(compcompound_compartment[templatecompcompound_ref])\n",
    "\n",
    "\n",
    "\n",
    "        or_rule = set()\n",
    "        for complex_ref in rxn['templatecomplex_refs']:\n",
    "            complex_id = complex_ref.split('/')[-1]\n",
    "            if complex_id in cpx_to_function_uids:\n",
    "                if len(cpx_to_function_uids[complex_id]) > 1:\n",
    "                    or_rule.add(function_complexes[complex_id])\n",
    "                elif len(cpx_to_function_uids[complex_id]) == 1:\n",
    "                    or_rule.add(list(cpx_to_function_uids[complex_id])[0])\n",
    "                else:\n",
    "                    logger.warning('%s', complex_id)\n",
    "\n",
    "        #print(rxn['id'], rxn_compartments, or_rule)\n",
    "\n",
    "        props = {\n",
    "            'compartment' : ';'.join(sorted(list(rxn_compartments))),\n",
    "            'reaction' : rxn_node.get('key'),\n",
    "            'base_cost' : rxn['base_cost'],\n",
    "            'direction' : rxn['direction'],\n",
    "            'forward_penalty' : rxn['forward_penalty'],\n",
    "            'reverse_penalty' : rxn['reverse_penalty'],\n",
    "            'type' : rxn['type'],\n",
    "            'GapfillDirection' : rxn['GapfillDirection'],\n",
    "        }\n",
    "\n",
    "        annotation_api.add_template_reaction_annotation(template_node, rxn_node, \n",
    "                                                        rxn_compartments, or_rule, props)\n",
    "                \n",
    "add_template(annotation_api, 'GramNegModelTemplate', templates['GramNegModelTemplate'])"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "def add_template(self, template_id, template):\n",
    "        roles = {}\n",
    "        function_uids = {}\n",
    "        compcompound_compartment = {}\n",
    "\n",
    "        for compcompound in template['compcompounds']:\n",
    "            compcompound_compartment[compcompound['id']] = compcompound['templatecompartment_ref'].split('/')[-1]\n",
    "\n",
    "        for role in template['roles']:\n",
    "            roles[role['id']] = role['name']\n",
    "        for role_id in roles:\n",
    "            function = roles[role_id]\n",
    "            #print(function)\n",
    "            function_doc = self.get_function(function)\n",
    "            if function_doc['value'] == function:\n",
    "                function_uids[function] = function_doc['_id']\n",
    "            else:\n",
    "                print('errooo!')\n",
    "        missing_function = set()\n",
    "        for role_id in roles:\n",
    "            function = roles[role_id]\n",
    "            if not function in function_uids:\n",
    "                missing_function.add(function)\n",
    "\n",
    "        print(len(missing_function))\n",
    "\n",
    "        cpx_to_function_uids = {}\n",
    "        for cpx in template['complexes']:\n",
    "            uids = set()\n",
    "            for complexrole in cpx['complexroles']:\n",
    "                role_id = complexrole['templaterole_ref'].split('/')[-1]\n",
    "                if role_id in roles:\n",
    "                    uids.add(function_uids[roles[role_id]])\n",
    "                else:\n",
    "                    print('errooo!')\n",
    "\n",
    "            cpx_to_function_uids[cpx['id']] = uids\n",
    "\n",
    "        reactions = {}\n",
    "        for rxn in template['reactions']:\n",
    "            rxn_id = rxn['reaction_ref'].split('/')[-1]\n",
    "            rxn_compartments = set()\n",
    "            for templateReactionReagent in rxn['templateReactionReagents']:\n",
    "                templatecompcompound_ref = templateReactionReagent['templatecompcompound_ref'].split('/')[-1]\n",
    "                rxn_compartments.add(compcompound_compartment[templatecompcompound_ref])\n",
    "            or_rule = set()\n",
    "            for complex_ref in rxn['templatecomplex_refs']:\n",
    "                complex_id = complex_ref.split('/')[-1]\n",
    "                and_rule = set()\n",
    "                if complex_id in cpx_to_function_uids:\n",
    "                    and_rule = frozenset(cpx_to_function_uids[complex_id])\n",
    "                or_rule.add(and_rule)\n",
    "\n",
    "            reactions[rxn_id] = {\n",
    "                'functions' : list(map(lambda x : list(x), or_rule)),\n",
    "                'compartment' : list(rxn_compartments),\n",
    "                'base_cost' : rxn['base_cost'],\n",
    "                'direction' : rxn['direction'],\n",
    "                'forward_penalty' : rxn['forward_penalty'],\n",
    "                'reverse_penalty' : rxn['reverse_penalty'],\n",
    "                'type' : rxn['type'],\n",
    "                'GapfillDirection' : rxn['GapfillDirection'],\n",
    "\n",
    "            }\n",
    "\n",
    "        template_doc = {\n",
    "            '_id' : template_id,\n",
    "            'reactions' : reactions\n",
    "        }\n",
    "\n",
    "        self.collection_templates.insert_one(template_doc)\n",
    "\n",
    "        return template_doc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Node id=185606 labels={'TemplateReactionAnnotation'} properties={'reaction': 'rxn00879', 'updated_at': 1568635675119, 'forward_penalty': 0, 'compartment': 'c', 'base_cost': 2, 'reverse_penalty': 0, 'created_at': 1568635675119, 'GapfillDirection': '=', 'type': 'conditional', 'key': 'rxn09225_c', 'direction': '='}>"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "add_template_reaction_annotation(annotation_api, template_node, rxn_node, rxn_compartments, or_rule, props)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<Node id=87034 labels={'ModelSeedReaction'} properties={'updated_at': 1568609814074, 'equation': '(1) cpd02570[0] <=> (2) cpd00020[0]', 'name': '4-hydroxy-4-methyl-2-oxoglutarate pyruvate-lyase (pyruvate-forming)', 'created_at': 1568609814074, 'definition': '(1) Parapyruvate[0] <=> (2) Pyruvate[0]', 'id': 'rxn00004', 'key': 'rxn00004'}>\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 152,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_ko_by_seed_id(annotation_api, seed_id):\n",
    "    rxn_doc = annotation_api.get_node('ModelSeedReaction', seed_id)\n",
    "    print(rxn_doc)\n",
    "    if not rxn_doc == None:\n",
    "        for r in rxn_doc.graph.relationships:\n",
    "            print(r)\n",
    "        if 'kegg_kos' in rxn_doc:\n",
    "            return set(rxn_doc['kegg_kos'])\n",
    "        else:\n",
    "            return set()\n",
    "    return None\n",
    "get_ko_by_seed_id(annotation_api, 'rxn00004')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_reaction_annotation_data(rxn_id):\n",
    "    kos = annotation_api.get_ko_by_seed_id(rxn_id)\n",
    "    function_data = {}\n",
    "    for ko in kos:\n",
    "        #print(ko)\n",
    "        functions, functions_data, metadata = get_functional_roles2(ko, annotation_api)\n",
    "        print(ko, metadata)\n",
    "        for f in functions:\n",
    "            function = functions_data[f]['value']\n",
    "            if not function in function_data:\n",
    "                function_data[function] = {\n",
    "                    'id' : f,\n",
    "                    'hits' : []\n",
    "                }\n",
    "            function_data[function]['hits'].append({\n",
    "                'score' : len(functions[f]),\n",
    "                'source' : ['KEGG', ko]\n",
    "            })\n",
    "    for template_doc in annotation_api.collection_templates.find():\n",
    "        if rxn_id in template_doc['reactions']:\n",
    "            template_id = template_doc['_id']\n",
    "            template_reaction_doc = template_doc['reactions'][rxn_id]\n",
    "            for and_rule in template_reaction_doc['functions']:\n",
    "                for function_id in and_rule:\n",
    "                    function_doc = annotation_api.collection_functions.find_one({'_id' : function_id})\n",
    "                    function = function_doc['value']\n",
    "                    if not function in function_data:\n",
    "                        function_data[function] = {\n",
    "                            'id' : function_id,\n",
    "                            'hits' : []\n",
    "                        }\n",
    "                    function_data[function]['hits'].append({\n",
    "                        'score' : 0,\n",
    "                        'source' : ['template', template_id]\n",
    "                    })\n",
    "                    #print(template_id, function_doc['value'])\n",
    "        \n",
    "    for function in function_data:\n",
    "        function_id = function_data[function]['id']\n",
    "        #print(function_id)\n",
    "        function_doc = annotation_api.collection_functions.find_one({'_id' : function_id})\n",
    "        function_metadata = annotation_api.collection_functions_data.find_one({'_id' : function_id})\n",
    "        \n",
    "        subsystem_tags = {}\n",
    "        if 'subsystem' in function_doc:\n",
    "            for subsystem_tag in function_doc['subsystem']:\n",
    "                subsystem_tags[subsystem_tag] = [function_doc['subsystem'][subsystem_tag]]\n",
    "                if 'subsystem_class' in function_doc and subsystem_tag in function_doc['subsystem_class']:\n",
    "                    subsystem_tags[subsystem_tag].append(function_doc['subsystem_class'][subsystem_tag])\n",
    "        #print(subsystem_tags)\n",
    "        \n",
    "        \n",
    "        source_tags = {}\n",
    "        if not function_metadata == None:\n",
    "            sources = function_metadata['sources']\n",
    "            for s in sources:\n",
    "                source_tags[s] = [len(sources[s]['genomes']), len(sources[s]['genes']), sources[s]['genes'][:10]]\n",
    "            #print(source_tags)\n",
    "        function_data[function]['sources'] = source_tags\n",
    "        function_data[function]['subsystems'] = subsystem_tags\n",
    "    return function_data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "rxn_record = annotation_api.get_node('ModelSeedReaction', 'rxn00148')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 101,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "(<Node id=87270 labels=set() properties={}>, <Node id=133756 labels=set() properties={}>)\n"
     ]
    }
   ],
   "source": [
    "def get_relationship(tx, key, label, target_label, relationship_type):\n",
    "    if not target_label == None:\n",
    "        target_label = \":\" + target_label\n",
    "    else:\n",
    "        target_label = \"\"\n",
    "    if not label == None:\n",
    "        label = \":\" + label\n",
    "    else:\n",
    "        label = \"\"\n",
    "    if not relationship_type == None:\n",
    "        relationship_type = \":\" + relationship_type\n",
    "    else:\n",
    "        relationship_type = \"\"\n",
    "    cypher_query = \"MATCH (n1\" + label + \" {key:$key})-[r\" + relationship_type + \"]->(n2\" + target_label + \") RETURN r as relationships\"\n",
    "    logger.debug('Query: %s', cypher_query)\n",
    "    result = tx.run(cypher_query, key = key)\n",
    "    return list(result.records())\n",
    "\n",
    "#rxn_doc = annotation_api.get_node('ModelSeedReaction', seed_id)\n",
    "def get_r_nodes(annotation_api, key, label, target_label=None, relationship_type = None):\n",
    "    result = None\n",
    "    with annotation_api.driver.session() as session:\n",
    "        query = session.read_transaction(get_relationship, key, label, target_label, relationship_type)\n",
    "    return query\n",
    "\n",
    "results = get_r_nodes(annotation_api, 'rxn00148', 'ModelSeedReaction')\n",
    "node_ids = set()\n",
    "for result in results:\n",
    "    r = result['relationships']\n",
    "    #r = r.data().get('r')\n",
    "    node_pair = (r.start_node, r.end_node)\n",
    "    node_ids.add(r.end_node.id)\n",
    "    print(node_pair)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "\"MATCH (n1:LigandReaction)-[r2]-(n2:KeggOrthology) WHERE id(n1)= 133756 RETURN n2\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{133756}"
      ]
     },
     "execution_count": 102,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "node_ids"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "99759"
      ]
     },
     "execution_count": 93,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 169,
   "metadata": {},
   "outputs": [],
   "source": [
    "kos = annotation_api.get_ko_by_seed_id('rxn00148')\n",
    "function_data = {}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<Node id=87270 labels={'ModelSeedReaction'} properties={'updated_at': 1568609816198, 'equation': '(1) cpd00002[0] + (1) cpd00020[0] <=> (1) cpd00008[0] + (1) cpd00061[0] + (1) cpd00067[0]', 'name': 'ATP:pyruvate 2-O-phosphotransferase', 'created_at': 1568609816198, 'definition': '(1) ATP[0] + (1) Pyruvate[0] <=> (1) ADP[0] + (1) Phosphoenolpyruvate[0] + (1) H+[0]', 'id': 'rxn00148', 'key': 'rxn00148'}>"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "rxn_record."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "a = Node(\"Person\", name=\"Alice\")\n",
    "b = Node(\"Person\", name=\"Bob\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 105,
   "metadata": {},
   "outputs": [],
   "source": [
    "ab = Relationship(a, \"KNOWS\", b)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "()"
      ]
     },
     "execution_count": 106,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "a.relationships"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 160,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 160,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "m = neo4j_graph.match((Node(\"ModelSeedReaction\", key=\"rxn00148\"), ))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 131,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(_133756:LigandReaction {created_at: 1568617528505, key: 'R00200', updated_at: 1568618373969})"
      ]
     },
     "execution_count": 131,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "node = neo4j_graph.nodes[133756]\n",
    "#node.__primarykey__ = \"key\"\n",
    "node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 290,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{<__main__.Neo4jAnnotationFunction at 0x7fb0c3d00400>,\n",
       " <__main__.Neo4jAnnotationFunction at 0x7fb0a7043160>}"
      ]
     },
     "execution_count": 290,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "        \n",
    "node = matcher.match('Function', key='Aspartokinase (EC 2.7.2.4) / Homoserine dehydrogenase (EC 1.1.1.3)').first()\n",
    "node\n",
    "function = Neo4jAnnotationFunction(node)\n",
    "function.sub_functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "True"
      ]
     },
     "execution_count": 273,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "node.has_label('Function')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 276,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Aspartokinase (EC 2.7.2.4) / Homoserine dehydrogenase (EC 1.1.1.3) Aspartokinase (EC 2.7.2.4) / Homoserine dehydrogenase (EC 1.1.1.3) set()\n",
      "Homoserine dehydrogenase (EC 1.1.1.3) homoserinedehydrogenaseec1113 {'Sulfur metabolism related cluster', 'Aspartate to Threonine Module', 'Aspartate to Homoserine module'}\n",
      "Aspartokinase (EC 2.7.2.4) aspartokinaseec2724 {'Aspartate to Threonine Module', 'Peptidase clustering with DAP', 'Lysine DAP biosynthetic pathway', 'Diaminopimelate Synthesis ', 'Aspartate to Homoserine module', 'Sulfur metabolism related cluster'}\n"
     ]
    }
   ],
   "source": [
    "print(function, function.search_value, function.function_group)\n",
    "for f in function.sub_functions:\n",
    "    print(f, f.search_value, f.function_group)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 225,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{(_74620:Function {created_at: 1563774166329, key: 'Aspartokinase (EC 2.7.2.4) / Homoserine dehydrogenase (EC 1.1.1.3)', updated_at: 1563778320381}): {(_75944:Function {created_at: 1563774195041, key: 'Aspartokinase (EC 2.7.2.4)', updated_at: 1563778378559}),\n",
       "  (_83913:Function {created_at: 1563774552943, key: 'Homoserine dehydrogenase (EC 1.1.1.3)', updated_at: 1563774578474})}}"
      ]
     },
     "execution_count": 225,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def get_gene_functions(gene_id):\n",
    "    functions = {}\n",
    "    matcher = NodeMatcher(annotation_api.neo4j_graph)\n",
    "    node = matcher.match('KBaseGene', key=gene_id).first()\n",
    "    if not node == None:\n",
    "        for rel in annotation_api.neo4j_graph.match((node, ), r_type=\"has_annotation\", ):\n",
    "            base_annotation = rel.end_node\n",
    "            functions[base_annotation] = set()\n",
    "            for sub_rel in annotation_api.neo4j_graph.match((base_annotation, ), r_type=\"has_subfunction\", ):\n",
    "                sub_function = sub_rel.end_node\n",
    "                functions[base_annotation].add(sub_function)\n",
    "                \n",
    "    return functions\n",
    "get_gene_functions('b3940@GCF_000005845.2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 284,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "({74620: {'b3940@GCF_000005845.2'},\n",
       "  75944: {'b3940@GCF_000005845.2'},\n",
       "  83913: {'b3940@GCF_000005845.2'}},\n",
       " {74620: {'value': 'Aspartokinase (EC 2.7.2.4) / Homoserine dehydrogenase (EC 1.1.1.3)'},\n",
       "  75944: {'value': 'Aspartokinase (EC 2.7.2.4)'},\n",
       "  83913: {'value': 'Homoserine dehydrogenase (EC 1.1.1.3)'}},\n",
       " {'total': 20, 'has_function': 1})"
      ]
     },
     "execution_count": 284,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'DVU2514@GCF_000195755.1'"
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "216826"
      ]
     },
     "execution_count": 49,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_node.identity"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "7194380194339715759"
      ]
     },
     "execution_count": 41,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "hash(gene_node)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'HI1573'"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gene_node['id']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "kbase = cobrakbase.KBaseAPI(token='UGOG6KLAWTCYI2ASYECYHNIIFTEXGA2J')\n",
    "genome1_rast   = kbase.get_object('GCF_000005845.2.RAST', 'jplfaria:narrative_1524466549180')\n",
    "genome2_script = kbase.get_object('GCF_000005845.2.RAST2', 'filipeliu:narrative_1562086644968')\n",
    "genome2_app    = kbase.get_object('test_nokmers1', 'jplfaria:narrative_1561579862828')\n",
    "genomes = {\n",
    "    'rast1' : cobrakbase.core.KBaseGenome(genome1_rast),\n",
    "    'rast2_script' : cobrakbase.core.KBaseGenome(genome2_script),\n",
    "    'rast2_app' : cobrakbase.core.KBaseGenome(genome2_app),\n",
    "}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "{'aliases': ['GeneID:944742', 'thrL', 'b0001', 'NP_414542.1', 'UniProtKB/Swiss-Prot:P0AD86', 'ASAP:ABE-0000006', 'JW4367', 'GI:16127995', 'ECK0001', 'EcoGene:EG11277'], 'cdss': ['NP_414542.1'], 'dna_sequence': 'ATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGA', 'dna_sequence_length': 66, 'function': 'Thr operon leader peptide', 'id': 'b0001', 'location': [['NC_000913', 190, '+', 66]], 'md5': '13fe4258b37e32edda386faa50ec5fdf', 'mrnas': ['mRNA_1'], 'protein_translation': 'MKRISTTITTTITITTGNGAG', 'protein_translation_length': 21, 'quality': {'hit_count': 7, 'weighted_hit_count': 9.1488}, 'type': 'gene'}\n"
     ]
    }
   ],
   "source": [
    "genome = genomes['rast1']\n",
    "for f in genome.features:\n",
    "    print(f)\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "Match (p: Person)\n",
    "WHERE p.id > 0\n",
    "Return p\n",
    "ORDER BY p.id\n",
    "Skip 0 LIMIT 30"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 112,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MATCH (n:RefSeqGenome) RETURN n ORDER BY n.key SKIP $skip LIMIT $page_size\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "dict_keys(['GCF_000005825.2', 'GCF_000005845.2', 'GCF_000006175.1', 'GCF_000006605.1', 'GCF_000006685.1', 'GCF_000006725.1', 'GCF_000006745.1', 'GCF_000006765.1', 'GCF_000006785.2', 'GCF_000006845.1'])"
      ]
     },
     "execution_count": 115,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "res = get_page_nodes(annotation_api, 'RefSeqGenome', 0, 10)\n",
    "os = {}\n",
    "for o in res:\n",
    "    n = o.data()['n']\n",
    "    os[n.get('key')] = dict(n.items())\n",
    "os.keys()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MATCH (n:RefSeqGenome) WHERE n.scientific_name =~ '.*coli.*' RETURN n ORDER BY n.key SKIP $skip LIMIT $page_size\n",
      "Escherichia coli str. K-12 substr. MG1655\n",
      "Escherichia coli O157:H7 str. Sakai\n",
      "Yersinia enterocolitica subsp. enterocolitica 8081\n",
      "Escherichia coli UMN026\n",
      "Escherichia coli IAI39\n",
      "Escherichia coli O83:H1 str. NRG 857C\n",
      "Syntrophobotulus glycolicus DSM 8271\n",
      "Yersinia enterocolitica subsp. palearctica Y11\n",
      "Escherichia coli O104:H4 str. 2011C-3493\n",
      "Brachyspira pilosicoli P43/6/78\n",
      "Campylobacter coli\n"
     ]
    }
   ],
   "source": [
    "def search_genomes(search_str, limit = None):\n",
    "    limit = 1000 if limit == None else limit\n",
    "    res = annotation_api.page_nodes('RefSeqGenome', 0, limit, \"WHERE n.scientific_name =~ '.*\" + search_str.strip() + \".*'\")\n",
    "    os = {}\n",
    "    if not res == None:\n",
    "        for o in res:\n",
    "            n = o.data()['n']\n",
    "            os[n.get('key')] = dict(n.items())\n",
    "    return os\n",
    "res = search_genomes('coli')\n",
    "for id in res:\n",
    "    print(res[id]['scientific_name'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "MATCH (n:RefSeqGenome) RETURN n ORDER BY n.key SKIP $skip LIMIT $page_size\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{'GCF_000005825.2': {'genetic_code': 11,\n",
       "  'notes': '',\n",
       "  'release': '2',\n",
       "  'created_at': 1563767037784,\n",
       "  'scientific_name': 'Bacillus pseudofirmus OF4',\n",
       "  'source': 'refseq',\n",
       "  'taxonomy': 'cellular organisms; Bacteria; Terrabacteria group; Firmicutes; Bacilli; Bacillales; Bacillaceae; Bacillus; Bacillus pseudofirmus',\n",
       "  'type': 'Representative',\n",
       "  'external_source_origination_date': '30-JUL-2015',\n",
       "  'count_features': 4211,\n",
       "  'workspace_id': 'jplfaria:narrative_1524466549180',\n",
       "  'dna_size': 4249248,\n",
       "  'original_source_file_name': 'GCF_000005825.2_ASM582v2_genomic.gbff',\n",
       "  'reference_annotation': 0,\n",
       "  'updated_at': 1563767040915,\n",
       "  'num_contigs': 3,\n",
       "  'assembly_ref': '19217/82559/1',\n",
       "  'domain': 'Bacteria',\n",
       "  'source_id': 'NC_013791 (2 more accessions)',\n",
       "  'id': 'GCF_000005825.2',\n",
       "  'taxon_ref': '12570/597018/1',\n",
       "  'genbank_handle_ref': 'KBH_310617',\n",
       "  'gc_content': 0.3986055885653179,\n",
       "  'key': 'GCF_000005825.2'}}"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def list_genomes(limit = None):\n",
    "    limit = 1000 if limit == None else limit\n",
    "    res = annotation_api.page_nodes('RefSeqGenome', 0, limit)\n",
    "    os = {}\n",
    "    if not res == None:\n",
    "        for o in res:\n",
    "            n = o.data()['n']\n",
    "            os[n.get('key')] = dict(n.items())\n",
    "    return os\n",
    "list_genomes(1)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_gene_data():\n",
    "    pass\n",
    "def get_genome():\n",
    "    pass"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 130,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'rast1': <cobrakbase.core.kbasegenomesgenome.KBaseGenome at 0x7fdaa8b10a58>,\n",
       " 'rast2_script': <cobrakbase.core.kbasegenomesgenome.KBaseGenome at 0x7fdaa9374f60>,\n",
       " 'rast2_app': <cobrakbase.core.kbasegenomesgenome.KBaseGenome at 0x7fdaa8527320>}"
      ]
     },
     "execution_count": 130,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "def add_genome(genome):"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "None\n"
     ]
    }
   ],
   "source": [
    "\n",
    "\n",
    "print(process_list([['1', '3', 1], '2']))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 198,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 199,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'assembly_ref': '19217/231473/2',\n",
       " 'contig_ids': 'NC_000913.3',\n",
       " 'dna_size': 4641652,\n",
       " 'domain': 'Bacteria',\n",
       " 'external_source_origination_date': '08-Aug-2016',\n",
       " 'gc_content': 0.508,\n",
       " 'genbank_handle_ref': 'KBH_3448578',\n",
       " 'genetic_code': 11,\n",
       " 'genome_tiers': 'ExternalDB',\n",
       " 'id': 'test_nokmers1',\n",
       " 'md5': '482a2b04485ec8c4b5f4eaba2c2002da',\n",
       " 'molecule_type': 'DNA',\n",
       " 'mrnas': '',\n",
       " 'notes': 'REVIEWED REFSEQ: This record has been curated by NCBI staff. The\\nreference sequence is identical to U00096.\\nOn Nov 3, 2013 this sequence version replaced NC_000913.2.\\nRefSeq Category: Reference Genome\\n            FGS: First Genome sequenced\\n            MOD: Model Organism\\n            PHY: Based on Phylogenetics\\n            UPR: UniProt Genome\\nCurrent U00096 annotation updates are derived from EcoGene\\nhttp://ecogene.org. Suggestions for updates can be sent to Dr.\\nKenneth Rudd (krudd@miami.edu). These updates are being generated\\nfrom a collaboration that also includes ASAP/ERIC, the Coli Genetic\\nStock Center, EcoliHub, EcoCyc, RegulonDB and UniProtKB/Swiss-Prot.\\nCOMPLETENESS: full length.',\n",
       " 'num_contigs': 1,\n",
       " 'original_source_file_name': 'GCF_000005845.2_ASM584v2_genomic.gbff',\n",
       " 'scientific_name': 'Escherichia coli str. K-12 substr. MG1655',\n",
       " 'source': 'RefSeq',\n",
       " 'source_id': 'NC_000913',\n",
       " 'taxon_ref': '12570/625052/2',\n",
       " 'taxonomy': 'cellular organisms; Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacterales; Enterobacteriaceae; Escherichia; Escherichia coli; Escherichia coli K-12',\n",
       " 'warnings': ''}"
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "process_dict(genome.data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 60,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{<Node id=414883 labels={'Function'} properties={'updated_at': 1569876120585, 'created_at': 1569876120585, 'key': 'fake_function1 @ very_fake_function2'}>}"
      ]
     },
     "execution_count": 60,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.add_annotation(['fake_function1 @ very_fake_function2'], 'Test')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{<Node id=185611 labels={'Function'} properties={'updated_at': 1569877637469, 'created_at': 1569877637469, 'key': 'fake function 1'}>}"
      ]
     },
     "execution_count": 64,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.add_annotation(['very fake function 2'], 'Test2')\n",
    "annotation_api.add_annotation(['fake function 1'], 'Test2')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 62,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{<Node id=414887 labels={'Function'} properties={'updated_at': 1569877575598, 'created_at': 1569877575598, 'key': 'fake function1 @ very fake function2'}>}"
      ]
     },
     "execution_count": 62,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.add_annotation(['fake function1 @ very fake function2'], 'Test3')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 63,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{<Node id=185610 labels={'Function'} properties={'updated_at': 1569877486050, 'created_at': 1569877486050, 'key': 'very fake function 2'}>}"
      ]
     },
     "execution_count": 63,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.add_annotation(['very fake function 2'], 'Test4')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "WARNING:__main__:ADD BASE Function: fake_function1 @ very_fake_function2\n",
      "WARNING:__main__:ADD SUB Function: very_fake_function2\n",
      "WARNING:__main__:ADD SEARCH SUB Function: veryfakefunction2\n",
      "WARNING:__main__:ADD SUB Function: fake_function1\n",
      "WARNING:__main__:ADD SEARCH SUB Function: fakefunction1\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "{<Node id=414883 labels={'Function'} properties={'updated_at': 1569876120585, 'created_at': 1569876120585, 'key': 'fake_function1 @ very_fake_function2'}>}"
      ]
     },
     "execution_count": 38,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "add_annotation(annotation_api, ['fake_function1 @ very_fake_function2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 162,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(_364723:KBaseGene {aliases: 'GeneID:944742 ### thrL ### b0001 ### NP_414542.1 ### UniProtKB/Swiss-Prot:P0AD86 ### ASAP:ABE-0000006 ### JW4367 ### GI:16127995 ### ECK0001 ### EcoGene:EG11277', cdss: 'NP_414542.1', created_at: 1569833182468, dna_sequence: 'ATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGA', dna_sequence_length: 66, function: 'Thr operon leader peptide', id: 'b0001', key: 'b0001@GCF_000005845.2', md5: '13fe4258b37e32edda386faa50ec5fdf', mrnas: 'mRNA_1', protein_translation: 'MKRISTTITTTITITTGNGAG', protein_translation_length: 21, type: 'gene', updated_at: 1569879680696})"
      ]
     },
     "execution_count": 162,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from py2neo import Graph, NodeMatcher\n",
    "matcher = NodeMatcher(annotation_api.neo4j_graph)\n",
    "node = matcher.match(\"KBaseGene\", key=feature_id).first()\n",
    "node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 147,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'aliases': 'GeneID:944742 ### thrL ### b0001 ### NP_414542.1 ### UniProtKB/Swiss-Prot:P0AD86 ### ASAP:ABE-0000006 ### JW4367 ### GI:16127995 ### ECK0001 ### EcoGene:EG11277',\n",
       " 'cdss': 'NP_414542.1',\n",
       " 'dna_sequence': 'ATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGA',\n",
       " 'dna_sequence_length': 66,\n",
       " 'function': 'Thr operon leader peptide',\n",
       " 'id': 'b0001',\n",
       " 'md5': '13fe4258b37e32edda386faa50ec5fdf',\n",
       " 'mrnas': 'mRNA_1',\n",
       " 'protein_translation': 'MKRISTTITTTITITTGNGAG',\n",
       " 'protein_translation_length': 21,\n",
       " 'type': 'gene',\n",
       " 'key': 'b0001@GCF_000005845.2'}"
      ]
     },
     "execution_count": 147,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 148,
   "metadata": {},
   "outputs": [],
   "source": [
    "node = feature_props"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 201,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "\n",
    "genome = genomes['rast2_app']\n",
    "add_kbase_genome(annotation_api, genome, 'GCF_000005845.2', 'BETA_RAST')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 202,
   "metadata": {},
   "outputs": [],
   "source": [
    "genome = genomes['rast1']\n",
    "add_kbase_genome(annotation_api, genome, 'GCF_000005845.2', 'KBASE_RAST')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def add_annotation_old(annotation_api, gene_genome_id, functions, source):\n",
    "    #[KBaseGene] -[has_function]-> [Function] <-[has_function]- [FunctionSource]\n",
    "\n",
    "    kbase_gene_node = self.get_node('KBaseGene', gene_genome_id)\n",
    "    function_source_node = self.get_node('FunctionSource', source)\n",
    "\n",
    "    if function_source_node == None:\n",
    "        print('not found', source)\n",
    "        return    \n",
    "    if kbase_gene_node == None:\n",
    "        print('not found', gene_genome_id)\n",
    "        return\n",
    "\n",
    "    for f in functions:\n",
    "        if not f == None:\n",
    "            function_node = annotation_api.add_node('Function', f)\n",
    "            annotation_api.link_nodes(function_source_node, function_node, 'has_function')\n",
    "            annotation_api.link_nodes(kbase_gene_node, function_node, 'has_function')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "aliases <class 'list'>\n",
      "cdss <class 'list'>\n",
      "location <class 'list'>\n",
      "mrnas <class 'list'>\n",
      "quality <class 'dict'>\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "'{aliases:, cdss:, dna_sequence:\"ATGAAACGCATTAGCACCACCATTACCACCACCATCACCATTACCACAGGTAACGGTGCGGGCTGA\", dna_sequence_length:66, function:\"Thr operon leader peptide\", id:\"b0001\", location:, md5:\"13fe4258b37e32edda386faa50ec5fdf\", mrnas:, protein_translation:\"MKRISTTITTTITITTGNGAG\", protein_translation_length:21, quality:, type:\"gene\"}'"
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_api.to_str_dict(feature.data)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 292,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(_41:RefSeqGenome {assembly_ref: '19217/231473/1', count_features: 4498, created_at: 1563767040940, dna_size: 4641652, domain: 'Bacteria', external_source_origination_date: '08-AUG-2016', gc_content: 0.5079070985933456, genbank_handle_ref: 'KBH_644247', genetic_code: 11, id: 'GCF_000005845.2', key: 'GCF_000005845.2', notes: '', num_contigs: 1, original_source_file_name: 'GCF_000005845.2_ASM584v2_genomic.gbff', reference_annotation: 1, release: '2', scientific_name: 'Escherichia coli str. K-12 substr. MG1655', source: 'refseq', source_id: 'NC_000913', taxon_ref: '12570/625052/2', taxonomy: 'cellular organisms; Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacterales; Enterobacteriaceae; Escherichia; Escherichia coli; Escherichia coli K-12', type: 'Reference', updated_at: 1563767040940, workspace_id: 'jplfaria:narrative_1524466549180'})"
      ]
     },
     "execution_count": 292,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "node"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{None}"
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "feature.functions_unsplit"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation_nodes = annotation_api.add_annotation(feature.functions_unsplit, 'KBASE_RAST')\n",
    "annotation_nodes"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "kbase = cobrakbase.KBaseAPI(token='64XQ7SABQILQWSEW3CQKZXJA63DXZBGH')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load Current RAST Genomes\n",
    "jplfaria:narrative_1524466549180\n",
    "\n",
    "NCBI Reference+Representative Complete B and A"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#genomes 1675\n"
     ]
    }
   ],
   "source": [
    "genome_objects = {}\n",
    "for o in kbase.list_objects('jplfaria:narrative_1524466549180'):\n",
    "    if o[1].endswith('RAST') and o[2].startswith(\"KBaseGenomes.Genome\"):\n",
    "        genome_id = o[1][:-5]\n",
    "        genome_objects[genome_id] = o[1]\n",
    "        \n",
    "print('#genomes', len(genome_objects))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 426,
   "metadata": {},
   "outputs": [],
   "source": [
    "for genome_id in genome_objects:\n",
    "    object_id = genome_objects[genome_id]\n",
    "    data = kbase.get_object(object_id, 'jplfaria:narrative_1524466549180')\n",
    "    write_json(data, '../../../data/kbase_data/jplfaria_narrative_1524466549180/' + object_id + '.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "1675\n"
     ]
    }
   ],
   "source": [
    "loaded_genomes = list_genomes(annotation_api)\n",
    "print(len(loaded_genomes))\n",
    "for genome_id in genome_objects:\n",
    "    break\n",
    "    if not genome_id in loaded_genomes:\n",
    "        object_id = genome_objects[genome_id]\n",
    "        genome_data = read_json('../../../data/kbase_data/jplfaria_narrative_1524466549180/' + object_id +'.json')\n",
    "        genome = cobrakbase.core.KBaseGenome(genome_data)\n",
    "        genome_node = annotation_api.add_kbase_genome(genome, genome_id, 'KBASE_RAST')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "from multiprocessing import Pool\n",
    "\n",
    "to_load = set(genome_objects)\n",
    "loaded_genomes = list_genomes(annotation_api)\n",
    "to_load = to_load - set(loaded_genomes)\n",
    "\n",
    "def load_data(genome_id):\n",
    "    object_id = genome_objects[genome_id]\n",
    "    genome_data = read_json('../../../data/kbase_data/jplfaria_narrative_1524466549180/' + object_id +'.json')\n",
    "    genome = cobrakbase.core.KBaseGenome(genome_data)\n",
    "    a = init_annotation_api(host, port, user, pwd)\n",
    "    genome_node = a.add_kbase_genome(genome, genome_id, 'KBASE_RAST')\n",
    "    return True\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    if len(to_load) > 0:\n",
    "        p = Pool(processes=5)\n",
    "        inputs = list(range(20))\n",
    "        i = p.map_async(load_data, to_load, chunksize = 2)\n",
    "        i.wait()\n",
    "        print(i.get())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Load Updated RAST Genomes\n",
    "filipeliu:narrative_1569597697005 KBASE2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "#genomes 1676\n"
     ]
    }
   ],
   "source": [
    "genome_objects = {}\n",
    "for o in kbase.list_objects('filipeliu:narrative_1569597697005'):\n",
    "    if o[1].endswith('RAST2') and o[2].startswith(\"KBaseGenomes.Genome\"):\n",
    "        genome_id = o[1][:-6]\n",
    "        genome_objects[genome_id] = o[1]\n",
    "        \n",
    "print('#genomes', len(genome_objects))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 428,
   "metadata": {},
   "outputs": [],
   "source": [
    "for genome_id in genome_objects:\n",
    "    object_id = genome_objects[genome_id]\n",
    "    data = kbase.get_object(object_id, 'filipeliu:narrative_1569597697005')\n",
    "    write_json(data, '../../../data/kbase_data/filipeliu_narrative_1569597697005/' + object_id + '.json')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "df = pd.read_csv('../../../data/modelseed2/missing_genomes.tsv', sep='\\t')\n",
    "to_load = set()\n",
    "for row_id, d in df.iterrows():\n",
    "    if d['rast2'] < 0.01:\n",
    "        to_load.add(d['genome_id'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "50"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "loaded_genomes = set()\n",
    "loaded_genomes.add('GCF_000027325.1')\n",
    "len(list(genome_objects)[:50])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#oaded_genomes |= to_load\n",
    "len(loaded_genomes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(loaded_genomes)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "len(to_load)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true,
    "jupyter": {
     "outputs_hidden": true
    }
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "547\n",
      "GCF_000767055.1\n",
      "GCF_000973105.1\n",
      "GCF_000023745.1\n",
      "GCF_000184435.1\n",
      "GCF_000017505.1\n",
      "GCF_000009125.1\n",
      "GCF_001302585.1\n",
      "GCF_000020065.1\n",
      "GCF_001719165.1\n",
      "GCF_000024605.1\n",
      "GCF_000155735.2\n",
      "GCF_000222305.1\n",
      "GCF_000304355.2\n",
      "GCF_000265425.1\n",
      "GCF_000968055.1\n",
      "GCF_001040945.1\n",
      "GCF_000017005.1\n",
      "GCF_900086555.1\n",
      "GCF_000017245.1\n",
      "GCF_000224085.1\n",
      "GCF_000023025.1\n",
      "GCF_000800295.1\n",
      "GCF_000069225.1\n",
      "GCF_000829315.1\n",
      "GCF_000513475.1\n",
      "GCF_000347595.1\n",
      "GCF_001274535.1\n",
      "GCF_000011385.1\n",
      "GCF_001887595.1\n",
      "GCF_000023465.1\n",
      "GCF_000015505.1\n",
      "GCF_000318015.1\n",
      "GCF_001005905.1\n",
      "GCF_000020985.1\n",
      "GCF_000006905.1\n",
      "GCF_000317795.1\n",
      "GCF_000012465.1\n",
      "GCF_001314225.1\n",
      "GCF_000953655.1\n",
      "GCF_900093775.1\n",
      "GCF_000953015.1\n",
      "GCF_001431725.1\n",
      "GCF_000176855.2\n",
      "GCF_000196455.1\n",
      "GCF_000011485.1\n",
      "GCF_000147815.2\n",
      "GCF_000025905.1\n",
      "GCF_000014965.1\n",
      "GCF_000255535.1\n",
      "GCF_000016405.1\n",
      "GCF_000010325.1\n",
      "GCF_000325745.1\n",
      "GCF_000063505.1\n",
      "GCF_000146185.1\n",
      "GCF_000583875.1\n",
      "GCF_000177535.2\n",
      "GCF_900116045.1\n",
      "GCF_000012425.1\n",
      "GCF_000196095.1\n",
      "GCF_000019845.1\n",
      "GCF_000217715.1\n",
      "GCF_000829395.1\n",
      "GCF_000619905.2\n",
      "GCF_000725365.1\n",
      "GCF_000328765.2\n",
      "GCF_000015045.1\n",
      "GCF_001878675.1\n",
      "GCF_000313175.2\n",
      "GCF_000018785.1\n",
      "GCF_000317675.1\n",
      "GCF_001460635.1\n",
      "GCF_000024765.1\n",
      "GCF_000349845.1\n",
      "GCF_000500935.1\n",
      "GCF_000025925.1\n",
      "GCF_000007785.1\n",
      "GCF_000737575.1\n",
      "GCF_000017705.1\n",
      "GCF_000021045.1\n",
      "GCF_000195995.1\n",
      "GCF_001042675.1\n",
      "GCF_000025645.1\n",
      "GCF_000203855.3\n",
      "GCF_000262715.1\n",
      "GCF_000007765.1\n",
      "GCF_000970085.1\n",
      "GCF_000195755.1\n",
      "GCF_000973545.1\n",
      "GCF_000230735.2\n",
      "GCF_000020945.1\n",
      "GCF_000276685.1\n",
      "GCF_000263735.1\n",
      "GCF_001611135.1\n",
      "GCF_000008685.2\n",
      "GCF_000340865.1\n",
      "GCF_000319575.2\n",
      "GCF_000152825.2\n",
      "GCF_000017265.1\n",
      "GCF_000011705.1\n",
      "GCF_000179035.2\n",
      "GCF_000224105.1\n",
      "GCF_000265465.1\n",
      "GCF_000025325.1\n",
      "GCF_000284155.1\n",
      "GCF_000512895.1\n",
      "GCF_000008525.1\n",
      "GCF_000803645.1\n",
      "GCF_000980815.1\n",
      "GCF_000300135.1\n",
      "GCF_000151105.2\n",
      "GCF_000072485.1\n",
      "GCF_000015885.1\n",
      "GCF_000016665.1\n",
      "GCF_000093165.1\n",
      "GCF_000319385.1\n",
      "GCF_000470655.1\n",
      "GCF_000025865.1\n",
      "GCF_000178975.2\n",
      "GCF_000746645.1\n",
      "GCF_001693675.1\n",
      "GCF_000800475.2\n",
      "GCF_000204565.1\n",
      "GCF_000284095.1\n",
      "GCF_000016425.1\n"
     ]
    }
   ],
   "source": [
    "from multiprocessing import Pool\n",
    "\n",
    "#to_load = set(genome_objects)\n",
    "#to_load = set(list(genome_objects)[900:])\n",
    "#to_load.add('GCF_000027325.1')\n",
    "#loaded_genomes = list_genomes(annotation_api)\n",
    "#to_load = to_load - set(loaded_genomes)\n",
    "print(len(to_load))\n",
    "def load_data(genome_id):\n",
    "    print(genome_id)\n",
    "    object_id = genome_objects[genome_id]\n",
    "    genome_data = read_json('../../../data/kbase_data/filipeliu_narrative_1569597697005/' + object_id +'.json')\n",
    "    genome = cobrakbase.core.KBaseGenome(genome_data)\n",
    "    a = init_annotation_api(host, port, user, pwd)\n",
    "    genome_node = a.add_kbase_genome(genome, genome_id, 'BETA_RAST')\n",
    "    loaded_genomes.add(genome_id)\n",
    "    return True\n",
    "\n",
    "if __name__ == '__main__':\n",
    "    if len(to_load) > 0:\n",
    "        p = Pool(processes=10)\n",
    "        i = p.map_async(load_data, to_load, chunksize = 1)\n",
    "        i.wait()\n",
    "        print(i.get())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 359,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'genetic_code': 11,\n",
       " 'notes': '',\n",
       " 'release': '2',\n",
       " 'created_at': 1563767040940,\n",
       " 'scientific_name': 'Escherichia coli str. K-12 substr. MG1655',\n",
       " 'source': 'refseq',\n",
       " 'taxonomy': 'cellular organisms; Bacteria; Proteobacteria; Gammaproteobacteria; Enterobacterales; Enterobacteriaceae; Escherichia; Escherichia coli; Escherichia coli K-12',\n",
       " 'type': 'Reference',\n",
       " 'external_source_origination_date': '08-AUG-2016',\n",
       " 'count_features': 4498,\n",
       " 'workspace_id': 'jplfaria:narrative_1524466549180',\n",
       " 'dna_size': 4641652,\n",
       " 'original_source_file_name': 'GCF_000005845.2_ASM584v2_genomic.gbff',\n",
       " 'reference_annotation': 1,\n",
       " 'updated_at': 1563767040940,\n",
       " 'num_contigs': 1,\n",
       " 'assembly_ref': '19217/231473/1',\n",
       " 'domain': 'Bacteria',\n",
       " 'source_id': 'NC_000913',\n",
       " 'id': 'GCF_000005845.2',\n",
       " 'taxon_ref': '12570/625052/2',\n",
       " 'genbank_handle_ref': 'KBH_644247',\n",
       " 'gc_content': 0.5079070985933456,\n",
       " 'key': 'GCF_000005845.2'}"
      ]
     },
     "execution_count": 359,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "class Neo4jGenome():\n",
    "    \n",
    "    def __init__(self, node):\n",
    "        self.node = node\n",
    "        self.data = dict(self.node.items())\n",
    "        self.features = {}\n",
    "        self.features_annotation = {}\n",
    "        #print(node)\n",
    "        self.populate_from_graph()\n",
    "        \n",
    "    def populate_from_graph(self):\n",
    "        for rel in self.node.graph.match((self.node, ), r_type=\"has_gene\", ):\n",
    "            gene_id = rel.end_node['id']\n",
    "            self.features[gene_id] = rel.end_node\n",
    "            self.features_annotation[gene_id] = set()\n",
    "            for srel in self.node.graph.match((rel.end_node, ), r_type=\"has_annotation\", ):\n",
    "                annotation = Neo4jAnnotationFunction(srel.end_node)\n",
    "                self.features_annotation[gene_id].add(annotation)\n",
    "                #print(annotation)\n",
    "            \n",
    "        for rel in self.node.graph.match((self.node, ), r_type=\"search_function\", ):\n",
    "            self.search_value = rel.end_node['key']\n",
    "            for srel in self.node.graph.match((rel.end_node, ), r_type=\"has_function\", ):\n",
    "                n = srel.end_node\n",
    "                if not n['key'] == self.value:\n",
    "                    self.synonyms.add(n['key'])\n",
    "                    \n",
    "        for rel in self.node.graph.match((None, self.node), r_type=\"has_function\", ):\n",
    "            n = rel.start_node\n",
    "            if n.has_label('FunctionGroup'):\n",
    "                self.function_group.add(n['key'])\n",
    "\n",
    "    def __str__(self):\n",
    "        return self.value\n",
    "    \n",
    "node = matcher.match(\"RefSeqGenome\", key='GCF_000005845.2').first()\n",
    "neo_object = Neo4jGenome(node)\n",
    "neo_object.data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 348,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Two-component response regulator CreC {'KBASE_RAST'}\n",
      "Sensory histidine kinase CreC of two-component signal transduction system CreBC {'BETA_RAST'}\n"
     ]
    }
   ],
   "source": [
    "node = matcher.match(\"KBaseGene\", key='b4399@GCF_000005845.2').first()\n",
    "\n",
    "for rel in node.graph.match((node, ), r_type=\"has_annotation\", ):\n",
    "    annotation = Neo4jAnnotationFunction(rel.end_node)\n",
    "    annotation.neo4j_load_source()\n",
    "    print(annotation.value, annotation.source)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 352,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "set()"
      ]
     },
     "execution_count": 352,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "annotation.sub_functions"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 374,
   "metadata": {},
   "outputs": [],
   "source": [
    "def load_kegg_orthologs(genome):\n",
    "    genome.features_ko = {}\n",
    "    for gene_id in genome.features:\n",
    "        genome.features_ko[gene_id] = {}\n",
    "        #print(gene_id, genome.features[gene_id])\n",
    "        for rel in genome.node.graph.match((genome.features[gene_id], ), r_type=\"has_ortholog\", ):\n",
    "            ko_node = rel.end_node\n",
    "            genome.features_ko[gene_id][ko_node['key']] = {}\n",
    "            for srel in ko_node.graph.match((ko_node, ), r_type=\"has_reaction\", ):\n",
    "                kegg_rn_node = srel.end_node\n",
    "                genome.features_ko[gene_id][ko_node['key']][kegg_rn_node['key']] = {}\n",
    "                for seed_rel in ko_node.graph.match((None, kegg_rn_node), r_type=\"has_crossreference_to\", ):\n",
    "                    seed_node = seed_rel.start_node\n",
    "                    genome.features_ko[gene_id][ko_node['key']][kegg_rn_node['key']][seed_node['key']] = seed_node\n",
    "    pass\n",
    "load_kegg_orthologs(neo_object)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 380,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "b4403 687\n",
      "\t K02533\n",
      "\t {'KBASE_RAST'} RNA methyltransferase, TrmH family, group 1 set()\n",
      "\t\t set()\n",
      "\t {'BETA_RAST'} Uncharacterized tRNA/rRNA methyltransferase LasT set()\n",
      "\t\t set()\n"
     ]
    }
   ],
   "source": [
    "for gene_id in neo_object.features:\n",
    "    gene_data = neo_object.features[gene_id]\n",
    "    print(gene_id, gene_data['dna_sequence_length'])\n",
    "    for ko in neo_object.features_ko[gene_id]:\n",
    "        print('\\t', ko)\n",
    "    for annotation in neo_object.features_annotation[gene_id]:\n",
    "        load_source(annotation)\n",
    "        if len(annotation.sub_functions) > 0:\n",
    "            print('\\t', annotation.source, annotation.value, annotation.synonyms)\n",
    "            for function in annotation.sub_functions:\n",
    "                print('\\t\\t', function.value, function.synonyms)\n",
    "                print('\\t\\t\\t', function.function_group)\n",
    "        else:\n",
    "            print('\\t', annotation.source, annotation.value, annotation.synonyms)\n",
    "            print('\\t\\t', annotation.function_group)\n",
    "    break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 378,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'K02533': {}}"
      ]
     },
     "execution_count": 378,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.4"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
