<!DOCTYPE html
	PUBLIC "-//W3C//DTD XHTML 1.0 Transitional//EN"
	 "http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" lang="en-US" xml:lang="en-US">
<head>
<title>PaperBLAST</title>
<link rel="shortcut icon" href="../static/favicon.ico" />
<style type="text/css">
<!--/* <![CDATA[ */
.autocomplete {
  /*the container must be positioned relative:*/
  position: relative;
  display: inline-block;
}

.autocomplete-items {
  position: absolute;
  border: 1px solid #d4d4d4;
  border-bottom: none;
  border-top: none;
  z-index: 99;
  /*position the autocomplete items to be the same width as the container:*/
  top: 100%;
  left: 0;
  right: 0;
}
.autocomplete-items div {
  padding: 10px;
  cursor: pointer;
  background-color: #fff;
  border-bottom: 1px solid #d4d4d4;
}
.autocomplete-items div:hover {
  /*when hovering an item:*/
  background-color: #e9e9e9;
}
.autocomplete-active {
  /*when navigating through the items using the arrow keys:*/
  background-color: DodgerBlue !important;
  color: #ffffff;
}
/* for showing protein sequences */
/* base style, and for unusual items */
.aa  { background-color: white; }
.aa1 { background-color: lightblue; } /* for AILMFWV */
.aa2 { background-color: red; } /* for KR */
.aa3 { background-color: magenta; } /* for ED */
.aa4 { background-color: lightgreen; } /* for NQST */
.aa5 { background-color: pink; } /* for C */
.aa6 { background-color: orange; } /* for G */
.aa7 { background-color: yellow; } /* for P */
.aa8 { background-color: cyan; } /* for HY */

/* for showing the alignment line */
.alnS, .alnS1, .alnS0 { font-size: smaller; }
.alnS1 { font-weight: bold; color: darkgreen; } /* site, matching */
.alnS0 { font-weight: bold; color: darkred; } /* site, not matching */

/* for an alignment column */
.alnCol, .alnLabel {
  vertical-align:top; float:left; display:inline-block;
  margin-top: 1em;
  font-family: "Courier New", monospace;
}
.alnCol { background-color: #EEEEEE; border: solid 1px #EEEEEE; }
/* In many browsers, the label prevents the columns from wrapping all the way to the left,
   so, make it subtly smaller */
.alnLabel { margin-right: 0.5em; padding-top: 0.04em; border: none; font-size: 95% }



/* ]]> */-->
</style>
<script src="../static/autocomplete_uniprot.js" type="text/javascript"></script>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
</head>
<body>
<div style="background-color: #40C0CB; display: block; position: absolute; top: 0px; left: -1px;
  width: 100%; padding: 0.25em; z-index: 400;">
<H2 style="margin: 0em;">
<A HREF="litSearch.cgi" style="color: gold; font-family: 'Montserrat', sans-serif; font-style:italic;
  text-shadow: 1px 1px 1px #000000; text-decoration: none;">
PaperBLAST &ndash; <small>Find papers about a protein or its homologs</small>
</A></H2></div>
<P style="margin: 0em;">&nbsp;</P>
<SCRIPT src="../static/pb.js"></SCRIPT>
<h2>PaperBLAST</h2>
<SCRIPT src="../static/pb.js"></SCRIPT>
<SCRIPT src="http://fit.genomics.lbl.gov/d3js/d3.min.js"></SCRIPT>
<SCRIPT src="http://fit.genomics.lbl.gov/images/fitblast.js"></SCRIPT>
<h3>PaperBLAST Hits for sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1 (255 a.a., MVKLQRFSEK...)</h3> <DIV style="float:right; padding-left: 10px; width:25%; background-color: #EEEEEE;">
                <P>Also see
                <A HREF="http://www.ncbi.nlm.nih.gov/Structure/cdd/wrpsb.cgi?seqinput=>sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1%0AMVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC" TITLE="Compare your sequence to NCBI's Conserved Domains Database">Conserved Domains</A>,
                <A HREF="sites.cgi?query=%3Esp%7CP53315%7CSOL4_YEAST%206-phosphogluconolactonase%204%20OS%3DSaccharomyces%20cerevisiae%20%28strain%20ATCC%20204508%20%2F%20S288c%29%20OX%3D559292%20GN%3DSOL4%20PE%3D1%20SV%3D1%0AMVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNI%0AFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLI%0AGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNN%0ASPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVT%0AWFLDDEAGALIPENC%0A" TITLE="Compare to proteins with known functional residues">SitesBLAST</A>,
                or
                <A TITLE="Fitness BLAST compares your sequence to bacterial proteins that have mutant phenotypes"
                   NAME="#fitness">Fitness BLAST:</A>
                <SPAN ID="fitblast_short"><SMALL>loading...</SMALL></SPAN>
                <SCRIPT>
                var server_root = "http://fit.genomics.lbl.gov/";
                var seq = "MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC";
                fitblast_load_short("fitblast_short", server_root, seq);
                </SCRIPT>
                </DIV>
             
<p>Found 79 similar proteins in the literature:</p>
<p style="margin-top: 1em; margin-bottom: 0em;"><a title="SwissProt" onmousedown="logger(this, 'curated::SOL4_YEAST / P53315')" href="http://www.uniprot.org/uniprot/P53315">SOL4_YEAST / P53315</a> <b>6-phosphogluconolactonase 4; 6PGL; EC 3.1.1.31</b> from <i>Saccharomyces cerevisiae (strain ATCC 204508 / S288c) (Baker's yeast)</i> (see <a onmousedown="logger(this, 'curatedpaper::SOL4_YEAST / P53315')" href="http://www.ncbi.nlm.nih.gov/pubmed/14562106,15454531">2 papers</a>)<BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::YGR248W')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=6583088">YGR248W</a> 6-phosphogluconolactonase with similarity to Sol3p (RefSeq) from <i>Saccharomyces cerevisiae</i><BR><a title="1:255/255 of query is similar to 1:255/255 of hit (E = 7e-150,  530 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=SOL4_YEAST / P53315&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS6583088" style="font-family: sans-serif; font-size: smaller;">100% identity, 100% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><b>function:</b> Hydrolysis of 6-phosphogluconolactone to 6-phosphogluconate. <BR>
<b>catalytic</b> <b>activity:</b> 6-phospho-D-glucono-1,5-lactone + H2O = 6-phospho-D-gluconate + H(+) <small>(<a href="https://www.rhea-db.org/reaction?id=12556">RHEA:12556</a>)</small><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YGR248W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4717488">Global translational impacts of the loss of the tRNA modification t6A in yeast</a><br /><small><a title="Thiaville PC, Legendre R, Rojas-Benítez D, Baudin-Baillieu A, Hatin I, Chalancon G, Glavic A, Namy O, de Crécy-Lagard V.">Thiaville,</a> Microbial cell (Graz, Austria) 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...genes involved in growth on non-fermentable carbon sources (see summary in SGD 98 ), Sol4 (<B><span style="color: red;">YGR248W</span></B>), which functions in the pentose phosphate pathway (see summary in SGD 98 ), and Smc6 (YLR383W), a component of the SMC5-SMC6 complex that plays a key role in the removal...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YGR248W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3163424">On the formalization and reuse of scientific research</a><br /><small><a title="King RD, Liakata M, Lu C, Oliver SG, Soldatova LN.">King,</a> Journal of the Royal Society, Interface 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...0.024 0.222 R_AGAT_SC YDL052C 0.009 0.733 0.034 0.805 R_FTHFCLm YER183C 0.022 0 0.014 0 R_G6PDA <B><span style="color: red;">YGR248W</span></B> 0.017 0 0.007 0 R_G6PDA YHR163W 0.222 0 0.005 0 R_G6PDA YNR034W 0.023 0 0.028 0 R_GLUN YIL033C 0.079 0 0.205 0 R_M1PD YNR073C 0.016 0 0.024 0 R_MACACI YLL060C...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YGR248W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3063666">New insight into the role of the Cdc34 ubiquitin-conjugating enzyme in cell cycle regulation via Ace2 and Sic1</a><br /><small><a title="Cocklin R, Heyen J, Larry T, Tyers M, Goebl M.">Cocklin,</a> Genetics 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...YGR138C YER130C YCR021C YDR171W YJL144W YGR249W YGR142W <B><span style="color: red;">YGR248W</span></B> YDR258C YOR134W YFL053W YPR158W YPR015C YGL037C YLL026W YNL077W YJL082W YBR214W YPL247C YNL007C...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YGR248W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2672114">Transcriptional response to mitochondrial NADH kinase deficiency in Saccharomyces cerevisiae</a><br /><small><a title="Stuart GR, Humble MM, Strand MK, Copeland WC.">Stuart,</a> Mitochondrion 2009 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...TKL2 -3.23 -4.06 -4.16 9.50 19.33 PPP transketolase; needed for synthesis of aromatic amino acids <B><span style="color: red;">YGR248W</span></B> SOL4 -3.17 -2.15 -2.66 4.77 6.74 6-phosphogluconolactonase with similarity to Sol3p YPR151C SUE1 -3.17 -1.76 -1.45 5.52 5.45 Mitochondrial protein; degradation of unstable forms of cytochrome c YDR258C HSP78 -3.16...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YGR248W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2673449">Construction and application of a protein and genetic interaction network (yeast interactome)</a><br /><small><a title="Stuart GR, Copeland WC, Strand MK.">Stuart,</a> Nucleic acids research 2009 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...xenobiotic agents and environmental or physiological stresses YPL054W LEE1 6.54 Zinc-finger protein of unknown function <B><span style="color: red;">YGR248W</span></B> SOL4 6.47 6-phosphogluconolactonase with similarity to Sol3p YEL033W YEL033W 6.42 Predicted to have metabolic role based on analysis of gene networks YLR162W YLR162W 6.38 Putative protein of unknown function; overexpression...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YGR248W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2667776">The high osmotic response and cell wall integrity pathways cooperate to regulate transcriptional responses to zymolyase-induced cell wall stress in Saccharomyces cerevisiae</a><br /><small><a title="García R, Rodríguez-Peña JM, Bermejo C, Nombela C, Arroyo J.">García,</a> The Journal of biological chemistry 2009 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...February 17, 2017 YGL038C YML128C YDL022W YER062C <B><span style="color: red;">YGR248W</span></B> YGL157W YLR327C YEL060C YOR374W YMR105C YFL014W YMR169C YOL053C-A YML131W YOR173W YGR088W YML100W...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...YLL051C YMR315W YPL088W YHR209W YJL155C YDL022W YER062C <B><span style="color: red;">YGR248W</span></B> YGL157W YLR327C YEL060C YOR374W YMR105C YOL112W YMR238W YKL161C YHR030C YOR208W YIL117C YMR173W...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YGR248W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2254601">Microarray analysis of the in vivo sequence preferences of a minor groove binding drug</a><br /><small><a title="Eckdahl TT, Brown AD, Hart SN, Malloy KJ, Shott M, Yiu G, Hoopes LL, Heyer LJ.">Eckdahl,</a> BMC genomics 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...unknown unknown YJL156W-A YJL156W-A -2.25 unknown unknown YMR105C PGM2 -2.23 glucose 1-phosphate utilization phosphoglucomutase activity <B><span style="color: red;">YGR248W</span></B> SOL4 -2.00 unknown unknown YHR087W YHR087W -1.95 unknown unknown YPR160W GPH1 -1.76 glycogen catabolism glycogen phosphorylase activity YLR178C TFS1 -1.76 regulation of proteolysis lipid binding activity YOR173W DCS2 -1.72 unknown...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YGR248W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1865649">Mg2+ deprivation elicits rapid Ca2+ uptake and activates Ca2+/calcineurin signaling in Saccharomyces cerevisiae</a><br /><small><a title="Wiesenberger G, Steinleitner K, Malli R, Graier WF, Vormann J, Schweyen RJ, Stadler JA.">Wiesenberger,</a> Eukaryotic cell 2007 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...S S Identification no. YBR005W YLR107W YPR108W <B><span style="color: red;">YGR248W</span></B> YLR136C YKL216W YOR230W YKL126W YDL124W YHL021C YKR049C YMR251W-A YGR268C YJL082W YML128C YLR350W YPR154W...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a href="litSearch.cgi?more=VIMSS6583088">More</a></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="SwissProt" onmousedown="logger(this, 'curated::SOL3_YEAST / P38858')" href="http://www.uniprot.org/uniprot/P38858">SOL3_YEAST / P38858</a> <b>6-phosphogluconolactonase 3; 6PGL; EC 3.1.1.31</b> from <i>Saccharomyces cerevisiae (strain ATCC 204508 / S288c) (Baker's yeast)</i> (see <a onmousedown="logger(this, 'curatedpaper::SOL3_YEAST / P38858')" href="http://www.ncbi.nlm.nih.gov/pubmed/14562095,14562106,15454531">3 papers</a>)<BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::YHR163W')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=6583364">YHR163W</a> 6-phosphogluconolactonase, catalyzes the second step of the pentose phosphate pathway; weak multicopy suppressor of los1-1 mutation; homologous to Sol2p and Sol1p (RefSeq) from <i>Saccharomyces cerevisiae</i><BR><a title="1:247/255 of query is similar to 1:240/249 of hit (E = 2e-70,  266 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=SOL3_YEAST / P38858&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=P38858" style="font-family: sans-serif; font-size: smaller;">49% identity, 97% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><b>function:</b> Hydrolysis of 6-phosphogluconolactone to 6-phosphogluconate. <BR>
<b>catalytic</b> <b>activity:</b> 6-phospho-D-glucono-1,5-lactone + H2O = 6-phospho-D-gluconate + H(+) <small>(<a href="https://www.rhea-db.org/reaction?id=12556">RHEA:12556</a>)</small><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::P38858')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4057618">A framework for application of metabolic modeling in yeast to predict the effects of nsSNV in human orthologs</a><br /><small><a title="Dingerdissen H, Weaver DS, Karp PD, Pan Y, Simonyan V, Mazumder R.">Dingerdissen,</a> Biology direct 2014 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Q9BV23 P53750 O43175 P40054 P08243 P49090 P56937 Q12452 P22830 P16622 Q86V88 P40081 Q16769 P43599 O95336 <B><span style="color: red;">P38858</span></B> Q8IXB1 P40564 Q15386 P53119 Q969P6 P04786 Q9Y3B8 P54964 Q9Y2H1 P53894 Q9Y2Z4 P48527 P00813 P53909 Q9UI42 P38836 Q9Y6R4 P53599 This table lists the pool of 113 reciprocal best match human-yeast ortholog...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YHR163W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4668057">A Tool for Multiple Targeted Genome Deletions that Is Precise, Scar-Free, and Suitable for Automation</a><br /><small><a title="Aubrey W, Riley MC, Young M, King RD, Oliver SG, Clare A.">Aubrey,</a> PloS one 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...added (negative control, 3 replicatesno colonies); (b) no cells added (sterility control, 3 replicatesno colonies); <B><span style="color: red;">YHR163W</span></B> PCR3 DNA (positive controlsee above, 2 replicatesabundant colonies). The controls were plated on the same medium as the other 96 transformations. As expected from deletion cassettes with long flanking homology,...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...were needed to produce any product. (PNG) Click here for additional data file. S8 Fig <B><span style="color: red;">YHR163W</span></B>, YPR073C, YGR125W, YJL134W. Confirmation gel for strains <B><span style="color: red;">YHR163W</span></B>, YPR073C, YGR125W, YJL134W after selection on FOA, with expected product sizes shown at bottom of lanes. (PNG) Click here for additional data...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YHR163W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3618361">A Whole Genome Screen for Minisatellite Stability Genes in Stationary-Phase Yeast Cells</a><br /><small><a title="Alver B, Jauert PA, Brosnan L, O'Hehir M, VanderSluis B, Myers CL, Kirkpatrick DT.">Alver,</a> G3 (Bethesda, Md.) 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...YGR001C YGR001c AGP3 YFL055w LEU3 YLR451w SNF6 YHL025w YGR031W YGR031w AIM23 YJL131c LIP5 YOR196c SOL3 <B><span style="color: red;">YHR163w</span></B> YGR051C YGR051c ALR2 YFL050c MET22 YOL064c SPE1 YKL184w YGR149W YGR149w APM4 YOL062c MON1 YGL124c SPS22 YCL048w YGR176W YGR176w APS1 YLR170c MPC54 YOR177c SRN2 YLR119w YGR207C YGR207c ARN2 YHL047c MRM2 YGL136c...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YHR163W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3163424">On the formalization and reuse of scientific research</a><br /><small><a title="King RD, Liakata M, Lu C, Oliver SG, Soldatova LN.">King,</a> Journal of the Royal Society, Interface 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...0.034 0.805 R_FTHFCLm YER183C 0.022 0 0.014 0 R_G6PDA YGR248W 0.017 0 0.007 0 R_G6PDA <B><span style="color: red;">YHR163W</span></B> 0.222 0 0.005 0 R_G6PDA YNR034W 0.023 0 0.028 0 R_GLUN YIL033C 0.079 0 0.205 0 R_M1PD YNR073C 0.016 0 0.024 0 R_MACACI YLL060C 0.011 0 0.014 0 R_POLYAO2 YMR020W...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YHR163W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1131933">Mapping of transcription start sites in Saccharomyces cerevisiae using 5' SAGE</a><br /><small><a title="Zhang Z, Dietrich FS.">Zhang,</a> Nucleic acids research 2005 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...YER050C RSM18 133(2) 297(1) 571(1) 5 +192 No YGR088W CTT1 8(2) 377(1) 7 +33 No <B><span style="color: red;">YHR163W</span></B> SOL3 91(2) 848(1) 8 +93 Yes YIL043C CBR1 82(1) 94(1) 102(3) 9 +114 Yes YIL053W RHR2 35(5) 36(1) 349(1) 526(1) 852(1) 9 +63 No YIL076W SEC28 177(4) 556(1) 9 +189...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YHR163W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1175969">Large-scale 13C-flux analysis reveals mechanistic principles of metabolic network robustness to null mutations in yeast</a><br /><small><a title="Blank LM, Kuepfer L, Sauer U.">Blank,</a> Genome biology 2005 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...1 1 1 sol2 /YCRX13W 0.99 0.98 1 idp1 /YDL066W 0.92 0.94 1.03 sol3 / <B><span style="color: red;">YHR163W</span></B> 0.71 0.94 1 idp2 /YLR174W 0.86 0.96 0.95 sol4 / YGR248W 0.95 0.91 1.01 lsc1 /YOR142W 1.05 0.93 1 tal1 / YLR354C 0.89 0.94 1 mae1 /YKL029C 1.01 0.96 1...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YHR163W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC383160">Molecular basis for anaerobic growth of Saccharomyces cerevisiae on xylose, investigated by global gene expression and metabolic flux analysis</a><br /><small><a title="Sonderegger M, Jeppsson M, Hahn-Hägerdal B, Sauer U.">Sonderegger,</a> Applied and environmental microbiology 2004 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...2.6 2.4 2.5 MEP3 YPR138C 2.9 PUT4 PYK2 SOL3 SUL2 YOR348C YOR347C <B><span style="color: red;">YHR163W</span></B> YLR092W YEL041W YER053C 6.6 12 2.6 2.2 4 2.7 5 9.1 2.2 2 3.5 2.6 3.4 3.1 2.6 2.1 26 2.5...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YHR163W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC524422">Large-scale mutagenesis of the yeast genome using a Tn7-derived multipurpose transposon</a><br /><small><a title="Kumar A, Seringhaus M, Biery MC, Sarnovsky RJ, Umansky L, Piccirillo S, Heidtman M, Cheung KH, Dobry CJ, Gerstein MB, Craig NL, Snyder M.">Kumar,</a> Genome research 2004 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...ENO1 RAD2 TN7-48D6 TN7-29G4 TN7-60C8 TN7-18E8 YHR141C <B><span style="color: red;">YHR163W</span></B> YIL123W YIL154C RPL42B SOL3 SIM1 IMP2 TN7-33E10 TN7-49C9 YJL117W YJR052W PHO86 RAD7 TN7-60F6...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YHR163W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC143595">Molecular analysis of a Saccharomyces cerevisiae mutant with improved ability to utilize xylose shows enhanced expression of proteins involved in transport, initial xylose metabolism, and the pentose phosphate pathway</a><br /><small><a title="Wahlbom CF, Cordero Otero RR, van Zyl WH, Hahn-Hägerdal B, Jönsson LJ.">Wahlbom,</a> Applied and environmental microbiology 2003 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...YHR104W YLR070C YGR194C Pentose phosphate pathway YNL241C <B><span style="color: red;">YHR163W</span></B> HXT4 HXT5 HXT7 GAL2 High-affinity glucose transporter Hexose transporter Hexose transporter...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a href="litSearch.cgi?more=P38858">More</a></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::PAS_chr3_1126')" href="http://www.ncbi.nlm.nih.gov/protein/XP_002493372.1">PAS_chr3_1126</a> 6-phosphogluconolactonase, catalyzes the second step of the pentose phosphate pathway from <i>Komagataella phaffii GS115</i><BR><a title="1:247/255 of query is similar to 1:238/249 of hit (E = 5e-41,  168 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=PAS_chr3_1126&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_002493372.1" style="font-family: sans-serif; font-size: smaller;">36% identity, 97% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PAS_chr3_1126')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4580311">Systems-level organization of yeast methylotrophic lifestyle</a><br /><small><a title="Rußmayer H, Buchetics M, Gruber C, Valli M, Grillitsch K, Modarres G, Guerrasio R, Klavins K, Neubauer S, Drexler H, Steiger M, Troyer C, Al Chalabi A, Krebiehl G, Sonntag D, Zellnig G, Daum G, Graf AB, Altmann F, Koellensperger G, Hann S, Sauer M, Mattanovich D, Gasser B.">Rußmayer,</a> BMC biology 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...ZWF1 PP7435_Chr2-0993/ PAS_chr2-1_0308 Glucose-6-phosphate dehydrogenase not changed not changed no no no WPVTRPDVLHKM SOL3 PP7435_Chr3-0037/ <B><span style="color: red;">PAS_chr3_1126</span></B> 6-phosphogluconolactonase not changed n.q. no no Twilight zone ALSGVSVSTSKY GND2 PP7435_Chr3-0944/ PAS_chr3_0277 6-phosphogluconate dehydrogenase not changed not changed yes yes no KGGNVSASTYDA Pentose phosphate pathway non-oxidative branch RPE1-1 PP7435_Chr3-0771 D-ribulose-5-phosphate...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::SPCC16C4.10')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=6577685">SPCC16C4.10</a> 6-phosphogluconolactonase (predicted) (RefSeq) from <i>Schizosaccharomyces pombe</i><BR><a title="7:249/255 of query is similar to 6:245/257 of hit (E = 1e-37,  157 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=SPCC16C4.10&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS6577685" style="font-family: sans-serif; font-size: smaller;">37% identity, 95% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::SPCC16C4.10')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5990653">The contribution of non-essential Schizosaccharomyces pombe genes to fitness in response to altered nutrient supply and target of rapamycin activity</a><br /><small><a title="Lie S, Banks P, Lawless C, Lydall D, Petersen J.">Lie,</a> Open biology 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...LSM8 NO 7.805 0.01662 SPAC3H5.08c unknown human WDR44 downstream effector for RAB11 WDR44 7.764 0.02212 <B><span style="color: red;">SPCC16C4.10</span></B> carbohydrate metabolic process PGLS 7.565 0.03079 msn5 nucleocytoplasmic transport XPO5 NO 7.185 0.02894 SPCC31H12.03c nucleocytoplasmic transport HNRNPUL1 7.077 0.0003582 hht1 chromatin organization HIST3H3 NO 6.83 0.001063 mug161 mRNA cis splicing...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::SPCC16C4.10')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4190384">Genome-wide metabolic re-annotation of Ashbya gossypii: new insights into its metabolism through a comparative analysis with Saccharomyces cerevisiae and Kluyveromyces lactis</a><br /><small><a title="Gomes D, Aguiar TQ, Dias O, Ferreira EC, Domingues L, Rocha I.">Gomes,</a> BMC genomics 2014 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...K. lactis , the AER268W gene from A. gossypii has reported similarities in AGD to <B><span style="color: red;">SPCC16C4.10</span></B> from Schizosaccharomyces pombe , which is associated to the referred enzyme. This phenotype was experimentally confirmed by a positive growth of A. gossypii on D-glucosamine (a product of chitin degradation)...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::SPCC16C4.10')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3713068">Cadmium-induced proteome remodeling regulated by Spc1/Sty1 and Zip1 in fission yeast</a><br /><small><a title="Guo L, Ghassemian M, Komives EA, Russell P.">Guo,</a> Toxicological sciences : an official journal of the Society of Toxicology 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...In growth assays, we found that <B><span style="color: red;">SPCC16C4.10</span></B> (6-phosphogluconolactonase) from the oxPPP pathway and SPCC794.10 (UTP-glucose-1phosphate uridylyltransferase) from...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...oxPPP and trehalose metabolism pathways. Mutants lacking <B><span style="color: red;">SPCC16C4.10</span></B> and SPCC794.10 were sensitive to cadmium. displayed large changes involving proteins in...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::SPCC16C4.10')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2563147">A genome-wide screen of genes involved in cadmium tolerance in Schizosaccharomyces pombe</a><br /><small><a title="Kennedy PJ, Vashisht AA, Hoe KL, Kim DU, Park HO, Hayles J, Russell P.">Kennedy,</a> Toxicological sciences : an official journal of the Society of Toxicology 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...S. cerevisiae ortholog SPAC1782.05 RRD2/YPL152W <B><span style="color: red;">SPCC16C4.10</span></B> SPAC1805.07c SOL1/YNR034W None SPBC27.02c SPBC800.09 SPAC13C5.04 ASK1/YKL052C SCD6/YPR129W YLR126C...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::AFLA_080390')" href="http://www.ncbi.nlm.nih.gov/protein/XP_002372956.1">AFLA_080390</a> 6-phosphogluconolactonase, putative from <i>Aspergillus flavus NRRL3357</i><BR><a title="4:247/255 of query is similar to 8:254/266 of hit (E = 9e-36,  150 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=AFLA_080390&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_002372956.1" style="font-family: sans-serif; font-size: smaller;">39% identity, 92% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AFLA_080390')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6642104">The anti-aflatoxigenic mechanism of cinnamaldehyde in Aspergillus flavus</a><br /><small><a title="Wang P, Ma L, Jin J, Zheng M, Pan L, Zhao Y, Sun X, Liu Y, Xing F.">Wang,</a> Scientific reports 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...pathway were up-regulated exposure to cinnamaldehyde including AFLA_041580 (estradiol 17 beta-dehydrogenase), AFLA_115890 (acyl-CoA oxidase) and <B><span style="color: red;">AFLA_080390</span></B> (6-phosphogluconolactonase Sol). These results suggested that the down-regulation of fatty acids -oxidation and the up-regulation of pentose phosphate pathway were also associated with the anti-aflatoxigenic mechanism of cinnamaldehyde. Cinnamaldehyde was...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AFLA_080390')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6071284">The Antioxidant Gallic Acid Inhibits Aflatoxin Formation in Aspergillus flavus by Modulating Transcription Factors FarB and CreA</a><br /><small><a title="Zhao X, Zhi QQ, Li JY, Keller NP, He ZM.">Zhao,</a> Toxins 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...genes responsible for the pentose phosphate pathway were upregulated by GA treatment, such as AFLA_086620, <B><span style="color: red;">AFLA_080390</span></B>, and AFLA_036840 ( Table 2 ) encoding proteins homologous to Zwf1, Sol3, and Gnd1 in Candida albicans , respectively [ 27 ]. Pentose phosphate pathway activity has been linked to...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...down no zwf1 (AFLA_086620) 0.6412 0.4931 up no 0.7590 6.7 10 3 up no sol3 (<B><span style="color: red;">AFLA_080390</span></B>) 0.2110 0.8837 up no 1.0991 1.0 10 4 up yes gnd1 (AFLA_036840) 0.3718 0.7936 up no 0.9072 1.3 10 3 up no a : 0.2% ( w / v )...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::AOL_s00083g229')" href="http://www.ncbi.nlm.nih.gov/protein/XP_011123710.1">AOL_s00083g229</a> hypothetical protein from <i>Arthrobotrys oligospora ATCC 24927</i><BR><a title="1:244/255 of query is similar to 1:241/256 of hit (E = 2e-34,  146 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=AOL_s00083g229&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_011123710.1" style="font-family: sans-serif; font-size: smaller;">38% identity, 95% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AOL_s00083g229')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3164635">Genomic and proteomic analyses of the fungus Arthrobotrys oligospora provide insights into nematode-trap formation</a><br /><small><a title="Yang J, Wang L, Ji X, Feng Y, Li X, Zou C, Xu J, Ren Y, Mi Q, Wu J, Liu S, Liu Y, Huang X, Wang H, Niu X, Li J, Liang L, Luo Y, Ji K, Zhou W, Yu Z, Li G, Liu Y, Li L, Qiao M, Feng L, Zhang KQ.">Yang,</a> PLoS pathogens 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...5. AOL_s00112g89; 6. AOL_s00004g627; 7. AOL_s00109g17; 8. AOL_s00110g144; 9. AOL_s00006g284; 10. AOL_s00078g394; 11. AOL_s00076g83; 12. <B><span style="color: red;">AOL_s00083g229</span></B>; 13. AOL_s00054g87; 14. AOL_s00004g628; 15. AOL_s00004g426; 16. AOL_s00054g899; 17. AOL_s00210g140; 18. AOL_s00112g112; 19. AOL_s00170g104; 20. AOL_s00043g45; 21. AOL_s00110g24; 22. AOL_s00075g130; 23. AOL_s00075g141; 24. AOL_s00004g494; 25. AOL_s00079g361; 26. AOL_s00215g818; 27. AOL_s00054g909;...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="SwissProt" onmousedown="logger(this, 'curated::SOL2_YEAST / P37262')" href="http://www.uniprot.org/uniprot/P37262">SOL2_YEAST / P37262</a> <b>6-phosphogluconolactonase-like protein 2; Suppressor of LOS1</b> from <i>Saccharomyces cerevisiae (strain ATCC 204508 / S288c) (Baker's yeast)</i> (see <a onmousedown="logger(this, 'curatedpaper::SOL2_YEAST / P37262')" href="http://www.ncbi.nlm.nih.gov/pubmed/14562095,14562106,15454531">3 papers</a>)<BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::YCR073W-A')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=6581300">YCR073W-A</a> Protein with a possible role in tRNA export; shows similarity to 6-phosphogluconolactonase non-catalytic domains but does not exhibit this enzymatic activity; homologous to Sol1p, Sol3p, and Sol4p (RefSeq) from <i>Saccharomyces cerevisiae</i><BR><a title="37:247/255 of query is similar to 88:296/315 of hit (E = 1e-32,  140 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=SOL2_YEAST / P37262&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS6581300" style="font-family: sans-serif; font-size: smaller;">39% identity, 67% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><b>function:</b> May be involved in regulation of tRNA subcellular distribution. <LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YCR073W-A')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5224485">A genome wide dosage suppressor network reveals genomic robustness</a><br /><small><a title="Patra B, Kon Y, Yadav G, Sevold AW, Frumkin JP, Vallabhajosyula RR, Hintze A, Østman B, Schossau J, Bhan A, Marzolf B, Tamashiro JK, Kaur A, Baliga NS, Grayhack EJ, Adami C, Galas DJ, Raval A, Phizicky EM, Ray A.">Patra,</a> Nucleic acids research 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...name Mutation(s) suppressed by paralog 2 YBR181C PRS6B* cdc24, cdc15 YPL090C RPS6A* cdc24, cdc15, poc4 <B><span style="color: red;">YCR073W-A</span></B> SOL2 cdc26 YNR034W SOL1 cdc26 YDR099W BMH2* cdc25 YER177W BMH1* cdc25 YDR471W RPL27B* pol5 YHR010W RPL27A* pol5, cdc25 YER056C-A RPL34A* cdc48 YIL052C RPL34B* cdc48 YER131W RPS26B* vrp1 YGL189C RPL26A* vrp1...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YCR073W-A')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4524479">Estimating the Fitness Effects of New Mutations in the Wild Yeast Saccharomyces paradoxus</a><br /><small><a title="Koufopanou V, Lomas S, Tsai IJ, Burt A.">Koufopanou,</a> Genome biology and evolution 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...or extensions, all within 50 bp from either end: YCL068C, YCL049C, YCL001W-B, YCR015C, YCR018C, YCR044C, <B><span style="color: red;">YCR073W-A</span></B>, YCR092C; multiple consecutive start codons in S. cerevisiae and only one in S. paradoxus : YCR038C, YCR042C, YCR073C). Finally, in the coding region of YCR028C there is polymorphism among European...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::YCR073W-A')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3413506">Yeast 5 - an expanded reconstruction of the Saccharomyces cerevisiae metabolic network</a><br /><small><a title="Heavner BD, Smallbone K, Barker B, Mendes P, Walker LP.">Heavner,</a> BMC systems biology 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...frames which are present in Yeast 4 but not Yeast 5 are: YAL014C, YAL030W, YAR042W, <B><span style="color: red;">YCR073W-A</span></B>, YDL019C, YDR313C, YDR331W, YDR468C, YEL011W, YEL013W, YER093C, YFR055W, YGR199W, YHR005C, YHR073W, YMR068W, YNL006W, YNR034W, YOL078W, YPL145C, YPL275W, YPL276W, YOR237W, YIL105C, YJL058C, YJR160C, YKL203C, YML082W, and YNL047C. These open reading frames...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::FFUJ_02823')" href="http://www.ncbi.nlm.nih.gov/protein/XP_023427926.1">FFUJ_02823</a> probable multicopy suppressor of los1-1 from <i>Fusarium fujikuroi IMI 58289</i><BR><a title="7:247/255 of query is similar to 11:248/260 of hit (E = 2e-32,  139 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=FFUJ_02823&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_023427926.1" style="font-family: sans-serif; font-size: smaller;">35% identity, 93% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::FFUJ_02823')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3694855">Deciphering the cryptic genome: genome-wide analyses of the rice pathogen Fusarium fujikuroi reveal complex regulation of secondary metabolism and novel metabolites</a><br /><small><a title="Wiemann P, Sieber CM, von Bargen KW, Studt L, Niehaus EM, Espino JJ, Huß K, Michielse CB, Albermann S, Wagner D, Bergner SV, Connolly LR, Fischer A, Reuter G, Kleigrewe K, Bald T, Wingfield BD, Ophir R, Freeman S, Hippler M, Smith KM, Brown DW, Proctor RH, Münsterkötter M, Freitag M, Humpf HU, Güldener U, Tudzynski B.">Wiemann,</a> PLoS pathogens 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...The gene codes of the 28 genes used for the calculation are: FFUJ_01340, FFUJ_01475, FFUJ_02452, <B><span style="color: red;">FFUJ_02823</span></B>, FFUJ_02928, FFUJ_02998, FFUJ_04170, FFUJ_04413, FFUJ_04421, FFUJ_04860, FFUJ_04893, FFUJ_05032, FFUJ_05937, FFUJ_06185, FFUJ_07582, FFUJ_08292, FFUJ_08315, FFUJ_08319, FFUJ_08541, FFUJ_08584, FFUJ_09575, FFUJ_09776, FFUJ_09852, FFUJ_13056, FFUJ_13472, FFUJ_13565, FFUJ_13760, FFUJ_13774. For phylogenetic analysis of SM genes,...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="SwissProt" onmousedown="logger(this, 'curated::SOL1_YEAST / P50278')" href="http://www.uniprot.org/uniprot/P50278">SOL1_YEAST / P50278</a> <b>6-phosphogluconolactonase-like protein 1</b> from <i>Saccharomyces cerevisiae (strain ATCC 204508 / S288c) (Baker's yeast)</i> (see <a onmousedown="logger(this, 'curatedpaper::SOL1_YEAST / P50278')" href="http://www.ncbi.nlm.nih.gov/pubmed/14562106,15454531">2 papers</a>)<BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::NP_014432, YNR034W')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=6585796">NP_014432, YNR034W</a> Protein with a possible role in tRNA export; shows similarity to 6-phosphogluconolactonase non-catalytic domains but does not exhibit this enzymatic activity; homologous to Sol2p, Sol3p, and Sol4p (RefSeq) from <i>Saccharomyces cerevisiae</i><BR><a title="37:247/255 of query is similar to 92:300/321 of hit (E = 3e-32,  139 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=SOL1_YEAST / P50278&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS6585796" style="font-family: sans-serif; font-size: smaller;">36% identity, 66% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><b>function:</b> May be involved in regulation of tRNA subcellular distribution. <LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_014432, YNR034W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5224485">A genome wide dosage suppressor network reveals genomic robustness</a><br /><small><a title="Patra B, Kon Y, Yadav G, Sevold AW, Frumkin JP, Vallabhajosyula RR, Hintze A, Østman B, Schossau J, Bhan A, Marzolf B, Tamashiro JK, Kaur A, Baliga NS, Grayhack EJ, Adami C, Galas DJ, Raval A, Phizicky EM, Ray A.">Patra,</a> Nucleic acids research 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...by paralog 2 YBR181C PRS6B* cdc24, cdc15 YPL090C RPS6A* cdc24, cdc15, poc4 YCR073W-A SOL2 cdc26 <B><span style="color: red;">YNR034W</span></B> SOL1 cdc26 YDR099W BMH2* cdc25 YER177W BMH1* cdc25 YDR471W RPL27B* pol5 YHR010W RPL27A* pol5, cdc25 YER056C-A RPL34A* cdc48 YIL052C RPL34B* cdc48 YER131W RPS26B* vrp1 YGL189C RPL26A* vrp1 YFR023W PES4 smc2...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_014432, YNR034W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5086634">Different Toxicity Mechanisms for Citrinin and Ochratoxin A Revealed by Transcriptomic Analysis in Yeast</a><br /><small><a title="Vanacloig-Pedros E, Proft M, Pascual-Ahuir A.">Vanacloig-Pedros,</a> Toxins 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Translational elongation factor EF-3 YML087C AIM33 6.2 1.01 10 4 Putative protein of unknown function <B><span style="color: red;">YNR034W</span></B> SOL1 6.2 7.19 10 7 Protein with a possible role in tRNA export YDR070C FMP16 6.1 3.24 10 4 Protein of unknown function YJR129C EFM3 6.1 4.06 10 2 S-adenosylmethionine-dependent...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_014432, YNR034W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3413506">Yeast 5 - an expanded reconstruction of the Saccharomyces cerevisiae metabolic network</a><br /><small><a title="Heavner BD, Smallbone K, Barker B, Mendes P, Walker LP.">Heavner,</a> BMC systems biology 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...YAR042W, YCR073W-A, YDL019C, YDR313C, YDR331W, YDR468C, YEL011W, YEL013W, YER093C, YFR055W, YGR199W, YHR005C, YHR073W, YMR068W, YNL006W, <B><span style="color: red;">YNR034W</span></B>, YOL078W, YPL145C, YPL275W, YPL276W, YOR237W, YIL105C, YJL058C, YJR160C, YKL203C, YML082W, and YNL047C. These open reading frames were removed from reaction annotations because of inadequate literature evidence supporting the Yeast 4...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_014432, YNR034W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3148214">Multiple independent fusions of glucose-6-phosphate dehydrogenase with enzymes in the pentose phosphate pathway</a><br /><small><a title="Stover NA, Dixon TA, Cavalcanti AR.">Stover,</a> PloS one 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Cryptosporidium . TBLASTN searches using both Zwf1p and Sol1p ( S. cerevisiae 6PGL homolog, NCBI: <B><span style="color: red;">NP_014432</span></B>), as well as the homologous proteins identified in other apicomplexans, were performed against the Cryptosporidium genome sequences to verify that the gene was not missed during gene model annotation. Eimeria...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_014432, YNR034W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3163424">On the formalization and reuse of scientific research</a><br /><small><a title="King RD, Liakata M, Lu C, Oliver SG, Soldatova LN.">King,</a> Journal of the Royal Society, Interface 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...0.014 0 R_G6PDA YGR248W 0.017 0 0.007 0 R_G6PDA YHR163W 0.222 0 0.005 0 R_G6PDA <B><span style="color: red;">YNR034W</span></B> 0.023 0 0.028 0 R_GLUN YIL033C 0.079 0 0.205 0 R_M1PD YNR073C 0.016 0 0.024 0 R_MACACI YLL060C 0.011 0 0.014 0 R_POLYAO2 YMR020W 0.016 0 0.023 0 R_PUNP1 YLR017W...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_014432, YNR034W')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2645918">Identification of human intracellular targets of the medicinal Herb St. John's Wort by chemical-genetic profiling in yeast</a><br /><small><a title="McCue PP, Phang JM.">McCue,</a> Journal of agricultural and food chemistry 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...completion of separation 11 YNL098C RAS2 sporulation 12 YGR162W TIF4631 ribosome biogenesis and assembly 13 <B><span style="color: red;">YNR034W</span></B> SOL1 tRNA export from nucleus 14 YER044C ERG28 ergosterol biosynthesis 15 YGR078C PAC10 tubulin folding 16 YGL249W ZIP2 synapsis 17 YIR033W MGA2 regulation of transcription from polymerase II promoter 18...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="SwissProt" onmousedown="logger(this, 'curated::6PGL_RAT / P85971')" href="http://www.uniprot.org/uniprot/P85971">6PGL_RAT / P85971</a> <b>6-phosphogluconolactonase; 6PGL; EC 3.1.1.31</b> from <i>Rattus norvegicus (Rat)</i> (see <a onmousedown="logger(this, 'curatedpaper::6PGL_RAT / P85971')" href="http://www.ncbi.nlm.nih.gov/pubmed/19343716">paper</a>)<BR><a title="7:251/255 of query is similar to 12:247/257 of hit (E = 3e-32,  139 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=6PGL_RAT / P85971&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=P85971" style="font-family: sans-serif; font-size: smaller;">34% identity, 95% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><b>function:</b> Hydrolysis of 6-phosphogluconolactone to 6-phosphogluconate. <BR>
<b>catalytic</b> <b>activity:</b> 6-phospho-D-glucono-1,5-lactone + H2O = 6-phospho-D-gluconate + H(+) <small>(<a href="https://www.rhea-db.org/reaction?id=12556">RHEA:12556</a>)</small><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::P85971')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3761203">Proteomic profiling and pathway analysis of the response of rat renal proximal convoluted tubules to metabolic acidosis</a><br /><small><a title="Schauer KL, Freund DM, Prenni JE, Curthoys NP.">Schauer,</a> American journal of physiology. Renal physiology 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...3.73 SpC SpC P98158 P04176 D4ACL3 D3ZPA9 Q5RKI0 <B><span style="color: red;">P85971</span></B> LRP2_RAT PH4H_RAT D4ACL3_RAT D3ZPA9_RAT WDR1_RAT 6PGL_RAT Lrp2 Pah Aco1 Ctsa Wdr1 Pgls 0.008 0.012 0.013...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::PAAG_05621')" href="http://www.ncbi.nlm.nih.gov/protein/XP_002792336.1">PAAG_05621</a> 6-phosphogluconolactonase from <i>Paracoccidioides lutzii Pb01</i><BR><a title="1:247/255 of query is similar to 5:254/266 of hit (E = 2e-31,  136 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=PAAG_05621&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_002792336.1" style="font-family: sans-serif; font-size: smaller;">34% identity, 93% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PAAG_05621')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4471430">Proteomic profile response of Paracoccidioides lutzii to the antifungal argentilactone</a><br /><small><a title="Prado RS, Bailão AM, Silva LC, de Oliveira CM, Marques MF, Silva LP, Silveira-Lacerda EP, Lima AP, Soares CM, Pereira M.">Prado,</a> Frontiers in microbiology 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...PAAG_06796 711.90 Associate energy conservation Cytochrome c PriAC=F2TJX0 PAAG_06268 1307.21 1.522 Glycolysis and gluconeogenesis 6-phosphogluconolactonase <B><span style="color: red;">PAAG_05621</span></B> 688.50 1.297 Glyoxylate cycle Malate synthase PAAG_04542 617.40 * Krebs cycle Succinyl-CoA: PAAG_05093 770.38 * Methyl citrate cycle 2-methylcitrate dehydratase PAAG_04559 20407.15 1.297 Oxidation of fatty acids Enoyl-CoA hydratase PAAG_06309...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PAAG_05621')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4014450">Transcriptional and proteomic responses to carbon starvation in Paracoccidioides</a><br /><small><a title="Lima Pde S, Casaletti L, Bailão AM, de Vasconcelos AT, Fernandes Gda R, Soares CM.">Lima,</a> PLoS neglected tropical diseases 2014 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...PAAG_02512 pyruvate decarboxylase # alcohol fermentation PAAG_08248 alcohol dehydrogenase # alcohol fermentation Pentose Phosphate pathway <B><span style="color: red;">PAAG_05621</span></B> 6-phosphogluconolactonase # pentose-phosphate shunt PAAG_05146 ribose 5-phosphate isomerase A # pentose-phosphate shunt CELL CYCLE and DNA PROCESSING PAAG_07608 DNA helicase # DNA repair PAAG_08917 histone H2a # DNA processing PAAG_07098...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::CNAG_02133')" href="http://www.ncbi.nlm.nih.gov/protein/XP_012049968.1">CNAG_02133</a> 6-phosphogluconolactonase from <i>Cryptococcus neoformans var. grubii H99</i><BR><a title="4:251/255 of query is similar to 10:269/309 of hit (E = 6e-31,  134 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=CNAG_02133&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_012049968.1" style="font-family: sans-serif; font-size: smaller;">35% identity, 80% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::CNAG_02133')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5959073">Cryptococcus neoformans can form titan-like cells in vitro in response to multiple signals</a><br /><small><a title="Trevijano-Contador N, de Oliveira HC, García-Rodas R, Rossi SA, Llorente I, Zaballos Á, Janbon G, Ariño J, Zaragoza Ó.">Trevijano-Contador,</a> PLoS pathogens 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...oxidative segment of the pentose-phosphate shunt were also induced, such as glucose-6-phosphate dehydrogenase (CNAG_03245), 6-phosphogluconolactonase (<B><span style="color: red;">CNAG_02133</span></B>), and two isoforms of 6-phosphogluconate dehydrogenase (CNAG_04099 and CNAG_07561). Finally, genes involved in trehalose metabolism, such as CNAG_03113, CNAG_03765, and CNAG_05292, encoding a trehalose phosphatase and two trehalose synthases, respectively,...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...7.8 1.7E-11 Pentose-phosphate shunt (p-value : 5 . 3E-04) CNAG_03245 glucose-6-phosphate deHase 0.6 2.8 3.2E-04 <B><span style="color: red;">CNAG_02133</span></B> 6-phosphogluconolactonase 0.6 2.6 8.7E-04 CNAG_01492 hypothetical 6-phosphogluconate deHase 0.5 2.7 1.0E-03 CNAG_04099 6-phosphogluconate deHase, decarboxylating 1.2 1.2E-02 3.3 3.8E-05 CNAG_07561 6-phosphogluconate deHase, decarboxylating 1 0.5 2.3 8.9E-03 Others CNAG_06313 phosphoglucomutase...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::CNAG_02133')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3524413">A defect in ATP-citrate lyase links acetyl-CoA production, virulence factor elaboration and virulence in Cryptococcus neoformans</a><br /><small><a title="Griffiths EJ, Hu G, Fries B, Caza M, Wang J, Gsponer J, Gates-Hollingsworth MA, Kozel TR, De Repentigny L, Kronstad JW.">Griffiths,</a> Molecular microbiology 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...TAGTGCCCG 3 16 CNAG_07561 6-phosphogluconate dehydrogenase AAGGTTGATG 1 14 CNAG_03146 alpha amylase GCACGGAGAT 5 13 <B><span style="color: red;">CNAG_02133</span></B> 6-phosphogluconolactonase CATTACTGCA 3 13 CNAG_01102 Glucose 1-dehydrogenase ATTCAAAAAA 1 13 CNAG_04640 ATP-citrate lyase, ACL1 ATAAGCTTTC 0 12 CNAG_07745 Mannitol-1-phosphate dehydrogenase GTCGTAGAGT 2 9 CNAG_03072 Enolase 3 TGTCAAAAAA 0 7 Transaldolase...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::PADG_07771')" href="http://www.ncbi.nlm.nih.gov/protein/XP_010763145.1">PADG_07771</a> 6-phosphogluconolactonase from <i>Paracoccidioides brasiliensis Pb18</i><BR><a title="1:247/255 of query is similar to 5:254/266 of hit (E = 7e-31,  134 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=PADG_07771&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_010763145.1" style="font-family: sans-serif; font-size: smaller;">35% identity, 93% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PADG_07771')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6371752">Proteomic Analysis of Paracoccidioides brasiliensis During Infection of Alveolar Macrophages Primed or Not by Interferon-Gamma</a><br /><small><a title="Chaves EGA, Parente-Rocha JA, Baeza LC, Araújo DS, Borges CL, de Oliveira MAP, Soares CMA.">Chaves,</a> Frontiers in microbiology 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...are ETC and ATP synthesis. TCA, tricarboxylic cicle acid; ETC, electron transport chain; PADG_04604, transketolase; <B><span style="color: red;">PADG_07771</span></B>, 6-phosphogluconolactonase; PADG_03651, 6-phosphogluconate dehydrogenase, decarboxylating 1; PADG_07420, transaldolase; PADG_08387, citrate synthase mitochondrial; PADG_11845, aconitate hydratase mitochondrial; PADG_00317, succinyl-CoA ligase subunit beta; PADG_04939, succinyl-CoA, 3-ketoacid-coenzyme A transferase; PADG_01762, oxoglutarate dehydrogenase; PADG_00052,...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PADG_07771')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5711425">Paracoccidioides brasiliensis presents metabolic reprogramming and secretes a serine proteinase during murine infection</a><br /><small><a title="Lacerda Pigosso L, Baeza LC, Vieira Tomazett M, Batista Rodrigues Faleiro M, Brianezi Dignani de Moura VM, Melo Bailão A, Borges CL, Alves Parente Rocha J, Rocha Fernandes G, Gauthier GM, Soares CMA.">Lacerda,</a> Virulence 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...synthase (PADG_04702), ECH enoyl-CoA hydratase (PADG_01209), ACD acyl-CoA dehydrogenase (PADG_07604) PLA phospholipase (PADG_05993), PGL phosphoglucolactonase (<B><span style="color: red;">PADG_07771</span></B>), TRX thioredoxins (PADG_03161 and PADG_05504), TrxR thioredoxin reductase (PADG_01551), SOD superoxide dismutase (PADG_07418 and PADG_01755), CCP cytochrome C peroxidase (PADG_03163). The MultiExperiment Viewer software V.4.8 ( www.tm4.org/mev/ ) was used...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="BRENDA" onmousedown="logger(this, 'curated::O95336')" href="http://www.brenda-enzymes.org/sequences.php?AC=O95336">O95336</a> <b>6-phosphogluconolactonase (EC 3.1.1.31)</b> from <i>Homo sapiens</i> (see <a onmousedown="logger(this, 'curatedpaper::O95336')" href="http://www.ncbi.nlm.nih.gov/pubmed/11457850,21443518">2 papers</a>)<BR><a title="MetaCyc" onmousedown="logger(this, 'curated::O95336')" href="https://metacyc.org/gene?orgid=META&amp;id=HS05370-MONOMER">O95336</a> <b>6-phosphogluconolactonase (EC 3.1.1.31)</b> from <i>Homo sapiens</i> (see <a onmousedown="logger(this, 'curatedpaper::O95336')" href="http://www.ncbi.nlm.nih.gov/pubmed/10518023">paper</a>)<BR><a title="7:251/255 of query is similar to 12:248/258 of hit (E = 1e-29,  130 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=O95336&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=O95336" style="font-family: sans-serif; font-size: smaller;">34% identity, 95% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::O95336')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6414445">A Pathogen and a Non-pathogen Spotted Fever Group Rickettsia Trigger Differential Proteome Signatures in Macrophages</a><br /><small><a title="Curto P, Santa C, Allen P, Manadas B, Simões I, Martinez JJ.">Curto,</a> Frontiers in cellular and infection microbiology 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...1.1.1.27 0.27 0.38 Pentose Phosphate Pathway Glucose-6-phosphate dehydrogenase (G6PD) P11413 1.1.149/1.1.1.343 0.22 0.29 6-phosphogluconolactonase (PGLS) <B><span style="color: red;">O95336</span></B> 3.1.1.31 0.80 0.52 Phosphogluconate dehydrogenase (PGD) P52209 1.1.1.44/1.1.1.343 0.48 0.25 Glucose-6-phosphate isomerase (GPI) P06744 5.3.1.9 0.41 0.40 Transketolase (TKT) P29401 2.2.1.1 0.64 0.60 Transaldolase 1 (TALDO1) P37837 2.2.1.2 0.57 0.43...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::O95336')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5907394">Young and Especially Senescent Endothelial Microvesicles Produce NADPH: The Fuel for Their Antioxidant Machinery</a><br /><small><a title="Bodega G, Alique M, Bohórquez L, Morán M, Magro L, Puebla L, Ciordia S, Mena MC, Arza E, Ramírez MR.">Bodega,</a> Oxidative medicine and cellular longevity 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...272 6 5 20.6 10 Glucose-6-phosphate 1-dehydrogenase P11413 G6PD 45 1 1 11.8 11 6-Phosphogluconolactonase <B><span style="color: red;">O95336</span></B> 6PGL Western blot 12 6-Phosphogluconate dehydrogenase, decarboxylating P52209 PGD 162 3 3 22.2 160 3 3 24.2 13 Transketolase P29401 TKT 238 7 4 14.4 185 6 4 19.3 14...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::O95336')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5431125">Subventricular zone involvement in Glioblastoma - A proteomic evaluation and clinicoradiological correlation</a><br /><small><a title="Gollapalli K, Ghantasala S, Kumar S, Srivastava R, Rapole S, Moiyadi A, Epari S, Srivastava S.">Gollapalli,</a> Scientific reports 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Gamma-synuclein O76070 0.39 0.28 0.71 67 Actin-related protein 2 P61160 2.01 1.42 0.71 68 6-phosphogluconolactonase <B><span style="color: red;">O95336</span></B> 2.21 1.56 0.70 69 Keratin, type I cytoskeletal 10 P13645 1.20 0.83 0.70 70 Heat shock protein beta-1 P04792 2.02 1.40 0.69 71 Peroxiredoxin-6 P30041 1.49 1.02 0.69 72 Trifunctional...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::O95336')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5548489">Estrogen receptor coregulator binding modulators (ERXs) effectively target estrogen receptor positive human breast cancers</a><br /><small><a title="Raj GV, Sareddy GR, Ma S, Lee TK, Viswanadhapalli S, Li R, Liu X, Murakami S, Chen CC, Lee WR, Mann M, Krishnan SR, Manandhar B, Gonugunta VK, Strand D, Tekmal RR, Ahn JM, Vadlamudi RK.">Raj,</a> eLife 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...homolog OS= Homo sapiens GN=PUS7 PE=1 SV=2 661 75186.30 20 14 26.30 1.00 7.98 10.99 <B><span style="color: red;">O95336</span></B> 6 PGL_HUMAN 6-phosphogluconolactonase OS= Homo sapiens GN=PGLS PE=1 SV=2 258 27601.60 13 8 41.50 0.99 6.00 6.04 Q8TD06 AGR3_HUMAN Anterior gradient protein three homolog OS= Homo sapiens GN=AGR3 PE=1 SV=1...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::O95336')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5382314">Decreased Levels of Foldase and Chaperone Proteins Are Associated with an Early-Onset Amyotrophic Lateral Sclerosis</a><br /><small><a title="Filareti M, Luotti S, Pasetto L, Pignataro M, Paolella K, Messina P, Pupillo E, Filosto M, Lunetta C, Mandrioli J, Fuda G, Calvo A, Chiò A, Corbo M, Bendotti C, Beghi E, Bonetto V.">Filareti,</a> Frontiers in molecular neuroscience 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...1.3 16 P00367 Glutamate dehydrogenase 1, mitochondrial 1.4 25 P11172 Uridine 5-monophosphate synthase 0.7 31 <B><span style="color: red;">O95336</span></B> 6-Phosphogluconolactonase 1.3 Inflammatory response 39a-b P06702 Protein S100 A9 0.7 42a-b P05109 Protein S100 A8 0.6 26 P04083 Annexin A1 0.6 Signal transduction 27 P63104 14-3-3 protein zeta/delta 1.3 28a...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::O95336')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4773818">Pharmacoproteomic analysis reveals that metapristone (RU486 metabolite) intervenes E-cadherin and vimentin to realize cancer metastasis chemoprevention</a><br /><small><a title="Yu S, Yan C, Yang X, He S, Liu J, Qin C, Huang C, Lu Y, Tian Z, Jia L.">Yu,</a> Scientific reports 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...dehydrogenase 16 5.338** 108 751 30.1 P07384 Calpain-1 catalytic subunit 19 7.247* 109 564 62 <B><span style="color: red;">O95336</span></B> 6-phosphogluconolactonase 11 11.246** 110 176 14.1 Q9H9T3-2 Isoform 2 of Elongator complex protein 3 6 5.245* 111 2382 35.4 Q53F91 Villin 1 variant 27 3.245** 112 247 31.4 B4DP27 cDNA...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::O95336')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5090863">Isolation and mass spectrometry analysis of urinary extraexosomal proteins</a><br /><small><a title="Hildonen S, Skarpen E, Halvorsen TG, Reubsaet L.">Hildonen,</a> Scientific reports 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...exosome (extracellular) na hydrolase glucose metabolisme Glycogen storage disease 10 (lysosomal storage disease) Y/Y PGLS <B><span style="color: red;">O95336</span></B> 6-phosphogluconolactonase 1133 P cytoplasma na hydrolase carbohydrate metabolisme Diabetic nephropathy Y/Y ALDOB P05062 Fructose-bisphosphate aldolase B 1327 P cytoplasma na lyase glycolysis Hereditary fructose intolerance Y/Y DCXR Q7Z4W1 L-xylulose reductase...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::O95336')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3796057">An in-depth comparison of the male pediatric and adult urinary proteomes</a><br /><small><a title="Froehlich JW, Vaezzadeh AR, Kirchner M, Briscoe AC, Hofmann O, Hide W, Steen H, Lee RS.">Froehlich,</a> Biochimica et biophysica acta 2014 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Q96DA0 Zymogen granule protein 16 homolog B 6.2E+02 Ad 1.5 P00450 Ceruloplasmin 3.9E+02 Ad 1.5 <B><span style="color: red;">O95336</span></B> 6-phosphogluconolactonase 6.9E+01 Ad 1.5 P05062 Fructose-bisphosphate aldolase B 3.2E+01 Ad 1.5 Q8WZ75 Roundabout homolog 4 5.5E+02 Ad 1.5 Q14393 Growth arrest-specific protein 6 3.0E+01 Ad 1.5 O00468 Agrin 6.3E+02 Ad...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::XP_003194407')" href="http://www.ncbi.nlm.nih.gov/protein/XP_003194407.1">XP_003194407</a> 6-phosphogluconolactonase, putative from <i>Cryptococcus gattii WM276</i><BR><a title="7:251/255 of query is similar to 55:311/351 of hit (E = 1e-29,  130 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=XP_003194407&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_003194407.1" style="font-family: sans-serif; font-size: smaller;">36% identity, 70% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_003194407')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4132117">Vaccine-mediated immune responses to experimental pulmonary Cryptococcus gattii infection in mice</a><br /><small><a title="Chaturvedi AK, Hameed RS, Wozniak KL, Hole CR, Leopold Wager CM, Weintraub ST, Lopez-Ribot JL, Wormley FL.">Chaturvedi,</a> PloS one 2014 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...hydratase 48 AFR93622 5 16 7 Phosphopyruvate hydratase 48 AFR97782 12 32 8 6-phosphogluconolactonase 40 <B><span style="color: red;">XP_003194407</span></B> 8 21 9 Mitochondrial precursor 32 XP_003194323 4 16 10 Hypothetical protein CGB_M1040C 15 XP_003197291 4 36 11 Hypothetical protein CGB_M1040C 15 XP_003197291 8 48 12 Translation initiation factor 5a...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::CG17333')" href="http://www.ncbi.nlm.nih.gov/protein/NP_572656.1">CG17333</a> uncharacterized protein from <i>Drosophila melanogaster</i><BR><a title="8:251/255 of query is similar to 14:239/243 of hit (E = 1e-26,  120 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=CG17333&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=NP_572656.1" style="font-family: sans-serif; font-size: smaller;">33% identity, 96% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::CG17333')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4869932">The Drosophila HNF4 nuclear receptor promotes glucose-stimulated insulin secretion and mitochondrial function in adults</a><br /><small><a title="Barry WE, Thummel CS.">Barry,</a> eLife 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...mutants, along with phosphoglucomutase (pgm ), which is involved in glycogen metabolism, and transaldolase and <B><span style="color: red;">CG17333</span></B> , which are involved in the pentose phosphate shunt. Additionally, the gluconeogenesis genes Pyruvate carboxylase (CG1514 ) and Phosphoenolpyruvate carboxykinase (Pepck, CG17725 ) show reduced expression in mutant animals, similar...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::CG17333')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3352090">Drosophila poly suggests a novel role for the Elongator complex in insulin receptor-target of rapamycin signalling</a><br /><small><a title="Bolukbasi E, Vass S, Cobbe N, Nelson B, Simossis V, Dunbar DR, Heck MM.">Bolukbasi,</a> Open biology 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...FBgn0023477 // CG2827 (1.27) transaldolase rep FBgn0004654 // CG3724 (2.15) phosphogluconate dehydrogenase rep FBgn0030239 // <B><span style="color: red;">CG17333</span></B> (1.34) 6-phosphogluconolactonase pred FBgn0037607 // CG8036 (1.39) transketolase pred GO:0006769 nicotinamide metabolic process same five genes 64.4 GO:0019362 pyridine nucleotide metabolic process same five genes 60.1 GO:0006006 glucose metabolic process...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::Synpcc7942_0529')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=1174062">Synpcc7942_0529</a> 6-phosphogluconolactonase (NCBI) from <i>Synechococcus elongatus PCC 7942</i><BR><a title="7:251/255 of query is similar to 1:239/239 of hit (E = 6e-25,  114 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=Synpcc7942_0529&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS1174062" style="font-family: sans-serif; font-size: smaller;">32% identity, 96% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::Synpcc7942_0529')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4672817">The essential gene set of a photosynthetic organism</a><br /><small><a title="Rubin BE, Wetmore KM, Price MN, Diamond S, Shultzaberger RK, Lowe LC, Curtin G, Arkin AP, Deutschbauer A, Golden SS.">Rubin,</a> Proceedings of the National Academy of Sciences of the United States of America 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...pathway encode 6-phosphogluconolactonase (pgl; <B><span style="color: red;">SynPCC7942_0529</span></B>), glucose6-phosphate 1-dehydrogenase (zwf; SynPCC7942_2334), and 6-phosphogluconate dehydrogenase...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::GRMZM2G148769, umc2374')" href="http://www.ncbi.nlm.nih.gov/protein/NP_001152205.1">GRMZM2G148769, umc2374</a> uncharacterized protein LOC100285843 from <i>Zea mays</i><BR><a title="3:251/255 of query is similar to 74:321/324 of hit (E = 4e-23,  108 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=GRMZM2G148769, umc2374&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=NP_001152205.1" style="font-family: sans-serif; font-size: smaller;">33% identity, 77% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::GRMZM2G148769, umc2374')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4668103">Proteomic Analysis of Silk Viability in Maize Inbred Lines and Their Corresponding Hybrids</a><br /><small><a title="Ma Z, Qin Y, Wang Y, Zhao X, Zhang F, Tang J, Fu Z.">Ma,</a> PloS one 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...10:1073115810736508 6.70E-120 68 Y/N A XZ-D 8 gi|224029461 unknown [Zea mays] 28955.8/5.08 386 100 13 <B><span style="color: red;">GRMZM2G148769</span></B> 2:221504148221501546 3.20E-88 136 Y/N A LZ-D 8 gi|224029461 unknown [Zea mays] 28955.8/5.08 195 100 9 <B><span style="color: red;">GRMZM2G148769</span></B> 2:221504148221501546 3.20E-88 181 Y/N A XZ-D 10 gi|224029461 unknown [Zea mays] 28955.8/5.08 299 100...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...2:221504148221501546 3.20E-88 256 Y/N A XZ-D 12 gi|224029461 unknown [Zea mays] 28955.8/5.08 201 100 12 <B><span style="color: red;">GRMZM2G148769</span></B> 2:221504148221501546 3.20E-88 276 Y/N + LZ-D 12 gi|224029461 unknown [Zea mays] 28955.8/5.08 238 100 9 <B><span style="color: red;">GRMZM2G148769</span></B> 2:221504148221501546 3.20E-88 167 2.6 + XZ-D 10 gi|194697638 unknown [Zea mays] 64089/4.8 270 100...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::GRMZM2G148769, umc2374')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3691554">Proteomic comparison of basal endosperm in maize miniature1 mutant and its wild-type Mn1</a><br /><small><a title="Silva-Sanchez C, Chen S, Zhu N, Li QB, Chourey PS.">Silva-Sanchez,</a> Frontiers in plant science 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...NF C0PGM6 Uncharacterized LOC100383510; K03064 26S proteasome regulatory subunit T4 1.37 0.04 zma:100383510 zma03050 B6UAK0 <B><span style="color: red;">umc2374</span></B>; LOC100285843 (EC:3.1.1.31); K01057 6-phosphogluconolactonase [EC:3.1.1.31] 1.49 0.3 zma:100285843 zma00030, zma01100, zma01110 B8A0Q6 TIDP3457; LOC100279930; K00234 succinate dehydrogenase (ubiquinone) flavoprotein subunit [EC:1.3.5.1] 1.51 0.04 zma:100279930 zma00020, zma00190, zma01100, zma01110 B6TYM9 cl13598_-2;...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...to alpha-galactosidase from Oryza sativa (Q9FXT4) 1.42 0.09 EC 3.2.1.22 zma00052, zma0056, zma00600, zma00603 B6UAK0 <B><span style="color: red;">umc2374</span></B>; LOC100285843 (EC:3.1.1.31); K01057 6-phosphogluconolactonase [EC:3.1.1.31] 1.49 0.3 zma:100285843 zma00030, zma01100, zma01110 B4G1X1 Uncharacterized LOC100274554; K07407 alpha-galactosidase [EC:3.2.1.22] 1.59 0.05 zma:100274554 zma00561, zma00600, zma00603 KEGG pathway identifiers: zma00010:glycolysis/gluconeogenesis; zma00020:citrate cycle (TCA...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::Bm1_33900')" href="http://www.ncbi.nlm.nih.gov/protein/XP_001898227.1">Bm1_33900</a> 6-phosphogluconolactonase family protein from <i>Brugia malayi</i><BR><a title="1:247/255 of query is similar to 13:257/268 of hit (E = 8e-23,  107 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=Bm1_33900&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_001898227.1" style="font-family: sans-serif; font-size: smaller;">32% identity, 92% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::Bm1_33900')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3501679">Identification of genes containing ecdysone response elements in the genome of Brugia malayi</a><br /><small><a title="Liu C, Enright T, Tzertzinis G, Unnasch TR.">Liu,</a> Molecular and biochemical parasitology 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...hypothetical protein AF, EE, MF, L3 unknown Bm1_32410 1841 4-hydroxyphenylpyruvate dioxygenase family protein Constitutive metabolism <B><span style="color: red;">Bm1_33900</span></B> 659 6 phosphogluconolactonase family protein Constitutive metabolism Bm1_38530 1083 PET Domain containing protein AF, EE, MF signalling Bm1_42085 2818 Tristetraproline AF, EE, MF, L3 transcription Bm1_42090 1198 MGC79044 protein-related AM...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::CNAG_01216')" href="http://www.ncbi.nlm.nih.gov/protein/XP_012049395.1">CNAG_01216</a> 6-phosphogluconolactonase from <i>Cryptococcus neoformans var. grubii H99</i><BR><a title="7:249/255 of query is similar to 14:247/294 of hit (E = 9e-23,  107 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=CNAG_01216&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_012049395.1" style="font-family: sans-serif; font-size: smaller;">30% identity, 83% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::CNAG_01216')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6867901">A Fungal Arrestin Protein Contributes to Cell Cycle Progression and Pathogenesis</a><br /><small><a title="Telzrow CL, Nichols CB, Castro-Lopez N, Wormley FL, Alspaugh JA.">Telzrow,</a> mBio 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...CNAG_07561 6-Phosphogluconate dehydrogenase, decarboxylating I 5.3 18.8 CNAG_06175 26S proteasome, regulatory subunit N2 4.7 0.0 <B><span style="color: red;">CNAG_01216</span></B> 6-Phosphogluconolactonase 4.0 0.0 CNAG_00602 Eukaryotic translation initiation factor 3, subunit I 4.0 0.0 CNAG_00136 Ubiquitin-activating enzyme E1 4.0 0.0 CNAG_06666 -1,4-Glucan phosphorylase 3.7 0.0 CNAG_02565 Homoaconitase, mitochondrial 3.7 0.0 CNAG_02035...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::NP_775547')" href="http://www.ncbi.nlm.nih.gov/protein/NP_775547.2">NP_775547</a> GDH/6PGL endoplasmic bifunctional protein isoform 1 precursor from <i>Mus musculus</i><BR><a title="37:247/255 of query is similar to 591:794/797 of hit (E = 5e-19, 95.5 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=NP_775547&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=NP_775547.2" style="font-family: sans-serif; font-size: smaller;">32% identity, 26% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_775547')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6105561">Enhanced hexose-6-phosphate dehydrogenase expression in adipose tissue may contribute to diet-induced visceral adiposity.</a><br /><small><a title="Liu L, Wang Y, Wang J, Dong Y, Chang S, Liu X, Lutfy K, Chen H, Friedman TC, Jiang M, Liu Y">Liu,</a> International journal of obesity (2005) 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: Studied expression levels and signal pathways involved with hexose-6-phosphate dehydrogenase in mouse model of diet induced visceral adiposity.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_775547')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5601182">Cellular and genetic models of H6PDH and 11β-HSD1 function in skeletal muscle.</a><br /><small><a title="Zielinska AE, Fletcher RS, Sherlock M, Doig CL, Lavery GG">Zielinska,</a> Cell biochemistry and function 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: We envisage these data and models finding utility when investigating the muscle-specific functions of the 11beta-HSD1/G6PT/H6PDH triad.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_775547')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3948972">Transgenic overexpression of hexose-6-phosphate dehydrogenase in adipose tissue causes local glucocorticoid amplification and lipolysis in male mice.</a><br /><small><a title="Wang Y, Liu L, Du H, Nagaoka Y, Fan W, Lutfy K, Friedman TC, Jiang M, Liu Y">Wang,</a> American journal of physiology. Endocrinology and metabolism 2014 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: These results suggest that increased H6PDH expression specifically in adipose tissue is sufficient to cause intra-adipose glucocorticoid production and adverse metabolic phenotypes.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_775547')" href="http://doi.org/10.1507/endocrj.ej13-0528">Mechanism of repression of 11β-hydroxysteroid dehydrogenase type 1 by growth hormone in 3T3-L1 adipocytes.</a><br /><small><a title="Muraoka T, Hizuka N, Fukuda I, Ishikawa Y, Ichihara A">Muraoka,</a> Endocrine journal 2014 (<a onmousedown="logger(this, 'pb::NP_775547')" href="http://www.ncbi.nlm.nih.gov/pubmed/24759003">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: Data suggest that Gh (growth hormone) represses H6pd through locally produced Igf1 (insulin-like growth factor 1); Gh directly represses Hsd11b1 (11-beta-hydroxysteroid dehydrogenase 1) mRNA rather than acting via Igf1 receptor.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_775547')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3795617">Tissue-specific dysregulation of hexose-6-phosphate dehydrogenase and glucose-6-phosphate transporter production in db/db mice as a model of type 2 diabetes.</a><br /><small><a title="Wang Y, Nakagawa Y, Liu L, Wang W, Ren X, Anghel A, Lutfy K, Friedman TC, Liu Y">Wang,</a> Diabetologia 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: Findings suggest that increased hepatic H6PDH and G6PT production contribute to 11beta-HSD1 upregulation of local glucocorticoid action that may be related to the development of type 2 diabetes.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_775547')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3219053">Hexose-6-phosphate dehydrogenase contributes to skeletal muscle homeostasis independent of 11β-hydroxysteroid dehydrogenase type 1.</a><br /><small><a title="Semjonous NM, Sherlock M, Jeyasuria P, Parker KL, Walker EA, Stewart PM, Lavery GG">Semjonous,</a> Endocrinology 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: These data are consistent with an 11beta-HSD1-independent function for H6PDH in which sarcoplasmic reticulum G6P metabolism and NAD/NADH redox status are important for maintaining muscle homeostasis.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_775547')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2879949">High glucose inhibits glucose-6-phosphate dehydrogenase, leading to increased oxidative stress and beta-cell apoptosis.</a><br /><small><a title="Zhang Z, Liew CW, Handy DE, Zhang Y, Leopold JA, Hu J, Guo L, Kulkarni RN, Loscalzo J, Stanton RC">Zhang,</a> FASEB journal : official publication of the Federation of American Societies for Experimental Biology 2010 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: High-glucose-mediated decrease in G6PD activity may provide a mechanistic explanation for the gradual loss of beta cells in patients with diabetes.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NP_775547')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2329282">Lack of hexose-6-phosphate dehydrogenase impairs lipid mobilization from mouse adipose tissue.</a><br /><small><a title="Bujalska IJ, Hewitt KN, Hauton D, Lavery GG, Tomlinson JW, Walker EA, Stewart PM">Bujalska,</a> Endocrinology 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: Lack of hexose-6-phosphate dehydrogenase impairs lipid mobilization from mouse adipose tissue.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a href="litSearch.cgi?more=NP_775547.2">More</a></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::XP_006239521')" href="http://www.ncbi.nlm.nih.gov/protein/XP_006239521.1">XP_006239521</a> GDH/6PGL endoplasmic bifunctional protein isoform X1 from <i>Rattus norvegicus</i><BR><a title="37:247/255 of query is similar to 583:786/789 of hit (E = 8e-19, 94.7 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=XP_006239521&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_006239521.1" style="font-family: sans-serif; font-size: smaller;">32% identity, 27% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006239521')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3741409">Direct regulation of glucose and not insulin on hepatic hexose-6-phosphate dehydrogenase and 11β-hydroxysteroid dehydrogenase type 1.</a><br /><small><a title="Fan Z, Du H, Zhang M, Meng Z, Chen L, Liu Y">Fan,</a> Molecular and cellular endocrinology 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: Activation of expression and activity of Hexose-6-phosphate dehydrogenase in primary hepatocytes was mediated through levels of glucose, not related to insulin levels so concluding that glucose instead of insulin directly regulates H6PDH.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006239521')" href="http://doi.org/10.1016/j.jsbmb.2011.05.006">Expression of hexose-6-phosphate dehydrogenase in rat tissues.</a><br /><small><a title="Marcolongo P, Senesi S, Giunti R, Csala M, Fulceri R, Bánhegyi G, Benedetti A">Marcolongo,</a> The Journal of steroid biochemistry and molecular biology 2011 (<a onmousedown="logger(this, 'pb::XP_006239521')" href="http://www.ncbi.nlm.nih.gov/pubmed/21620971">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: It was found that H6PD was present in all investigated tissues; both expression and activity remained within an order of magnitude. Correlation was found between the dehydrogenase activity and protein or mRNA levels.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006239521')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2219311">Hexose-6-phosphate dehydrogenase and 11beta-hydroxysteroid dehydrogenase-1 tissue distribution in the rat.</a><br /><small><a title="Gomez-Sanchez EP, Romero DG, de Rodriguez AF, Warden MP, Krozowski Z, Gomez-Sanchez CE">Gomez-Sanchez,</a> Endocrinology 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: H6PDH was amply expressed in most tissues examined in which 11beta-HSD1 was also expressed, with the notable exception of the renal interstitial cells</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006239521')" href="http://doi.org/10.1210/en.2006-1472">The glucose-6-phosphate transporter-hexose-6-phosphate dehydrogenase-11beta-hydroxysteroid dehydrogenase type 1 system of the adipose tissue.</a><br /><small><a title="Marcolongo P, Piccirella S, Senesi S, Wunderlich L, Gerin I, Mandl J, Fulceri R, Bánhegyi G, Benedetti A">Marcolongo,</a> Endocrinology 2007 (<a onmousedown="logger(this, 'pb::XP_006239521')" href="http://www.ncbi.nlm.nih.gov/pubmed/17303657">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: adipocytes are equipped with a functional G6PT-hexose-6-phosphate dehydrogenase-11beta-hydroxysteroid dehydrogenase type 1 system</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006239521')" href="http://doi.org/10.1093/jn/137.12.2616">Sucrose access differentially modifies 11beta-hydroxysteroid dehydrogenase-1 and hexose-6-phosphate dehydrogenase message in liver and adipose tissue in rats.</a><br /><small><a title="London E, Lala G, Berger R, Panzenbeck A, Kohli AA, Renner M, Jackson A, Raynor T, Loya K, Castonguay TW">London,</a> The Journal of nutrition 2007 (<a onmousedown="logger(this, 'pb::XP_006239521')" href="http://www.ncbi.nlm.nih.gov/pubmed/18029473">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: This study showed that sucrose can promote increased 11Beta-hydroxysteroid dehydrogenase-1 (11beta-HSD-1) and Hexose-6-phosphate dehydrogenase (H6PDH) message in mesenteric fat while concomitantly decreasing the enzyme messages in liver.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006239521')" href="http://doi.org/10.1016/j.mce.2005.10.018">Cooperativity between 11beta-hydroxysteroid dehydrogenase type 1 and hexose-6-phosphate dehydrogenase is based on a common pyridine nucleotide pool in the lumen of the endoplasmic reticulum.</a><br /><small><a title="Czegle I, Piccirella S, Senesi S, Csala M, Mandl J, Bánhegyi G, Fulceri R, Benedetti A">Czegle,</a> Molecular and cellular endocrinology 2006 (<a onmousedown="logger(this, 'pb::XP_006239521')" href="http://www.ncbi.nlm.nih.gov/pubmed/16337333">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: The results demonstrate the existence of a separate intraluminal pyridine nucleotide pool in the hepatic endoplasmic reticulum and a close cooperation between 11betaHSD1 and H6PDH based on their co-localization.</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="SwissProt" onmousedown="logger(this, 'curated::6PGL4_ARATH / Q9FIN1')" href="http://www.uniprot.org/uniprot/Q9FIN1">6PGL4_ARATH / Q9FIN1</a> <b>Probable 6-phosphogluconolactonase 4; 6PGL4; EC 3.1.1.31</b> from <i>Arabidopsis thaliana (Mouse-ear cress)</i> (see <a onmousedown="logger(this, 'curatedpaper::6PGL4_ARATH / Q9FIN1')" href="http://www.ncbi.nlm.nih.gov/pubmed/24008768">paper</a>)<BR><a title="BRENDA" onmousedown="logger(this, 'curated::Q9FIN1')" href="http://www.brenda-enzymes.org/sequences.php?AC=Q9FIN1">Q9FIN1</a> <b>6-phosphogluconolactonase (EC 3.1.1.31)</b> from <i>Arabidopsis thaliana</i> (see <a onmousedown="logger(this, 'curatedpaper::Q9FIN1')" href="http://www.ncbi.nlm.nih.gov/pubmed/19457984">paper</a>)<BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::AT5G24410')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=10106433">AT5G24410</a> glucosamine/galactosamine-6-phosphate isomerase-related (RefSeq) from <i>Arabidopsis thaliana</i><BR><a title="37:253/255 of query is similar to 43:259/261 of hit (E = 9e-19, 94.4 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=6PGL4_ARATH / Q9FIN1&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS10106433" style="font-family: sans-serif; font-size: smaller;">32% identity, 83% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><b>function:</b> Catalyzes the hydrolysis of 6-phosphogluconolactone to 6- phosphogluconate. <BR>
<b>catalytic</b> <b>activity:</b> 6-phospho-D-glucono-1,5-lactone + H2O = 6-phospho-D-gluconate + H(+) <small>(<a href="https://www.rhea-db.org/reaction?id=12556">RHEA:12556</a>)</small><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24410')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3142401">Divergent evolutionary fates of major photosynthetic gene networks following gene and whole genome duplications</a><br /><small><a title="Coate JE, Doyle JJ.">Coate,</a> Plant signaling & behavior 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...-0.39 0.37 AT3G49360 0.39 -0.51 AT5G24420 0.31 -0.33 <B><span style="color: red;">AT5G24410</span></B> -0.33 0.40 AT3G04790 0.84 -0.70 AT5G44520 0.70 -0.77 AT2G01290 0.68 -0.65 AT1G71100 -0.11 0.24...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24410')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1621065">Comprehensive transcriptional profiling of NaCl-stressed Arabidopsis roots reveals novel classes of responsive genes</a><br /><small><a title="Jiang Y, Deyholos MK.">Jiang,</a> BMC plant biology 2006 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...decreased (Table 3 ), including D-ribulose-5-phosphate 3-epimerase (At3g01850, At1g63290), ribose 5-phosphate isomerase (At2g01290, At3g04790), 6-phosphogluconolactonase (<B><span style="color: red;">At5g24410</span></B>, At3g02360), and glucose-6-phosphate-1-dehydrogenase (G6PDH) ( Additional File 1 ). The PPP is an important source of NADPH, especially in non-photosynthetic tissues, and therefore might be expected to have increased activity...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="SwissProt" onmousedown="logger(this, 'curated::6PGL3_ARATH / Q84WW2')" href="http://www.uniprot.org/uniprot/Q84WW2">6PGL3_ARATH / Q84WW2</a> <b>6-phosphogluconolactonase 3, chloroplastic; 6PGL3; Protein EMBRYO DEFECTIVE 2024; EC 3.1.1.31</b> from <i>Arabidopsis thaliana (Mouse-ear cress)</i> (see <a onmousedown="logger(this, 'curatedpaper::6PGL3_ARATH / Q84WW2')" href="http://www.ncbi.nlm.nih.gov/pubmed/15266054,17951448,19457984,23621281,24008768">5 papers</a>)<BR><a title="BRENDA" onmousedown="logger(this, 'curated::Q84WW2')" href="http://www.brenda-enzymes.org/sequences.php?AC=Q84WW2">Q84WW2</a> <b>6-phosphogluconolactonase (EC 3.1.1.31)</b> from <i>Arabidopsis thaliana</i> (see <a onmousedown="logger(this, 'curatedpaper::Q84WW2')" href="http://www.ncbi.nlm.nih.gov/pubmed/19457984">paper</a>)<BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::AT5G24400')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=10106432">AT5G24400</a> emb2024 (embryo defective 2024); 6-phosphogluconolactonase/ catalytic (RefSeq) from <i>Arabidopsis thaliana</i><BR><a title="RefSeq" onmousedown="logger(this, 'curated::EMB2024, NP_568445')" href="http://www.ncbi.nlm.nih.gov/protein/NP_568445.1">EMB2024, NP_568445</a> NagB/RpiA/CoA transferase-like superfamily protein from <i>Arabidopsis thaliana</i><BR><a title="35:249/255 of query is similar to 112:323/325 of hit (E = 3e-18, 92.8 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=6PGL3_ARATH / Q84WW2&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS10106432" style="font-family: sans-serif; font-size: smaller;">32% identity, 66% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><b>function:</b> Catalyzes the hydrolysis of 6-phosphogluconolactone to 6- phosphogluconate (PubMed:19457984). Involved in the regulation of cellular redox state; enzymatic activity is required for this function (PubMed:19457984). Required for sugar-dependent expression of nitrate assimilation genes in the nucleus of root cells (PubMed:23621281). <BR>
<b>catalytic</b> <b>activity:</b> 6-phospho-D-glucono-1,5-lactone + H2O = 6-phospho-D-gluconate + H(+) <small>(<a href="https://www.rhea-db.org/reaction?id=12556">RHEA:12556</a>)</small><BR>
<b>subunit:</b> Interacts with TRXM2. <BR>
<b>disruption</b> <b>phenotype:</b> Embryonic lethality when homozygous (PubMed:15266054, PubMed:19457984, PubMed:23621281). Embryo development arrest at cotyledon stage (PubMed:15266054). <LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24400')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6660741">The plastidial pentose phosphate pathway is essential for postglobular embryo development in Arabidopsis</a><br /><small><a title="Andriotis VME, Smith AM.">Andriotis,</a> Proceedings of the National Academy of Sciences of the United States of America 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...1 ) ( 26 ). One of the 5 PGL genes in Arabidopsis , PGL3 (<B><span style="color: red;">At5g24400</span></B>), encodes an enzyme with dual plastidial and peroxisomal localization ( 8 , 27 , 28 ), whereas the other 4 encode cytosolic proteins. Mutations eliminating PGL3 are embryo-lethal due to...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Eco-type Position At5g17630 XPT Xul5P/phosphate translocator 2 xpt-2 SAIL_378_C01 Col +68 xpt-4 GT_5_112515 Ler +1017 <B><span style="color: red;">At5g24400</span></B> PGL3 6-Phosphogluconolactonase 6 pgl3-1 SALK_005685 Col +975 pgl3-2 FLAG_219G10 WS +250 pgl3-3 emb2024 Col +215/437 At1g64190 PGD1 6-Phosphogluconate dehydrogenase 7 pgd1-1 GABI_762C02 Col +11 pgd1-2 SALK_121521 Col +1445 NA SALK_002712...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24400')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4047471">Protein abundance changes and ubiquitylation targets identified after inhibition of the proteasome with syringolin A</a><br /><small><a title="Svozil J, Hirsch-Hoffmann M, Dudler R, Gruissem W, Baerenfaller K.">Svozil,</a> Molecular & cellular proteomics : MCP 2014 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...AT5G02490 AT5G52440 AT5G60160 AT5G24650 AT1G32900 <B><span style="color: red;">AT5G24400</span></B> AT5G47860 AT1G64550 AT2G22170 AT2G25840 AT2G31810 AT2G35860 AT2G45060 AT2G47450 AT3G02770 AT3G15840...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24400')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3999085">Widespread dual targeting of proteins in land plants: when, where, how and why</a><br /><small><a title="Carrie C, Whelan J.">Carrie,</a> Plant signaling & behavior 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Chloroplasts and cytoplasm 6-Phosphogluconolactonase (PGL6) At5g23050 <B><span style="color: red;">At5g24400</span></B> Arabidopsis thaliana Acyl-activating enzyme 5 (AAE5) His triad family protein...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24400')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3355810">Defining the plant peroxisomal proteome: from Arabidopsis to rice</a><br /><small><a title="Kaur N, Hu J.">Kaur,</a> Frontiers in plant science 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...intermembrane space assembly machinery 40 SKL LOC_Os04g44550 PKL Carrie et al. ( 2010 ) 6PGL <B><span style="color: red;">At5g24400</span></B> 6-Phosphoglu conolactonase SKL LOC_Os08g43370 SSI Reumann et al. ( 2007 ) LACS7 At5g27600 Long-chain acyl-CoA synthetase 7 RLx 5 HI/SKL LOC_Os11g04980 RLx 5 HL/SKL Eubel et al. ( 2008 ),...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24400')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2944999">Indispensable Roles of Plastids in Arabidopsis thaliana Embryogenesis</a><br /><small><a title="Hsu SC, Belmonte MF, Harada JJ, Inoue K.">Hsu,</a> Current genomics 2010 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...At4g23100 C gamma-glutamylcysteine synthetase # (M) [ 88 ] [ 89 ] [ 90 ] <B><span style="color: red;">At5g24400</span></B> C 6-phosphogluconolactonase (M) [ 91 ] [ 91 ] At5g52920 C4 similar to pyruvate kinase isozyme G # (M) [ 92 ] [ 92 ] At2g19450 C4 diacylglycerol O-acyltransferase /...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24400')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2935873">Caesium and strontium accumulation in shoots of Arabidopsis thaliana: genetic and physiological aspects</a><br /><small><a title="Kanter U, Hauser A, Michalke B, Dräxl S, Schäffner AR.">Kanter,</a> Journal of experimental botany 2010 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...8.1 14.6 SrDW SrDW1.1 At1g43780 55.6 5.1 14.5 SrDW1.2 At1g71350 102.7 4.3 12 + SrDW5.1 <B><span style="color: red;">At5g24400</span></B> 26.8 4.4 12.3 Positive (+) effects indicate that Sq-1 alleles at that marker increased levels of caesium and strontium ions, negative () effects indicate that Sorbo alleles increased the caesium...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24400')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3243405">Peroxisome biogenesis and function</a><br /><small><a title="Kaur N, Reumann S, Hu J.">Kaur,</a> The arabidopsis book 2009 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...dehydrogenase 17 of 41 (At3g02360), and 6-phosphoglucono-lactonase (<B><span style="color: red;">At5g24400</span></B>) (Nyathi and Baker, 2006; Reumann et al., 2007; Eubel et al., 2008; Reumann...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::EMB2024, NP_568445')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6804568">Integrated physiologic, proteomic, and metabolomic analyses of Malus halliana adaptation to saline-alkali stress</a><br /><small><a title="Jia XM, Zhu YF, Hu Y, Zhang R, Cheng L, Zhu ZL, Zhao T, Zhang X, Wang YX.">Jia,</a> Horticulture research 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...(Table 2 ). In addition, we found high expression of PDH2 (2.246-fold), ALDH2B4 (3.503-fold), and <B><span style="color: red;">EMB2024</span></B> (6.350-fold) in relation to the EMP and the pentose phosphate pathway (PPP). <B><span style="color: red;">EMB2024</span></B> is involved in step 2 and synthesizes 6-phospho- d -gluconate from glucose-6-phosphate in the PPP, the former...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...subunit beta-2 Q9C6Z3 2.246 1.168 ALDH2B4 Aldehyde dehydrogenase family 2 member B4 Q9SU63 3.503 1.809 <B><span style="color: red;">EMB2024</span></B> Probable 6-phosphogluconolactonase 5 Q84WW2 6.350 2.667 LPAT1 1-acyl-sn-glycerol-3-phosphate acyltransferase 1 Q8GXU8 0.499 1.003 AOC2 Allene oxide cyclase 2 Q9LS02 0.318 1.653 Metabolites Metabolite Adducts VIP Fold-change P-value Sucrose M+Na 5.28798...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::EMB2024, NP_568445')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6660741">The plastidial pentose phosphate pathway is essential for postglobular embryo development in Arabidopsis</a><br /><small><a title="Andriotis VME, Smith AM.">Andriotis,</a> Proceedings of the National Academy of Sciences of the United States of America 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Ler +1017 At5g24400 PGL3 6-Phosphogluconolactonase 6 pgl3-1 SALK_005685 Col +975 pgl3-2 FLAG_219G10 WS +250 pgl3-3 <B><span style="color: red;">emb2024</span></B> Col +215/437 At1g64190 PGD1 6-Phosphogluconate dehydrogenase 7 pgd1-1 GABI_762C02 Col +11 pgd1-2 SALK_121521 Col +1445 NA SALK_002712 Col 107 At5g41670 PGD3 6-Phosphogluconate dehydrogenase 7 pgd3-1 SAIL_528_E08 Col +20 pgd3-2 SALK_040050...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::EMB2024, NP_568445')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6369206">Unveiling Kiwifruit Metabolite and Protein Changes in the Course of Postharvest Cold Storage</a><br /><small><a title="Salzano AM, Renzone G, Sobolev AP, Carbone V, Petriccione M, Capitani D, Vitale M, Novi G, Zambrano N, Pasquariello MS, Mannina L, Scaloni A.">Salzano,</a> Frontiers in plant science 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...proteins assisting pentose-phosphate metabolism displayed a mixed quantitative trend overtime, i.e., 6-phosphogluconate dehydrogenase (AT3G02360), 6-phosphogluconolactonase (<B><span style="color: red;">EMB2024</span></B>), and transketolase isoforms (AT3G60750 and AT2G45290). In the whole, the quantitative behavior of specific proteins involved in glycolysis, the Krebs cycle, alcoholic fermentation, energy production and additional carbon metabolism pathways...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::EMB2024, NP_568445')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4549554">Peroxisomes contribute to reactive oxygen species homeostasis and cell division induction in Arabidopsis protoplasts</a><br /><small><a title="Tiew TW, Sheahan MB, Rose RJ.">Tiew,</a> Frontiers in plant science 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...( BSMDR ) and a 6-phosphoglucunolactonase encoded by a gene product embryo defective 2024 ( <B><span style="color: red;">EMB2024</span></B> ) also showed little change in expression during culture. Values are means SEM with n 3. ( p &lt; 0.05, p &lt; 0.01, t -tests). Ascorbate and Glutathione are Abundant...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::EMB2024, NP_568445')" href="http://doi.org/10.1093/mp/sst126">Dual-targeting of Arabidopsis 6-phosphogluconolactonase 3 (PGL3) to chloroplasts and peroxisomes involves interaction with Trx m2 in the cytosol.</a><br /><small><a title="Hölscher C, Meyer T, von Schaewen A">Hölscher,</a> Molecular plant 2014 (<a onmousedown="logger(this, 'pb::EMB2024, NP_568445')" href="http://www.ncbi.nlm.nih.gov/pubmed/24008768">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: targeting of Arabidopsis 6-phosphogluconolactonase 3 (PGL3) to chloroplasts and peroxisomes</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::EMB2024, NP_568445')" href="http://doi.org/10.1093/pcp/pcp070">Characterization of Arabidopsis 6-phosphogluconolactonase T-DNA insertion mutants reveals an essential role for the oxidative section of the plastidic pentose phosphate pathway in plant growth and development.</a><br /><small><a title="Xiong Y, DeFraia C, Williams D, Zhang X, Mou Z">Xiong,</a> Plant & cell physiology 2009 (<a onmousedown="logger(this, 'pb::EMB2024, NP_568445')" href="http://www.ncbi.nlm.nih.gov/pubmed/19457984">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: Knockdown of PGL3 leads to a dramatic decrease in plant size, a significant increase in total glucose-6-phosphate dehydrogenase activity and a marked decrease in cellular redox potential. [PGL3]</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::XP_003212309')" href="http://www.ncbi.nlm.nih.gov/protein/XP_003212309.2">XP_003212309</a> GDH/6PGL endoplasmic bifunctional protein from <i>Meleagris gallopavo</i><BR><a title="37:247/255 of query is similar to 584:787/790 of hit (E = 3e-18, 92.8 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=XP_003212309&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_003212309.2" style="font-family: sans-serif; font-size: smaller;">33% identity, 27% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >Purification and properties of glucose 6-phosphate dehydrogenase from turkey erythrocytes.<br /><small><a title="Yilmaz H, Ciftçi M, Beydemir S, Bakan E, Küfrevioğlu OI">Yilmaz,</a> Indian journal of biochemistry & biophysics 2003 (<a onmousedown="logger(this, 'pb::XP_003212309')" href="http://www.ncbi.nlm.nih.gov/pubmed/22900294">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: enzyme activity determined using substrates and inhibitors</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="SwissProt" onmousedown="logger(this, 'curated::G6PE_HUMAN / O95479')" href="http://www.uniprot.org/uniprot/O95479">G6PE_HUMAN / O95479</a> <b>GDH/6PGL endoplasmic bifunctional protein; EC 1.1.1.47; EC 1.1.1.363; EC 3.1.1.31</b> from <i>Homo sapiens (Human)</i> (see <a onmousedown="logger(this, 'curatedpaper::G6PE_HUMAN / O95479')" href="http://www.ncbi.nlm.nih.gov/pubmed/12858176,23132696">2 papers</a>)<BR><a title="MetaCyc" onmousedown="logger(this, 'curated::O95479')" href="https://metacyc.org/gene?orgid=META&amp;id=HS00614-MONOMER">O95479</a> <b>GDH/6PGL endoplasmic bifunctional protein (EC 1.1.1.47; EC 3.1.1.31)</b> from <i>Homo sapiens</i> (see <a onmousedown="logger(this, 'curatedpaper::O95479')" href="http://www.ncbi.nlm.nih.gov/pubmed/12858176,19804362">2 papers</a>)<BR><a title="RefSeq" onmousedown="logger(this, 'curated::XP_006711115')" href="http://www.ncbi.nlm.nih.gov/protein/XP_006711115.1">XP_006711115</a> GDH/6PGL endoplasmic bifunctional protein isoform X2 from <i>Homo sapiens</i><BR><a title="37:247/255 of query is similar to 585:788/791 of hit (E = 3e-18, 92.8 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=G6PE_HUMAN / O95479&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_006711115.1" style="font-family: sans-serif; font-size: smaller;">30% identity, 27% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><b>function:</b> Oxidizes glucose-6-phosphate and glucose, as well as other hexose-6-phosphates<BR>
<b>catalytic</b> <b>activity:</b> D-glucose + NAD(+) = D-glucono-1,5-lactone + H(+) + NADH <small>(<a href="https://www.rhea-db.org/reaction?id=14293">RHEA:14293</a>)</small><BR>
<b>catalytic</b> <b>activity:</b> D-glucose + NADP(+) = D-glucono-1,5-lactone + H(+) + NADPH <small>(<a href="https://www.rhea-db.org/reaction?id=14405">RHEA:14405</a>)</small><BR>
<b>catalytic</b> <b>activity:</b> D-glucose 6-phosphate + NAD(+) = 6-phospho-D-glucono-1,5- lactone + H(+) + NADH <small>(<a href="https://www.rhea-db.org/reaction?id=38215">RHEA:38215</a>)</small><BR>
<b>catalytic</b> <b>activity:</b> D-glucose 6-phosphate + NADP(+) = 6-phospho-D-glucono-1,5- lactone + H(+) + NADPH <small>(<a href="https://www.rhea-db.org/reaction?id=15841">RHEA:15841</a>)</small><BR>
<b>catalytic</b> <b>activity:</b> 6-phospho-D-glucono-1,5-lactone + H2O = 6-phospho-D-gluconate + H(+) <small>(<a href="https://www.rhea-db.org/reaction?id=12556">RHEA:12556</a>)</small><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006711115')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5901385">Hexose-6-phosphate dehydrogenase controls cancer cell proliferation and migration through pleiotropic effects on the unfolded-protein response, calcium homeostasis, and redox balance.</a><br /><small><a title="Tsachaki M, Mladenovic N, Štambergová H, Birk J, Odermatt A">Tsachaki,</a> FASEB journal : official publication of the Federation of American Societies for Experimental Biology 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: Knockdown of H6PD resulted in an increase in ER lumen oxidation, and down-regulation of many components of the unfolded protein response, including the transcription factors activating transcription factor-4, activating transcription factor-6, split X-box binding protein-1, and CCAAT/enhancer binding protein homologous protein.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006711115')" href="http://doi.org/10.1002/1873-3468.12638">Mutations in the tetramer interface of human glucose-6-phosphate dehydrogenase reveals kinetic differences between oligomeric states.</a><br /><small><a title="Ranzani AT, Cordeiro AT">Ranzani,</a> FEBS letters 2017 (<a onmousedown="logger(this, 'pb::XP_006711115')" href="http://www.ncbi.nlm.nih.gov/pubmed/28370139">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: Data indicate the tetramer as the most active form of glucose-6-phosphate dehydrogenase (G6PDH).</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006711115')" href="http://doi.org/10.1530/JOE-16-0006">Cortisol promotes endoplasmic glucose production via pyridine nucleotide redox.</a><br /><small><a title="Wang Z, Mick GJ, Xie R, Wang X, Xie X, Li G, McCormick KL">Wang,</a> The Journal of endocrinology 2016 (<a onmousedown="logger(this, 'pb::XP_006711115')" href="http://www.ncbi.nlm.nih.gov/pubmed/26860459">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: cortisol reduces glucose-6-phosphate (G6P) flux through H6PDH by increasing luminal NADPH, thereby allowing more G6P for hydrolysis via G6Pase</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006711115')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4745784">Inhibition of glucose-6-phosphate dehydrogenase sensitizes cisplatin-resistant cells to death.</a><br /><small><a title="Catanzaro D, Gaude E, Orso G, Giordano C, Guzzo G, Rasola A, Ragazzi E, Caparrotta L, Frezza C, Montopoli M">Catanzaro,</a> Oncotarget 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: Inhibition of glucose-6-phosphate dehydrogenase sensitizes cisplatin-resistant cells to death.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006711115')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4599835">Association Analysis between the Polymorphisms of HSD11B1 and H6PD and Risk of Polycystic Ovary Syndrome in Chinese Population.</a><br /><small><a title="Ju R, Wu W, Tang Q, Wu D, Xia Y, Wu J, Wang X">Ju,</a> PloS one 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: The G allele of rs6688832 in H6PD might exert potential genetic protective role against the development of PCOS, especially in overweight women. PCOS patients with AG genotype of rs6688832 might confer risk to the phenotype of hyperandrogenemia of PCOS.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006711115')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4021576">Reduced cellular Mg²⁺ content enhances hexose 6-phosphate dehydrogenase activity and expression in HepG2 and HL-60 cells.</a><br /><small><a title="Voma C, Barfell A, Croniger C, Romani A">Voma,</a> Archives of biochemistry and biophysics 2014 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: Cytoplasmic Mg2+ regulates glucose 6-phosphate utilization by reticular H6PD.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006711115')" href="http://doi.org/10.1055/s-0033-1345186">The expression of 11β-HSDs, GR, and H6PDH in subcutaneous adipose tissue from polycystic ovary syndrome subjects.</a><br /><small><a title="Li S, Tao T, Wang L, Mao X, Zheng J, Zhao A, Liu W">Li,</a> Hormone and metabolic research = Hormon- und Stoffwechselforschung = Hormones et metabolisme 2013 (<a onmousedown="logger(this, 'pb::XP_006711115')" href="http://www.ncbi.nlm.nih.gov/pubmed/23979790">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: The aim of the work was to investigate the expression of HSD11B1, HSD11B2, H6PDH, and glucocorticoids receptor (GR) mRNA in subcutaneous adipose tissue (SAT) from obese women with or without polycystic ovary syndrome.</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_006711115')" href="http://doi.org/10.1016/j.exer.2012.01.001">Effects of antioxidant supplementation on mRNA expression of glucose-6-phosphate dehydrogenase, β-actin and 18S rRNA in the anterior capsule of the lens in cataract patients.</a><br /><small><a title="Hayashi R, Hayashi S, Arai K, Chikuda M, Obara Y">Hayashi,</a> Experimental eye research 2012 (<a onmousedown="logger(this, 'pb::XP_006711115')" href="http://www.ncbi.nlm.nih.gov/pubmed/22285204">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: Antioxidant supplementation was noted to increase G6PDH in the pentose phosphate cycle and 18S rRNA in the ribosome. There were no significant changes in the gene expression levels of beta-ACT</UL><LI style="list-style-type: none;" margin-left: 6em; ><a href="litSearch.cgi?more=XP_006711115.1">More</a><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::O95479')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5295240">Quantitative Proteomic Profiling of Tachyplesin I Targets in U251 Gliomaspheres</a><br /><small><a title="Li X, Dai J, Tang Y, Li L, Jin G.">Li,</a> Marine drugs 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...galactosidase PGK1 P00558 Phosphoglycerate kinase 1 84.41 31 0.686 0.647 0.461 0.437 carbohydrate kinase H6PD <B><span style="color: red;">O95479</span></B> GDH/6PGL endoplasmic bifunctional protein 35.65 23 0.708 0.729 0.477 0.483 dehydrogenase CSRP1 P21291 Cysteine and glycine-rich protein 1 64.25 9 0.711 0.670 0.426 0.427 actin family cytoskeletal protein CPVL Q9H3G5...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::APA386B_1156')" href="http://www.ncbi.nlm.nih.gov/protein/WP_003623561.1">APA386B_1156</a> 6-phosphogluconolactonase from <i>Acetobacter pasteurianus 386B</i><BR><a title="8:254/255 of query is similar to 15:248/248 of hit (E = 3e-18, 92.4 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=APA386B_1156&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_003623561.1" style="font-family: sans-serif; font-size: smaller;">28% identity, 97% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::APA386B_1156')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3751514">Complete genome sequence and comparative analysis of Acetobacter pasteurianus 386B, a strain well-adapted to the cocoa bean fermentation ecosystem</a><br /><small><a title="Illeghems K, De Vuyst L, Weckx S.">Illeghems,</a> BMC genomics 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...(APA386B_1157); (20) transketolase (APA386B_1152 and APA386B_1521); (21) transaldolase (APA386B_1153) (22) glucose-6-phosphate dehydrogenase (APA386B_729); (23) 6-phosphogluconolactonase (<B><span style="color: red;">APA386B_1156</span></B>); (24) glycerol kinase (Apa396B_92); (25) glycerol-3-phosphate dehydrogenase (APA386B_1931 and APA386B_94); (26) pyruvaldehyde oxidoreductase (APA386B_1946); (27) lactoylgluthathione lyase (APA386B_2164 and APA386B_1276); (28) hydroxyacylglutathione hydrolase (APA386B_2280); (29) lactate dehydrogenase (APA386B_910 and APA386B_1053);...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::Gmet_2618')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=1091130">Gmet_2618</a> 6-phosphogluconolactonase (NCBI) from <i>Geobacter metallireducens GS-15</i><BR><a title="37:251/255 of query is similar to 33:237/241 of hit (E = 1e-17, 90.5 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=Gmet_2618&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS1091130" style="font-family: sans-serif; font-size: smaller;">29% identity, 84% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::Gmet_2618')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2700814">The genome sequence of Geobacter metallireducens: features of metabolism, physiology and regulation common and dissimilar to Geobacter sulfurreducens</a><br /><small><a title="Aklujkar M, Krushkal J, DiBartolo G, Lapidus A, Land ML, Lovley DR.">Aklujkar,</a> BMC microbiology 2009 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...the G. metallireducens genome includes a cluster of oxidative pentose phosphate pathway enzyme genes: 6-phosphogluconolactonase (<B><span style="color: red;">Gmet_2618</span></B>, 30% identical to the Pseudomonas putida enzyme [ 82 ]), glucose-6-phosphate dehydrogenase (Gmet_2619, 50% identical to the Nostoc punctiforme enzyme [ 83 ]), and 6-phosphogluconate dehydrogenase (Gmet_2620, 36% identical to...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::cg1780')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=5916711">cg1780</a> 6-phosphogluconolactonase (RefSeq) from <i>Corynebacterium glutamicum ATCC 13032</i><BR><a title="RefSeq" onmousedown="logger(this, 'curated::NCgl1516')" href="http://www.ncbi.nlm.nih.gov/protein/NP_600792.1">NCgl1516</a> 6-phosphogluconolactonase from <i>Corynebacterium glutamicum ATCC 13032</i><BR><a title="1:208/255 of query is similar to 1:193/235 of hit (E = 1e-17, 90.5 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=cg1780&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS5916711" style="font-family: sans-serif; font-size: smaller;">32% identity, 82% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::cg1780')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3489674">Transcriptional regulation of the operon encoding stress-responsive ECF sigma factor SigH and its anti-sigma factor RshA, and control of its regulatory network in Corynebacterium glutamicum</a><br /><small><a title="Busche T, Silar R, Pičmanová M, Pátek M, Kalinowski J.">Busche,</a> BMC genomics 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...cg1411 rbsA ABC-type ribose transporter, ATPase subunit (TC 3.A.1.2.1) 1.92 cg2636 catA Catechol 1,2-dioxygenase 1.88 <B><span style="color: red;">cg1780</span></B> pgl 6-Phosphogluconolactonase 1.87 cg1413 rbsB ABC-type ribose transporter, substrate-binding lipoprotein (TC 3.A.1.2.1) 1.85 cg0498 hemC Porphobilinogen deaminase 1.84 cg2665 Hypothetical protein 1.82 cg2181* ABC-type putative dipeptide/oligopeptide transporter, substrate-binding lipoprotein 1.80...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::cg1780')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2230225">Co-ordinated regulation of gluconate catabolism and glucose uptake in Corynebacterium glutamicum by two functionally equivalent transcriptional regulators, GntR1 and GntR2</a><br /><small><a title="Frunzke J, Engels V, Hasenbein S, Gätgens C, Bott M.">Frunzke,</a> Molecular microbiology 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...0.88 0.89 1.09 1.15 0.74 cg1776 Transaldolase, tal 2.01 1.21 0.81 1.06 1.02 1.12 0.97 <B><span style="color: red;">cg1780</span></B> 6-phosphogluconolactonase, devB 2.51 1.28 0.89 1.14 1.02 0.91 1.08 cg1774 Transketolase, tkt 2.87 1.13 0.83 1.07 1.09 1.03 1.12 cg2836 Succinyl-CoA synthetase chain, sucD 4.48 0.93 0.52 1.41 0.75 0.81...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NCgl1516')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4767432">Altered acetylation and succinylation profiles in Corynebacterium glutamicum in response to conditions inducing glutamate overproduction</a><br /><small><a title="Mizuno Y, Nagano-Shoji M, Kubo S, Kawamura Y, Yoshida A, Kawasaki H, Nishiyama M, Yoshida M, Kosono S.">Mizuno,</a> MicrobiologyOpen 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...phosphoglyceromutase (NCgl0390); Eno, enolase (NCgl0935); Pyk, pyruvate kinase (NCgl2008); Zwf, glucose6phosphate 1dehydrogenase (NCgl1514); DevB, 6phosphogluconolactonase (<B><span style="color: red;">NCgl1516</span></B>); Gnd, 6phosphogluconate dehydrogenase (NCgl1396); Rpi, ribose5phosphate isomerase B (NCgl2337); Tkt, transketolase (NCgl1512); Tal, transaldolase (NCgl1513); AceE, pyruvate dehydrogenase subunit E1 (NCgl2167); GltA, type II citrate synthase (NCgl0795); Acn, aconitate hydratase...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="SwissProt" onmousedown="logger(this, 'curated::6PGL1_ARATH / Q9LMX8')" href="http://www.uniprot.org/uniprot/Q9LMX8">6PGL1_ARATH / Q9LMX8</a> <b>Probable 6-phosphogluconolactonase 1; 6PGL1; EC 3.1.1.31</b> from <i>Arabidopsis thaliana (Mouse-ear cress)</i> (see <a onmousedown="logger(this, 'curatedpaper::6PGL1_ARATH / Q9LMX8')" href="http://www.ncbi.nlm.nih.gov/pubmed/19457984,24008768">2 papers</a>)<BR><a title="BRENDA" onmousedown="logger(this, 'curated::Q9LMX8')" href="http://www.brenda-enzymes.org/sequences.php?AC=Q9LMX8">Q9LMX8</a> <b>6-phosphogluconolactonase (EC 3.1.1.31)</b> from <i>Arabidopsis thaliana</i> (see <a onmousedown="logger(this, 'curatedpaper::Q9LMX8')" href="http://www.ncbi.nlm.nih.gov/pubmed/19457984">paper</a>)<BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::AT1G13700')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=10079653">AT1G13700</a> glucosamine/galactosamine-6-phosphate isomerase family protein (RefSeq) from <i>Arabidopsis thaliana</i><BR><a title="20:249/255 of query is similar to 29:259/268 of hit (E = 3e-17, 89.4 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=6PGL1_ARATH / Q9LMX8&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS10079653" style="font-family: sans-serif; font-size: smaller;">31% identity, 86% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><b>function:</b> Catalyzes the hydrolysis of 6-phosphogluconolactone to 6- phosphogluconate. <BR>
<b>catalytic</b> <b>activity:</b> 6-phospho-D-glucono-1,5-lactone + H2O = 6-phospho-D-gluconate + H(+) <small>(<a href="https://www.rhea-db.org/reaction?id=12556">RHEA:12556</a>)</small><BR>
<b>disruption</b> <b>phenotype:</b> No visible phenotype under normal growth conditions. <LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT1G13700')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6375920">Proteomic and metabolomic analyses provide insight into the off-flavour of fruits from citrus trees infected with 'Candidatus Liberibacter asiaticus'</a><br /><small><a title="Yao L, Yu Q, Huang M, Hung W, Grosser J, Chen S, Wang Y, Gmitter FG.">Yao,</a> Horticulture research 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...EC 1.1.1.41 0.57 4.29E02 gi|641851263 AT1G68750 PPC4 Phosphoenolpyruvate carboxylase 4 EC 4.1.1.31 0.63 3.05E02 gi|641851457 <B><span style="color: red;">AT1G13700</span></B> PGL1 6-Phosphogluconolactonase 1 EC 3.1.1.31 0.54 3.89E03 gi|641853224 AT1G48030 mtLPD1 Mitochondrial lipoamide dehydrogenase 1 EC 1.8.1.4 0.65 5.78E05 gi|641853915 AT5G15490 UGD3 UDP-glucose dehydrogenase EC 1.1.1.22 0.58 2.33E02 gi|641854939 AT3G14310 PME3...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT1G13700')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5710523">Cell growth and homeostasis are disrupted in arabidopsis rns2-2 mutants missing the main vacuolar RNase activity</a><br /><small><a title="Morriss SC, Liu X, Floyd BE, Bassham DC, MacIntosh GC.">Morriss,</a> Annals of botany 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...AT5G24800 AT1G18270 0.524407 0.522557 AT3G45970 <B><span style="color: red;">AT1G13700</span></B> 0.485587 0.48172 AT3G52400 AT3G62860 AT4G29950 AT3G10020 AT2G02710 AT2G28400 AT5G66650 AT5G51390...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...glucosamine/galactosamine-6-phosphate isomerase family protein gene <B><span style="color: red;">AT1G13700</span></B> and the AT3G45730 AT3G04640 AT1G05575 AT5G49450 AT1G02660 AT1G19530 AT3G10930...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT1G13700')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5440562">A Conserved Carbon Starvation Response Underlies Bud Dormancy in Woody and Herbaceous Species</a><br /><small><a title="Tarancón C, González-Grandío E, Oliveros JC, Nicolas M, Cubas P.">Tarancón,</a> Frontiers in plant science 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...TPS11 II-III-IV At1g54710 ATG18H II At4g24040 TREHALASE III At3g62770 ATG18A III At5g54730 ATG18F II-III Metabolism <B><span style="color: red;">At1g13700</span></B> 6-PHOSPHOGLUCONOLACTONASE 1 IV Lipid degradation At1g54100 ALDEHYDE DEHYDROGENASE H7B4 II-III At4g16760 ACYL-COENZYME A OXIDASE 1 (ACX1) III At3g23920 -AMYLASE 1 (BAM 1) II At5g65110 ACX2 III At5g55700 BAM 4 II-III...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT1G13700')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3600680">Expression profiling of tomato pre-abscission pedicels provides insights into abscission zone properties including competence to respond to abscission signals</a><br /><small><a title="Nakano T, Fujisawa M, Shima Y, Ito Y.">Nakano,</a> BMC plant biology 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Solyc03g034020 A_96_P202249 BI925250 AT4G36550 binding / ubiquitin-protein ligase 2.1 0.014 1.6 0.009 Solyc01g010250 A_96_P215689 BP892102 <B><span style="color: red;">AT1G13700</span></B> glucosamine/galactosamine-6-phosphate isomerase family protein 2.1 0.006 1.7 0.002 Solyc12g042470 A_96_P109252 TA51045_4081 AT1G59950 aldo/keto reductase, putative 4.2 0.018 3.8 0.045 Solyc00g071180 A_96_P016051 AF083253 AT3G12490 ATCYSB | cysteine protease inhibitor, putative /...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT1G13700')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3697946">Iron-dependent modifications of the flower transcriptome, proteome, metabolome, and hormonal content in an Arabidopsis ferritin mutant</a><br /><small><a title="Sudre D, Gutierrez-Carbonell E, Lattanzio G, Rellán-Álvarez R, Gaymard F, Wohlgemuth G, Fiehn O, Alvarez-Fernández A, Zamarreño AM, Bacaicoa E, Duy D, García-Mina JM, Abadía J, Philippar K, López-Millán AF, Briat JF.">Sudre,</a> Journal of experimental botany 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...protein At5g58840 1.8 Subtilisin-like serine endopeptidase family protein At4g12910 1.4 scpl20, serine carboxy peptidase-like 20 <B><span style="color: red;">At1g13700</span></B> Carbohydrate metabolism 1.5 Glucosamine/galacto samine-6-phosphate isomerase family protein At1g19610 Stress+redox 2.0 LCR78, PDF1.4, pathogenesis-related protein/plant defensin (PDF) family At1g22900 1.7 Disease resistance-responsive (dirigent-like protein)/molecular function unknown At2g15120 4.9 Pseudogene, disease-...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT1G13700')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3355747">The bZIP Transcription Factor PERIANTHIA: A Multifunctional Hub for Meristem Control</a><br /><small><a title="Maier AT, Stehling-Sun S, Offenburger SL, Lohmann JU.">Maier,</a> Frontiers in plant science 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...DUF1005 (InterPro:IPR010410) 263236_at 0.53 AT1G10470 ARR4 (RESPONSE REGULATOR 4); transcription regulator/two-component response regulator 256098_at 0.50 <B><span style="color: red;">AT1G13700</span></B> Glucosamine/galactosamine-6-phosphate isomerase family protein 259466_at 0.44 AT1G19050 ARR7 (RESPONSE REGULATOR 7); transcription regulator/two-component response regulator 260662_at 0.49 AT1G19540 Isoflavone reductase. putative 260856_at 0.46 AT1G21910 AP2 domain-containing transcription factor family protein...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::SYNPCC7002_A0928')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=5418447">SYNPCC7002_A0928</a> 6-phosphogluconolactonase (RefSeq) from <i>Synechococcus sp. PCC 7002</i><BR><a title="4:248/255 of query is similar to 5:234/252 of hit (E = 9e-17, 87.8 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=SYNPCC7002_A0928&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS5418447" style="font-family: sans-serif; font-size: smaller;">28% identity, 96% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::SYNPCC7002_A0928')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3133671">Transcription Profiling of the Model Cyanobacterium Synechococcus sp. Strain PCC 7002 by Next-Gen (SOLiD™) Sequencing of cDNA</a><br /><small><a title="Ludwig M, Bryant DA.">Ludwig,</a> Frontiers in microbiology 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...1.62 0.50 0.83 pgi Glucose-6-phosphate isomerase SYNPCC7002_A1459 0.25 3.08 2.64 0.85 0.86 zwf Glucose-6-phosphate 1-dehydrogenase <B><span style="color: red;">SYNPCC7002_A0928</span></B> 0.33 2.54 0.32 0.13 0.79 pgl 6-Phosphogluconolactonase SYNPCC7002_A0221 0.39 3.55 3.46 0.97 0.65 gnd 6-Phosphogluconate DH, decarboxylating SYNPCC7002_A1269 $ 2.45 0.09 0.06 0.66 1.39 rpiA Ribose 5-phosphate isomerase A SYNPCC7002_A0324...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::GOX1707')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=857324">GOX1707</a> 6-Phosphogluconolactonase (NCBI) from <i>Gluconobacter oxydans 621H</i><BR><a title="34:232/255 of query is similar to 36:228/246 of hit (E = 1e-16, 87.4 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=GOX1707&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS857324" style="font-family: sans-serif; font-size: smaller;">31% identity, 78% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::GOX1707')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6191907">Global mRNA decay and 23S rRNA fragmentation in Gluconobacter oxydans 621H</a><br /><small><a title="Kranz A, Steinmann A, Degner U, Mengus-Kaya A, Matamouros S, Bott M, Polen T.">Kranz,</a> BMC genomics 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...complex (GOX1073); pdc , pyruvate decarboxylase (GOX1081); pgk , phosphoglycerate kinase (GOX0507); pgl , 6-phosphogluconolactonase (<B><span style="color: red;">GOX1707</span></B>); ppc , phosphoenolpyruvate carboxylase (GOX0102); pyk , pyruvate kinase (GOX2250); rpe , ribulosephosphate 3-epimerase (GOX1352); rpi , ribose 5-phosphate isomerase (GOX1708); sldA , polyol dehydrogenase subunit SldA (GOX0854); sldB ,...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::sll1479')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=13305">sll1479</a> glucose-6-P-dehydrogenase (NCBI ptt file) from <i>Synechocystis sp. PCC 6803</i><BR><a title="38:248/255 of query is similar to 51:249/254 of hit (E = 1e-16, 87.0 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=sll1479&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS13305" style="font-family: sans-serif; font-size: smaller;">31% identity, 83% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::sll1479')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4959675">Diurnal Regulation of Cellular Processes in the Cyanobacterium Synechocystis sp. Strain PCC 6803: Insights from Transcriptomic, Fluxomic, and Physiological Analyses</a><br /><small><a title="Saha R, Liu D, Hoynes-O'Connor A, Liberton M, Yu J, Bhattacharyya-Pakrasi M, Balassy A, Zhang F, Moon TS, Maranas CD, Pakrasi HB.">Saha,</a> mBio 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...cycle under autotrophic conditions. Genes such as tktA ( sll1070 ), encoding transketolase, devB ( <B><span style="color: red;">sll1479</span></B> ), encoding 6-phosphogluconolactonase, and cfxE ( sll0807 ), encoding pentose-5-phosphate-3-epimerase, were upregulated in the earlier light period (with peaks at L1), while rpiA ( slr0194 ), encoding ribose 5-phosphate isomerase,...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::sll1479')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3196541">Time-course analysis of cyanobacterium transcriptome: detecting oscillatory genes</a><br /><small><a title="Layana C, Diambra L.">Layana,</a> PloS one 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...strongly synchronized between replicates. In group F (top panels) the genes sll0329 (30), sll1196 (31), <B><span style="color: red;">sll1479</span></B> (33), slr0884 (36), slr1734 (38), slr1793 (39) and slr1843 (40) are synchronized between experiments. Furthermore, genes sll0329, sll1196 and slr1793 are almost co-expressed in both experiments. In group H (medium...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...sll1196 * Phosphofructokinase Glycolysis 32 F sll1234 # Adenosylhomocysteinase Amino acids and amines 33 F <B><span style="color: red;">sll1479</span></B> * 6-phosphogluconolactonase Pentose phosphate pathway 34 F slr0301 Phosphoenolpyruvate synthase Pyruvate metabolism 35 F slr0394 # Phosphoglycerate kinase Glycolysis 36 F slr0884 * , # GAPDH 1 Glycolysis 37 F...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::sll1479')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1064041">Global analysis of circadian expression in the cyanobacterium Synechocystis sp. strain PCC 6803</a><br /><small><a title="Kucho K, Okamoto K, Tsuchiya Y, Nomura S, Nango M, Kanehisa M, Ishiura M.">Kucho,</a> Journal of bacteriology 2005 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...slr0646 sll1898 2.8 3.2 6 9 2.5 2.8 9 9 2.6 3.0 8 9 <B><span style="color: red;">sll1479</span></B> sll0242 sll1899 sll1461 slr0884 1.4 1.4 2.6 1.6 2.6 9 9 9 11 9 1.9 1.6 2.5 1.7 3.0 10 11 11 10 12...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...circadian expression patterns: 6-phosphogluconolactonase (Glc, <B><span style="color: red;">sll1479</span></B>), transaldolase (TalB, slr1793), and the rate-limiting NADPH-producing enzymes...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="SwissProt" onmousedown="logger(this, 'curated::6PGL5_ARATH / Q8LEV7')" href="http://www.uniprot.org/uniprot/Q8LEV7">6PGL5_ARATH / Q8LEV7</a> <b>Probable 6-phosphogluconolactonase 5; 6PGL5; EC 3.1.1.31</b> from <i>Arabidopsis thaliana (Mouse-ear cress)</i> (see <a onmousedown="logger(this, 'curatedpaper::6PGL5_ARATH / Q8LEV7')" href="http://www.ncbi.nlm.nih.gov/pubmed/19457984,24008768">2 papers</a>)<BR><a title="BRENDA" onmousedown="logger(this, 'curated::Q8LEV7')" href="http://www.brenda-enzymes.org/sequences.php?AC=Q8LEV7">Q8LEV7</a> <b>6-phosphogluconolactonase (EC 3.1.1.31)</b> from <i>Arabidopsis thaliana</i> (see <a onmousedown="logger(this, 'curatedpaper::Q8LEV7')" href="http://www.ncbi.nlm.nih.gov/pubmed/19457984">paper</a>)<BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::AT5G24420')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=10106434">AT5G24420</a> glucosamine/galactosamine-6-phosphate isomerase-related (RefSeq) from <i>Arabidopsis thaliana</i><BR><a title="18:249/255 of query is similar to 28:250/252 of hit (E = 2e-16, 87.0 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=6PGL5_ARATH / Q8LEV7&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS10106434" style="font-family: sans-serif; font-size: smaller;">30% identity, 91% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><b>function:</b> Catalyzes the hydrolysis of 6-phosphogluconolactone to 6- phosphogluconate. <BR>
<b>catalytic</b> <b>activity:</b> 6-phospho-D-glucono-1,5-lactone + H2O = 6-phospho-D-gluconate + H(+) <small>(<a href="https://www.rhea-db.org/reaction?id=12556">RHEA:12556</a>)</small><BR>
<b>disruption</b> <b>phenotype:</b> No visible phenotype under normal growth conditions. <LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24420')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5512810">Transcriptome analysis uncovers Arabidopsis F-BOX STRESS INDUCED 1 as a regulator of jasmonic acid and abscisic acid stress gene expression</a><br /><small><a title="Gonzalez LE, Keller K, Chan KX, Gessel MM, Thines BC.">Gonzalez,</a> BMC genomics 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...1.4 7.0 59.7 3.9 TRAF-like family protein At4g17470 5.3 11.3 111.4 1.9 Alpha/beta-hydrolases superfamily protein <B><span style="color: red;">At5g24420</span></B> PGL5 3.0 8.0 48.5 2.9 Cytosolic 6-phosphogluconolactonase At3g28270 AFL1 1.0 2.0 32.0 5.6 Peripheral membrane protein, growth regulation in drought At3g44970 1.0 0.9 1.9 1.9 Cytochrome P450 superfamily protein At1g18710...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24420')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4174771">Transcriptome Analysis of Induced Systemic Drought Tolerance Elicited by Pseudomonas chlororaphis O6 in Arabidopsis thaliana</a><br /><small><a title="Cho SM, Kang BR, Kim YC.">Cho,</a> The plant pathology journal 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...3-hydroxylase-like protein 1.28 250558_at At5g07990 xyloglucan endo-1,4-beta-D-glucanase-like protein 1.31 253628_at At4g30280 6-phosphogluconolactonase-like protein 1.34 249732_at <B><span style="color: red;">At5g24420</span></B> reticuline oxidase-like protein 1.52 254432_at At4g20830 11-beta-hydroxysteroid dehydrogenase-like 1.55 248520_at At5g50600 UDP-glucose glucosyltransferase 1.89 260799_at At1g78270 tyrosine transaminase like protein 2.12 254232_at At4g23600 oleosin 2.22 249353_at At5g40420 leucoanthocyanidin dioxygenase-like protein...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24420')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3563046">Differential Contribution of Transcription Factors to Arabidopsis thaliana Defense Against Spodoptera littoralis</a><br /><small><a title="Schweizer F, Bodenhausen N, Lassueur S, Masclaux FG, Reymond P.">Schweizer,</a> Frontiers in plant science 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...transmembrane protein At3g28300 3.70* 1.06 1.09 AT14A, transmembrane protein At3g28290 3.58* 1.12 0.60 PGL5, 6-phosphogluconolactonase <B><span style="color: red;">At5g24420</span></B> 3.56*** 0.83 0.79* List of the 50 most highly induced genes in response to S. littoralis in Col-0. Three-week-old plants were challenged for 8days with first-instar larvae. Values are calculated...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24420')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3142401">Divergent evolutionary fates of major photosynthetic gene networks following gene and whole genome duplications</a><br /><small><a title="Coate JE, Doyle JJ.">Coate,</a> Plant signaling & behavior 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...-0.11 0.51 AT1G71170 -0.39 0.37 AT3G49360 0.39 -0.51 <B><span style="color: red;">AT5G24420</span></B> 0.31 -0.33 AT5G24410 -0.33 0.40 AT3G04790 0.84 -0.70 AT5G44520 0.70 -0.77 AT2G01290 0.68 -0.65...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24420')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2845782">Coronatine-insensitive 1 (COI1) mediates transcriptional responses of Arabidopsis thaliana to external potassium supply</a><br /><small><a title="Armengaud P, Breitling R, Amtmann A.">Armengaud,</a> Molecular plant 2010 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...0.6 0.0 1.8 1.0 At4g15100 SCPL30 Serine carboxypeptidase S10 family 5.0 0.5 0.1 1.9 1.1 <B><span style="color: red;">At5g24420</span></B> 6PGL 6-phosphogluconolactonase-like 8.1 1.7 0.0 5.2 1.3 At1g76790 O-methyltransferase, family 2 9.2 0.5 0.3 1.7 0.9 At4g08870 ARGAH2 Putative arginase 9.2 1.2 0.0 3.6 1.2 At1g04710 Putative acetyl-CoA acyltransferase 9.4...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24420')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2662457">Function annotation of an SBP-box gene in Arabidopsis based on analysis of co-expression networks and promoters</a><br /><small><a title="Wang Y, Hu Z, Yang Y, Chen X, Chen G.">Wang,</a> International journal of molecular sciences 2009 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...AT5G59000 zinc finger (C3HC4-type RING finger) family protein AT4G39675 unknown protein AT5G03700 PAN domain-containing protein <B><span style="color: red;">AT5G24420</span></B> glucosamine/galactosamine-6-phosphate isomerase-related AT2G21045 similar to unknown protein AT3G24240 leucine-rich repeat transmembrane protein kinase, putative AT5G50760 auxin-responsive family protein AT1G72260 toxin receptor binding AT3G15950 (TSA1-LIKE); unknown protein AT3G15270 SPL5 AT1G17860 trypsin...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT5G24420')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2375892">A spatial dissection of the Arabidopsis floral transcriptome by MPSS</a><br /><small><a title="Peiffer JA, Kaushik S, Sakai H, Arteaga-Vazquez M, Sanchez-Leon N, Ghazal H, Vielle-Calzada JP, Meyers BC.">Peiffer,</a> BMC plant biology 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...parameters. A gene of unknown function, At1g05550, was identified within the integuments (Figure 5a ). <B><span style="color: red;">At5g24420</span></B> was characterized within the funiculus and both integuments of the ovule, confirming MPSS-based predictions of ovule expression (Figure 5b ). Expression of At5g49180 was found in both the integuments and...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...At1g05550; GUS is expressed in both integuments and the nucellus, at the chalazal region. (B) <B><span style="color: red;">At5g24420</span></B>; GUS is expressed in the funiculus and both integuments throughout the ovule. (C) At5g49180; GUS is expressed in both integuments and the funiculus, but not in the female gametophyte. (D)...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::alr1602')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=163955">alr1602</a> glucose-6-P-dehydrogenase (NCBI ptt file) from <i>Nostoc sp. PCC 7120</i><BR><a title="9:249/255 of query is similar to 10:236/240 of hit (E = 3e-16, 85.9 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=alr1602&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS163955" style="font-family: sans-serif; font-size: smaller;">30% identity, 95% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::alr1602')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3232833">FraH is required for reorganization of intracellular membranes during heterocyst differentiation in Anabaena sp. strain PCC 7120</a><br /><small><a title="Merino-Puerto V, Mariscal V, Schwarz H, Maldener I, Mullineaux CW, Herrero A, Flores E.">Merino-Puerto,</a> Journal of bacteriology 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...cyanobacteria. In Anabaena sp. strain PCC 7120, both genes, <B><span style="color: red;">alr1602</span></B> and alr1603 (fraH), can be expressed as independent transcripts as well as in a common...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::cce_4743')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=5431433">cce_4743</a> 6-phosphogluconolactonase (RefSeq) from <i>Cyanothece sp. ATCC 51142</i><BR><a title="1:249/255 of query is similar to 13:247/251 of hit (E = 4e-16, 85.9 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=cce_4743&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS5431433" style="font-family: sans-serif; font-size: smaller;">29% identity, 98% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::cce_4743')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6400067">Analysis of Protein Complexes in the Unicellular Cyanobacterium Cyanothece ATCC 51142</a><br /><small><a title="Aryal UK, Ding Z, Hedrick V, Sobreira TJP, Kihara D, Sherman LA.">Aryal,</a> Journal of proteome research 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Oxidative PP enzymes including glucose 6-phosphate dehydrogenase; Zwf (cce_2536), 6-phosphogluconate dehydrogenase; Gnd (cce_3746), 6-phosphogluconolactonase; Pgl (<B><span style="color: red;">cce_4743</span></B>), ribulose-phosphate 3-epimerase; Rpe (cce_0798), and transaldolases; TalC (cce_4208) and TalA (cce_4687) were identified as complexes. For example, Zwf, a branching enzyme of the PP pathway was identified with R app...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::cce_4743')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3329150">Genome-scale modeling of light-driven reductant partitioning and carbon fluxes in diazotrophic unicellular cyanobacterium Cyanothece sp. ATCC 51142</a><br /><small><a title="Vu TT, Stolyar SM, Pinchuk GE, Hill EA, Kucek LA, Brown RN, Lipton MS, Osterman A, Fredrickson JK, Konopka AE, Beliaev AS, Reed JL.">Vu,</a> PLoS computational biology 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...ammonium- and light-limited conditions. Enzymes of the oxidative PPP branch, namely glucose-6-phosphate dehydrogenase (cce_25352536), 6-phosphogluconolactonase (<B><span style="color: red;">cce_4743</span></B>) and 6-phosphogluconate dehydrogenase (cce_3746), showed increased abundances under ammonium limited conditions. Similarly, two-fold increase in abundance levels was observed for gluconeogenesis proteins, including fructose 1,6-bisphosphatase (cce_4758), glucose-6-phosphate isomerase (cce_0666), glyceraldehyde-3-phosphate...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::VCA0897')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=56238">VCA0897</a> devB protein (NCBI ptt file) from <i>Vibrio cholerae O1 biovar eltor str. N16961</i><BR><a title="38:251/255 of query is similar to 32:237/238 of hit (E = 2e-15, 83.6 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=VCA0897&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS56238" style="font-family: sans-serif; font-size: smaller;">29% identity, 84% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::VCA0897')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6606805">RadD Contributes to R-Loop Avoidance in Sub-MIC Tobramycin</a><br /><small><a title="Negro V, Krin E, Aguilar Pierlé S, Chaze T, Giai Gianetto Q, Kennedy SP, Matondo M, Mazel D, Baharoglu Z.">Negro,</a> mBio 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...stress No reads (0) 1.2 VC2294 nqrB Iron and oxidative stress No reads (0) 5.1 <B><span style="color: red;">VCA0897</span></B> pgl Pentose phosphate pathway 26.3 (0.009) 2.1 VCA0609 Unknown 23.4 (1.60E15) 2.9 VCA0634 Putative tRNA modification 9.3 (0.011) 1.5 VC2517 Putative ABC-type transport 6.7 (1.00E08) 1.5 VC2234* rnhA RNase H1,...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::VCA0897')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5816637">Transcriptomics reveals a cross-modulatory effect between riboflavin and iron and outlines responses to riboflavin biosynthesis and uptake in Vibrio cholerae</a><br /><small><a title="Sepúlveda-Cisternas I, Lozano Aguirre L, Fuentes Flores A, Vásquez Solis de Ovando I, García-Angulo VA.">Sepúlveda-Cisternas,</a> Scientific reports 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...aminotransferase 1.820 VCA0825 ectA L-24-diaminobutyric acid acetyltransferase 1.691 VCA0867 ompW outer membrane protein OmpW 1.639 <B><span style="color: red;">VCA0897</span></B> devB devB protein 1.127 VCA0898 gnd 6-phosphogluconate dehydrogenase decarboxylating 1.401 1.262 VCA0907 hutZ heme binding 1.430 1.047 VCA0908 hutX Unknown, linked to hutZ 1.626 1.091 VCA0909 hutW unknown, linked to...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::VCA0897')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1201252">Transcriptional profiling of Vibrio cholerae recovered directly from patient specimens during early and late stages of human infection</a><br /><small><a title="Larocque RC, Harris JB, Dziejman M, Li X, Khan AI, Faruque AS, Faruque SM, Nair GB, Ryan ET, Qadri F, Mekalanos JJ, Calderwood SB.">Larocque,</a> Infection and immunity 2005 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...5.0 104 1.1 105 Energy metabolism VC2413 VC2447 <B><span style="color: red;">VCA0897</span></B> Pyruvate dehydrogenase, E2 component, dihydrolipoamide Enolase DevB protein 3.24 3.05 2.04 3.5 104 2.8...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::Rv1445c')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=33218">Rv1445c</a> 6-phosphogluconolactonase (NCBI) from <i>Mycobacterium tuberculosis H37Rv</i><BR><a title="39:252/255 of query is similar to 38:246/247 of hit (E = 2e-15, 83.2 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=Rv1445c&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS33218" style="font-family: sans-serif; font-size: smaller;">31% identity, 84% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::Rv1445c')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4802306">Integration of Metabolic Modeling with Gene Co-expression Reveals Transcriptionally Programmed Reactions Explaining Robustness in Mycobacterium tuberculosis</a><br /><small><a title="Puniya BL, Kulshreshtha D, Mittal I, Mobeen A, Ramachandran S.">Puniya,</a> Scientific reports 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Rv1121 ) Glycolysis (PGI, TPI), Pentose Phosphate Pathway (RPE, TALA, TKT1, TKT2) PGL (6-phosphogluconolactonase) ( <B><span style="color: red;">Rv1445c</span></B> ) Glycolysis ( PGI , TPI), Pentose Phosphate Pathway (TKT1, TKT2) GLUDC (Glutamate decarboxylase) ( Rv3432c ) Alanine and Aspartate Metabolism (ASNS1), Glycolysis ( GAPD , PGK ) GND (phosphogluconate...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::CT154_14190')" href="http://www.ncbi.nlm.nih.gov/protein/WP_034927080.1">CT154_14190</a> 6-phosphogluconolactonase from <i>Komagataeibacter xylinus</i><BR><a title="34:247/255 of query is similar to 41:246/249 of hit (E = 4e-15, 82.4 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=CT154_14190&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_034927080.1" style="font-family: sans-serif; font-size: smaller;">28% identity, 84% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::CT154_14190')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5908849">Complete genome analysis of Gluconacetobacter xylinus CGMCC 2955 for elucidating bacterial cellulose biosynthesis and metabolic regulation</a><br /><small><a title="Liu M, Liu L, Jia S, Li S, Zou Y, Zhong C.">Liu,</a> Scientific reports 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...6pgd , CT154_14710), glucose-6-phosphate 1-dehydrogenase ( 6pgd-1 , CT154_14205 and CT154_05345), 6-phosphogluconolactonase ( 6pgl , <B><span style="color: red;">CT154_14190</span></B>), 6-phosphogluconate dehydrogenase ( 6pgad , CT154_14210), ribose-5-phosphate isomerase A ( rpi A , CT154_14185), ribulose-phosphate 3-epimerase ( rpe , CT154_13290), transketolase ( tkt , CT154_07695, CT154_08455, CT154_12910, and CT154_14220), and...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="SwissProt" onmousedown="logger(this, 'curated::6PGL2_ARATH / Q8LG70')" href="http://www.uniprot.org/uniprot/Q8LG70">6PGL2_ARATH / Q8LG70</a> <b>Probable 6-phosphogluconolactonase 2; 6PGL2; EC 3.1.1.31</b> from <i>Arabidopsis thaliana (Mouse-ear cress)</i> (see <a onmousedown="logger(this, 'curatedpaper::6PGL2_ARATH / Q8LG70')" href="http://www.ncbi.nlm.nih.gov/pubmed/19457984,24008768">2 papers</a>)<BR><a title="BRENDA" onmousedown="logger(this, 'curated::Q8LG70')" href="http://www.brenda-enzymes.org/sequences.php?AC=Q8LG70">Q8LG70</a> <b>6-phosphogluconolactonase (EC 3.1.1.31)</b> from <i>Arabidopsis thaliana</i> (see <a onmousedown="logger(this, 'curatedpaper::Q8LG70')" href="http://www.ncbi.nlm.nih.gov/pubmed/19457984">paper</a>)<BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::AT3G49360')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=10097201">AT3G49360</a> glucosamine/galactosamine-6-phosphate isomerase family protein (RefSeq) from <i>Arabidopsis thaliana</i><BR><a title="2:253/255 of query is similar to 4:253/259 of hit (E = 1e-14, 80.9 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=6PGL2_ARATH / Q8LG70&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS10097201" style="font-family: sans-serif; font-size: smaller;">28% identity, 97% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><b>function:</b> Catalyzes the hydrolysis of 6-phosphogluconolactone to 6- phosphogluconate. <BR>
<b>catalytic</b> <b>activity:</b> 6-phospho-D-glucono-1,5-lactone + H2O = 6-phospho-D-gluconate + H(+) <small>(<a href="https://www.rhea-db.org/reaction?id=12556">RHEA:12556</a>)</small><BR>
<b>disruption</b> <b>phenotype:</b> No visible phenotype under normal growth conditions. <LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AT3G49360')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3142401">Divergent evolutionary fates of major photosynthetic gene networks following gene and whole genome duplications</a><br /><small><a title="Coate JE, Doyle JJ.">Coate,</a> Plant signaling & behavior 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...-0.51 0.85 AT5G41670 -0.11 0.51 AT1G71170 -0.39 0.37 <B><span style="color: red;">AT3G49360</span></B> 0.39 -0.51 AT5G24420 0.31 -0.33 AT5G24410 -0.33 0.40 AT3G04790 0.84 -0.70 AT5G44520 0.70 -0.77...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::AL01_06130')" href="http://www.ncbi.nlm.nih.gov/protein/WP_077396539.1">AL01_06130</a> 6-phosphogluconolactonase from <i>Bombella intestini</i><BR><a title="38:211/255 of query is similar to 40:208/246 of hit (E = 4e-14, 79.3 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=AL01_06130&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_077396539.1" style="font-family: sans-serif; font-size: smaller;">29% identity, 68% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AL01_06130')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5112900">Whole-Genome Sequence Analysis of Bombella intestini LMG 28161T, a Novel Acetic Acid Bacterium Isolated from the Crop of a Red-Tailed Bumble Bee, Bombus lapidarius</a><br /><small><a title="Li L, Illeghems K, Van Kerrebroeck S, Borremans W, Cleenwerck I, Smagghe G, De Vuyst L, Vandamme P.">Li,</a> PloS one 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...(AL01_07015); 34, 6-phosphogluconate dehydrogenase (AL01_06120); 35, ribose-5-phosphate isomerase (AL01_06135); 36, glucose-6-phosphate dehydrogenase (AL01_02790); 37, 6-phosphogluconolactonase (<B><span style="color: red;">AL01_06130</span></B>); 38, ribulose-phosphate 3-epimerase (AL01_09060); 39, polyol:NADP oxidoreductase (AL01_07080); 40, phosphomannomutase (AL_0102400); 41, mannose-1-phosphate guanyltransferase (AL01_07360); 42, carbohydrate kinase (AL01_03675); 43, mannose06-phosphate isomerase (AL01_00140); 44, 2-dehydro-3-deoxyphosphogluconate aldolase (AL01_04330). All genes encoding...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::SCO1939')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=242296">SCO1939</a> 6-phosphogluconolactonase (NCBI) from <i>Streptomyces coelicolor A3(2)</i><BR><a title="21:255/255 of query is similar to 28:255/261 of hit (E = 5e-14, 79.0 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=SCO1939&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS242296" style="font-family: sans-serif; font-size: smaller;">29% identity, 90% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::SCO1939')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3294797">Functional analysis of the N-acetylglucosamine metabolic genes of Streptomyces coelicolor and role in control of development and antibiotic production</a><br /><small><a title="Świątek MA, Tenconi E, Rigali S, van Wezel GP.">Świątek,</a> Journal of bacteriology 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...(SCO3070; 35% aa identity to NagA), a 6-phosphogluconolactonase (<B><span style="color: red;">SCO1939</span></B>; 29% aa identity to NagB), and an FGGY family carbohydrate kinase (SCO6260; 35% aa...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::SCO1939')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3277767">Metabolic switches and adaptations deduced from the proteomes of Streptomyces coelicolor wild type and phoP mutant grown in batch culture</a><br /><small><a title="Thomas L, Hodgson DA, Wentzel A, Nieselt K, Ellingsen TE, Moore J, Morrissey ER, Legaie R, STREAM Consortium, Wohlleben W, Rodríguez-García A, Martín JF, Burroughs NJ, Wellington EM, Smith MC.">Thomas,</a> Molecular & cellular proteomics : MCP 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...pentose phosphate pathway, i.e. 6-phosphogluconolactonase (<B><span style="color: red;">SCO1939</span></B>) and a transketolase for the conversion of Derythrose-4-phosphate to...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::SCO1939')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4546429">Conservation of thiol-oxidative stress responses regulated by SigR orthologues in actinomycetes</a><br /><small><a title="Kim MS, Dufour YS, Yoo JS, Cho YB, Park JH, Nam GB, Kim HM, Lee KL, Donohue TJ, Roe JH.">Kim,</a> Molecular microbiology 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...pentose phosphate pathway S1 9 SCO1937 zwf2 Glucose 6-phosphate 1-dehydrogenase 9 SCO1938 Hypothetical protein 9 <B><span style="color: red;">SCO1939</span></B> pgl 6-phosphogluconolactonase 10 SCO1958 * uvrA ABC excision nuclease subunit A (A), S1 14 SCO1995 * Unknown (S1, A) 11 SCO1996 coaE Dephospho-CoA kinase (A) 11 SCO1997 * Unknown (S1,...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::XP_828707')" href="http://www.ncbi.nlm.nih.gov/protein/XP_828707.1">XP_828707</a> 6-phosphogluconolactonase from <i>Trypanosoma brucei brucei TREU927</i><BR><a title="37:246/255 of query is similar to 38:258/266 of hit (E = 7e-14, 78.2 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=XP_828707&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_828707.1" style="font-family: sans-serif; font-size: smaller;">28% identity, 79% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::XP_828707')" href="http://doi.org/10.1002/prot.24019">Insights into internal dynamics of 6-phosphogluconolactonase from Trypanosoma brucei studied by nuclear magnetic resonance and molecular dynamics.</a><br /><small><a title="Calligari PA, Salgado GF, Pelupessy P, Lopes P, Ouazzani J, Bodenhausen G, Abergel D">Calligari,</a> Proteins 2012 (<a onmousedown="logger(this, 'pb::XP_828707')" href="http://www.ncbi.nlm.nih.gov/pubmed/22275079">PubMed</a>)</small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a title="from Gene Reference into Function (NCBI)" href="https://www.ncbi.nlm.nih.gov/gene/about-generif" style="color: black; text-decoration: none; font-style: italic;">GeneRIF</a>: The distribution of eigenvalues are rather similar in apo-and holo-Tb6PGL, therefore indicating similar overall flexibilities in both proteins.</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::MAP1174c')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=535110">MAP1174c</a> DevB (NCBI) from <i>Mycobacterium avium subsp. paratuberculosis str. k10</i><BR><a title="2:255/255 of query is similar to 3:247/251 of hit (E = 9e-14, 77.8 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=MAP1174c&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS535110" style="font-family: sans-serif; font-size: smaller;">28% identity, 100% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::MAP1174c')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2223459">Profiling bovine antibody responses to Mycobacterium avium subsp. paratuberculosis infection by using protein arrays</a><br /><small><a title="Bannantine JP, Paustian ML, Waters WR, Stabel JR, Palmer MV, Li L, Kapur V.">Bannantine,</a> Infection and immunity 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...MAP0864 MAP0865 MAP0866 MAP0904 MAP0961c MAP1087 MAP1121c <B><span style="color: red;">MAP1174c</span></B> MAP1204 MAP1233 MAP1345 MAP1388 MAP1416c MAP1417c MAP1609c MAP1636c MAP1643 MAP1655c MAP1730c...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...D3 A1 H4 MAP1417c MAP2663c MAP3121 MAP4129 <B><span style="color: red;">MAP1174c</span></B> MAP4199 MAP3902c MAP2764c MAP3131 MAP0261c MAP2360c MAP3903c MAP2761c MAP1121c MAP0182c MAP2156 MAP0863...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::MAP1174c')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2265687">Early antibody response against Mycobacterium avium subspecies paratuberculosis antigens in subclinical cattle</a><br /><small><a title="Bannantine JP, Bayles DO, Waters WR, Palmer MV, Stabel JR, Paustian ML.">Bannantine,</a> Proteome science 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...146 prob. peptide transp. system permease, N-terminal MAP1121c 0.32 498 165 putative substrate binding protein <B><span style="color: red;">MAP1174c</span></B> 1.64 756 251 glucose-6-phosphate 1-dehydrogenase MAP1204 2.61 735 244 putative exported p60 protein homologue MAP1233 0.04 723 240 conserved hypothetical protein MAP1345 0.69 600 199 no BLAST hits MAP1388 0.08...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...0.874 33.469 75.392 68.964 B11 MAP1121c 0.758 2.438 0 2.094 0.699 3.364 0 1.848 A10 <B><span style="color: red;">MAP1174c</span></B> 0 2.301 3.155 4.411 0.623 1.921 5.81 4.955 C11 MAP1204 0.78 40.8 43.856 55.046 0 28.367 61.122 58.985 F2 MAP1233 2.568 12.375 22.507 32.493 1.932 3.212 24.157 30.447 C5 MAP1345...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::blr6761')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=321972">blr6761</a> blr6761 (NCBI ptt file) from <i>Bradyrhizobium japonicum USDA 110</i><BR><a title="39:247/255 of query is similar to 40:241/250 of hit (E = 1e-13, 77.4 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=blr6761&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS321972" style="font-family: sans-serif; font-size: smaller;">31% identity, 82% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::blr6761')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2045231">Transcriptional and physiological responses of Bradyrhizobium japonicum to desiccation-induced stress</a><br /><small><a title="Cytryn EJ, Sangurdekar DP, Streeter JG, Franck WL, Chang WS, Stacey G, Emerich DW, Joshi T, Xu D, Sadowsky MJ.">Cytryn,</a> Journal of bacteriology 2007 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...bll7185 bll5979 blr6459 bll2817 blr6758 blr6759 blr6760 <B><span style="color: red;">blr6761</span></B> blr6762 bll0324 blr3200 blr3201 bll3208 blr2271 blr0725 blr4948 blr0612 blr8148 blr3575 blr8147...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...of glucose-6phosphate dehydrogenase (blr6760), 6-phosphoglucolactonase (<B><span style="color: red;">blr6761</span></B>), and 6-phosphogluconate dehydrogenase (blr6759) (Table 2), all of which encode...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="BRENDA" onmousedown="logger(this, 'curated::Q9GRG6')" href="http://www.brenda-enzymes.org/sequences.php?AC=Q9GRG6">Q9GRG6</a> <b>6-phosphogluconolactonase (EC 3.1.1.31)</b> from <i>Trypanosoma brucei</i> (see <a onmousedown="logger(this, 'curatedpaper::Q9GRG6')" href="http://www.ncbi.nlm.nih.gov/pubmed/10867008,11457850">2 papers</a>)<BR><a title="37:246/255 of query is similar to 38:258/266 of hit (E = 1e-13, 77.4 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=Q9GRG6&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=BRENDA::Q9GRG6" style="font-family: sans-serif; font-size: smaller;">28% identity, 79% coverage</a></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::AMED_2804')" href="http://www.ncbi.nlm.nih.gov/protein/YP_003765001.1">AMED_2804</a> 6-phosphogluconolactonase from <i>Amycolatopsis mediterranei U32</i><BR><a title="38:208/255 of query is similar to 38:208/259 of hit (E = 4e-13, 75.5 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=AMED_2804&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=YP_003765001.1" style="font-family: sans-serif; font-size: smaller;">31% identity, 66% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AMED_2804')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5640789">A systematic study of the whole genome sequence of Amycolatopsis methanolica strain 239T provides an insight into its physiological and taxonomic properties which correlate with its position in the genus</a><br /><small><a title="Tang B, Xie F, Zhao W, Wang J, Dai S, Zheng H, Ding X, Cen X, Liu H, Yu Y, Zhou H, Zhou Y, Zhang L, Goodfellow M, Zhao GP.">Tang,</a> Synthetic and systems biotechnology 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...AMED_2808 AORI_2795 AJAP_24815 pgi AMETH_4094 AMED_2807 AORI_2794 AJAP_24820 zwf AMETH_4095 AMED_2806 AORI_2793 AJAP_24825 pgl AMETH_4097 <B><span style="color: red;">AMED_2804</span></B> AORI_2791 AJAP_24835 Glycolysis/gluconeogenesis glpX AMETH_5852 AMED_8059 AORI_6881 AJAP_04695 AMETH_4516 AMETH_4530 fbaA AMETH_6838 AMED_9110 AORI_7850 AJAP_40945 AMETH_4655 AMETH_6121 AMED_3923 AORI_5186 AJAP_13540 AMETH_4531 Each line is the corresponding ortholog. On the other...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::ACPL_1861')" href="http://www.ncbi.nlm.nih.gov/protein/WP_014688831.1">ACPL_1861</a> 6-phosphogluconolactonase from <i>Actinoplanes sp. SE50/110</i><BR><a title="38:249/255 of query is similar to 38:235/238 of hit (E = 5e-13, 75.5 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=ACPL_1861&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_014688831.1" style="font-family: sans-serif; font-size: smaller;">30% identity, 83% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::ACPL_1861')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4479805">Reconstruction and in silico analysis of an Actinoplanes sp. SE50/110 genome-scale metabolic model for acarbose production</a><br /><small><a title="Wang Y, Xu N, Ye C, Liu L, Shi Z, Wu J.">Wang,</a> Frontiers in microbiology 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...synthesis, and genes ACPL_8304 and ACPL_887 participated in L-phenylalanine synthesis. In addition, genes ACPL_1328 and <B><span style="color: red;">ACPL_1861</span></B> belong to the pentose phosphate pathway, but there was no flux of the pentose phosphate pathway in sucrose medium. When a BLASTp search was performed with the database of essential...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...pathway and the inhibition of glycolysis. Thus, genes in the pentose phosphate pathway, such as <B><span style="color: red;">ACPL_1861</span></B>, ACPL_6461, ACPL_1328, were candidates for overexpression to improve acarbose production. In addition, the flux of the reaction catalyzed by alanine aminotransferase (EC: 2.6.1.2), which forms glutamate, increased with increasing acarbose...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::TP0477')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=26299">TP0477</a> glucose-6-phosphate 1-dehydrogenase, putative (NCBI ptt file) from <i>Treponema pallidum subsp. pallidum str. Nichols</i><BR><a title="36:252/255 of query is similar to 29:226/241 of hit (E = 5e-13, 75.5 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=TP0477&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS26299" style="font-family: sans-serif; font-size: smaller;">26% identity, 85% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::TP0477')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3373638">Footprint of positive selection in Treponema pallidum subsp. pallidum genome sequences suggests adaptive microevolution of the syphilis pathogen</a><br /><small><a title="Giacani L, Chattopadhyay S, Centurion-Lara A, Jeffrey BM, Le HT, Molini BJ, Lukehart SA, Sokurenko EV, Rockey DD.">Giacani,</a> PLoS neglected tropical diseases 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...204 232 Yes TPChic0425 TP0425 Hypothetical protein F GG insertion 453695:454240 181 86 No TPChic0477 <B><span style="color: red;">TP0477</span></B> Hypothetical protein R C insertion 507750:508454 234 241 No TPChic0479 TP0479 Hypothetical protein R C insertion 510135:510692 185 224 Yes TPChic0486 TP0486 Hypothetical protein R T deletion 518618:520213 581 484...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::AORI_2791')" href="http://www.ncbi.nlm.nih.gov/protein/WP_016333147.1">AORI_2791</a> 6-phosphogluconolactonase from <i>Amycolatopsis keratiniphila</i><BR><a title="38:244/255 of query is similar to 38:235/251 of hit (E = 5e-13, 75.5 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=AORI_2791&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_016333147.1" style="font-family: sans-serif; font-size: smaller;">28% identity, 81% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AORI_2791')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5640789">A systematic study of the whole genome sequence of Amycolatopsis methanolica strain 239T provides an insight into its physiological and taxonomic properties which correlate with its position in the genus</a><br /><small><a title="Tang B, Xie F, Zhao W, Wang J, Dai S, Zheng H, Ding X, Cen X, Liu H, Yu Y, Zhou H, Zhou Y, Zhang L, Goodfellow M, Zhao GP.">Tang,</a> Synthetic and systems biotechnology 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...AORI_2795 AJAP_24815 pgi AMETH_4094 AMED_2807 AORI_2794 AJAP_24820 zwf AMETH_4095 AMED_2806 AORI_2793 AJAP_24825 pgl AMETH_4097 AMED_2804 <B><span style="color: red;">AORI_2791</span></B> AJAP_24835 Glycolysis/gluconeogenesis glpX AMETH_5852 AMED_8059 AORI_6881 AJAP_04695 AMETH_4516 AMETH_4530 fbaA AMETH_6838 AMED_9110 AORI_7850 AJAP_40945 AMETH_4655 AMETH_6121 AMED_3923 AORI_5186 AJAP_13540 AMETH_4531 Each line is the corresponding ortholog. On the other hand,...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::CC2056')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=98871">CC2056</a> 6-phospho-glucono-lactonase (NCBI ptt file) from <i>Caulobacter crescentus CB15</i><BR><a title="2:241/255 of query is similar to 6:230/232 of hit (E = 1e-12, 74.3 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=CC2056&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS98871" style="font-family: sans-serif; font-size: smaller;">27% identity, 94% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::CC2056')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1855722">Genetic analysis of a novel pathway for D-xylose metabolism in Caulobacter crescentus</a><br /><small><a title="Stephens C, Christen B, Fuchs T, Sundaram V, Watanabe K, Jenal U.">Stephens,</a> Journal of bacteriology 2007 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Glycolysis and gluconeogenesis Xyl Gluc zwf (CC2057) <B><span style="color: red;">CC2056</span></B> ppc (CC1493) CC3065 1 1 2 1 Glucose-6-phosphate 1-dehydrogenase (EC 1.1.1.49)...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::CC2056')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC344409">Transcriptional profiling of Caulobacter crescentus during growth on complex and minimal media</a><br /><small><a title="Hottes AK, Meewan M, Yang D, Arana N, Romero P, McAdams HH, Stephens C.">Hottes,</a> Journal of bacteriology 2004 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...glk CC2055, phosphogluconate dehydratase, edd <B><span style="color: red;">CC2056</span></B>, 6-phosphogluconolactonase, pgl CC2057, glucose-6-phosphate 1-dehydrogenase, zwf Transport CC0028,...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::CC2056')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1170934">Probing the structure of complex macromolecular interactions by homolog specificity scanning: the P1 and P7 plasmid partition systems</a><br /><small><a title="Radnedge L, Youngren B, Davis M, Austin S.">Radnedge,</a> The EMBO journal 1998 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Biolabs (Beverly, MA). Bacterial strains and bacteriophage Strain <B><span style="color: red;">CC2056</span></B> [xis6ind - recA56 trpam thi lacam (Radnedge et al., 1996)] was used for the partition...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::AJAP_24835')" href="http://www.ncbi.nlm.nih.gov/protein/WP_038515527.1">AJAP_24835</a> 6-phosphogluconolactonase from <i>Amycolatopsis japonica</i><BR><a title="38:244/255 of query is similar to 38:235/251 of hit (E = 3e-12, 72.8 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=AJAP_24835&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_038515527.1" style="font-family: sans-serif; font-size: smaller;">27% identity, 81% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AJAP_24835')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5640789">A systematic study of the whole genome sequence of Amycolatopsis methanolica strain 239T provides an insight into its physiological and taxonomic properties which correlate with its position in the genus</a><br /><small><a title="Tang B, Xie F, Zhao W, Wang J, Dai S, Zheng H, Ding X, Cen X, Liu H, Yu Y, Zhou H, Zhou Y, Zhang L, Goodfellow M, Zhao GP.">Tang,</a> Synthetic and systems biotechnology 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...AJAP_24815 pgi AMETH_4094 AMED_2807 AORI_2794 AJAP_24820 zwf AMETH_4095 AMED_2806 AORI_2793 AJAP_24825 pgl AMETH_4097 AMED_2804 AORI_2791 <B><span style="color: red;">AJAP_24835</span></B> Glycolysis/gluconeogenesis glpX AMETH_5852 AMED_8059 AORI_6881 AJAP_04695 AMETH_4516 AMETH_4530 fbaA AMETH_6838 AMED_9110 AORI_7850 AJAP_40945 AMETH_4655 AMETH_6121 AMED_3923 AORI_5186 AJAP_13540 AMETH_4531 Each line is the corresponding ortholog. On the other hand, in...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::BBMN68_1187')" href="http://www.ncbi.nlm.nih.gov/protein/WP_077321619.1">BBMN68_1187</a> 6-phosphogluconolactonase from <i>Bifidobacterium longum subsp. longum BBMN68</i><BR><a title="38:246/255 of query is similar to 75:305/310 of hit (E = 4e-12, 72.4 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=BBMN68_1187&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_077321619.1" style="font-family: sans-serif; font-size: smaller;">26% identity, 67% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::BBMN68_1187')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3517610">Mechanism analysis of acid tolerance response of bifidobacterium longum subsp. longum BBMN 68 by gene expression profile using RNA-sequencing</a><br /><small><a title="Jin J, Zhang B, Guo H, Cui J, Jiang L, Song S, Sun M, Ren F.">Jin,</a> PloS one 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...shunt continuously provided sufficient substrates and energy for ATR in BBMN68. In addition, BBMN68_1185 , <B><span style="color: red;">BBMN68_1187</span></B> , and BBMN68_1188 , which are responsible for the conversion from glucose-6-phosphate to ribulose-5-phosphate, were upregulated by more than 2 fold after acid-adaptation. This shift enhanced the organism's ability to...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Gene Gene description BBMN68_1185 237.79 512.44 2.16 * G COG0364 zwf K00036, glucose-6-phosphate 1-dehydrogenase [EC:1.1.1.49] <B><span style="color: red;">BBMN68_1187</span></B> 80.61 173.56 2.15 * G COG0363 magB K01057, 6-phosphogluconolactonase [EC:3.1.1.31] BBMN68_1188 815.25 1577.33 1.93 G COG0362 gnd K00033, 6-phosphogluconate dehydrogenase [EC:1.1.1.44] BBMN68_408 1291.78 1065.78 1.21 G COG0021 tkTA K00615, transketolase...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::BAB2_0459')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=1055506">BAB2_0459</a> 6- phosphogluconolactonase:Glucosamine/galactosami ne-6-phosphate isomerase (NCBI) from <i>Brucella melitensis biovar Abortus 2308</i><BR><a title="7:219/255 of query is similar to 8:205/232 of hit (E = 5e-12, 72.0 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=BAB2_0459&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS1055506" style="font-family: sans-serif; font-size: smaller;">29% identity, 84% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::BAB2_0459')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5587712">Iron-dependent reconfiguration of the proteome underlies the intracellular lifestyle of Brucella abortus</a><br /><small><a title="Roset MS, Alefantis TG, DelVecchio VG, Briones G.">Roset,</a> Scientific reports 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...NO 1.86 0.16 Pentose Phosphate Shunt GI:82700516 BAB1_1740 tkt transketolase C NO 1.25 0.04 GI:83269362 <B><span style="color: red;">BAB2_0459</span></B> pgl 6-phosphogluconolactonase U NO 0.63 0.05 GI:83269363 BAB2_0460 glucose-6-phosphate 1-dehydrogenase U NO 0.70 0.09 ATP-binding cassette ABC transporter complex GI:82699142 BAB1_0238 ABC-type sugar transport system, periplasmic comp. U YES 0.60...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::BAB2_0459')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2877026">Proteomics-based confirmation of protein expression and correction of annotation errors in the Brucella abortus genome</a><br /><small><a title="Lamontagne J, Béland M, Forest A, Côté-Martin A, Nassif N, Tomaki F, Moriyón I, Moreno E, Paramithiotis E.">Lamontagne,</a> BMC genomics 2010 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Unknown BAB1_0540 Formyl transferase, BAB1_1170 GltA BAB1_1840 MmsA BAB2_0457 FolD N-terminal BAB1_1174 FabZ BAB1_1872 PrfA <B><span style="color: red;">BAB2_0459</span></B> Pgl BAB1_0544 DegT/DnrJ/EryC1/StrS BAB1_1187 Endoribonuclease BAB1_1874 LysC BAB2_0460 Zwf BAB1_0561 Man-6-P isomerase L-PSP BAB1_1879 GrxC BAB2_0483 ShuT type II BAB1_1188 GDPD BAB1_1887 HemC BAB2_0513 GcvT BAB1_0570 XylA BAB1_1205 ElaB-domain BAB1_1895...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::RL0752')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=2203022">RL0752</a> putative 6-phosphogluconolactonase (NCBI) from <i>Rhizobium leguminosarum bv. viciae 3841</i><BR><a title="4:215/255 of query is similar to 5:201/232 of hit (E = 5e-12, 72.0 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=RL0752&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS2203022" style="font-family: sans-serif; font-size: smaller;">24% identity, 83% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::RL0752')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5165102">Role of O2 in the Growth of Rhizobium leguminosarum bv. viciae 3841 on Glucose and Succinate</a><br /><small><a title="Wheatley RM, Ramachandran VK, Geddes BA, Perry BJ, Yost CK, Poole PS.">Wheatley,</a> Journal of bacteriology 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Berkeley Locus pRL80070 pRL120710 RL0037 RL0054 RL0182 RL0183 RL0751 <B><span style="color: red;">RL0752</span></B> RL0753 RL0761 RL1010 RL1351 RL2234 RL2239 RL2241 RL2243 Role of O2 in the Growth of...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::RL0752')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5118466">The Use of Transposon Insertion Sequencing to Interrogate the Core Functional Genome of the Legume Symbiont Rhizobium leguminosarum</a><br /><small><a title="Perry BJ, Akter MS, Yost CK.">Perry,</a> Frontiers in microbiology 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...1-dehydrogenase PGN 1 19 NE 1 14.5 NE 6.3.0 Putative XRE family transcriptional regulator 1.2 <B><span style="color: red;">RL0752</span></B> pgl 6-phosphogluconolactonase GI 0 0 ES 0 0 ES 3.5.6 Putative 6-phosphogluconolactonase 1.3 RL0751 edd Phosphogluconate dehydratase VGI 0 0 ES 1 8.73 NE 3.3.3 Putative phosphogluconate dehydratase 1.4 RL4162...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::PP1023, PP_1023')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=216720">PP1023, PP_1023</a> 6-phosphogluconolactonase (NCBI ptt file) from <i>Pseudomonas putida KT2440</i><BR><a title="2:227/255 of query is similar to 17:227/242 of hit (E = 1e-11, 71.2 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=PP1023, PP_1023&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS216720" style="font-family: sans-serif; font-size: smaller;">25% identity, 89% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PP1023, PP_1023')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4855977">Integrated analysis of gene expression and metabolic fluxes in PHA-producing Pseudomonas putida grown on glycerol</a><br /><small><a title="Beckers V, Poblete-Castro I, Tomasch J, Wittmann C.">Beckers,</a> Microbial cell factories 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...0.85 0.09 zwf2 PP4042 0.48 0.14 0.10 0.48 zwf3 PP5351 0.23 0.05 0.10 0.21 pgl <B><span style="color: red;">PP1023</span></B> 6-P-gluconate dehydrogenase 0.65 0.42 1.15 0.77 gnd PP4043 6-P-gluconate dehydrogenase 0.08 0.15 0.1 0.23 gnuK PP3416 Carbohydrate kinase 0.06 0.52 0.10 0.7 kguK PP3378 Dehydroglucokinase 0.02 0.63 0.10 0.21 kguD...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PP1023, PP_1023')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4077159">Production of medium chain length polyhydroxyalkanoate in metabolic flux optimized Pseudomonas putida</a><br /><small><a title="Borrero-de Acuña JM, Bielecka A, Häussler S, Schobert M, Jahn M, Wittmann C, Jahn D, Poblete-Castro I.">Borrero-de,</a> Microbial cell factories 2014 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...zwf1 PP1022 G6P dehydrogenase 0.9 0.2 zwf2 PP4042 0.1 -0.3 zwf3 PP5351 0.2 0.4 pgl <B><span style="color: red;">PP1023</span></B> 6-phosphogluconate dehydrogenase 0.5 -0.9 gnd PP4043 6-phosphoglucolacto dehydrogenase -0.6 -0.1 gnuK PP3416 Carbohydrate kinase -1.1 -0.6 kguK PP3378 Dehydroglucokinase 0.45 -0.4 kguD PP3376 2-Ketogluconate 6-phosphate reductase -0.7 -0.5 rpiA PP5150...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...zwf1 PP1022 G6P dehydrogenase 5.2 3.6 zwf2 PP4042 0.2 -0.6 zwf3 PP5351 -0.1 -0.3 pgl <B><span style="color: red;">PP1023</span></B> 6-phosphogluconate dehydrogenase 9.3 7.8 gnd PP4043 6-phosphoglucolacto dehydrogenase 0.2 -0.3 gnuK PP3416 Carbohydrate kinase -0.1 0.3 kguK PP3378 Dehydroglucokinase 0.6 0.5 kguD PP3376 2-Ketogluconate 6-phosphate reductase 0.1 -0.1 rpiA PP5150...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PP1023, PP_1023')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2293218">A set of activators and repressors control peripheral glucose pathways in Pseudomonas putida to yield a common central intermediate</a><br /><small><a title="del Castillo T, Duque E, Ramos JL.">del,</a> Journal of bacteriology 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...9 17 This study This study This study <B><span style="color: red;">PP1023</span></B> PP1516 PP2333 PP2741 PP3415 PP3416 PP3417 PP3418 PP3642 a Gene Function pgl 6-Phosphogluconolactonase RND efflux...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...organizationa PP1009 PP1010 PP1011 PP1012 PP1022 <B><span style="color: red;">PP1023</span></B> PP1024 PP4488 gap-1 edd glk gltR-2 zwf-1 pgl eda Glyceraldehyde-3-P-dehydrogenase 6-Phosphogluconate...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PP1023, PP_1023')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2563689">Genome-scale reconstruction and analysis of the Pseudomonas putida KT2440 metabolic network facilitates applications in biotechnology</a><br /><small><a title="Puchałka J, Oberhardt MA, Godinho M, Bielecka A, Regenhardt D, Timmis KN, Papin JA, Martins dos Santos VA.">Puchałka,</a> PLoS computational biology 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...22.26 0.83 11.16 1 Triose-phosphate isomerase PP4715 d -Fructose 7.7 29.74 0.83 3.5 6-Phosphoglucono lactonase <B><span style="color: red;">PP1023</span></B> 2 Glucose dehydrogenase (membrane) PP1444 d -Glucose 7.05 28.51 0.83 4.17 6-Phosphoglucono lactonase <B><span style="color: red;">PP1023</span></B> 3 Isocitrate dehydrogenase PP4011 or PP4012 l -Serine 22.41 23.01 6.66 10.67 Formate dehydrogenase PP0490 or...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PP1023, PP_1023')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2583600">A two-component regulatory system integrates redox state and population density sensing in Pseudomonas putida</a><br /><small><a title="Fernández-Piñar R, Ramos JL, Rodríguez-Herva JJ, Espinosa-Urgel M.">Fernández-Piñar,</a> Journal of bacteriology 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...metabolism PP_0794 (fruK) PP_1010 (edd) <B><span style="color: red;">PP_1023</span></B> (pgl) 1-Phosphofructokinase 6-Phosphogluconate dehydratase 6-Phosphogluconolactonase 2.8 2.6 2.2 Amino acid...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PP1023, PP_1023')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1797225">The RpoT regulon of Pseudomonas putida DOT-T1E and its role in stress endurance against solvents</a><br /><small><a title="Duque E, Rodríguez-Herva JJ, de la Torre J, Domínguez-Cuevas P, Muñoz-Rojas J, Ramos JL.">Duque,</a> Journal of bacteriology 2007 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...PP0290 PP0291 PP0379 PP0663 PP0700 PP0794 PP0861 PP1009 <B><span style="color: red;">PP1023</span></B> PP1024 PP1399 PP1400 PP1545 PP1673 PP1827 PP1891 PP2307 PP2310 PP2794 PP2885 PP2886 PP2887 PP3397...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PP1023, PP_1023')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC532441">Genetic evidence that catabolites of the Entner-Doudoroff pathway signal C source repression of the sigma54 Pu promoter of Pseudomonas putida</a><br /><small><a title="Velázquez F, di Bartolo I, de Lorenzo V.">Velázquez,</a> Journal of bacteriology 2004 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...regulator HexR, PP1022 determines glucose-6-phosphate dehydrogenase, <B><span style="color: red;">PP1023</span></B> encodes phosphogluconolactonase, and PP1025 corresponds to leuA, the gene for...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="BRENDA" onmousedown="logger(this, 'curated::Q9X2N2')" href="http://www.brenda-enzymes.org/sequences.php?AC=Q9X2N2">Q9X2N2</a> <b>6-phosphogluconolactonase (EC 3.1.1.31)</b> from <i>Pseudomonas aeruginosa</i> (see <a onmousedown="logger(this, 'curatedpaper::Q9X2N2')" href="http://www.ncbi.nlm.nih.gov/pubmed/10869070">paper</a>)<BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::PA3182')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=59632">PA3182</a> 6-phosphogluconolactonase (NCBI) from <i>Pseudomonas aeruginosa PAO1</i><BR><a title="2:223/255 of query is similar to 12:218/238 of hit (E = 1e-11, 70.9 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=Q9X2N2&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS59632" style="font-family: sans-serif; font-size: smaller;">27% identity, 87% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PA3182')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6591848">Seawater salt-trapped Pseudomonas aeruginosa survives for years and gets primed for salinity tolerance</a><br /><small><a title="Elabed H, González-Tortuero E, Ibacache-Quiroga C, Bakhrouf A, Johnston P, Gaddour K, Blázquez J, Rodríguez-Rojas A.">Elabed,</a> BMC microbiology 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...0.00038 Aspartate ammonia-lyase aspA PA5429 2.56 0.0029 Sulfite reductase cysI PA1838 2.47 0.0029 6-phosphogluconolactonase pgI <B><span style="color: red;">PA3182</span></B> 2.44 0.0018 Soluble pyridine nucleotide transhydrogenase sth PA2991 2.35 0.0041 Probable ferredoxin PA1551* 2.29 ID327 0.0017 Cytochrome C-oxidase, cbb3-type, CcoP subunit ccoP2 PA1555* 2.22 0.0041 Probable cytochrome c oxidase subunit...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PA3182')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6814747">Reverse diauxie phenotype in Pseudomonas aeruginosa biofilm revealed by exometabolomics and label-free proteomics</a><br /><small><a title="Yung YP, McGill SL, Chen H, Park H, Carlson RP, Hanley L.">Yung,</a> NPJ biofilms and microbiomes 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...2.1 MaiA PA2007 Maleylacetoacetate isomerase 5 39.2% 2.0 HutI PA5092 Imidazolonepropionase 3 11.2% 1.8 Pgl <B><span style="color: red;">PA3182</span></B> 6-phosphogluconolactonase 9 52.1% 1.5 Hfq PA4944 RNA-binding protein in P. aeruginosa 16 51.3% 1.5 Biofilm LldD PA4771 L-lactate dehydrogenase 12 39.60% 5.3 FolE2 PA5539 GTP cyclohydrolase 6 36.20% 2.9 PvdA...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PA3182')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6444534">Proteome profiling of Pseudomonas aeruginosa PAO1 identifies novel responders to copper stress</a><br /><small><a title="Wright BW, Kamath KS, Krisp C, Molloy MP.">Wright,</a> BMC microbiology 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...PA3920 4.0 0.003 2 Probable metal transporting P-type ATPase Cytoplasmic Membrane [Class 3] Class 3 <B><span style="color: red;">PA3182</span></B> 3.0 0.008 2 6-phosphogluconolactonase Unknown [Class 3] Class 1 PA0283 2.8 0.009 10 Sulfate-binding protein precursor Periplasmic [Class 2]; Periplasmic [Class 3] Class 2 PA3450 2.8 0.001 6 Probable antioxidant...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PA3182')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6235884">The development of a new parameter for tracking post-transcriptional regulation allows the detailed map of the Pseudomonas aeruginosa Crc regulon</a><br /><small><a title="Corona F, Reales-Calderón JA, Gil C, Martínez JL.">Corona,</a> Scientific reports 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...FAD-dependent oxidoreductase 1,55 0,2 2,13 CCM/catabolism eda PA3181 2-keto-3-deoxy-6-phosphogluconate aldolase 0,29 1,6 4,84 CCM/catabolism pgl <B><span style="color: red;">PA3182</span></B> 6-phosphogluconolactonase 0,64 1,71 5,53 CCM/catabolism zwf PA3183 Glucose-6-phosphate 1-dehydrogenase 1,18 1,63 5,83 Transport oprB PA3186 Glucose/carbohydrate outer membrane porin OprB precursor 2,35 2,76 5,5 Transport gltK PA3187 Probable ATP-binding component...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PA3182')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3632916">Polymyxin resistance of Pseudomonas aeruginosa phoQ mutants is dependent on additional two-component regulatory systems</a><br /><small><a title="Gutu AD, Sgambati N, Strasbourger P, Brannon MK, Jacobs MA, Haugen E, Kaul RK, Johansen HK, Høiby N, Moskowitz SM.">Gutu,</a> Antimicrobial agents and chemotherapy 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...PA1777 PA1802 PA2023 PA2397 PA2653 PA2853 PA2854 PA3078 <B><span style="color: red;">PA3182</span></B> PA3257 PA3553 PA3556 PA3737 PA4020 Gene name and/or function Gutu et al. tional major peaks...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PA3182')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1169510">A cystic fibrosis epidemic strain of Pseudomonas aeruginosa displays enhanced virulence and antimicrobial resistance</a><br /><small><a title="Salunkhe P, Smart CH, Morgan JA, Panagea S, Walshaw MJ, Hart CA, Geffers R, Tümmler B, Winstanley C.">Salunkhe,</a> Journal of bacteriology 2005 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...TABLE 2--Continued ORFa PA2747 PA2939 PA3032 PA3104 PA3181 <B><span style="color: red;">PA3182</span></B> PA3183 PA3326 PA3327 PA3328 PA3329 PA3330 PA3331 PA3332 PA3333 PA3334 PA3335 PA3347 PA3361...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...PA1317 PA1319 PA1404 PA1431 PA1432 PA2365 PA2445 PA2446 PA3181 <B><span style="color: red;">PA3182</span></B> PA3183 PA3188 PA3190 Gene name GENE EXPRESSION IN A P. AERUGINOSA CF EPIDEMIC STRAIN VOL....&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PA3182')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC151497">Identification, timing, and signal specificity of Pseudomonas aeruginosa quorum-controlled genes: a transcriptome analysis</a><br /><small><a title="Schuster M, Lostroh CP, Ogi T, Greenberg EP.">Schuster,</a> Journal of bacteriology 2003 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...PA2717 PA2747 PA2927 PA2939 PA3022 PA3032 PA3104 PA3181 <B><span style="color: red;">PA3182</span></B> PA3183 PA3188 PA3189 PA3190 PA3194 PA3195 PA3274 PA3311 Description b VOL. 185, 2003...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::Pro0844')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=564268">Pro0844</a> 6-phosphogluconolactonase (NCBI) from <i>Prochlorococcus marinus str. SS120</i><BR><a title="37:251/255 of query is similar to 37:237/237 of hit (E = 1e-11, 70.9 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=Pro0844&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS564268" style="font-family: sans-serif; font-size: smaller;">30% identity, 84% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::Pro0844')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2565063">Glucose uptake and its effect on gene expression in prochlorococcus</a><br /><small><a title="Gómez-Baena G, López-Lozano A, Gil-Martínez J, Lucena JM, Diez J, Candau P, García-Fernández JM.">Gómez-Baena,</a> PloS one 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...2.7.1.2 Pro1065 Glucose-6-phosphate isomerase pgi 5.3.1.9 Pro0946 Glucose-6-phosphate dehydrogenase zwf 1.1.1.49 Pro1124 6-phosphogluconolactonase nagB 3.1.1.31 <B><span style="color: red;">Pro0844</span></B> 6-phosphogluconate dehydrogenase gnd 1.1.1.44 Pro0843 Phosphoribulokinase prk 2.7.1.19 Pro0861 Pentose-5-phosphate-3-epimerase rpe 5.1.3.1 Pro0839 Ribose-5-phosphate isomerase rpi 5.3.1.6 Pro1644 Transketolase tktA 2.2.1.1 Pro1770 Transaldolase tal 2.2.1.2 Pro0519 Triosephosphate isomerase tpi 5.3.1.1...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::TC0458')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=370773">TC0458</a> glucosamine-6-phosphate isomerase, putative (NCBI ptt file) from <i>Chlamydia muridarum Nigg</i><BR><a title="37:219/255 of query is similar to 44:210/256 of hit (E = 2e-11, 70.1 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=TC0458&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS370773" style="font-family: sans-serif; font-size: smaller;">30% identity, 71% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::TC0458')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3019893">Identification of immunodominant antigens by probing a whole Chlamydia trachomatis open reading frame proteome microarray using sera from immunized mice</a><br /><small><a title="Cruz-Fisher MI, Cheng C, Sun G, Pal S, Teng A, Molina DM, Kayala MA, Vigil A, Baldi P, Felgner PL, Liang X, de la Maza LM.">Cruz-Fisher,</a> Infection and immunity 2011 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...TC0422 TC0425 TC0437 TC0438 TC0439 TC0446 TC0448 TC0449 <B><span style="color: red;">TC0458</span></B> TC0473 TC0478 TC0479 TC0491 TC0500 TC0503 TC0504 TC0512 TC0519 TC0521 TC0522 TC0527 TC0536 TC0539...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::BB0222')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=43864">BB0222</a> glucose-6-phosphate 1-dehydrogenase, putative (NCBI ptt file) from <i>Borrelia burgdorferi B31</i><BR><a title="32:233/255 of query is similar to 25:216/235 of hit (E = 2e-11, 69.7 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=BB0222&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS43864" style="font-family: sans-serif; font-size: smaller;">30% identity, 79% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::BB0222')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2651281">A manganese transporter, BB0219 (BmtA), is required for virulence by the Lyme disease spirochete, Borrelia burgdorferi</a><br /><small><a title="Ouyang Z, He M, Oman T, Yang XF, Norgard MV.">Ouyang,</a> Proceedings of the National Academy of Sciences of the United States of America 2009 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...6-Gene Operon. In the Bb genome, bb0219, bb0220, bb0221, <B><span style="color: red;">bb0222</span></B>, bb0223, and bb0224 are oriented in the same direction (Fig. 1A). In this region, bb0219 is...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...from bb0220 by 40 bp, bb0223 is separated from <B><span style="color: red;">bb0222</span></B> by 20 bp, and no intergenic regions are predicted between the other adjacent genes (22). To determine...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::BB0222')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1698074">Identification of potential virulence determinants by Himar1 transposition of infectious Borrelia burgdorferi B31</a><br /><small><a title="Botkin DJ, Abbott AN, Stewart PE, Rosa PA, Kawabata H, Watanabe H, Norris SJ.">Botkin,</a> Infection and immunity 2006 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...including genes that encode a putative glucose-6-phosphate-1-dehydrogenase (<B><span style="color: red;">BB0222</span></B>), an alanyl t-RNA synthetase (BB0220), and a putative zinc permease (BB0219)....&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::PPUBIRD1_1073')" href="http://www.ncbi.nlm.nih.gov/protein/WP_041166853.1">PPUBIRD1_1073</a> 6-phosphogluconolactonase from <i>Pseudomonas putida BIRD-1</i><BR><a title="2:227/255 of query is similar to 12:222/237 of hit (E = 7e-11, 68.2 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=PPUBIRD1_1073&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_041166853.1" style="font-family: sans-serif; font-size: smaller;">25% identity, 89% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::PPUBIRD1_1073')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4720416">Understanding butanol tolerance and assimilation in Pseudomonas putida BIRD-1: an integrated omics approach</a><br /><small><a title="Cuenca Mdel S, Roca A, Molina-Santiago C, Duque E, Armengaud J, Gómez-Garcia MR, Ramos JL.">Cuenca,</a> Microbial biotechnology 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...genes involved in pentose phosphate pathways, such as Zwf, Edd and PgI (PPUBIRD1_1071, PPUBIRD1_1060 and <B><span style="color: red;">PPUBIRD1_1073</span></B>, respectively) were found to be strongly downregulated when butanol was used as sole carbon source. Membrane proteome involved in butanol assimilation QedH protein abundance was strongly upregulated (41.5fold change) in...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::EIO_1651, KVC_1673, KVH_07945, KvSKV_07915')" href="http://www.ncbi.nlm.nih.gov/protein/WP_013384425.1">EIO_1651, KVC_1673, KVH_07945, KvSKV_07915</a> 6-phosphogluconolactonase from <i>Ketogulonicigenium vulgare</i><BR><a title="RefSeq" onmousedown="logger(this, 'curated::KVU_1123')" href="http://www.ncbi.nlm.nih.gov/protein/YP_005794958.1">KVU_1123</a> 6-phosphogluconolactonase from <i>Ketogulonicigenium vulgare WSH-001</i><BR><a title="28:241/255 of query is similar to 27:220/223 of hit (E = 7e-11, 68.2 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=EIO_1651, KVC_1673, KVH_07945, KvSKV_07915&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_013384425.1" style="font-family: sans-serif; font-size: smaller;">28% identity, 84% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::EIO_1651, KVC_1673, KVH_07945, KvSKV_07915')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6293013">Reconstruction and analysis of carbon metabolic pathway of Ketogulonicigenium vulgare SPU B805 by genome and transcriptome</a><br /><small><a title="Wang CY, Li Y, Gao ZW, Liu LC, Wu YC, Zhang MY, Zhang TY, Zhang YX.">Wang,</a> Scientific reports 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Zwf PPP 1.1.1.49 KVC_1674 KvSKV_07920 KVH_07950 KVU_1124 EIO_1652 Pgl PPP 3.1.1.31 KVC_1673 KvSKV_07915 KVH_07945 KVU_1123 <B><span style="color: red;">EIO_1651</span></B> Gnd PPP 1.1.1.44 KVC_0890 KvSKV_03940 KVH_03970 KVU_0365 EIO_0834 RpiA PPP 5.3.1.6 KVC_0880 KvSKV_03890 KVH_03920 KVU_0356 EIO_0823 Rpe PPP 5.1.3.1 KVC_2798 KvSKV_03940 KVH_13715 KVU_2194 EIO_2692 TktA PPP 2.2.1.1 KVC_2247 KvSKV_10835 KVH_10920...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Null KVH_eda* KVU_PB0157 EIO_3350 Zwf PPP 1.1.1.49 KVC_1674 KvSKV_07920 KVH_07950 KVU_1124 EIO_1652 Pgl PPP 3.1.1.31 <B><span style="color: red;">KVC_1673</span></B> KvSKV_07915 KVH_07945 KVU_1123 EIO_1651 Gnd PPP 1.1.1.44 KVC_0890 KvSKV_03940 KVH_03970 KVU_0365 EIO_0834 RpiA PPP 5.3.1.6 KVC_0880 KvSKV_03890 KVH_03920 KVU_0356 EIO_0823 Rpe PPP 5.1.3.1 KVC_2798 KvSKV_03940 KVH_13715 KVU_2194 EIO_2692 TktA PPP...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...KVU_PB0157 EIO_3350 Zwf PPP 1.1.1.49 KVC_1674 KvSKV_07920 KVH_07950 KVU_1124 EIO_1652 Pgl PPP 3.1.1.31 KVC_1673 KvSKV_07915 <B><span style="color: red;">KVH_07945</span></B> KVU_1123 EIO_1651 Gnd PPP 1.1.1.44 KVC_0890 KvSKV_03940 KVH_03970 KVU_0365 EIO_0834 RpiA PPP 5.3.1.6 KVC_0880 KvSKV_03890 KVH_03920 KVU_0356 EIO_0823 Rpe PPP 5.1.3.1 KVC_2798 KvSKV_03940 KVH_13715 KVU_2194 EIO_2692 TktA PPP 2.2.1.1 KVC_2247...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...KVH_eda* KVU_PB0157 EIO_3350 Zwf PPP 1.1.1.49 KVC_1674 KvSKV_07920 KVH_07950 KVU_1124 EIO_1652 Pgl PPP 3.1.1.31 KVC_1673 <B><span style="color: red;">KvSKV_07915</span></B> KVH_07945 KVU_1123 EIO_1651 Gnd PPP 1.1.1.44 KVC_0890 KvSKV_03940 KVH_03970 KVU_0365 EIO_0834 RpiA PPP 5.3.1.6 KVC_0880 KvSKV_03890 KVH_03920 KVU_0356 EIO_0823 Rpe PPP 5.1.3.1 KVC_2798 KvSKV_03940 KVH_13715 KVU_2194 EIO_2692 TktA PPP 2.2.1.1...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::KVU_1123')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6293013">Reconstruction and analysis of carbon metabolic pathway of Ketogulonicigenium vulgare SPU B805 by genome and transcriptome</a><br /><small><a title="Wang CY, Li Y, Gao ZW, Liu LC, Wu YC, Zhang MY, Zhang TY, Zhang YX.">Wang,</a> Scientific reports 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...EIO_3350 Zwf PPP 1.1.1.49 KVC_1674 KvSKV_07920 KVH_07950 KVU_1124 EIO_1652 Pgl PPP 3.1.1.31 KVC_1673 KvSKV_07915 KVH_07945 <B><span style="color: red;">KVU_1123</span></B> EIO_1651 Gnd PPP 1.1.1.44 KVC_0890 KvSKV_03940 KVH_03970 KVU_0365 EIO_0834 RpiA PPP 5.3.1.6 KVC_0880 KvSKV_03890 KVH_03920 KVU_0356 EIO_0823 Rpe PPP 5.1.3.1 KVC_2798 KvSKV_03940 KVH_13715 KVU_2194 EIO_2692 TktA PPP 2.2.1.1 KVC_2247 KvSKV_10835...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::ZMO1478')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=958284">ZMO1478</a> 6-phosphogluconolactonase (RefSeq) from <i>Zymomonas mobilis subsp. mobilis ZM4</i><BR><a title="39:219/255 of query is similar to 39:208/235 of hit (E = 1e-10, 67.8 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=ZMO1478&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS958284" style="font-family: sans-serif; font-size: smaller;">30% identity, 71% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::ZMO1478')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6190654">Proteomic and metabolomic analysis of the cellular biomarkers related to inhibitors tolerance in Zymomonas mobilis ZM4</a><br /><small><a title="Chang D, Yu Z, Ul Islam Z, French WT, Zhang Y, Zhang H.">Chang,</a> Biotechnology for biofuels 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...metabolism ZMO0070, ZMO1422, ZMO1593 Posttranslational modification, protein turnover, and chaperones ZMO0472, ZMO1124, Rho Transcription Pgi, <B><span style="color: red;">ZMO1478</span></B> Carbohydrate transport and metabolism Mfd, ZMO0127 Replication, recombination, and repair ZMO1439, ZMO1522 Cell wall/membrane/envelope biogenesis ZMO0495, ZMO0912 Uncharacterized protein ZMO1754, ZMO1113 Energy production and conversion ZMO1485 Nucleotide transport and metabolism...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::ZMO1478')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5770649">Identification and Characterization of 5' Untranslated Regions (5'UTRs) in Zymomonas mobilis as Regulatory Biological Parts</a><br /><small><a title="Cho SH, Haning K, Shen W, Blome C, Li R, Yang S, Contreras LM.">Cho,</a> Frontiers in microbiology 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...with Clostridium Down in ethanol stress Up Ethanol ZMO1198 5-aminolevulinate synthase Down in xylose stress <B><span style="color: red;">ZMO1478</span></B> 6-phosphogluconolactonase Acetate ZMO0275 ABC transporter ZMO1399 fatty acid hydroxylase ZMO1432 Fusaric acid resistance protein ZMO0367 glucose-6-phosphate 1-dehydrogenase Up in ethanol stress Down ZMO1412 MucR family transcriptional regulator Down in ethanol...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...linkage of lactone, resulting in production of 6-phophogluconate in the pentose phosphate pathway. Up-regulation of <B><span style="color: red;">ZMO1478</span></B> (6-phosphogluconolactonase) transcript in acetate stress has been previously reported (Yang et al., 2014a ), and in this study, UTR_<B><span style="color: red;">ZMO1478</span></B> (6-phosphogluconolactonase) was shown to activate GFP expression under acetate stress (10...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::ZMO1478')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4578398">Transcriptome analysis of Zymomonas mobilis ZM4 reveals mechanisms of tolerance and detoxification of phenolic aldehyde inhibitors from lignocellulose pretreatment</a><br /><small><a title="Yi X, Gu H, Gao Q, Liu ZL, Bao J.">Yi,</a> Biotechnology for biofuels 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...ZMO0997 (KDPG aldolase), ZMO1237 ( d -isomer specific 2-hydroxyacid dehydrogenase), ZMO1307 (fumarase), ZMO1360 (pyruvate decarboxylase), <B><span style="color: red;">ZMO1478</span></B> (6-phosphogluconolactonase), ZMO1496 (phosphoenolpyruvate carboxylase), ZMO1596 (Fe-containing alcohol dehydrogenase), ZMO1608 (phosphopyruvate hydratase), and ZMO1963 (citrate synthase). The primers for qRT-PCR of the genes are listed in Additional file 1 Our hierarchical...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::ZMO1478')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4033097">Elucidation of Zymomonas mobilis physiology and stress responses by quantitative proteomics and transcriptomics</a><br /><small><a title="Yang S, Pan C, Hurst GB, Dice L, Davison BH, Brown SD.">Yang,</a> Frontiers in microbiology 2014 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...mostly related to the central carbon metabolism, which included genes such as 6-phosphogluconolactonase gene pgl (<B><span style="color: red;">ZMO1478</span></B>), phosphoglycerate mutase gene pgmA (ZMO1240), gluconolactonase gene gnl (ZMO1649), pyruvate decarboxylase gene pdC (ZMO1360), and pyruvate dehydrogenase gene pdhA and pdhB (ZMO1606, ZMO1605) (Additional Files 13 , 14B ). Genes...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::ZMO1478')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3495753">Transcriptome profiling of Zymomonas mobilis under ethanol stress</a><br /><small><a title="He MX, Wu B, Shui ZX, Hu QC, Wang WG, Tan FR, Tang XY, Zhu QL, Pan K, Li Q, Su XH.">He,</a> Biotechnology for biofuels 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...glf 0.92 ZMO0369 glk (glucokinase) 0.86 ZMO0367 zwf 0.79 ZMO1649 gnl 0.34 ZMO1757 gntK 0.48 <B><span style="color: red;">ZMO1478</span></B> pgl 0.67 ZMO0368 edd 0.89 ZMO0997 eda 0.88 ZMO0177 gap 0.97 ZMO0178 pgk 0.8 ZMO1240 gpm 0.91 ZMO1608 eno 0.87 ZMO0152 pyk 0.87 ZMO1496 Ppc 0.64 ZMO1237 ldhA 0.54 ZMO1360...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::ZMO1478')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6870993">The genome sequence of the ethanologenic bacterium Zymomonas mobilis ZM4</a><br /><small><a title="Seo JS, Chong H, Park HS, Yoon KO, Jung C, Kim JJ, Hong JH, Kim H, Kim JH, Kil JI, Park CJ, Oh HM, Lee JS, Jin SJ, Um HW, Lee HJ, Oh SJ, Kim JY, Kang HL, Lee SY, Lee KJ, Kang HS.">Seo,</a> Nature biotechnology 2005 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...zwf , ZMO0367) oxidizes glucose-6-phosphate to 6-phosphonolactone. The lactone is dehydrated to 6-phosphogluconate by lactonase (<B><span style="color: red;">ZMO1478</span></B>). 6-phosphogluconate is dehydrated by 6-phosphogluconate dehydratase ( edd , ZMO0368) to yield 2-keto-3-deoxy-6-phosphogluconate (KDPG). KDPG aldolase ( eda , ZMO0997) cleaves KDPG to form pyruvate and glyceraldehyde-3-phosphate ( Fig. 2...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::AMETH_4097')" href="http://www.ncbi.nlm.nih.gov/protein/WP_020421295.1">AMETH_4097</a> 6-phosphogluconolactonase from <i>Amycolatopsis methanolica 239</i><BR><a title="38:249/255 of query is similar to 38:248/259 of hit (E = 3e-10, 66.2 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=AMETH_4097&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_020421295.1" style="font-family: sans-serif; font-size: smaller;">26% identity, 82% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::AMETH_4097')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5640789">A systematic study of the whole genome sequence of Amycolatopsis methanolica strain 239T provides an insight into its physiological and taxonomic properties which correlate with its position in the genus</a><br /><small><a title="Tang B, Xie F, Zhao W, Wang J, Dai S, Zheng H, Ding X, Cen X, Liu H, Yu Y, Zhou H, Zhou Y, Zhang L, Goodfellow M, Zhao GP.">Tang,</a> Synthetic and systems biotechnology 2016 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...AMETH_4093 AMED_2808 AORI_2795 AJAP_24815 pgi AMETH_4094 AMED_2807 AORI_2794 AJAP_24820 zwf AMETH_4095 AMED_2806 AORI_2793 AJAP_24825 pgl <B><span style="color: red;">AMETH_4097</span></B> AMED_2804 AORI_2791 AJAP_24835 Glycolysis/gluconeogenesis glpX AMETH_5852 AMED_8059 AORI_6881 AJAP_04695 AMETH_4516 AMETH_4530 fbaA AMETH_6838 AMED_9110 AORI_7850 AJAP_40945 AMETH_4655 AMETH_6121 AMED_3923 AORI_5186 AJAP_13540 AMETH_4531 Each line is the corresponding ortholog. On the...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::CT186')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=189">CT186</a> Glucose-6-P Dehyrogenase (DevB family) (NCBI ptt file) from <i>Chlamydia trachomatis D/UW-3/CX</i><BR><a title="37:255/255 of query is similar to 44:251/256 of hit (E = 7e-10, 65.1 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=CT186&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS189" style="font-family: sans-serif; font-size: smaller;">28% identity, 86% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::CT186')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC427450">Chlamydia trachomatis lacks an adaptive response to changes in carbon source availability</a><br /><small><a title="Nicholson TL, Chiu K, Stephens RS.">Nicholson,</a> Infection and immunity 2004 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...pathway CT063 gnd CT121 araA CT185 zwf <B><span style="color: red;">CT186</span></B> devB CT213 rpiA CT313 tal CT331 dxs CT750 tktB 6-Phosphogluconate dehydrogenase Ribulose-3-phosphate epimerase...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::NMA1608')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=142675">NMA1608</a> 6-phosphogluconolactonase (NCBI ptt file) from <i>Neisseria meningitidis Z2491</i><BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::NMB1391')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=140467">NMB1391</a> oxidoreductase, Sol/DevB family (NCBI) from <i>Neisseria meningitidis MC58</i><BR><a title="39:231/255 of query is similar to 37:217/231 of hit (E = 9e-10, 64.7 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=NMA1608&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS142675" style="font-family: sans-serif; font-size: smaller;">26% identity, 76% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NMA1608')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1418650">Identification of a meningococcal L-glutamate ABC transporter operon essential for growth in low-sodium environments</a><br /><small><a title="Monaco C, Talà A, Spinosa MR, Progida C, De Nitto E, Gaballo A, Bruni CB, Bucci C, Alifano P.">Monaco,</a> Infection and immunity 2006 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...NMB1961 NMA0227 NMA0228 NMA0229 NMA0230 NMA0233 NMA1609 <B><span style="color: red;">NMA1608</span></B> NMA1607 NMA1605 NMA1604 NMA0493 NMA0492 NMA0491 NMA0490 NMB1962 NMB1963 NMB1964 NMB1965 NMB1966...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NMB1391')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1418650">Identification of a meningococcal L-glutamate ABC transporter operon essential for growth in low-sodium environments</a><br /><small><a title="Monaco C, Talà A, Spinosa MR, Progida C, De Nitto E, Gaballo A, Bruni CB, Bucci C, Alifano P.">Monaco,</a> Infection and immunity 2006 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Z2491) NMB2141 NMB2142 NMB2143 NMB2144 NMB2147 NMB1392 <B><span style="color: red;">NMB1391</span></B> NMB1390 NMB1389 NMB1388 NMB1958 NMB1959 NMB1960 NMB1961 NMA0227 NMA0228 NMA0229 NMA0230 NMA0233...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::H16_B2565')" href="http://www.ncbi.nlm.nih.gov/protein/WP_011617961.1">H16_B2565</a> 6-phosphogluconolactonase from <i>Cupriavidus necator H16</i><BR><a title="MicrobesOnline" onmousedown="logger(this, 'curated::H16_B2565')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=2042548">H16_B2565</a> 6-Phosphogluconolactonase (NCBI) from <i>Ralstonia eutropha H16</i><BR><a title="39:231/255 of query is similar to 36:213/226 of hit (E = 4e-09, 62.4 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=H16_B2565&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_011617961.1" style="font-family: sans-serif; font-size: smaller;">29% identity, 76% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::H16_B2565')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6572756">The genetic basis of 3-hydroxypropanoate metabolism in Cupriavidus necator H16</a><br /><small><a title="Arenas-López C, Locker J, Orol D, Walter F, Busche T, Kalinowski J, Minton NP, Kovács K, Winzer K.">Arenas-López,</a> Biotechnology for biofuels 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...or no changes, none of which were statistically significant. These included the genes for 6-phosphogluconolactonase (<B><span style="color: red;">H16_B2565</span></B>, no change), 6-phosphogluconate dehydratase (H16_B2567, no change) and 2-keto-3-deoxy-6-phosphogluconate aldolase (B16_B1213, 3.4-fold downregulated; p adj >0.05) (Additional file 3 : Table S1). A second 6-phosphogluconate dehydratase gene, H16_A1178, was strongly...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::RSP_2735')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=1139795">RSP_2735</a> 6-phosphogluconolactonase (NCBI) from <i>Rhodobacter sphaeroides 2.4.1</i><BR><a title="38:235/255 of query is similar to 35:219/223 of hit (E = 7e-09, 61.6 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=RSP_2735&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS1139795" style="font-family: sans-serif; font-size: smaller;">27% identity, 78% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::RSP_2735')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4323418">CceR and AkgR regulate central carbon and energy metabolism in alphaproteobacteria</a><br /><small><a title="Imam S, Noguera DR, Donohue TJ.">Imam,</a> mBio 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Fig.4 from reference 28 . pgi (glucose-6-phosphate isomerase; RSP_2736), zwf (glucose-6-phosphate 1-dehydrogenase; RSP_2734), pgl (6-phosphogluconolactonase; <B><span style="color: red;">RSP_2735</span></B>), edd (6-phosphogluconate dehydratase; RSP_2646), eda (KDPG/KHG bifunctional aldolase; RSP_2645), fbaB (fructose-bisphosphate aldolase; RSP_4045), pckA (phosphoenolpyruvate carboxykinase; RSP_1680), sucAB (-ketoglutarate dehydrogenase; RSP_0965-4), sucCD (succinyl-CoA synthetase; RSP_0967-6), sdh (succinate dehydrogenase; RSP_0974-9), atpHAGDC...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::RSP_2735')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4344238">An integrated approach to reconstructing genome-scale transcriptional regulatory networks</a><br /><small><a title="Imam S, Noguera DR, Donohue TJ.">Imam,</a> PLoS computational biology 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...enzyme (RSP_1593), PdhAB (RSP_2968-RSP_4047-RSP_4050), succinate dehydrogenase (RSP_09746), as well as glycolytic enzymes Zwf (RSP_2734), Pgl (<B><span style="color: red;">RSP_2735</span></B>), Pgi (RSP_2736) and FbaB (RSP_4045), potentially making this TF a major regulator of carbon metabolism under many conditions ( Fig. 6 ). This predicted RSP_1663 regulon might make it functionally...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::RSP_2735')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3441632">Convergence of the transcriptional responses to heat shock and singlet oxygen stresses</a><br /><small><a title="Dufour YS, Imam S, Koo BM, Green HA, Donohue TJ.">Dufour,</a> PLoS genetics 2012 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Entner-Doudoroff RSP_2646 Fermentation RSP_3164 Glycolysis/gluconeogenesis RSP_2736, RSP_4045, RSP_4211 Other RSP_0392, RSP_2294 Pentose phosphate pathway RSP_2734, <B><span style="color: red;">RSP_2735</span></B> Sugars RSP_2937, RSP_3138 Biosynthesis of cofactors, prosthetic groups, and carriers Glutathione and analogs RSP_3272 Heme, porphyrin, and cobalamin RSP_1197, RSP_1692, RSP_2831 Menaquinone and ubiquinone RSP_1175, RSP_1338, RSP_1492, RSP_1869 Other RSP_0750,...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::SO2488')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=201629">SO2488</a> 6-phosphogluconolactonase (NCBI ptt file) from <i>Shewanella oneidensis MR-1</i><BR><a title="4:219/255 of query is similar to 7:205/232 of hit (E = 1e-08, 60.5 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=SO2488&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS201629" style="font-family: sans-serif; font-size: smaller;">26% identity, 85% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::SO2488')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2980473">An empirical strategy for characterizing bacterial proteomes across species in the absence of genomic sequences</a><br /><small><a title="Turse JE, Marshall MJ, Fredrickson JK, Lipton MS, Callister SJ.">Turse,</a> PloS one 2010 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...3 SO2345 gapA-2 7 3 SO2347 gapA-3 4 SO2486 eda 3 2 SO2487 edd 5 <B><span style="color: red;">SO2488</span></B> pgl 2 SO2489 zwf 4 SO2491 pykA 10 4 SO2644 ppsA 15 6 3 SO3440 eno 8 3 2 2 SO3547 pgi 3 SO3991 fbp 4 3 2 TCA Cycle...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...3 3 2 3 4 SO2486 CN32_1869 eda SO2487 CN32_1868 edd 2 2 2 3 <B><span style="color: red;">SO2488</span></B> CN32_1867 pgl SO2489 CN32_1866 zwf SO2491 CN32_1864 pyk A 3 8 4 6 5 3 5 SO2644 CN32_2243 pps A 11 7 9 8 8 4 5 3 SO3440 CN32_2757...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::SO2488')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1482949">Global transcriptome analysis of the cold shock response of Shewanella oneidensis MR-1 and mutational analysis of its classical cold shock proteins</a><br /><small><a title="Gao H, Yang ZK, Wu L, Thompson DK, Zhou J.">Gao,</a> Journal of bacteriology 2006 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...so2487 (edd, 6-phosphogluconate dehydratase; 2.6-fold), <B><span style="color: red;">so2488</span></B> (pgl, 6-phosphogluconolactonase; 2.9-fold), and so2489 (zwf, glucose-6-phosphate 1-dehydrogenase;...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::BCAL3043')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=5925399">BCAL3043</a> 6-phosphogluconolactonase (RefSeq) from <i>Burkholderia cenocepacia J2315</i><BR><a title="1:240/255 of query is similar to 1:217/226 of hit (E = 2e-08, 60.1 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=BCAL3043&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS5925399" style="font-family: sans-serif; font-size: smaller;">22% identity, 94% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::BCAL3043')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5688073">Identification of small RNAs abundant in Burkholderia cenocepacia biofilms reveal putative regulators with a potential role in carbon and iron metabolism</a><br /><small><a title="Sass A, Kiekens S, Coenye T.">Sass,</a> Scientific reports 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Underlined: reverse complement of ribosome binding sites. ( C ) RNA duplex between ncS03 and <B><span style="color: red;">BCAL3043</span></B> (6-phosphonolactonase). Both CU-rich regions of ncS03 are involved in duplex formation. mRNA nucleotide positions are relative to start codon (boxed). ( D ) Overexpressing ncS27 attenuates growth. ( E )...&rdquo;<LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...e.g. gluconate permease (BCAL3365), galactonate transporter (BCAL0184, BCAM2500), glycerol kinase (BCAL0925), glycerol-phosphate transporter (BCAL3041), 6-phosphogluconolactonase (<B><span style="color: red;">BCAL3043</span></B>) and salicylate hydroxylase(BCAM1274). Predicted targets of ncS16 include a higher percentage of outer membrane and cell envelope components, as well as genes for transport of inorganic compounds. The predicted targets...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::Rmet_5800')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=1778177">Rmet_5800</a> 6-phosphogluconolactonase (RefSeq) from <i>Ralstonia metallidurans CH34</i><BR><a title="RefSeq" onmousedown="logger(this, 'curated::Rmet_5800')" href="http://www.ncbi.nlm.nih.gov/protein/WP_011520195.1">Rmet_5800</a> 6-phosphogluconolactonase from <i>Cupriavidus metallidurans CH34</i><BR><a title="39:231/255 of query is similar to 36:213/226 of hit (E = 3e-08, 59.3 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=Rmet_5800&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS1778177" style="font-family: sans-serif; font-size: smaller;">27% identity, 76% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::Rmet_5800')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2864759">The complete genome sequence of Cupriavidus metallidurans strain CH34, a master survivalist in harsh and anthropogenic environments</a><br /><small><a title="Janssen PJ, Van Houdt R, Moors H, Monsieurs P, Morin N, Michaux A, Benotmane MA, Leys N, Vallaeys T, Lapidus A, Monchy S, Médigue C, Taghavi S, McCorkle S, Dunn J, van der Lelie D, Mergeay M.">Janssen,</a> PloS one 2010 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...to 6-phosphogluconate (6GP) in a two-step reaction involving the pgl and zwf genes (encoded by <B><span style="color: red;">Rmet_5800</span></B> and Rmet_5801, respectively). The 6GP is metabolized through the alternative glycolytic Entner-Doudoroff (ED) pathway involving a 6-phosphogluconate dehydratase (EDD; Rmet_5802) and a 2-keto-3-deoxy-6-phosphogluconate aldolase (EDA; Rmet_4768), thus generating two pyruvate...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::NE1691')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=368371">NE1691</a> Glucosamine/galactosamine-6-phosphate isomerase (NCBI ptt file) from <i>Nitrosomonas europaea ATCC 19718</i><BR><a title="64:229/255 of query is similar to 62:214/228 of hit (E = 5e-08, 58.5 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=NE1691&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS368371" style="font-family: sans-serif; font-size: smaller;">26% identity, 65% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::NE1691')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2546632">Candidate stress genes of Nitrosomonas europaea for monitoring inhibition of nitrification by heavy metals</a><br /><small><a title="Park S, Ely RL.">Park,</a> Applied and environmental microbiology 2008 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...103 Carbohydrate transport and metabolism, <B><span style="color: red;">NE1691</span></B> Phosphogluconolactonase/glucosamine-6-phosphate isomerase/deaminase 3.8 6.2 104 Fatty acid biosynthesis,...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::C7W88_RS12355')" href="http://www.ncbi.nlm.nih.gov/protein/WP_118073754.1">C7W88_RS12355</a> 6-phosphogluconolactonase from <i>Novosphingobium sp. THN1</i><BR><a title="38:215/255 of query is similar to 35:179/208 of hit (E = 1e-07, 57.4 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=C7W88_RS12355&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_118073754.1" style="font-family: sans-serif; font-size: smaller;">29% identity, 70% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::C7W88_RS12355')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6691742">Microcystin-LR Degradation and Gene Regulation of Microcystin-Degrading Novosphingobium sp. THN1 at Different Carbon Concentrations</a><br /><small><a title="Wang J, Wang C, Li Q, Shen M, Bai P, Li J, Lin Y, Gan N, Li T, Zhao J.">Wang,</a> Frontiers in microbiology 2019 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...0.87915 1.302 ribose-phosphate pyrophosphokinase prsA C7W88_RS01250 2.4087 2.743 phosphoglucomutase pgm C7W88_RS11010 0.78001 1.1774 6-phosphogluconolactonase devB <B><span style="color: red;">C7W88_RS12355</span></B> 2.1998 2.9099 amino sugar and nucleotide sugar metabolism UDP-N-acetyl-D-mannosaminuronic acid dehydrogenase wecC C7W88_RS02750 0.52148 0.51175 phosphoglucosamine mutase glmM C7W88_RS03760 0.61432 0.51499 fructokinase scrK C7W88_RS12310 2.1873 1.8955 UDP-N-acetyl-D-galactosamine dehydrogenase wbpO C7W88_RS02740...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::CPA40_RS06275')" href="http://www.ncbi.nlm.nih.gov/protein/WP_107044159.1">CPA40_RS06275</a> glucosamine-6-phosphate deaminase from <i>Bifidobacterium callitrichos</i><BR><a title="34:211/255 of query is similar to 30:212/270 of hit (E = 2e-07, 56.6 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=CPA40_RS06275&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_107044159.1" style="font-family: sans-serif; font-size: smaller;">31% identity, 66% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::CPA40_RS06275')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6096940">The comparative genomics of Bifidobacterium callitrichos reflects dietary carbohydrate utilization within the common marmoset gut</a><br /><small><a title="Albert K, Rani A, Sela DA.">Albert,</a> Microbial genomics 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...CPA40_RS10465 BCAL_RS06425 UDP- N -acetylglucosamine diphosphorylase 2.7.7.23 COO72_RS04345 CPA40_RS08380 BCAL_RS00645 N -Acetylglucosamine-6-phosphate deacetylase 3.5.1.25 COO72_RS10300 <B><span style="color: red;">CPA40_RS06275</span></B> BCAL_RS02715 Glucosamine-1-phosphate N -acetyltransferase 2.3.1.157 COO72_RS04345 CPA40_RS08380 BCAL_RS00645 Phosphoglucosamine mutase 5.4.2.10 COO72_RS05530 CPA40_RS04580 BCAL_RS02085 Glucosamine-6-phosphate deaminase 3.5.99.6 COO72_RS08105 CPA40_RS05730 BCAL_RS02710 Glutaminefructose-6-phosphate transaminase (isomerizing) 2.6.1.16 COO72_RS05020 CPA40_RS04675 BCAL_RS02180 Non-reducing end -...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::WP_007113049')" href="http://www.ncbi.nlm.nih.gov/protein/WP_007113049.1">WP_007113049</a> 6-phosphogluconolactonase from <i>Halomonas boliviensis LC1</i><BR><a title="39:219/255 of query is similar to 32:197/221 of hit (E = 4e-07, 55.8 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=WP_007113049&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_007113049.1" style="font-family: sans-serif; font-size: smaller;">26% identity, 71% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::WP_007113049')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC4522284">Production of poly(3-hydroxybutyrate) by Halomonas boliviensis in an air-lift reactor</a><br /><small><a title="Rivera-Terceros P, Tito-Claros E, Torrico S, Carballo S, Van-Thuoc D, Quillaguamán J.">Rivera-Terceros,</a> Journal of biological research (Thessalonike, Greece) 2015 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...aldolase 4.1.2.14 WP_007113045 0800 1 - - - WP_007113048 2 + + + 6-phosphogluconolactonase 3.1.1.31 <B><span style="color: red;">WP_007113049</span></B> 0363 1 + + + Glucose-6-phosphate 1-dehydrogenase 1.1.1.49 WP_007113050 0364 1 + + + Transketolase 2.2.1.1 WP_007112132 0021 1 + + + Transaldolase 2.2.1.2 WP_007111145 0176 1 - + +...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::lpg0417')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=756881">lpg0417</a> 6-phosphogluconolactonase (NCBI) from <i>Legionella pneumophila subsp. pneumophila str. Philadelphia 1</i><BR><a title="2:226/255 of query is similar to 7:214/243 of hit (E = 5e-06, 52.0 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=lpg0417&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS756881" style="font-family: sans-serif; font-size: smaller;">24% identity, 88% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::lpg0417')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5928044">The LetA/S two-component system regulates transcriptomic changes that are essential for the culturability of Legionella pneumophila in water</a><br /><small><a title="Mendis N, McBride P, Saoud J, Mani T, Faucher SP.">Mendis,</a> Scientific reports 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...GDP mannose pyrophosphorylase (mannose-1-phosphate guanylyltransferase/mannose-6-phosphate isomerase) rfbA 1.57 2.14 Lpg0939 2-dehydro-3-deoxy-phosphogluconate aldolase eda 1.82 1.71 <B><span style="color: red;">Lpg0417</span></B> 6-phosphogluconolactonase pgl 1.91 2.46 Lpg0805 phosphoenolpyruvate synthase 2.32 2.85 Lpg2352 malate dehydrogenase mdh 1.39 2.28 Lpg2792 triosephosphate isomerase (TIM) tpiA 2.72 2.66 Lpg0138 glyceraldehyde 3-phosphate dehydrogenase gap 1.44 2.40 Cell...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="MicrobesOnline" onmousedown="logger(this, 'curated::lpp0484')" href="http://www.microbesonline.org/cgi-bin/fetchLocus.cgi?locus=753824">lpp0484</a> hypothetical protein (NCBI) from <i>Legionella pneumophila str. Paris</i><BR><a title="2:160/255 of query is similar to 7:148/243 of hit (E = 9e-06, 51.2 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=lpp0484&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=VIMSS753824" style="font-family: sans-serif; font-size: smaller;">28% identity, 62% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::lpp0484')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC5338858">The Legionella pneumophila genome evolved to accommodate multiple regulatory mechanisms controlled by the CsrA-system</a><br /><small><a title="Sahr T, Rusniok C, Impens F, Oliva G, Sismeiro O, Coppée JY, Buchrieser C.">Sahr,</a> PLoS genetics 2017 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...Glycerol-3-phosphate dehydrogenase, GlpK (Lpp1369), glycerol kinase, Tpi (Lpp2838), Triosephosphate isomerase, Zwf (Lpp0483), Glucose-6-phosphate 1-dehydrogenase, Pgl (<B><span style="color: red;">Lpp0484</span></B>), 6-Phosphogluconolactonase, Edd (Lpp0485), 6-Phosphogluconate dehydratase, Gap (Lpp0153), Glyceraldehyde 3-phosphate dehydrogenase, Pgk (Lpp0152), 3-Phosphoglycerate kinase, Eno (Lpp2020), Enolase, Pyk (Lpp0151), Pyruvate kinase, PpsA (Lpp0567), Phosphoenolpyruvate synthase, Pdh (Lpp1461), Pyruvate dehydrogenase complex,...&rdquo;</UL><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::lpp0484')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2903384">Isotopologue profiling of Legionella pneumophila: role of serine and glucose as carbon substrates</a><br /><small><a title="Eylert E, Herrmann V, Jules M, Gillmaier N, Lautner M, Buchrieser C, Eisenreich W, Heuner K.">Eylert,</a> The Journal of biological chemistry 2010 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...transcripts were determined via RT-PCR. lpp0483, zwf, <B><span style="color: red;">lpp0484</span></B>, pgl; lpp0485, edd; lpp0486, glk; lpp0487, eda; lpp0488 (putative sugar transport protein);...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::BCAL_RS02710')" href="http://www.ncbi.nlm.nih.gov/protein/WP_043164840.1">BCAL_RS02710</a> glucosamine-6-phosphate deaminase from <i>Bifidobacterium callitrichos DSM 23973</i><BR><a title="34:211/255 of query is similar to 30:212/270 of hit (E = 2e-04, 47.0 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=BCAL_RS02710&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=WP_043164840.1" style="font-family: sans-serif; font-size: smaller;">29% identity, 66% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::BCAL_RS02710')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC6096940">The comparative genomics of Bifidobacterium callitrichos reflects dietary carbohydrate utilization within the common marmoset gut</a><br /><small><a title="Albert K, Rani A, Sela DA.">Albert,</a> Microbial genomics 2018 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...2.3.1.157 COO72_RS04345 CPA40_RS08380 BCAL_RS00645 Phosphoglucosamine mutase 5.4.2.10 COO72_RS05530 CPA40_RS04580 BCAL_RS02085 Glucosamine-6-phosphate deaminase 3.5.99.6 COO72_RS08105 CPA40_RS05730 <B><span style="color: red;">BCAL_RS02710</span></B> Glutaminefructose-6-phosphate transaminase (isomerizing) 2.6.1.16 COO72_RS05020 CPA40_RS04675 BCAL_RS02180 Non-reducing end - l -arabinofuranosidase 3.2.1.55 COO72_RS05470 CPA40_RS04135 BCAL_RS02145 Glucose-6-phosphate isomerase 5.3.1.9 COO72_RS12210 CPA40_RS08670 BCAL_RS05265 UDP-glucose 6-dehydrogenase 1.1.1.22 COO72_RS07800 CPA40_RS09840 BCAL_RS07900 Phosphoglucomutase (-...&rdquo;</UL></UL></p>

<p style="margin-top: 1em; margin-bottom: 0em;"><a title="RefSeq" onmousedown="logger(this, 'curated::TGME49_078830')" href="http://www.ncbi.nlm.nih.gov/protein/XP_018635111.1">TGME49_078830</a> glucose-6-phosphate 1-dehydrogenase from <i>Toxoplasma gondii ME49</i><BR><a title="143:218/255 of query is similar to 253:330/878 of hit (E = 4e-04, 45.4 bits)" href="showAlign.cgi?def1=sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1&amp;def2=TGME49_078830&amp;seq1=MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNIFPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLIGGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNNSPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVTWFLDDEAGALIPENC&amp;acc2=XP_018635111.1" style="font-family: sans-serif; font-size: smaller;">36% identity, 9% coverage</a><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; ><a onmousedown="logger(this, 'pb::TGME49_078830')" href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3757501">Extensive lysine acetylation occurs in evolutionarily conserved metabolic pathways and parasite-specific functions during Plasmodium falciparum intraerythrocytic development</a><br /><small><a title="Miao J, Lawrence M, Jeffers V, Zhao F, Parker D, Ge Y, Sullivan WJ, Cui L.">Miao,</a> Molecular microbiology 2013 </small><UL style="margin-top: 0em; margin-bottom: 0em;"><LI style="list-style-type: none;" margin-left: 6em; >&ldquo;...aldolase PFI1340w fumarate hydratase, putative TGME49_067330 fumarase, putative (Mitochondrial localization) PF14_0511 glucose-6-phosphate dehydrogenase-6- phosphogluconolactonase (G6PDH) <B><span style="color: red;">TGME49_078830</span></B> glucose-6-phosphate dehydrogenase, putative PF08_0132 glutamate dehydrogenase, putative (GDHc) TGME49_049390 NAD-specific glutamate dehydrogenase, putative PF13_0257 glutamate--tRNA ligase, putative TGME49_063870 glutamyl-tRNA synthetase, putative PF14_0598 glyceraldehyde-3-phosphate dehydrogenase (GAPDH) TGME49_089690; TGME49_069190 glyceraldehyde-3-phosphate dehydrogenase PF11_0183...&rdquo;</UL></UL></p>

<h3>Query Sequence</h3><p style="font-family: monospace;"><small>>sp|P53315|SOL4_YEAST 6-phosphogluconolactonase 4 OS=Saccharomyces cerevisiae (strain ATCC 204508 / S288c) OX=559292 GN=SOL4 PE=1 SV=1<br />MVKLQRFSEKKSLIHEFGKFILEKQESALTGDADAVFNIAISGGSMNQALYESLVNDKNI<br />FPHIKWPQWRIFFCDERLVPFEDPQSNYGQFKKTVLDPLVHQGNQLNLGPTVYTINESLI<br />GGGETANRKIAEEYASMLPASFDLILLGCGEDGHTCSLFPGVEFNYLVEEMDRKVLWCNN<br />SPKAPKDRITFTLAVVAEAKSVCFLVRGAAKKAIMHDVLIVKNSELPSVLVNEMVGTKVT<br />WFLDDEAGALIPENC</small></p><h3><a href="litSearch.cgi">New Search</a></h3>
<H3><A NAME="stats">Statistics</A></H3>

The PaperBLAST database links 513,873 different protein sequences to 985,746 scientific articles. Searches against EuropePMC were last performed on February 4 2020.

<H3><A NAME="works">How It Works</A></H3>

<P>PaperBLAST builds a database of protein sequences that are linked
to scientific articles. These links come from automated text searches
against the articles in <A HREF="http://europepmc.org/" title="Europe PMC (PubMedCentral)">EuropePMC</A>
and from manually-curated information from <A
HREF="https://www.ncbi.nlm.nih.gov/gene/about-generif" title="Gene
Reference into Function (NCBI)">GeneRIF</A>, <A
HREF="http://www.uniprot.org/" title="The manually annotated and reviewed section of UniProt">UniProtKB/Swiss-Prot</A>,
<A HREF="http://www.brenda-enzymes.org/index.php" title="The Comprehensive Enzyme Information System">BRENDA</A>,
<A HREF="http://www.cazy.org/" title="Carbohydrate-Active enZYmes Database">CAZy</A> (as made available by <A HREF="http://csbl.bmb.uga.edu/dbCAN/download.php">dbCAN</A>),
<A HREF="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3245046/" title="Database of experimentally characterized protein annotations">CharProtDB</A>,
<A HREF="http://metacyc.org/" title="MetaCyc Metabolic Pathway Database">MetaCyc</A>,
<A HREF="http://ecocyc.org" title="EcoCyc: Encyclopedia of E. coli Genes and Metabolic Pathways">EcoCyc</A>,
<A HREF="http://rebase.neb.com/rebase/rebase.html" title="The Restriction Enzyme Database">REBASE</A>,
and the <A HREF="http://fit.genomics.lbl.gov/" title="Reannotations from genome-wide fitness data">Fitness Browser</A>.
Given this database and a protein sequence query,
PaperBLAST uses <A
HREF="https://en.wikipedia.org/wiki/BLAST">protein-protein BLAST</A>
to find similar sequences with E &lt; 0.001.

<P>To build the database, we query EuropePMC with locus tags, with <A
HREF="https://www.ncbi.nlm.nih.gov/refseq/">RefSeq</A> protein
identifiers, and with <A HREF="http://www.uniprot.org/">UniProt</A>
accessions.  We obtain the locus tags from RefSeq or from <A
HREF="http://www.microbesonline.org/">MicrobesOnline</A>.  We use
queries of the form "locus_tag AND genus_name" to try to ensure that
the paper is actually discussing that gene.  Because EuropePMC indexes
most recent biomedical papers, even if they are not open access, some
of the links may be to papers that you cannot read or that our
computers cannot read.  We query each of these identifiers that
appears in the open access part of EuropePMC, as well as every locus
tag that appears in the 500 most-referenced genomes, so that a gene
may appear in the PaperBLAST results even though none of the papers
that mention it are open access. We also incorporate text-mined links
from EuropePMC that link open access articles to UniProt or RefSeq
identifiers.  (This yields some additional links because EuropePMC
uses different heuristics for their text mining than we do.)

<P>For every article that mentions a locus tag, a RefSeq protein
identifier, or a UniProt accession, we try to select one or two
snippets of text that refer to the protein. If we cannot get access to
the full text, we try to select a snippet from the abstract, but
unfortunately, unique identifiers such as locus tags are rarely
provided in abstracts.

<P>PaperBLAST also incorporates manually-curated links between protein sequences and
articles: <UL> <LI>Proteins from NCBI's RefSeq are included if a
<A HREF="https://www.ncbi.nlm.nih.gov/gene/about-generif">GeneRIF</A>
entry links the gene to an article in <A
HREF="http://www.ncbi.nlm.nih.gov/pubmed/">PubMed</A><sup>&reg;</sup>.
GeneRIF also provides a short summary of the article's claim about the
protein, which is shown instead of a snippet.  <LI>Proteins
from Swiss-Prot (the curated part of <A HREF="http://uniprot.org">UniProt</A>) are included if the curators
identified experimental evidence for the protein's function (evidence
code ECO:0000269). For these proteins, the fields of the Swiss-Prot entry that
describe the protein's function are shown (with bold headings).
<LI>Proteins from <A HREF="http://www.brenda-enzymes.org/index.php">BRENDA</A>, a curated database of enzymes, are included if they are linked to a paper in PubMed and their full sequence is known.
  <LI>Every protein from <A HREF="http://ecocyc.org">EcoCyc</A>, a curated
database of the proteins in <i> Escherichia coli</i> K-12, is included, regardless
of whether they are characterized or not.
<LI>Proteins from the <A HREF="http://metacyc.org">MetaCyc</A> metabolic pathway database are included if they are linked to a paper in PubMed and their full sequence is known.
<LI>Every protein from <A HREF="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3245046/">CharProtDB</A>, a database of experimentally characterized protein annotations, is included.
<LI>Proteins from the <A HREF="http://www.cazy.org/">CAZy</A> database of carbohydrate-active enzymes are included if they are associated with an Enzyme Classification number. Even though CAZy does not provide links from individual protein sequences to papers, these should all be experimentally-characterized proteins.
<LI>Proteins from the <A HREF="http://rebase.neb.com/rebase/rebase.html">REBASE</A> database of restriction enzymes are included if they have known specificity.
<LI>Every protein with an evidence-based reannotation (based on mutant phenotypes) in the <A HREF="http://fit.genomics.lbl.gov/">Fitness Browser</A> is included.
  </UL>Except for GeneRIF,
the curated entries include a short curated
description of the protein's function.  Many of these entries also
link to articles in <A
HREF="http://www.ncbi.nlm.nih.gov/pubmed/">PubMed</A>.

<P>For more information see the
<A title="PaperBLAST: Text Mining Papers for Information about Homologs" HREF="http://msystems.asm.org/content/2/4/e00039-17">PaperBLAST paper</A> (<i>mSystems</i> 2017)
or the <A HREF="https://github.com/morgannprice/PaperBLAST">code</A>.
You can download PaperBLAST's database <A HREF="https://github.com/morgannprice/PaperBLAST#download">here</A>.

<P>Changes since the paper was written:

<UL>
<LI>April 2019: EuropePMC now returns table entries in their search results. This has expanded PaperBLAST's database, but most of the new entries are of low relevance, and the resulting snippets are often just lists of locus tags with annotations.
<LI>February 2018: the alignment page reports the conservation of the hit's functional sites (if available from from Swiss-Prot or UniProt)
<LI>January 2018: incorporated BRENDA.
<LI>December 2017: incorporated MetaCyc, CharProtDB, CAZy, REBASE, and the reannotations from the Fitness Browser.
<LI>September 2017: EuropePMC no longer returns some table entries in their search results. This has shrunk PaperBLAST's database, but has also reduced the number of low-relevance hits.
</UL>

<H3><A NAME="secret">Secrets</A></H3>

<P>PaperBLAST cannot provide snippets for many of the papers that are
published in non-open-access journals. This limitation applies even if
the paper is marked as "free" on the publisher's web site and is
available in PubmedCentral or EuropePMC. If a journal that you publish
in is marked as "secret," please consider publishing elsewhere.

<H3><A NAME="omission">Omissions from the PaperBLAST Database</A></H3>

<P>Many important articles are missing from PaperBLAST, either because
the article's full text is not in EuropePMC (as for many older
articles), or because the paper does not mention a protein identifier such as a locus tag, or because of PaperBLAST's heuristics. If you notice an
article that characterizes a protein's function but is missing from
PaperBLAST, please notify the curators at <A
HREF="http://www.uniprot.org/update">UniProt</A>
or add an entry to <A
HREF="https://www.ncbi.nlm.nih.gov/gene/submit-generif">GeneRIF</A>.
Entries in either of these databases will eventually be incorporated
into PaperBLAST.  Note that to add an entry to UniProt, you will need
to find the UniProt identifier for the protein.  If the protein is not
already in UniProt, you can ask them to create an entry.  To add an
entry to GeneRIF, you will need an NCBI Gene identifier, but
unfortunately many prokaryotic proteins in RefSeq do not have
corresponding Gene identifers.

<H3>References</H3>

<P><A HREF="http://msystems.asm.org/content/2/4/e00039-17">PaperBLAST: Text-mining papers for information about homologs.</A><BR>
 <small>M. N. Price and A. P. Arkin (2017). mSystems, 10.1128/mSystems.00039-17.</small>

<P><A HREF="https://doi.org/10.1093/nar/gkx1005">Europe PMC in 2017.</A><BR>
<small>M. Levchenko et al (2017). Nucleic Acids Research, 10.1093/nar/gkx1005.</small>

<P><A HREF="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1480312/">Gene indexing: characterization and analysis of NLM's GeneRIFs.</A><BR>
<small>J. A. Mitchell et al (2003). AMIA Annu Symp Proc 2003:460-464.</small>

<P><A HREF="https://doi.org/10.1093/nar/gkw1099">UniProt: the universal protein knowledgebase.</A><BR>
<small>The UniProt Consortium (2016). Nucleic Acids Research, 10.1093/nar/gkw1099.</small>

<P><A HREF="https://doi.org/10.1093/nar/gkw952">BRENDA in 2017: new perspectives and new tools in BRENDA.</A><BR>
<small>S. Placzek et al (2017). Nucleic Acids Research, 10.1093/nar/gkw952.</small>

<P><A HREF="https://doi.org/10.1093/nar/gkw1003">The EcoCyc database: reflecting new knowledge about Escherichia coli K-12.</A><BR>
<small>I. M. Keeseler et al (2016). Nucleic Acids Research, 10.1093/nar/gkw1003.</small>

<P><A HREF="https://doi.org/10.1093/nar/gkx935">The MetaCyc database of metabolic pathways and enzymes.</A><BR>
<small>R. Caspi et al (2018). Nucleic Acids Research, 10.1093/nar/gkx935.</small>

<P><A HREF="https://academic.oup.com/nar/article/40/D1/D237/2903195">CharProtDB: a database of experimentally characterized protein annotations.</A><BR>
<small>R. Madupu et al (2012). Nucleic Acids Research, 10.1093/nar/gkr1133.</small>

<P><A HREF="https://doi.org/10.1093/nar/gkt1178">The carbohydrate-active enzymes database (CAZy) in 2013.</A><BR>
<small>V. Lombard et al (2014). Nucleic Acids Research, 10.1093/nar/gkt1178.</small>

<P><A HREF="https://doi.org/10.1093/nar/gku1046">REBASE - a database for DNA restriction and modification: enzymes, genes and genomes.</A><BR>
<small>R. J. Roberts et al (2015). Nucleic Acids Research, 10.1093/nar/gku1046.</small>

<P><A HREF="http://dx.doi.org/10.1101/072470">Deep annotation of protein function across diverse bacteria from mutant phenotypes.</A><BR>
<small>M. N. Price et al (2016). bioRxiv, 10.1101/072470.</small>

<P>
<small>
<center>by <A HREF="http://morgannprice.org/">Morgan Price</A>,
<A HREF="http://genomics.lbl.gov/">Arkin group</A><BR>
Lawrence Berkeley National Laboratory
</center>
</small>
</P>

</body>
</html>